{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# PP2A - Single gene analysis in GP2 data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Description\n",
    "\n",
    "Using \"individual level data\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "### 0. Getting Started\n",
    "\n",
    "- Loading Python libraries\n",
    "- Defining functions\n",
    "- Installing packages\n",
    "\n",
    "### 1. Copy data from workspace to cloud environment\n",
    "\n",
    "### 2. Extract PP2A\n",
    "\n",
    "### 3. Annotate PP2A variants\n",
    "\n",
    "### 4. Extract coding/non-syn variants\n",
    "\n",
    "### 5. Calculate frequency in cases versus controls\n",
    "\n",
    "### 6. Calculate frequency (homozygotes) in cases versus controls\n",
    "\n",
    "### 7. Save out results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Getting Started"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "### Loading Python libraries"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Defining functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {
    "code_folding": []
   },
   "outputs": [],
   "source": [
    "# Use the os package to interact with the environment\n",
    "import os\n",
    "\n",
    "# Bring in Pandas for DatAACame functionality\n",
    "import pandas as pd\n",
    "\n",
    "# Numpy for basics\n",
    "import numpy as np\n",
    "\n",
    "# Use StringIO for working with file contents\n",
    "from io import StringIO\n",
    "\n",
    "# Enable IPython to display matplotlib graphs\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline\n",
    "\n",
    "# Enable interaction with the FireCloud API\n",
    "from firecloud import api as fapi\n",
    "\n",
    "# Import the iPython HTML rendering for displaying links to Google Cloud Console\n",
    "from IPython.core.display import display, HTML\n",
    "\n",
    "# Import urllib modules for building URLs to Google Cloud Console\n",
    "import urllib.parse\n",
    "\n",
    "# BigQuery for querying data\n",
    "from google.cloud import bigquery\n",
    "\n",
    "#Import Sys\n",
    "import sys as sys"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {
    "code_folding": [
     0
    ]
   },
   "outputs": [],
   "source": [
    "# Utility routine for printing a shell command before executing it\n",
    "def shell_do(command):\n",
    "    print(f'Executing: {command}', file=sys.stderr)\n",
    "    !$command\n",
    "    \n",
    "def shell_return(command):\n",
    "    print(f'Executing: {command}', file=sys.stderr)\n",
    "    output = !$command\n",
    "    return '\\n'.join(output)\n",
    "\n",
    "# Utility routine for printing a query before executing it\n",
    "def bq_query(query):\n",
    "    print(f'Executing: {query}', file=sys.stderr)\n",
    "    return pd.read_gbq(query, project_id=BILLING_PROJECT_ID, dialect='standard')\n",
    "\n",
    "# Utility routine for display a message and a link\n",
    "def display_html_link(description, link_text, url):\n",
    "    html = f'''\n",
    "    <p>\n",
    "    </p>\n",
    "    <p>\n",
    "    {description}\n",
    "    <a target=_blank href=\"{url}\">{link_text}</a>.\n",
    "    </p>\n",
    "    '''\n",
    "\n",
    "    display(HTML(html))\n",
    "\n",
    "# Utility routines for reading files from Google Cloud Storage\n",
    "def gcs_read_file(path):\n",
    "    \"\"\"Return the contents of a file in GCS\"\"\"\n",
    "    contents = !gsutil -u {BILLING_PROJECT_ID} cat {path}\n",
    "    return '\\n'.join(contents)\n",
    "    \n",
    "def gcs_read_csv(path, sep=None):\n",
    "    \"\"\"Return a DatAACame from the contents of a delimited file in GCS\"\"\"\n",
    "    return pd.read_csv(StringIO(gcs_read_file(path)), sep=sep, engine='python')\n",
    "\n",
    "# Utility routine for displaying a message and link to Cloud Console\n",
    "def link_to_cloud_console_gcs(description, link_text, gcs_path):\n",
    "    url = '{}?{}'.format(\n",
    "        os.path.join('https://console.cloud.google.com/storage/browser',\n",
    "                     gcs_path.replace(\"gs://\",\"\")),\n",
    "        urllib.parse.urlencode({'userProject': BILLING_PROJECT_ID}))\n",
    "\n",
    "    display_html_link(description, link_text, url)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Set paths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {
    "code_folding": [
     0
    ]
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Billing and Workspace\n",
      "Workspace Name: PP2A\n",
      "Billing Project: terra-02e7cd23\n",
      "Workspace Bucket, where you can upload and download data: gs://fc-a2382a6f-59cf-4cf3-b70e-8cd83806d5e0\n",
      "\n",
      "GP2 v3.0\n",
      "Path to GP2 v3.0 Clinical Data: gs://gp2tier2/release3_31102022/clinical_data\n",
      "Path to GP2 v3.0 Raw Genotype Data: gs://gp2tier2/release3_31102022/raw_genotypes\n",
      "Path to GP2 v3.0 Imputed Genotype Data: gs://gp2tier2/release3_31102022/imputed_genotypes\n",
      "AMP-PD v3.0\n",
      "Path to AMP-PD v3.0 Clinical Data: gs://amp-pd-data/releases/2022_v3release_1115/clinical\n",
      "Path to AMP-PD v3.0 WGS Data: gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink\n",
      "Path to AMP-PD v3.0 WGS Data: gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Set up billing project and data path variables\n",
    "BILLING_PROJECT_ID = os.environ['GOOGLE_PROJECT']\n",
    "WORKSPACE_NAMESPACE = os.environ['WORKSPACE_NAMESPACE']\n",
    "WORKSPACE_NAME = os.environ['WORKSPACE_NAME']\n",
    "WORKSPACE_BUCKET = os.environ['WORKSPACE_BUCKET']\n",
    "\n",
    "WORKSPACE_ATTRIBUTES = fapi.get_workspace(WORKSPACE_NAMESPACE, WORKSPACE_NAME).json().get('workspace',{}).get('attributes',{})\n",
    "\n",
    "## Print the information to check we are in the proper release and billing \n",
    "## This will be different for you, the user, depending on the billing project your workspace is on\n",
    "print('Billing and Workspace')\n",
    "print(f'Workspace Name: {WORKSPACE_NAME}')\n",
    "print(f'Billing Project: {BILLING_PROJECT_ID}')\n",
    "print(f'Workspace Bucket, where you can upload and download data: {WORKSPACE_BUCKET}')\n",
    "print('')\n",
    "\n",
    "\n",
    "## GP2 v3.0\n",
    "## Explicitly define release v3.0 path \n",
    "GP2_RELEASE_PATH = 'gs://gp2tier2/release3_31102022'\n",
    "GP2_CLINICAL_RELEASE_PATH = f'{GP2_RELEASE_PATH}/clinical_data'\n",
    "GP2_RAW_GENO_PATH = f'{GP2_RELEASE_PATH}/raw_genotypes'\n",
    "GP2_IMPUTED_GENO_PATH = f'{GP2_RELEASE_PATH}/imputed_genotypes'\n",
    "print('GP2 v3.0')\n",
    "print(f'Path to GP2 v3.0 Clinical Data: {GP2_CLINICAL_RELEASE_PATH}')\n",
    "print(f'Path to GP2 v3.0 Raw Genotype Data: {GP2_RAW_GENO_PATH}')\n",
    "print(f'Path to GP2 v3.0 Imputed Genotype Data: {GP2_IMPUTED_GENO_PATH}')\n",
    "\n",
    "\n",
    "## AMP-PD v3.0\n",
    "## Explicitly define release v3.0 path \n",
    "AMP_RELEASE_PATH = 'gs://amp-pd-data/releases/2022_v3release_1115'\n",
    "AMP_CLINICAL_RELEASE_PATH = f'{AMP_RELEASE_PATH}/clinical'\n",
    "\n",
    "AMP_WGS_RELEASE_PATH = 'gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS'\n",
    "AMP_WGS_RELEASE_PLINK_PATH = os.path.join(AMP_WGS_RELEASE_PATH, 'plink')\n",
    "AMP_WGS_RELEASE_PLINK_PFILES = os.path.join(AMP_WGS_RELEASE_PLINK_PATH, 'pfiles')\n",
    "\n",
    "print('AMP-PD v3.0')\n",
    "print(f'Path to AMP-PD v3.0 Clinical Data: {AMP_CLINICAL_RELEASE_PATH}')\n",
    "print(f'Path to AMP-PD v3.0 WGS Data: {AMP_WGS_RELEASE_PLINK_PATH}')\n",
    "print(f'Path to AMP-PD v3.0 WGS Data: {AMP_WGS_RELEASE_PLINK_PFILES}')\n",
    "print('')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Install packages"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Install Plink 1.9 and Plink 2.0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Plink1.9 is already installed in /home/jupyter/tools/\n",
      "Plink2 is already installed in /home/jupyter/tools/\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "mkdir -p ~/tools\n",
    "cd ~/tools\n",
    "\n",
    "if test -e /home/jupyter/tools/plink; then\n",
    "echo \"Plink1.9 is already installed in /home/jupyter/tools/\"\n",
    "\n",
    "else\n",
    "echo -e \"Downloading plink \\n    -------\"\n",
    "wget -N http://s3.amazonaws.com/plink1-assets/plink_linux_x86_64_20190304.zip \n",
    "unzip -o plink_linux_x86_64_20190304.zip\n",
    "echo -e \"\\n plink downloaded and unzipped in /home/jupyter/tools \\n \"\n",
    "\n",
    "fi\n",
    "\n",
    "\n",
    "if test -e /home/jupyter/tools/plink2; then\n",
    "echo \"Plink2 is already installed in /home/jupyter/tools/\"\n",
    "\n",
    "else\n",
    "echo -e \"Downloading plink2 \\n    -------\"\n",
    "wget -N https://s3.amazonaws.com/plink2-assets/alpha3/plink2_linux_avx2_20220603.zip\n",
    "unzip -o plink2_linux_avx2_20220603.zip\n",
    "echo -e \"\\n plink2 downloaded and unzipped in /home/jupyter/tools \\n \"\n",
    "\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annovar\n",
      "annovar.latest.tar.gz\n",
      "LICENSE\n",
      "plink\n",
      "plink2\n",
      "plink2_linux_avx2_20220603.zip\n",
      "plink_linux_x86_64_20190304.zip\n",
      "prettify\n",
      "toy.map\n",
      "toy.ped\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls /home/jupyter/tools/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Remote restrictions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "# chmod plink 1.9 \n",
    "chmod u+x /home/jupyter/tools/plink"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "# chmod plink 2.0\n",
    "chmod u+x /home/jupyter/tools/plink2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Install ANNOVAR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annovar is already installed in /home/jupyter/tools/\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Install ANNOVAR:\n",
    "# https://www.openbioinformatics.org/annovar/annovar_download_form.php\n",
    "\n",
    "if test -e /home/jupyter/tools/annovar; then\n",
    "\n",
    "echo \"annovar is already installed in /home/jupyter/tools/\"\n",
    "else\n",
    "echo \"annovar is not installed\"\n",
    "cd /home/jupyter/tools/\n",
    "\n",
    "wget http://www.openbioinformatics.org/annovar/download/0wgxR2rIVP/annovar.latest.tar.gz\n",
    "\n",
    "tar xvfz annovar.latest.tar.gz\n",
    "\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annovar\n",
      "annovar.latest.tar.gz\n",
      "LICENSE\n",
      "plink\n",
      "plink2\n",
      "plink2_linux_avx2_20220603.zip\n",
      "plink_linux_x86_64_20190304.zip\n",
      "prettify\n",
      "toy.map\n",
      "toy.ped\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls /home/jupyter/tools/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Install ANNOVAR: Download sources of annotation "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "NOTICE: Web-based checking to see whether ANNOVAR new version is available ... Done\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_refGene.txt.gz ... OK\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_refGeneMrna.fa.gz ... OK\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_refGeneVersion.txt.gz ... OK\n",
      "NOTICE: Uncompressing downloaded files\n",
      "NOTICE: Finished downloading annotation files for hg38 build version, with files saved at the 'humandb' directory\n",
      "NOTICE: Web-based checking to see whether ANNOVAR new version is available ... Done\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_clinvar_20140902.txt.gz ... OK\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_clinvar_20140902.txt.idx.gz ... OK\n",
      "NOTICE: Uncompressing downloaded files\n",
      "NOTICE: Finished downloading annotation files for hg38 build version, with files saved at the 'humandb' directory\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "cd /home/jupyter/tools/annovar/\n",
    "\n",
    "perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar refGene humandb/\n",
    "perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar clinvar_20140902 humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb cytoBand humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar ensGene humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar exac03 humandb/ \n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar avsnp147 humandb/ \n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar dbnsfp30a humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar gnomad211_genome humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar ljb26_all humandb/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annotate_variation.pl\n",
      "coding_change.pl\n",
      "convert2annovar.pl\n",
      "example\n",
      "humandb\n",
      "retrieve_seq_from_fasta.pl\n",
      "table_annovar.pl\n",
      "variants_reduction.pl\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls /home/jupyter/tools/annovar/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## AAC"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Copy data from GP2 bucket to workspace"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Making a working directory\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: mkdir -p /home/jupyter/PP2A_GP2/\n"
     ]
    }
   ],
   "source": [
    "# Make a directory\n",
    "print(\"Making a working directory\")\n",
    "WORK_DIR = f'/home/jupyter/PP2A_GP2/'\n",
    "shell_do(f'mkdir -p {WORK_DIR}') # f' significa f-string - contiene expresiones para ejecutar el codigo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -u terra-02e7cd23 ls gs://gp2tier2/release3_31102022/imputed_genotypes\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gs://gp2tier2/release3_31102022/imputed_genotypes/AAC/\r\n",
      "gs://gp2tier2/release3_31102022/imputed_genotypes/AFR/\r\n",
      "gs://gp2tier2/release3_31102022/imputed_genotypes/AJ/\r\n",
      "gs://gp2tier2/release3_31102022/imputed_genotypes/AMR/\r\n",
      "gs://gp2tier2/release3_31102022/imputed_genotypes/CAS/\r\n",
      "gs://gp2tier2/release3_31102022/imputed_genotypes/EAS/\r\n",
      "gs://gp2tier2/release3_31102022/imputed_genotypes/EUR/\r\n",
      "gs://gp2tier2/release3_31102022/imputed_genotypes/MDE/\r\n",
      "gs://gp2tier2/release3_31102022/imputed_genotypes/SAS/\r\n"
     ]
    }
   ],
   "source": [
    "# Check directory where GP2 data is\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} ls {GP2_IMPUTED_GENO_PATH}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -u terra-02e7cd23 -m cp -r gs://gp2tier2/release3_31102022/imputed_genotypes/AAC/chr9* /home/jupyter/PP2A_GP2/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Copying gs://gp2tier2/release3_31102022/imputed_genotypes/AAC/chr9_AAC_release3.log...\n",
      "Copying gs://gp2tier2/release3_31102022/imputed_genotypes/AAC/chr9_AAC_release3.psam...\n",
      "Copying gs://gp2tier2/release3_31102022/imputed_genotypes/AAC/chr9_AAC_release3.pgen...\n",
      "Copying gs://gp2tier2/release3_31102022/imputed_genotypes/AAC/chr9_AAC_release3.pvar...\n",
      "/ [4/4 files][535.9 MiB/535.9 MiB] 100% Done                                    \n",
      "Operation completed over 4 objects/535.9 MiB.                                    \n"
     ]
    }
   ],
   "source": [
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp -r {GP2_IMPUTED_GENO_PATH}/AAC/chr9* {WORK_DIR}')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Create a covariate file with GP2 data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -u terra-02e7cd23 ls gs://gp2tier2/release3_31102022/clinical_data\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gs://gp2tier2/release3_31102022/clinical_data/\r\n",
      "gs://gp2tier2/release3_31102022/clinical_data/master_key_release3_final.csv\r\n",
      "gs://gp2tier2/release3_31102022/clinical_data/release3_31102022_data_dictionary.csv\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -u terra-02e7cd23 -m cp -r gs://gp2tier2/release3_31102022/clinical_data/master_key_release3_final.csv /home/jupyter/PP2A_GP2/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Copying gs://gp2tier2/release3_31102022/clinical_data/master_key_release3_final.csv...\n",
      "/ [1/1 files][  1.6 MiB/  1.6 MiB] 100% Done                                    \n",
      "Operation completed over 1 objects/1.6 MiB.                                      \n"
     ]
    }
   ],
   "source": [
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} ls {GP2_CLINICAL_RELEASE_PATH}')\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp -r {GP2_CLINICAL_RELEASE_PATH}/master_key_release3_final.csv {WORK_DIR}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['GP2ID', 'GP2sampleID', 'manifest_id', 'phenotype', 'pheno_for_qc',\n",
       "       'other_pheno', 'sex_for_qc', 'age', 'age_of_onset', 'age_at_diagnosis',\n",
       "       'age_at_death', 'race_for_qc', 'family_history_for_qc', 'region_for_qc',\n",
       "       'study', 'pruned', 'pruned_reason', 'label', 'related'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cov = pd.read_csv(f'{WORK_DIR}/master_key_release3_final.csv')\n",
    "cov.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>SEX</th>\n",
       "      <th>AGE</th>\n",
       "      <th>PHENO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SYNAPS-KZ_000001_s1</td>\n",
       "      <td>SYNAPS-KZ_000001_s1</td>\n",
       "      <td>2</td>\n",
       "      <td>64.0</td>\n",
       "      <td>Control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SYNAPS-KZ_000002_s1</td>\n",
       "      <td>SYNAPS-KZ_000002_s1</td>\n",
       "      <td>1</td>\n",
       "      <td>53.0</td>\n",
       "      <td>Control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SYNAPS-KZ_000003_s1</td>\n",
       "      <td>SYNAPS-KZ_000003_s1</td>\n",
       "      <td>2</td>\n",
       "      <td>59.0</td>\n",
       "      <td>Control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SYNAPS-KZ_000004_s1</td>\n",
       "      <td>SYNAPS-KZ_000004_s1</td>\n",
       "      <td>1</td>\n",
       "      <td>57.0</td>\n",
       "      <td>Control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SYNAPS-KZ_000005_s1</td>\n",
       "      <td>SYNAPS-KZ_000005_s1</td>\n",
       "      <td>1</td>\n",
       "      <td>59.0</td>\n",
       "      <td>Control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16852</th>\n",
       "      <td>PPMI_000789_s1</td>\n",
       "      <td>PPMI_000789_s1</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16853</th>\n",
       "      <td>PPMI_000781_s1</td>\n",
       "      <td>PPMI_000781_s1</td>\n",
       "      <td>1</td>\n",
       "      <td>57.0</td>\n",
       "      <td>Other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16854</th>\n",
       "      <td>PPMI_000818_s1</td>\n",
       "      <td>PPMI_000818_s1</td>\n",
       "      <td>1</td>\n",
       "      <td>60.0</td>\n",
       "      <td>Other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16855</th>\n",
       "      <td>PPMI_000841_s1</td>\n",
       "      <td>PPMI_000841_s1</td>\n",
       "      <td>2</td>\n",
       "      <td>68.0</td>\n",
       "      <td>Other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16856</th>\n",
       "      <td>PPMI_000808_s1</td>\n",
       "      <td>PPMI_000808_s1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Other</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>16857 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                       FID                  IID  SEX   AGE    PHENO\n",
       "0      SYNAPS-KZ_000001_s1  SYNAPS-KZ_000001_s1    2  64.0  Control\n",
       "1      SYNAPS-KZ_000002_s1  SYNAPS-KZ_000002_s1    1  53.0  Control\n",
       "2      SYNAPS-KZ_000003_s1  SYNAPS-KZ_000003_s1    2  59.0  Control\n",
       "3      SYNAPS-KZ_000004_s1  SYNAPS-KZ_000004_s1    1  57.0  Control\n",
       "4      SYNAPS-KZ_000005_s1  SYNAPS-KZ_000005_s1    1  59.0  Control\n",
       "...                    ...                  ...  ...   ...      ...\n",
       "16852       PPMI_000789_s1       PPMI_000789_s1    2   NaN    Other\n",
       "16853       PPMI_000781_s1       PPMI_000781_s1    1  57.0    Other\n",
       "16854       PPMI_000818_s1       PPMI_000818_s1    1  60.0    Other\n",
       "16855       PPMI_000841_s1       PPMI_000841_s1    2  68.0    Other\n",
       "16856       PPMI_000808_s1       PPMI_000808_s1    1   NaN    Other\n",
       "\n",
       "[16857 rows x 5 columns]"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cov_reduced = cov[['GP2sampleID','GP2sampleID','sex_for_qc', 'age', 'phenotype']]\n",
    "cov_reduced.columns = ['FID','IID', 'SEX','AGE','PHENO']\n",
    "cov_reduced"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [],
   "source": [
    "conditions = [\n",
    "    (cov_reduced['PHENO'] == \"Case\"),\n",
    "    (cov_reduced['PHENO'] == \"Control\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.7/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  \n"
     ]
    }
   ],
   "source": [
    "choices = [2,1]\n",
    "cov_reduced['PHENOTYPE'] = np.select(conditions, choices, default=-9).astype(np.int64)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.7/site-packages/pandas/core/frame.py:4913: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  errors=errors,\n"
     ]
    }
   ],
   "source": [
    "cov_reduced.reset_index(inplace=True)\n",
    "cov_reduced.drop(columns=[\"index\"], inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [],
   "source": [
    "cov_reduced.drop(columns=['PHENO'], inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [],
   "source": [
    "sex = cov_reduced[['FID','IID','SEX']]\n",
    "sex.to_csv(f'{WORK_DIR}/SEX.txt',index=False, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [],
   "source": [
    "pheno = cov_reduced[['FID','IID','PHENOTYPE']]\n",
    "pheno.to_csv(f'{WORK_DIR}/PHENO.txt',index=False, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extract PP2A"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a3.3LM AVX2 Intel (3 Jun 2022)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2022 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to pheno_PP2A_AAC.log.\n",
      "Options in effect:\n",
      "  --chr 9\n",
      "  --from-bp 129110950\n",
      "  --make-bed\n",
      "  --out pheno_PP2A_AAC\n",
      "  --pfile chr9_AAC_release3\n",
      "  --to-bp 129148946\n",
      "\n",
      "Start time: Thu Jun  1 15:44:57 2023\n",
      "14998 MiB RAM detected; reserving 7499 MiB for main workspace.\n",
      "Using up to 4 compute threads.\n",
      "1216 samples (742 females, 474 males; 1216 founders) loaded from\n",
      "chr9_AAC_release3.psam.\n",
      "871030 variants loaded from chr9_AAC_release3.pvar.\n",
      "1 binary phenotype loaded (150 cases, 1065 controls).\n",
      "310 variants remaining after main filters.\n",
      "Writing pheno_PP2A_AAC.fam ... done.\n",
      "Writing pheno_PP2A_AAC.bim ... done.\n",
      "Writing pheno_PP2A_AAC.bed ... 0%\b\bdone.\n",
      "End time: Thu Jun  1 15:44:58 2023\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "# PODXL: gene positions on hg38 (from https://useast.ensembl.org/index.html)\n",
    "/home/jupyter/tools/plink2 \\\n",
    "--pfile chr9_AAC_release3 \\\n",
    "--chr 9 \\\n",
    "--from-bp 129110950  \\\n",
    "--to-bp 129148946 \\\n",
    "--make-bed \\\n",
    "--out pheno_PP2A_AAC"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Visualize plink files bim, fam and bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9\tchr9:129110977:A:C\t0\t129110977\tC\tA\n",
      "9\tchr9:129111023:G:C\t0\t129111023\tC\tG\n",
      "9\tchr9:129111059:G:A\t0\t129111059\tA\tG\n",
      "9\tchr9:129111157:A:T\t0\t129111157\tT\tA\n",
      "9\tchr9:129111181:T:C\t0\t129111181\tC\tT\n",
      "9\tchr9:129111184:G:C\t0\t129111184\tC\tG\n",
      "9\tchr9:129111359:T:G\t0\t129111359\tG\tT\n",
      "9\tchr9:129111392:G:C\t0\t129111392\tC\tG\n",
      "9\tchr9:129111606:T:C\t0\t129111606\tC\tT\n",
      "9\tchr9:129111885:C:T\t0\t129111885\tT\tC\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Visualize bim file\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "head pheno_PP2A_AAC.bim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\tBCM_000096_s1\t0\t0\t1\t2\n",
      "0\tBCM_000142_s1\t0\t0\t1\t2\n",
      "0\tBCM_000150_s1\t0\t0\t2\t2\n",
      "0\tBCM_000173_s1\t0\t0\t1\t2\n",
      "0\tBCM_000204_s1\t0\t0\t2\t2\n",
      "0\tBCM_000215_s1\t0\t0\t1\t2\n",
      "0\tBCM_000230_s1\t0\t0\t1\t2\n",
      "0\tBCM_000270_s1\t0\t0\t2\t1\n",
      "0\tBCM_000277_s1\t0\t0\t2\t1\n",
      "0\tBCM_000286_s1\t0\t0\t1\t2\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Visualize fam file\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "head pheno_PP2A_AAC.fam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "l\u001b\u0001������������������������������������?����������������������3��??����������/������������������������������������������������������������������?�������������������������>>���������������������������������������:���������������������������������������������������ﾾ��믿�����������������﾿�����������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������쿿����������������������������������������������������������������������������������������������������������>������������������������������������������������ο��������?�����������~�����������������������������������>��������������ﹿ���������������������������������������������������������쿿����������������������������������������������������������������������������������������������������������>������������������������������������������������ο��������?�����������~�����������������������������������>��������������ﹿ�������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������쿿����������������������������������������������������������������������������������������������������������>������������������������������������������������ο��������?�����������~�����������������������������������>��������������ﹿ\u0000\u0004\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0002\u0000\u0002\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0004\u0000\u0000\b\u0000\u0000\u0000\u0000\u0000\u0000\u0002\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000 \b\u0000\u0000�\u0000\u0000\b\u0000\u0000\u0000\b\u0000\u0002\u0000�\u0000\u0000\u0000\b\u0000�\u0000\u0000\b\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0004\u0000\u0000\u0000\u0000\u0002\u0000\u0000\u0000\u0001\u0000\u0000\u0000@\u0000\u0000\u0000\u0000\u0000\u0000\u0002\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0010\u0004\u0000\u0000\u0000 \u0000\u0000\u0000\u0000\u0000 \u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0002\u0000\u0000\u0000\u0000\u0000\u0002\u0000\b\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0002\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000 \u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\b\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000@�\u0000\u0000\u0000�\u0006\u0000\b\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000@\u0000\b\u0000\u0000\u0000�\u0000\u0000\u0010\u0000\u0000\u0000\u0000\u0000\u0000 \u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000 \u0000\u0000\u0002\u0000\u0000 \u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000���������������������������������������������������������쿿����������������������������������������������������������������������������������������������������������>������������������������������������������������ο��������?�����������~�����������������������������������>��������������ﹿ�������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������\u0000\u0000\u0000\u0000�\b�(\b\u0000\u0000\u0000\u0000\u0000\u0002\u0000�\b�\b\u0000\u0002\u0002\u0000\u0000\u0000�\u0000\n",
      "\u0000�\u0002\u0000 \b�\u0000\u0002\u0000\u0002\u0000�\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0002,�\u0000\f",
      "\u0000\u0000 \"\u0000� \u0000\u0000\u0000�\u0000\u0000\u0000\u0000�\u00020�\b\u0000\u0000�\u0000\u0000\b�\u0000\u0002\b\u0002\u0003\u0000�\u0000\u0000�\b �\u0002\u0000\b\u0000\u0000\u0000�\u0000\u0000�2\u0000 \u0000�\u0000\u0000\u0000��\b\u0002��\u0000\u0000\u0000\u0000\u0000 �\b\u0000\"\u0000�\u0000\u0000\b\u0002\u0000\u0000\u0000\u0000\u0000\u0000\b\u0000\u0000\u0000\u0000\b\u0002\u0000���\u0000 \u0000 \u0000\u0000\u0000(\u0000\u0000 \u0000\u0000\u0000\u0000\b\u0002� \n",
      "\u0000\u0000\u0002\u0000\b\u0000\u0000\u0000\u0002\u0000\n",
      "\b\u0002�\u0000\u0000�\u0000 \u0000\u0003\u0000\"\u0000\u0002\u0000\u0000\u0000� \u0000\u0002\u0000�\b\u0000\b\u0000\u0000\u0000\u0000�\u0002 \u0000\u0000\b\u0000\u0002��\"\u0000�\u0002\b� \u0000 \b\u0000\u0000\u0000\u0000�\u0000��\u0000\u0002\u0002��\u0000 �\u0002��\u0000\u0000\n",
      "\u0000\u0000\u0000�\u0000\u0000\u0000\b�\u0000��\u0000\b\u0010\u0000\u0000\u0000 \u0000\u0000 \u0000\u0000\u0000\u0002 \b\b\b0\u0002\b\"�  \u0002���\u0002\u0000�\u0000\u0000\u0000\u0000\u0000\b����������������������������������������������������������������?�����������������������������������������������n�������������>��������������������������������������������������������/��������������������������������������������������������������������������﾿�������������������������뿪껣�>��?��3:�\u000e*?;��>�຾������:�#�Ͼ��/.���ﾯ��+\u000b",
      "������ʯ������(���쯮��쫻�>����\"�,�:���\b���î����ꎊ�?�;������������ ������:<��+���\u000b",
      "���>��쮿����迳���*����.�������ϯ��0�*�器������?���?�>�.���\u0003+���˾���ʬ����/���������������2>?��¢����������¬����������/ 닲���̮��;�����ί�ﺻ;�,�ˢ�2��������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������껣�:��;��3:�\u000e*/;��>�຾������8�#�Ͼ��/.���ﾯ��+\u000b",
      "�������/������(���쯮��쫻�>����\"�,�:���\b���������ꎊ�?�;���\u000e�������� ������:<��+���\u000b",
      "���>��쪿����迳���*����,�������ϯ��0�*�器������?���?�>�.���\u0003+��������ʬ����/>��������������2>>��¢����������¬����������/ 닲���̮��;�����ʯ�ﺫ;�,�ˢ�2�������������������������������������������������������������������������������������������������������������������n�������������>�������������������������������������������������������/���������������������������������������������������������������������������﾿��������������������������������������������~���������������������������������������������������������������������������������������������������������������������������������������������������������������������������￿���>���������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������쿿����������������������������������������������������������������������������������������������������������>������������������������������������������������ο��������?�����������~�����������������������������������>��������������ﹿ�&���﫮����������r￿�������.���꺲����hꦮ���󾿲�\u001b:࿻�������⮻����󬨮���#����(������¿����+����������={�ﻮ����Ǫ�������������:ӯ����\u000f�����������?��;�;:��.�����(���ث�>��-����������ޯ����������ʲ���,ʷ�箮���.����꾾��趨��������H�������������������:�������������Ϻ�\u0000\u0004\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0002\u0000\u0002\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0004\u0000\u0000\b\u0000\u0000\u0000\u0000\u0000\u0000\u0002\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000 \b\u0000\u0000�\u0000\u0000\b\u0000\u0000\u0000\b\u0000\u0002\u0000�\u0000\u0000\u0000\b\u0000�\u0000\u0000\b\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0004\u0000\u0000\u0000\u0000\u0002@\u0000\u0000\u0001\u0000\u0000\u0000@\u0000\u0000\u0000\u0000\u0000\u0000\u0002\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0010\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000 \u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0002\u0000\b\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0002\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000 \u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\b\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000@�\u0000\u0000\u0000�\u0006\u0000\b\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000@\u0000\b\u0000\u0000\u0000�\u0000\u0000\u0010\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000 \u0000\u0000\u0002\u0000\u0000 \u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000����������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������?����������������������������������������������������������������������������������������������������﫯����������~������������뿿���뻲����������������뻾������������������������꯫�����������������+�����������?{�������������������������������������������������������/������>�����?��,�����������￻�������������<������������������������������~����������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������껣�>��?��3:�\u000f*�?��>�⺾�쿏��:�+�Ͽ��/����ﾯ��?�������ʿ�����������쯮�����?����\"���:���\b���î�������?��������������(���껯>>��+���\u000b",
      "���>����������������*����.�������ϯ��:��器������?��㿫>�.���\u000b",
      "+���뿿��ά����/����������������>?��¢�����������¬����������/ 뫲���̮��?�����ί������,�ˢ�2������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������?������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������\u0000\u0004\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0002\u0000\u0002\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0004\u0000\u0000\b\u0000\u0000\u0000\u0000\u0000\u0000\u0002\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000 \b\u0000\u0000�\u0000\u0000\b\u0000\u0000\u0000\b\u0000\u0002\u0000�\u0000\u0000\u0000\b\u0000�\u0000\u0000\b\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0004\u0000\u0000\u0000\u0000\u0002@\u0000\u0000\u0001\u0000\u0000\u0000@\u0000\u0000\u0000\u0000\u0000\u0000\u0002\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0010\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000 \u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0002\u0000\b\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0002\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000 \u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\b\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000@�\u0000\u0000\u0000�\u0006\u0000\b\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000@\u0000\b\u0000\u0000\u0000�\u0000\u0000\u0010\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000 \u0000\u0000\u0002\u0000\u0000 \u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000������������������������������������?����������������������3��??����������/���������������������������������������{����������������������������?�����������������������������������������������������������������ﯫ�:���������������������������������������������������ﾾ��믿�����������������������������﫯����������~������������뿿���뻲����������������뻾������������������������꯫�����������������+�����������?{�������������������������������������������������������/������>�����?��,�����������￻�������������<������������������������������~��������������������������������������������\u0000\u0004\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0002\u0000\u0002\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0004\u0000\u0000\b\u0000\u0000\u0000\u0000\u0000\u0000\u0002\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000 \b\u0000\u0000\u0000\u0000\u0000\b\u0000\u0000\u0000\b\u0000\u0002\u0000�\u0000\u0000\u0000\b\u0000�\u0000\u0000\b\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0004\u0000\u0000\u0000\u0000\u0002@\u0000\u0000\u0001\u0000\u0000\u0000@\u0000\u0000\u0000\u0000\u0000\u0000\u0002\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0010\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000 \u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0002\u0000\b\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0002\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000 \u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\b\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000@�\u0000\u0000\u0000�\u0006\u0000\b\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000@\u0000\b\u0000\u0000\u0000�\u0000\u0000\u0010\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000 \u0000\u0000\u0002\u0000\u0000 \u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000�껣�>��?���;�\u000f*�?��>�⺾���:�+�Ͽ��?����ﾯ��?�������ʿ������������쯮�����?����\"���:������î�������?��������������8���껯>>��+��닮��>����������������*����.����������:���視������?��㿫>�.���\u000b",
      "+���뿿�����?����������������>���ª�����������¬����������/ 뫲���쮯�?�����ί������,�ˢ�3�����������������������������������������������������������쿿����������������������������������������������������������������������������������������������������������>������������������������������������������������ο��������?�����������~�����������������������������������>��������������ﹿ������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������﫯����������~������������뿿���뻲����������������뻾������������������������꯫�����������������+�����������?{�������������������������������������������������������/������>�����?��,�����������￻�������������<������������������������������~���������������������������������������������껣�>��?��3:�\u000e*?7��>�຾������:�#�Ͼ��/.���ﾯ��+\u000b",
      "������ʯ������(���쯮��쫻�>����\"�,�:���\b���î����ꎊ�?�;������������ ������:<��+���\u000b",
      "���>��쮿����迳���*����.�������ϯ��0쪪�器������?���?�>�.���\u0003+���˾���ʬ����/���������������2>?��¢����������¬����������/ 닲���̮��;�����ί�ﺻ;�,�ˢ�2�������������������������������������������������������������������������������������������������������������������n�������������>�������������������������������������������������������/���������������������������������������������������������������������������﾿����������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������~���������������������������������������������������������������������������������������������������������������������������������������������������������������������������￿���>���������������������������������������������������������������������������������������������������������&���﫮�����������￿�������.���꺲����(ꦮ���󾿲�+:࿻�������⮻����󬨮���#����(�����������+����������>{�ﻬ����˪�������������:ӯ����\u000f�����������?��;�;:��.�����(���諯>��+,����������쯻���������ʲ���,ʻ�㮮���.����꾾��躨��������H��������������������:�������������Ϻ��껣�>��?��3:�\u000e*?;��>�຾������:�#�Ͼ��/.���ﾯ��+\u000b",
      "������ʯ������(���쯮��쫻�>����\"�,�:���\b���î����ꎊ�?�;������������ ������:<��+���\u000b",
      "���>��쮿����迳���*����.�������ϯ��0즪�器������?���?�>�.���\u0003+���˾���ʬ����/���������������2>?��¢����������¬����������/ 닲���̮��;�����ί�ﺻ;�,�ˢ�2�������������������������������������������������������������������������������������������������������������������n�������������>�������������������������������������������������������/���������������������������������������������������������������������������﾿�������������������������뿪껣�>��?��3:�\u000e*?;��>�຾������:�#�Ͼ��/.���ﾯ��+\u000b",
      "������ʯ������(���쯮��쫻�>����\"�,�:���\b���î����ꎊ�?�;������������ ������:<��+���\u000b",
      "���>��쮿����迳���*����.�������ϯ��0즪�器������?���?�>�.���\u0003+���˾���ʬ����/���������������2>?��¢����������¬����������/ 닲���̮��;�����ί�ﺻ;�,�ˢ�2�ê껣�>��?��3:�\u000e*?;��>�຾������:�#�Ͼ��/.���ﾯ��+\u000b",
      "������ʯ������(���쯮��쫻�>����\"�,�:���\b���î����ꎊ�?�;������������ ������:<��+���\u000b",
      "���>��쮿����迳���*����.�������ϯ��0즪�器������?���?�>�.���\u0003+���˾���ʬ����/���������������2>?��¢����������¬����������/ 닲���̮��;�����ί�ﺻ;�,�ˢ�2��������������������~���������������������������������������������������������������������������������������������������������������������������������������������������������������������������￿���>������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������o���������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������껣�>��?��3:�\u000e*?;��>�຾������:�#�Ͼ��/.���ﾯ��+\u000b",
      "������ʯ������(���쯮��쫻�>����\"�,�:���\b���î����ꎊ�?�;������������ ������:<��+���\u000b",
      "���>��쮿����迳���*����.�������ϯ��0즪�器������?���?�>�.���\u0003+���˾���ʬ����/���������������2>?��¢����������¬����������/ 닲���̮��;�����ί�ﺻ;�,�ˢ�2�������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������껣�>��?��3:�\u000e*?;��>�຾������:�#�Ͼ��/.���ﾯ��+\u000b",
      "������ʯ������(���쯮��쫻�>����\"�,�:���\b���î����ꎊ�?�;������������ ������:<��+���\u000b",
      "���>��쮿����迳���*����.�������ϯ��0즪�器������?���?�>�.���\u0003+���˾���ʬ����/���������������2>?��¢����������¬����������/ 닲���̮��;�����ί�ﺻ;�,�ˢ�2���>�����������������������������������������8�����������/;쿻��������������������������.������������������������������������������������������������������������?�����;��;>�������������������>����������������������ͻ��ήο��������?�����������~��������������������z������������>�������������ﹿ���������������������������������������������������������쿿�������������������������������������������������������������������������������������������������������������>������������������������������������������������ο��������?�����������~�����������������������������������>��������������ﹿ����������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������껣�>��?��3:�\u000e*?;��>�຾������:�#�Ͼ��/.���ﾯ��+\u000b",
      "������ʯ������(���쯮��쫻�>����\"�,�:���\b���î����ꎊ�?�;������������ ������:<��+���\u000b",
      "���>��쮿����迳���*����.�������ϯ��0즪�器������?���?�>�.���\u0003+���˾���ʬ����/���������������2>?��¢����������¬����������/ 닲���̮��;�����ί�ﺻ;�,�ˢ�2�ï�������������������������������������������������������������������������������������������������������;�����������?{�����������������������������������������������������������������>����������������������������������������������������������������������~��������������������������������������������껣�>��?��3:�\u000f*�?��>�⺾�쿏��:�+�Ͽ��/����ﾯ��?�������ʿ�����������쯮�����?����\"���:���\b���î�������?��������������(���껯>>��+���\u000b",
      "���>����������������*����.�������ϯ��:��器������?��㿫>�.���\u000b",
      "+���뿿��ά����/����������������>?��¢�����������¬����������/ 뫲���̮��?�����ί������,�ˢ�2�˪껣�>��?��3:�\u000e*?;��>�຾������:�#�Ͼ��/.���ﾯ��+\u000b",
      "������ʯ������(���쯮��쫻�>����\"�,�:���\b���î����ꎊ�?�;������������ ������:<��+���\u000b",
      "���>��쮿����迳���*����.�������ϯ��0즪�器������?���?�>�.���\u0003+���˾���ʬ����/���������������2>?��¢����������¬����������/ 닲���̮��;�����ί�ﺻ;�,�ˢ�2�ê껣�>��?��3:�\u000e*?;��>�຾������:�#�Ͼ��/.���ﾯ��+\u000b",
      "������ʯ������(���쯯��쫻�>����\"�,�:���\b���î����ꎊ�?�;������������ ������:<��+���\u000b",
      "���>��쮿����迳���*����.�������ϯ��0즪�器������?���?�>�.���\u0003+���˾���ʬ����/���������������2>?��¢����������¬����������/ 닲���̮��;�����ί�ﺻ;�,�ˢ�2�������������������������������������������������������������������������������������������������������������������n�������������>�������������������������������������������������������/���������������������������������������������������������������������������﾿�������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������&���﫮�����������￿�������.���꺲����(ꦮ���󾿲�+:࿻�������⮻����󬨮���#����(�����������+��������>{�ﻬ����˪�������������:ӯ����\u000f�����������?��;�;:��.�����(���諯>��+,����������쯻���������ʲ���,ʻ�㮮���.����꾾��躨��������H��������������������:�������������Ϻ�\u0002\u0004��\u0002.��+(�\"(�\f",
      "\"22��..��0�\u0000��\u0000\u0002\" \u0000\u0002�è��\u000b",
      "\"���\b�\u0006� *\u000b",
      "��\"0�\u0002\u0000�\u0000;��\u0000����\u0000��(�\u0000 ���\b 8:\f",
      "�\u0003��\u0000\n",
      "\u0000\b(��\u0002\b�+���2���\u0002�\n",
      "0.�+h8\u0000���>J�貨 �\u000e���\n",
      "8(�(#���\b\b �0� �\u000e����\n",
      "�: �����\n",
      "\"��:\u0002�*��\u0002ˊ;\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Visualize fam file\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "head pheno_PP2A_AAC.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to pheno_PP2A_AAC.log.\n",
      "Options in effect:\n",
      "  --bfile pheno_PP2A_AAC\n",
      "  --out pheno_PP2A_AAC\n",
      "  --recode vcf-fid\n",
      "\n",
      "14998 MB RAM detected; reserving 7499 MB for main workspace.\n",
      "310 variants loaded from .bim file.\n",
      "1216 people (474 males, 742 females) loaded from .fam.\n",
      "1215 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1216 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.995185.\n",
      "310 variants and 1216 people pass filters and QC.\n",
      "Among remaining phenotypes, 150 are cases and 1065 are controls.  (1 phenotype\n",
      "is missing.)\n",
      "--recode vcf-fid to pheno_PP2A_AAC.vcf ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "# Turn binary files into VCF\n",
    "/home/jupyter/tools/plink \\\n",
    "--bfile pheno_PP2A_AAC \\\n",
    "--recode vcf-fid \\\n",
    "--out pheno_PP2A_AAC"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Annotate PP2A variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[bgzip] can't create pheno_PP2A_AAC.vcf.gz: File exists\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "export PATH=$PATH:/home/jupyter/tools/rvtests/third/tabix-0.2.6/\n",
    "\n",
    "### Bgzip and Tabix\n",
    "bgzip pheno_PP2A_AAC.vcf\n",
    "\n",
    "tabix -f -p vcf pheno_PP2A_AAC.vcf.gz\n",
    "tabix -f -p vcf pheno_PP2A_AAC.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 pheno_PP2A_AAC.vcf.gz > pheno_PP2A_AAC.annovar.avinput>\n",
      "NOTICE: Finished reading 317 lines from VCF file\n",
      "NOTICE: A total of 310 locus in VCF file passed QC threshold, representing 285 SNPs (213 transitions and 72 transversions) and 25 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 346560 SNP genotypes (259008 transitions and 87552 transversions) and 30400 indels/substitutions for 1216 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl pheno_PP2A_AAC.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile pheno_PP2A_AAC.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile pheno_PP2A_AAC.annovar.refGene -exonsort -nofirstcodondel pheno_PP2A_AAC.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to pheno_PP2A_AAC.annovar.refGene.variant_function, pheno_PP2A_AAC.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 310 unique variants in 310 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 6 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile pheno_PP2A_AAC.annovar pheno_PP2A_AAC.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to pheno_PP2A_AAC.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to pheno_PP2A_AAC.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 310 unique variants in 310 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 0\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to pheno_PP2A_AAC.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from pheno_PP2A_AAC.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to pheno_PP2A_AAC.annovar.hg38_multianno.vcf\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "### Annotate variants using ANNOVAR: https://annovar.openbioinformatics.org/en/latest/ \n",
    "perl /home/jupyter/tools/annovar/table_annovar.pl pheno_PP2A_AAC.vcf.gz /home/jupyter/tools/annovar/humandb/ -buildver hg38 \\\n",
    "-out pheno_PP2A_AAC.annovar \\\n",
    "-remove -protocol refGene,clinvar_20140902 \\\n",
    "-operation g,f \\\n",
    "--nopolish \\\n",
    "-nastring . \\\n",
    "-vcfinput"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extract coding and non-syn variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>Otherinfo1219</th>\n",
       "      <th>Otherinfo1220</th>\n",
       "      <th>Otherinfo1221</th>\n",
       "      <th>Otherinfo1222</th>\n",
       "      <th>Otherinfo1223</th>\n",
       "      <th>Otherinfo1224</th>\n",
       "      <th>Otherinfo1225</th>\n",
       "      <th>Otherinfo1226</th>\n",
       "      <th>Otherinfo1227</th>\n",
       "      <th>Otherinfo1228</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9</td>\n",
       "      <td>129110977</td>\n",
       "      <td>129110977</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>PTPA</td>\n",
       "      <td>NM_021131:c.-624A&gt;C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9</td>\n",
       "      <td>129111023</td>\n",
       "      <td>129111023</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>PTPA</td>\n",
       "      <td>NM_021131:c.-578G&gt;C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>9</td>\n",
       "      <td>129111059</td>\n",
       "      <td>129111059</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>PTPA</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>./.</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>9</td>\n",
       "      <td>129111157</td>\n",
       "      <td>129111157</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>PTPA</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>./.</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9</td>\n",
       "      <td>129111181</td>\n",
       "      <td>129111181</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>PTPA</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 1239 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Chr      Start        End Ref Alt Func.refGene Gene.refGene  \\\n",
       "0    9  129110977  129110977   A   C         UTR5         PTPA   \n",
       "1    9  129111023  129111023   G   C         UTR5         PTPA   \n",
       "2    9  129111059  129111059   G   A     intronic         PTPA   \n",
       "3    9  129111157  129111157   A   T     intronic         PTPA   \n",
       "4    9  129111181  129111181   T   C     intronic         PTPA   \n",
       "\n",
       "    GeneDetail.refGene ExonicFunc.refGene AAChange.refGene  ... Otherinfo1219  \\\n",
       "0  NM_021131:c.-624A>C                  .                .  ...           0/0   \n",
       "1  NM_021131:c.-578G>C                  .                .  ...           0/0   \n",
       "2                    .                  .                .  ...           0/1   \n",
       "3                    .                  .                .  ...           0/1   \n",
       "4                    .                  .                .  ...           0/0   \n",
       "\n",
       "   Otherinfo1220 Otherinfo1221 Otherinfo1222  Otherinfo1223  Otherinfo1224  \\\n",
       "0            0/0           0/0           0/0            0/0            0/0   \n",
       "1            0/0           0/0           0/0            0/0            0/0   \n",
       "2            0/0           ./.           0/1            0/0            0/1   \n",
       "3            0/0           ./.           0/1            0/0            0/1   \n",
       "4            0/0           0/0           0/0            0/0            0/0   \n",
       "\n",
       "  Otherinfo1225 Otherinfo1226 Otherinfo1227 Otherinfo1228  \n",
       "0           0/1           0/0           0/0           0/0  \n",
       "1           0/0           0/0           0/0           0/0  \n",
       "2           0/0           0/0           0/0           0/1  \n",
       "3           0/0           0/0           0/0           0/1  \n",
       "4           0/0           0/0           0/0           0/0  \n",
       "\n",
       "[5 rows x 1239 columns]"
      ]
     },
     "execution_count": 97,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Visualize multianno file\n",
    "PP2A_AAC = pd.read_csv(f'{WORK_DIR}/pheno_PP2A_AAC.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "PP2A_AAC.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Chr              310\n",
       "Start            310\n",
       "End              310\n",
       "Ref              310\n",
       "Alt              310\n",
       "                ... \n",
       "Otherinfo1224    310\n",
       "Otherinfo1225    310\n",
       "Otherinfo1226    310\n",
       "Otherinfo1227    310\n",
       "Otherinfo1228    310\n",
       "Length: 1239, dtype: int64"
      ]
     },
     "execution_count": 98,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PP2A_AAC.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>Otherinfo1219</th>\n",
       "      <th>Otherinfo1220</th>\n",
       "      <th>Otherinfo1221</th>\n",
       "      <th>Otherinfo1222</th>\n",
       "      <th>Otherinfo1223</th>\n",
       "      <th>Otherinfo1224</th>\n",
       "      <th>Otherinfo1225</th>\n",
       "      <th>Otherinfo1226</th>\n",
       "      <th>Otherinfo1227</th>\n",
       "      <th>Otherinfo1228</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>9</td>\n",
       "      <td>129111606</td>\n",
       "      <td>129111606</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>PTPA</td>\n",
       "      <td>.</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>PTPA:NM_001271832:exon1:c.T6C:p.A2A,PTPA:NM_17...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>9</td>\n",
       "      <td>129123135</td>\n",
       "      <td>129123135</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>PTPA</td>\n",
       "      <td>.</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>PTPA:NM_001193397:exon3:c.C108T:p.S36S,PTPA:NM...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>183</th>\n",
       "      <td>9</td>\n",
       "      <td>129134882</td>\n",
       "      <td>129134882</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>PTPA</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>PTPA:NM_001271832:exon5:c.A461C:p.K154T,PTPA:N...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>199</th>\n",
       "      <td>9</td>\n",
       "      <td>129136522</td>\n",
       "      <td>129136522</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>PTPA</td>\n",
       "      <td>.</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>PTPA:NM_001271832:exon6:c.C525T:p.A175A,PTPA:N...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>293</th>\n",
       "      <td>9</td>\n",
       "      <td>129147457</td>\n",
       "      <td>129147457</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>PTPA</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>PTPA:NM_001271832:exon9:c.C878T:p.S293L,PTPA:N...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 1239 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chr      Start        End Ref Alt Func.refGene Gene.refGene  \\\n",
       "8      9  129111606  129111606   T   C       exonic         PTPA   \n",
       "94     9  129123135  129123135   C   T       exonic         PTPA   \n",
       "183    9  129134882  129134882   A   C       exonic         PTPA   \n",
       "199    9  129136522  129136522   C   T       exonic         PTPA   \n",
       "293    9  129147457  129147457   C   T       exonic         PTPA   \n",
       "\n",
       "    GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "8                    .     synonymous SNV   \n",
       "94                   .     synonymous SNV   \n",
       "183                  .  nonsynonymous SNV   \n",
       "199                  .     synonymous SNV   \n",
       "293                  .  nonsynonymous SNV   \n",
       "\n",
       "                                      AAChange.refGene  ... Otherinfo1219  \\\n",
       "8    PTPA:NM_001271832:exon1:c.T6C:p.A2A,PTPA:NM_17...  ...           0/0   \n",
       "94   PTPA:NM_001193397:exon3:c.C108T:p.S36S,PTPA:NM...  ...           0/1   \n",
       "183  PTPA:NM_001271832:exon5:c.A461C:p.K154T,PTPA:N...  ...           0/0   \n",
       "199  PTPA:NM_001271832:exon6:c.C525T:p.A175A,PTPA:N...  ...           0/0   \n",
       "293  PTPA:NM_001271832:exon9:c.C878T:p.S293L,PTPA:N...  ...           0/0   \n",
       "\n",
       "     Otherinfo1220 Otherinfo1221 Otherinfo1222  Otherinfo1223  Otherinfo1224  \\\n",
       "8              0/0           0/0           0/0            0/1            0/0   \n",
       "94             0/0           0/0           0/0            0/0            0/0   \n",
       "183            0/0           0/0           0/0            0/0            0/0   \n",
       "199            0/0           0/0           0/0            0/0            0/0   \n",
       "293            0/0           0/0           0/0            0/0            0/0   \n",
       "\n",
       "    Otherinfo1225 Otherinfo1226 Otherinfo1227 Otherinfo1228  \n",
       "8             0/0           0/0           0/0           0/0  \n",
       "94            0/0           0/0           0/0           0/0  \n",
       "183           0/0           0/0           0/0           0/0  \n",
       "199           0/0           0/0           0/0           0/0  \n",
       "293           0/0           0/0           0/0           0/0  \n",
       "\n",
       "[5 rows x 1239 columns]"
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter exonic variants\n",
    "coding_AAC = PP2A_AAC[PP2A_AAC['Func.refGene'] == 'exonic']\n",
    "coding_AAC.count()\n",
    "coding_AAC.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "183    1239\n",
       "293    1239\n",
       "dtype: int64"
      ]
     },
     "execution_count": 100,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter exonic and non-syn vars\n",
    "coding_nonsynonymous_AAC = PP2A_AAC[(PP2A_AAC['Func.refGene'] == 'exonic') & (PP2A_AAC['ExonicFunc.refGene'] == 'nonsynonymous SNV')]\n",
    "coding_nonsynonymous_AAC.count(1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>Otherinfo1219</th>\n",
       "      <th>Otherinfo1220</th>\n",
       "      <th>Otherinfo1221</th>\n",
       "      <th>Otherinfo1222</th>\n",
       "      <th>Otherinfo1223</th>\n",
       "      <th>Otherinfo1224</th>\n",
       "      <th>Otherinfo1225</th>\n",
       "      <th>Otherinfo1226</th>\n",
       "      <th>Otherinfo1227</th>\n",
       "      <th>Otherinfo1228</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>183</th>\n",
       "      <td>9</td>\n",
       "      <td>129134882</td>\n",
       "      <td>129134882</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>PTPA</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>PTPA:NM_001271832:exon5:c.A461C:p.K154T,PTPA:N...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 1239 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chr      Start        End Ref Alt Func.refGene Gene.refGene  \\\n",
       "183    9  129134882  129134882   A   C       exonic         PTPA   \n",
       "\n",
       "    GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "183                  .  nonsynonymous SNV   \n",
       "\n",
       "                                      AAChange.refGene  ... Otherinfo1219  \\\n",
       "183  PTPA:NM_001271832:exon5:c.A461C:p.K154T,PTPA:N...  ...           0/0   \n",
       "\n",
       "     Otherinfo1220 Otherinfo1221 Otherinfo1222  Otherinfo1223  Otherinfo1224  \\\n",
       "183            0/0           0/0           0/0            0/0            0/0   \n",
       "\n",
       "    Otherinfo1225 Otherinfo1226 Otherinfo1227 Otherinfo1228  \n",
       "183           0/0           0/0           0/0           0/0  \n",
       "\n",
       "[1 rows x 1239 columns]"
      ]
     },
     "execution_count": 101,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding_nonsynonymous_AAC.head(1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "     Chr      Start        End Ref Alt Func.refGene Gene.refGene  \\\n",
      "183    9  129134882  129134882   A   C       exonic         PTPA   \n",
      "293    9  129147457  129147457   C   T       exonic         PTPA   \n",
      "\n",
      "    GeneDetail.refGene ExonicFunc.refGene  \\\n",
      "183                  .  nonsynonymous SNV   \n",
      "293                  .  nonsynonymous SNV   \n",
      "\n",
      "                                      AAChange.refGene  ... Otherinfo1219  \\\n",
      "183  PTPA:NM_001271832:exon5:c.A461C:p.K154T,PTPA:N...  ...           0/0   \n",
      "293  PTPA:NM_001271832:exon9:c.C878T:p.S293L,PTPA:N...  ...           0/0   \n",
      "\n",
      "     Otherinfo1220 Otherinfo1221 Otherinfo1222  Otherinfo1223  Otherinfo1224  \\\n",
      "183            0/0           0/0           0/0            0/0            0/0   \n",
      "293            0/0           0/0           0/0            0/0            0/0   \n",
      "\n",
      "    Otherinfo1225 Otherinfo1226 Otherinfo1227 Otherinfo1228  \n",
      "183           0/0           0/0           0/0           0/0  \n",
      "293           0/0           0/0           0/0           0/0  \n",
      "\n",
      "[2 rows x 1239 columns]\n"
     ]
    }
   ],
   "source": [
    "print(coding_nonsynonymous_AAC)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>Otherinfo1219</th>\n",
       "      <th>Otherinfo1220</th>\n",
       "      <th>Otherinfo1221</th>\n",
       "      <th>Otherinfo1222</th>\n",
       "      <th>Otherinfo1223</th>\n",
       "      <th>Otherinfo1224</th>\n",
       "      <th>Otherinfo1225</th>\n",
       "      <th>Otherinfo1226</th>\n",
       "      <th>Otherinfo1227</th>\n",
       "      <th>Otherinfo1228</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>183</th>\n",
       "      <td>9</td>\n",
       "      <td>129134882</td>\n",
       "      <td>129134882</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>PTPA</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>PTPA:NM_001271832:exon5:c.A461C:p.K154T,PTPA:N...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>293</th>\n",
       "      <td>9</td>\n",
       "      <td>129147457</td>\n",
       "      <td>129147457</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>PTPA</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>PTPA:NM_001271832:exon9:c.C878T:p.S293L,PTPA:N...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 1239 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chr      Start        End Ref Alt Func.refGene Gene.refGene  \\\n",
       "183    9  129134882  129134882   A   C       exonic         PTPA   \n",
       "293    9  129147457  129147457   C   T       exonic         PTPA   \n",
       "\n",
       "    GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "183                  .  nonsynonymous SNV   \n",
       "293                  .  nonsynonymous SNV   \n",
       "\n",
       "                                      AAChange.refGene  ... Otherinfo1219  \\\n",
       "183  PTPA:NM_001271832:exon5:c.A461C:p.K154T,PTPA:N...  ...           0/0   \n",
       "293  PTPA:NM_001271832:exon9:c.C878T:p.S293L,PTPA:N...  ...           0/0   \n",
       "\n",
       "     Otherinfo1220 Otherinfo1221 Otherinfo1222  Otherinfo1223  Otherinfo1224  \\\n",
       "183            0/0           0/0           0/0            0/0            0/0   \n",
       "293            0/0           0/0           0/0            0/0            0/0   \n",
       "\n",
       "    Otherinfo1225 Otherinfo1226 Otherinfo1227 Otherinfo1228  \n",
       "183           0/0           0/0           0/0           0/0  \n",
       "293           0/0           0/0           0/0           0/0  \n",
       "\n",
       "[2 rows x 1239 columns]"
      ]
     },
     "execution_count": 103,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding_nonsynonymous_AAC.to_csv(f'{WORK_DIR}/coding_nonsynonymous.txt', sep = '\\t', index = False)\n",
    "coding_nonsynonymous_AAC.head(9)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Chr              285\n",
       "Start            285\n",
       "End              285\n",
       "Ref              285\n",
       "Alt              285\n",
       "                ... \n",
       "Otherinfo1224    285\n",
       "Otherinfo1225    285\n",
       "Otherinfo1226    285\n",
       "Otherinfo1227    285\n",
       "Otherinfo1228    285\n",
       "Length: 1239, dtype: int64"
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter intronic variants\n",
    "intronic_AAC = PP2A_AAC[PP2A_AAC['Func.refGene'] == 'intronic']\n",
    "intronic_AAC.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Chr              16\n",
       "Start            16\n",
       "End              16\n",
       "Ref              16\n",
       "Alt              16\n",
       "                 ..\n",
       "Otherinfo1224    16\n",
       "Otherinfo1225    16\n",
       "Otherinfo1226    16\n",
       "Otherinfo1227    16\n",
       "Otherinfo1228    16\n",
       "Length: 1239, dtype: int64"
      ]
     },
     "execution_count": 105,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter UTR3 variants\n",
    "UTR3_AAC = PP2A_AAC[PP2A_AAC['Func.refGene'] == ('UTR3')]\n",
    "UTR3_AAC.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Chr              4\n",
       "Start            4\n",
       "End              4\n",
       "Ref              4\n",
       "Alt              4\n",
       "                ..\n",
       "Otherinfo1224    4\n",
       "Otherinfo1225    4\n",
       "Otherinfo1226    4\n",
       "Otherinfo1227    4\n",
       "Otherinfo1228    4\n",
       "Length: 1239, dtype: int64"
      ]
     },
     "execution_count": 106,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter UTR5 variants\n",
    "UTR5_AAC = PP2A_AAC[PP2A_AAC['Func.refGene'] == ('UTR5')]\n",
    "UTR5_AAC.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Calculate freq - cases vs controls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [],
   "source": [
    "reduced_coding_nonsynonymous_AAC = coding_nonsynonymous_AAC[[\"Chr\", \"Start\", \"End\", \"Gene.refGene\"]]\n",
    "reduced_coding_nonsynonymous_AAC.to_csv(f'{WORK_DIR}/reduced_coding_nonsynonymous_AAC.txt', sep = '\\t', index = False, header= False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9\t129134882\t129134882\tPTPA\n",
      "9\t129147457\t129147457\tPTPA\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "head reduced_coding_nonsynonymous_AAC.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to coding_nonsynonymous_pheno_PP2A_AAC.log.\n",
      "Options in effect:\n",
      "  --adjust\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile pheno_PP2A_AAC\n",
      "  --extract range reduced_coding_nonsynonymous_AAC.txt\n",
      "  --out coding_nonsynonymous_pheno_PP2A_AAC\n",
      "\n",
      "14998 MB RAM detected; reserving 7499 MB for main workspace.\n",
      "310 variants loaded from .bim file.\n",
      "1216 people (474 males, 742 females) loaded from .fam.\n",
      "1215 phenotype values loaded from .fam.\n",
      "--extract range: 308 variants excluded.\n",
      "--extract range: 2 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1216 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.996299.\n",
      "2 variants and 1216 people pass filters and QC.\n",
      "Among remaining phenotypes, 150 are cases and 1065 are controls.  (1 phenotype\n",
      "is missing.)\n",
      "Writing C/C --assoc report to coding_nonsynonymous_pheno_PP2A_AAC.assoc ...\n",
      "0%\b\bdone.\n",
      "--adjust: Genomic inflation est. lambda (based on median chisq) = 2.37854.\n",
      "0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b--adjust values (2 variants) written to\n",
      "coding_nonsynonymous_pheno_PP2A_AAC.assoc.adjusted .\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "/home/jupyter/tools/plink --bfile pheno_PP2A_AAC --extract range reduced_coding_nonsynonymous_AAC.txt --assoc --out coding_nonsynonymous_pheno_PP2A_AAC --allow-no-sex --adjust"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to coding_nonsynonymous_AAC_rare_0.05.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile pheno_PP2A_AAC\n",
      "  --extract range reduced_coding_nonsynonymous_AAC.txt\n",
      "  --make-bed\n",
      "  --max-maf 0.05\n",
      "  --out coding_nonsynonymous_AAC_rare_0.05\n",
      "\n",
      "14998 MB RAM detected; reserving 7499 MB for main workspace.\n",
      "310 variants loaded from .bim file.\n",
      "1216 people (474 males, 742 females) loaded from .fam.\n",
      "1215 phenotype values loaded from .fam.\n",
      "--extract range: 308 variants excluded.\n",
      "--extract range: 2 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1216 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.996299.\n",
      "0 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "2 variants and 1216 people pass filters and QC.\n",
      "Among remaining phenotypes, 150 are cases and 1065 are controls.  (1 phenotype\n",
      "is missing.)\n",
      "--make-bed to coding_nonsynonymous_AAC_rare_0.05.bed +\n",
      "coding_nonsynonymous_AAC_rare_0.05.bim + coding_nonsynonymous_AAC_rare_0.05.fam\n",
      "... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "/home/jupyter/tools/plink --bfile pheno_PP2A_AAC --extract range reduced_coding_nonsynonymous_AAC.txt --make-bed --max-maf 0.05 --out coding_nonsynonymous_AAC_rare_0.05 --allow-no-sex"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP</th>\n",
       "      <th>UNADJ</th>\n",
       "      <th>GC</th>\n",
       "      <th>BONF</th>\n",
       "      <th>HOLM</th>\n",
       "      <th>SIDAK_SS</th>\n",
       "      <th>SIDAK_SD</th>\n",
       "      <th>FDR_BH</th>\n",
       "      <th>FDR_BY</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:129134882:A:C</td>\n",
       "      <td>0.1446</td>\n",
       "      <td>0.3442</td>\n",
       "      <td>0.2893</td>\n",
       "      <td>0.2893</td>\n",
       "      <td>0.2684</td>\n",
       "      <td>0.2684</td>\n",
       "      <td>0.2893</td>\n",
       "      <td>0.4339</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:129147457:C:T</td>\n",
       "      <td>0.8387</td>\n",
       "      <td>0.8950</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.8387</td>\n",
       "      <td>0.9740</td>\n",
       "      <td>0.8387</td>\n",
       "      <td>0.8387</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHR                 SNP   UNADJ      GC    BONF    HOLM  SIDAK_SS  \\\n",
       "0    9  chr9:129134882:A:C  0.1446  0.3442  0.2893  0.2893    0.2684   \n",
       "1    9  chr9:129147457:C:T  0.8387  0.8950  1.0000  0.8387    0.9740   \n",
       "\n",
       "   SIDAK_SD  FDR_BH  FDR_BY  \n",
       "0    0.2684  0.2893  0.4339  \n",
       "1    0.8387  0.8387  1.0000  "
      ]
     },
     "execution_count": 112,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PP2A_AAC_freq = pd.read_csv(f'{WORK_DIR}/coding_nonsynonymous_pheno_PP2A_AAC.assoc.adjusted', delim_whitespace=True)\n",
    "PP2A_AAC_freq.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to coding_nonsynonymous_pheno_PP2A_AAC.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile pheno_PP2A_AAC\n",
      "  --ci 0.95\n",
      "  --extract range reduced_coding_nonsynonymous_AAC.txt\n",
      "  --out coding_nonsynonymous_pheno_PP2A_AAC\n",
      "\n",
      "14998 MB RAM detected; reserving 7499 MB for main workspace.\n",
      "310 variants loaded from .bim file.\n",
      "1216 people (474 males, 742 females) loaded from .fam.\n",
      "1215 phenotype values loaded from .fam.\n",
      "--extract range: 308 variants excluded.\n",
      "--extract range: 2 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1216 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.996299.\n",
      "2 variants and 1216 people pass filters and QC.\n",
      "Among remaining phenotypes, 150 are cases and 1065 are controls.  (1 phenotype\n",
      "is missing.)\n",
      "Writing C/C --assoc report to coding_nonsynonymous_pheno_PP2A_AAC.assoc ...\n",
      "0%\b\bdone.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "/home/jupyter/tools/plink --bfile pheno_PP2A_AAC --extract range reduced_coding_nonsynonymous_AAC.txt --assoc --out coding_nonsynonymous_pheno_PP2A_AAC --allow-no-sex --ci 0.95"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP</th>\n",
       "      <th>BP</th>\n",
       "      <th>A1</th>\n",
       "      <th>F_A</th>\n",
       "      <th>F_U</th>\n",
       "      <th>A2</th>\n",
       "      <th>CHISQ</th>\n",
       "      <th>P</th>\n",
       "      <th>OR</th>\n",
       "      <th>SE</th>\n",
       "      <th>L95</th>\n",
       "      <th>U95</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:129134882:A:C</td>\n",
       "      <td>129134882</td>\n",
       "      <td>C</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.007049</td>\n",
       "      <td>A</td>\n",
       "      <td>2.12800</td>\n",
       "      <td>0.1446</td>\n",
       "      <td>0.000</td>\n",
       "      <td>inf</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:129147457:C:T</td>\n",
       "      <td>129147457</td>\n",
       "      <td>T</td>\n",
       "      <td>0.02013</td>\n",
       "      <td>0.018430</td>\n",
       "      <td>C</td>\n",
       "      <td>0.04142</td>\n",
       "      <td>0.8387</td>\n",
       "      <td>1.094</td>\n",
       "      <td>0.443</td>\n",
       "      <td>0.4593</td>\n",
       "      <td>2.607</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHR                 SNP         BP A1      F_A       F_U A2    CHISQ  \\\n",
       "0    9  chr9:129134882:A:C  129134882  C  0.00000  0.007049  A  2.12800   \n",
       "1    9  chr9:129147457:C:T  129147457  T  0.02013  0.018430  C  0.04142   \n",
       "\n",
       "        P     OR     SE     L95    U95  \n",
       "0  0.1446  0.000    inf  0.0000    NaN  \n",
       "1  0.8387  1.094  0.443  0.4593  2.607  "
      ]
     },
     "execution_count": 114,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PP2A_AAC_freq = pd.read_csv(f'{WORK_DIR}/coding_nonsynonymous_pheno_PP2A_AAC.assoc', delim_whitespace=True)\n",
    "PP2A_AAC_freq.head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Calculate freq of HMZ in cases versus controls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to coding_nonsynonymous_pheno_PP2A_AAC.log.\n",
      "Options in effect:\n",
      "  --bfile pheno_PP2A_AAC\n",
      "  --extract range reduced_coding_nonsynonymous_AAC.txt\n",
      "  --out coding_nonsynonymous_pheno_PP2A_AAC\n",
      "  --recode A\n",
      "\n",
      "14998 MB RAM detected; reserving 7499 MB for main workspace.\n",
      "310 variants loaded from .bim file.\n",
      "1216 people (474 males, 742 females) loaded from .fam.\n",
      "1215 phenotype values loaded from .fam.\n",
      "--extract range: 308 variants excluded.\n",
      "--extract range: 2 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1216 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.996299.\n",
      "2 variants and 1216 people pass filters and QC.\n",
      "Among remaining phenotypes, 150 are cases and 1065 are controls.  (1 phenotype\n",
      "is missing.)\n",
      "--recode A to coding_nonsynonymous_pheno_PP2A_AAC.raw ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "/home/jupyter/tools/plink --bfile pheno_PP2A_AAC --extract range reduced_coding_nonsynonymous_AAC.txt --recode A --out coding_nonsynonymous_pheno_PP2A_AAC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP</th>\n",
       "      <th>BP</th>\n",
       "      <th>A1</th>\n",
       "      <th>F_A</th>\n",
       "      <th>F_U</th>\n",
       "      <th>A2</th>\n",
       "      <th>CHISQ</th>\n",
       "      <th>P</th>\n",
       "      <th>OR</th>\n",
       "      <th>SE</th>\n",
       "      <th>L95</th>\n",
       "      <th>U95</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:129134882:A:C</td>\n",
       "      <td>129134882</td>\n",
       "      <td>C</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.007049</td>\n",
       "      <td>A</td>\n",
       "      <td>2.12800</td>\n",
       "      <td>0.1446</td>\n",
       "      <td>0.000</td>\n",
       "      <td>inf</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:129147457:C:T</td>\n",
       "      <td>129147457</td>\n",
       "      <td>T</td>\n",
       "      <td>0.02013</td>\n",
       "      <td>0.018430</td>\n",
       "      <td>C</td>\n",
       "      <td>0.04142</td>\n",
       "      <td>0.8387</td>\n",
       "      <td>1.094</td>\n",
       "      <td>0.443</td>\n",
       "      <td>0.4593</td>\n",
       "      <td>2.607</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHR                 SNP         BP A1      F_A       F_U A2    CHISQ  \\\n",
       "0    9  chr9:129134882:A:C  129134882  C  0.00000  0.007049  A  2.12800   \n",
       "1    9  chr9:129147457:C:T  129147457  T  0.02013  0.018430  C  0.04142   \n",
       "\n",
       "        P     OR     SE     L95    U95  \n",
       "0  0.1446  0.000    inf  0.0000    NaN  \n",
       "1  0.8387  1.094  0.443  0.4593  2.607  "
      ]
     },
     "execution_count": 116,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PP2A_AAC_freq = pd.read_csv(f'{WORK_DIR}/coding_nonsynonymous_pheno_PP2A_AAC.assoc', delim_whitespace=True)\n",
    "PP2A_AAC_freq.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129134882:A:C_C</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000096_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000142_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000150_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000173_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000204_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   FID            IID  PAT  MAT  SEX  PHENOTYPE  chr9:129134882:A:C_C  \\\n",
       "0    0  BCM_000096_s1    0    0    1          2                   0.0   \n",
       "1    0  BCM_000142_s1    0    0    1          2                   0.0   \n",
       "2    0  BCM_000150_s1    0    0    2          2                   0.0   \n",
       "3    0  BCM_000173_s1    0    0    1          2                   0.0   \n",
       "4    0  BCM_000204_s1    0    0    2          2                   0.0   \n",
       "\n",
       "   chr9:129147457:C:T_T  \n",
       "0                   0.0  \n",
       "1                   0.0  \n",
       "2                   0.0  \n",
       "3                   0.0  \n",
       "4                   0.0  "
      ]
     },
     "execution_count": 117,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PP2A_AAC_recode = pd.read_csv(f'{WORK_DIR}/coding_nonsynonymous_pheno_PP2A_AAC.raw', delim_whitespace=True)\n",
    "PP2A_AAC_recode.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to coding_nonsynonymous_pheno_PP2A_AAC.log.\n",
      "Options in effect:\n",
      "  --bfile pheno_PP2A_AAC\n",
      "  --extract range reduced_coding_nonsynonymous_AAC.txt\n",
      "  --out coding_nonsynonymous_pheno_PP2A_AAC\n",
      "  --recode A\n",
      "\n",
      "14998 MB RAM detected; reserving 7499 MB for main workspace.\n",
      "310 variants loaded from .bim file.\n",
      "1216 people (474 males, 742 females) loaded from .fam.\n",
      "1215 phenotype values loaded from .fam.\n",
      "--extract range: 308 variants excluded.\n",
      "--extract range: 2 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1216 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.996299.\n",
      "2 variants and 1216 people pass filters and QC.\n",
      "Among remaining phenotypes, 150 are cases and 1065 are controls.  (1 phenotype\n",
      "is missing.)\n",
      "--recode A to coding_nonsynonymous_pheno_PP2A_AAC.raw ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "/home/jupyter/tools/plink --bfile  pheno_PP2A_AAC --extract range reduced_coding_nonsynonymous_AAC.txt --recode A --out coding_nonsynonymous_pheno_PP2A_AAC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129134882:A:C_C</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [FID, IID, PAT, MAT, SEX, PHENOTYPE, chr9:129134882:A:C_C, chr9:129147457:C:T_T]\n",
       "Index: []"
      ]
     },
     "execution_count": 119,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Explore HMZs for chr9:129147457:C:T_T in cases\n",
    "PP2A_hom_casos_AAC = PP2A_AAC_recode[(PP2A_AAC_recode['chr9:129147457:C:T_T'] == 2) & (PP2A_AAC_recode['PHENOTYPE'] == 2)]\n",
    "PP2A_hom_casos_AAC.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(0, 8)"
      ]
     },
     "execution_count": 120,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PP2A_hom_casos_AAC.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129134882:A:C_C</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [FID, IID, PAT, MAT, SEX, PHENOTYPE, chr9:129134882:A:C_C, chr9:129147457:C:T_T]\n",
       "Index: []"
      ]
     },
     "execution_count": 121,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Explore HMZs for chr9:129147457:C:T_T in controls\n",
    "PP2A_hom_controles_AAC = PP2A_AAC_recode[(PP2A_AAC_recode['chr9:129147457:C:T_T'] == 2) & (PP2A_AAC_recode['PHENOTYPE'] == 1)]\n",
    "PP2A_hom_controles_AAC.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(0, 8)"
      ]
     },
     "execution_count": 122,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PP2A_hom_controles_AAC.shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "COMP HET"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(60, 8)\n"
     ]
    }
   ],
   "source": [
    "# Identifying individuals het for the alternative allele\n",
    "het = PP2A_AAC_recode[(PP2A_AAC_recode['chr9:129147457:C:T_T'] == 1) | (PP2A_AAC_recode['chr9:129134882:A:C_C'] == 1)]\n",
    "het = het.dropna()\n",
    "print(het.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129134882:A:C_C</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>0</td>\n",
       "      <td>BIOFIND_000173_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>0</td>\n",
       "      <td>BLAACPD-KPM_000016_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>0</td>\n",
       "      <td>BLAACPD-UAB_000014_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>0</td>\n",
       "      <td>BLAACPD-UAB_000032_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>0</td>\n",
       "      <td>BLAACPD-UC_000008_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    FID                    IID  PAT  MAT  SEX  PHENOTYPE  \\\n",
       "22    0      BIOFIND_000173_s1    0    0    2          1   \n",
       "30    0  BLAACPD-KPM_000016_s1    0    0    2          1   \n",
       "66    0  BLAACPD-UAB_000014_s1    0    0    1          2   \n",
       "70    0  BLAACPD-UAB_000032_s1    0    0    2          1   \n",
       "85    0   BLAACPD-UC_000008_s1    0    0    1          2   \n",
       "\n",
       "    chr9:129134882:A:C_C  chr9:129147457:C:T_T  \n",
       "22                   0.0                   1.0  \n",
       "30                   1.0                   0.0  \n",
       "66                   0.0                   1.0  \n",
       "70                   0.0                   1.0  \n",
       "85                   0.0                   1.0  "
      ]
     },
     "execution_count": 124,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "het.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1.0    60\n",
       "Name: compound, dtype: int64"
      ]
     },
     "execution_count": 125,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "het['compound'] = het['chr9:129147457:C:T_T'] + het['chr9:129134882:A:C_C']\n",
    "het['compound'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(0, 9)\n"
     ]
    }
   ],
   "source": [
    "comp_het_info_cases = het[(het['PHENOTYPE'] == 2) & (het['compound'] == 2)]\n",
    "print(comp_het_info_cases.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(54, 9)\n"
     ]
    }
   ],
   "source": [
    "comp_het_info_control = het[(het['PHENOTYPE'] == 1) & (het['compound'] == 1)]\n",
    "print(comp_het_info_control.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {},
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: '/home/jupyter/P2PA_GP2/comp_het_info_cases.txt'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m/tmp/ipykernel_7473/3726340871.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mcomp_het_info_cases\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mto_csv\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'/home/jupyter/P2PA_GP2/comp_het_info_cases.txt'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msep\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m'\\t'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mindex\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0mcomp_het_info_cases\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhead\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/conda/lib/python3.7/site-packages/pandas/core/generic.py\u001b[0m in \u001b[0;36mto_csv\u001b[0;34m(self, path_or_buf, sep, na_rep, float_format, columns, header, index, index_label, mode, encoding, compression, quoting, quotechar, line_terminator, chunksize, date_format, doublequote, escapechar, decimal, errors, storage_options)\u001b[0m\n\u001b[1;32m   3480\u001b[0m             \u001b[0mdoublequote\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdoublequote\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3481\u001b[0m             \u001b[0mescapechar\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mescapechar\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3482\u001b[0;31m             \u001b[0mstorage_options\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mstorage_options\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3483\u001b[0m         )\n\u001b[1;32m   3484\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/conda/lib/python3.7/site-packages/pandas/io/formats/format.py\u001b[0m in \u001b[0;36mto_csv\u001b[0;34m(self, path_or_buf, encoding, sep, columns, index_label, mode, compression, quoting, quotechar, line_terminator, chunksize, date_format, doublequote, escapechar, errors, storage_options)\u001b[0m\n\u001b[1;32m   1103\u001b[0m             \u001b[0mformatter\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfmt\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1104\u001b[0m         )\n\u001b[0;32m-> 1105\u001b[0;31m         \u001b[0mcsv_formatter\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msave\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1106\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1107\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mcreated_buffer\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/conda/lib/python3.7/site-packages/pandas/io/formats/csvs.py\u001b[0m in \u001b[0;36msave\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    241\u001b[0m             \u001b[0merrors\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0merrors\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    242\u001b[0m             \u001b[0mcompression\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcompression\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 243\u001b[0;31m             \u001b[0mstorage_options\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstorage_options\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    244\u001b[0m         ) as handles:\n\u001b[1;32m    245\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/conda/lib/python3.7/site-packages/pandas/io/common.py\u001b[0m in \u001b[0;36mget_handle\u001b[0;34m(path_or_buf, mode, encoding, compression, memory_map, is_text, errors, storage_options)\u001b[0m\n\u001b[1;32m    705\u001b[0m                 \u001b[0mencoding\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mioargs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mencoding\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    706\u001b[0m                 \u001b[0merrors\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0merrors\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 707\u001b[0;31m                 \u001b[0mnewline\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    708\u001b[0m             )\n\u001b[1;32m    709\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: '/home/jupyter/P2PA_GP2/comp_het_info_cases.txt'"
     ]
    }
   ],
   "source": [
    "comp_het_info_cases.to_csv('/home/jupyter/P2PA_GP2/comp_het_info_cases.txt', sep = '\\t', index=True)\n",
    "comp_het_info_cases.head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129134882:A:C_C</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>0</td>\n",
       "      <td>BLAACPD-UAB_000014_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>0</td>\n",
       "      <td>BLAACPD-UC_000008_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>147</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_003677_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>164</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_004266_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>166</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_004346_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1196</th>\n",
       "      <td>0</td>\n",
       "      <td>PPMI_000087_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      FID                    IID  PAT  MAT  SEX  PHENOTYPE  \\\n",
       "66      0  BLAACPD-UAB_000014_s1    0    0    1          2   \n",
       "85      0   BLAACPD-UC_000008_s1    0    0    1          2   \n",
       "147     0      CORIELL_003677_s1    0    0    2          2   \n",
       "164     0      CORIELL_004266_s1    0    0    2          2   \n",
       "166     0      CORIELL_004346_s1    0    0    2          2   \n",
       "1196    0         PPMI_000087_s1    0    0    2          2   \n",
       "\n",
       "      chr9:129134882:A:C_C  chr9:129147457:C:T_T  \n",
       "66                     0.0                   1.0  \n",
       "85                     0.0                   1.0  \n",
       "147                    0.0                   1.0  \n",
       "164                    0.0                   1.0  \n",
       "166                    0.0                   1.0  \n",
       "1196                   0.0                   1.0  "
      ]
     },
     "execution_count": 129,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Explore HETs for chr9:129147457:C:T_T in cases\n",
    "PP2A_het_casos_129147457_AAC = PP2A_AAC_recode[(PP2A_AAC_recode['chr9:129147457:C:T_T'] == 1) & (PP2A_AAC_recode['PHENOTYPE'] == 2)]\n",
    "PP2A_het_casos_129147457_AAC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129134882:A:C_C</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>0</td>\n",
       "      <td>BIOFIND_000173_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>0</td>\n",
       "      <td>BLAACPD-UAB_000032_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>0</td>\n",
       "      <td>BLAACPD-UC_000021_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>171</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_004672_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>175</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_004696_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>181</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_004736_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_005187_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>204</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_005874_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>205</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_005948_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>222</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006457_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>280</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007000_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>344</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007332_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>405</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007450_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>433</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007492_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>458</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007529_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>481</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007562_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>520</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007619_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>534</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007635_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>538</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007641_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>599</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007722_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>606</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007733_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>609</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007736_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>631</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007802_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>708</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007917_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>729</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007956_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>763</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008004_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>778</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008023_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>869</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008157_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>914</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008216_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>936</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008249_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>965</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008287_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>984</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008318_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1006</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008345_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1009</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008349_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1018</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008359_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1073</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008445_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1102</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008484_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1113</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008501_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1165</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008575_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      FID                    IID  PAT  MAT  SEX  PHENOTYPE  \\\n",
       "22      0      BIOFIND_000173_s1    0    0    2          1   \n",
       "70      0  BLAACPD-UAB_000032_s1    0    0    2          1   \n",
       "95      0   BLAACPD-UC_000021_s1    0    0    2          1   \n",
       "171     0      CORIELL_004672_s1    0    0    2          1   \n",
       "175     0      CORIELL_004696_s1    0    0    1          1   \n",
       "181     0      CORIELL_004736_s1    0    0    1          1   \n",
       "193     0      CORIELL_005187_s1    0    0    2          1   \n",
       "204     0      CORIELL_005874_s1    0    0    1          1   \n",
       "205     0      CORIELL_005948_s1    0    0    1          1   \n",
       "222     0      CORIELL_006457_s1    0    0    2          1   \n",
       "280     0      CORIELL_007000_s1    0    0    2          1   \n",
       "344     0      CORIELL_007332_s1    0    0    1          1   \n",
       "405     0      CORIELL_007450_s1    0    0    1          1   \n",
       "433     0      CORIELL_007492_s1    0    0    2          1   \n",
       "458     0      CORIELL_007529_s1    0    0    1          1   \n",
       "481     0      CORIELL_007562_s1    0    0    1          1   \n",
       "520     0      CORIELL_007619_s1    0    0    1          1   \n",
       "534     0      CORIELL_007635_s1    0    0    1          1   \n",
       "538     0      CORIELL_007641_s1    0    0    2          1   \n",
       "599     0      CORIELL_007722_s1    0    0    2          1   \n",
       "606     0      CORIELL_007733_s1    0    0    1          1   \n",
       "609     0      CORIELL_007736_s1    0    0    2          1   \n",
       "631     0      CORIELL_007802_s1    0    0    1          1   \n",
       "708     0      CORIELL_007917_s1    0    0    2          1   \n",
       "729     0      CORIELL_007956_s1    0    0    2          1   \n",
       "763     0      CORIELL_008004_s1    0    0    1          1   \n",
       "778     0      CORIELL_008023_s1    0    0    2          1   \n",
       "869     0      CORIELL_008157_s1    0    0    2          1   \n",
       "914     0      CORIELL_008216_s1    0    0    2          1   \n",
       "936     0      CORIELL_008249_s1    0    0    1          1   \n",
       "965     0      CORIELL_008287_s1    0    0    1          1   \n",
       "984     0      CORIELL_008318_s1    0    0    2          1   \n",
       "1006    0      CORIELL_008345_s1    0    0    1          1   \n",
       "1009    0      CORIELL_008349_s1    0    0    2          1   \n",
       "1018    0      CORIELL_008359_s1    0    0    2          1   \n",
       "1073    0      CORIELL_008445_s1    0    0    2          1   \n",
       "1102    0      CORIELL_008484_s1    0    0    1          1   \n",
       "1113    0      CORIELL_008501_s1    0    0    2          1   \n",
       "1165    0      CORIELL_008575_s1    0    0    2          1   \n",
       "\n",
       "      chr9:129134882:A:C_C  chr9:129147457:C:T_T  \n",
       "22                     0.0                   1.0  \n",
       "70                     0.0                   1.0  \n",
       "95                     0.0                   1.0  \n",
       "171                    0.0                   1.0  \n",
       "175                    0.0                   1.0  \n",
       "181                    0.0                   1.0  \n",
       "193                    0.0                   1.0  \n",
       "204                    0.0                   1.0  \n",
       "205                    0.0                   1.0  \n",
       "222                    0.0                   1.0  \n",
       "280                    0.0                   1.0  \n",
       "344                    0.0                   1.0  \n",
       "405                    0.0                   1.0  \n",
       "433                    0.0                   1.0  \n",
       "458                    0.0                   1.0  \n",
       "481                    0.0                   1.0  \n",
       "520                    0.0                   1.0  \n",
       "534                    0.0                   1.0  \n",
       "538                    0.0                   1.0  \n",
       "599                    0.0                   1.0  \n",
       "606                    0.0                   1.0  \n",
       "609                    0.0                   1.0  \n",
       "631                    0.0                   1.0  \n",
       "708                    0.0                   1.0  \n",
       "729                    0.0                   1.0  \n",
       "763                    0.0                   1.0  \n",
       "778                    0.0                   1.0  \n",
       "869                    0.0                   1.0  \n",
       "914                    0.0                   1.0  \n",
       "936                    0.0                   1.0  \n",
       "965                    0.0                   1.0  \n",
       "984                    0.0                   1.0  \n",
       "1006                   0.0                   1.0  \n",
       "1009                   0.0                   1.0  \n",
       "1018                   0.0                   1.0  \n",
       "1073                   0.0                   1.0  \n",
       "1102                   0.0                   1.0  \n",
       "1113                   0.0                   1.0  \n",
       "1165                   0.0                   1.0  "
      ]
     },
     "execution_count": 130,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Explore HETs for chr9:129147457:C:T_T in controls\n",
    "PP2A_het_controles_129147457_AAC = PP2A_AAC_recode[(PP2A_AAC_recode['chr9:129147457:C:T_T'] == 1) & (PP2A_AAC_recode['PHENOTYPE'] == 1)]\n",
    "PP2A_het_controles_129147457_AAC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: rpy2 in /home/jupyter/.local/lib/python3.7/site-packages (3.5.12)\n",
      "Requirement already satisfied: cffi>=1.10.0 in /opt/conda/lib/python3.7/site-packages (from rpy2) (1.15.0)\n",
      "Requirement already satisfied: jinja2 in /opt/conda/lib/python3.7/site-packages (from rpy2) (3.1.2)\n",
      "Requirement already satisfied: pytz in /opt/conda/lib/python3.7/site-packages (from rpy2) (2022.6)\n",
      "Requirement already satisfied: tzlocal<5.0 in /home/jupyter/.local/lib/python3.7/site-packages (from rpy2) (4.3)\n",
      "Requirement already satisfied: typing-extensions in /opt/conda/lib/python3.7/site-packages (from rpy2) (4.1.1)\n",
      "Requirement already satisfied: pycparser in /opt/conda/lib/python3.7/site-packages (from cffi>=1.10.0->rpy2) (2.21)\n",
      "Requirement already satisfied: pytz-deprecation-shim in /home/jupyter/.local/lib/python3.7/site-packages (from tzlocal<5.0->rpy2) (0.1.0.post0)\n",
      "Requirement already satisfied: backports.zoneinfo in /home/jupyter/.local/lib/python3.7/site-packages (from tzlocal<5.0->rpy2) (0.2.1)\n",
      "Requirement already satisfied: MarkupSafe>=2.0 in /opt/conda/lib/python3.7/site-packages (from jinja2->rpy2) (2.0.1)\n",
      "Requirement already satisfied: tzdata in /home/jupyter/.local/lib/python3.7/site-packages (from pytz-deprecation-shim->tzlocal<5.0->rpy2) (2023.3)\n"
     ]
    }
   ],
   "source": [
    "!pip install rpy2\n",
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The rpy2.ipython extension is already loaded. To reload it, use:\n",
      "  %reload_ext rpy2.ipython\n"
     ]
    }
   ],
   "source": [
    "import rpy2\n",
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "metadata": {},
   "outputs": [],
   "source": [
    "PP2A_het_casos_129147457_AAC.to_csv('PP2A_het_casos_129147457_AAC.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129134882:A:C_C</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>66</td>\n",
       "      <td>0</td>\n",
       "      <td>BLAACPD-UAB_000014_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>85</td>\n",
       "      <td>0</td>\n",
       "      <td>BLAACPD-UC_000008_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>147</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_003677_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>164</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_004266_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>166</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_004346_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1196</td>\n",
       "      <td>0</td>\n",
       "      <td>PPMI_000087_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Unnamed: 0  FID                    IID  PAT  MAT  SEX  PHENOTYPE  \\\n",
       "0          66    0  BLAACPD-UAB_000014_s1    0    0    1          2   \n",
       "1          85    0   BLAACPD-UC_000008_s1    0    0    1          2   \n",
       "2         147    0      CORIELL_003677_s1    0    0    2          2   \n",
       "3         164    0      CORIELL_004266_s1    0    0    2          2   \n",
       "4         166    0      CORIELL_004346_s1    0    0    2          2   \n",
       "5        1196    0         PPMI_000087_s1    0    0    2          2   \n",
       "\n",
       "   chr9:129134882:A:C_C  chr9:129147457:C:T_T  \n",
       "0                   0.0                   1.0  \n",
       "1                   0.0                   1.0  \n",
       "2                   0.0                   1.0  \n",
       "3                   0.0                   1.0  \n",
       "4                   0.0                   1.0  \n",
       "5                   0.0                   1.0  "
      ]
     },
     "execution_count": 147,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "temp = pd.read_csv('PP2A_het_casos_129147457_AAC.csv')\n",
    "temp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(6, 8)"
      ]
     },
     "execution_count": 151,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PP2A_het_casos_129147457_AAC.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "metadata": {},
   "outputs": [],
   "source": [
    "PP2A_het_controles_129147457_AAC.to_csv('PP2A_het_controles_129147457_AAC.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129134882:A:C_C</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>BIOFIND_000173_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>70</td>\n",
       "      <td>0</td>\n",
       "      <td>BLAACPD-UAB_000032_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>95</td>\n",
       "      <td>0</td>\n",
       "      <td>BLAACPD-UC_000021_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>171</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_004672_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>175</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_004696_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>181</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_004736_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>193</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_005187_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>204</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_005874_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>205</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_005948_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>222</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006457_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>280</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007000_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>344</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007332_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>405</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007450_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>433</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007492_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>458</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007529_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>481</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007562_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>520</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007619_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>534</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007635_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>538</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007641_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>599</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007722_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>606</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007733_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>609</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007736_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>631</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007802_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>708</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007917_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>729</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007956_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>763</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008004_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>778</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008023_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>869</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008157_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>914</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008216_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>936</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008249_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>965</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008287_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>984</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008318_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>1006</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008345_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>1009</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008349_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>1018</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008359_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>1073</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008445_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>1102</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008484_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>1113</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008501_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>1165</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008575_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Unnamed: 0  FID                    IID  PAT  MAT  SEX  PHENOTYPE  \\\n",
       "0           22    0      BIOFIND_000173_s1    0    0    2          1   \n",
       "1           70    0  BLAACPD-UAB_000032_s1    0    0    2          1   \n",
       "2           95    0   BLAACPD-UC_000021_s1    0    0    2          1   \n",
       "3          171    0      CORIELL_004672_s1    0    0    2          1   \n",
       "4          175    0      CORIELL_004696_s1    0    0    1          1   \n",
       "5          181    0      CORIELL_004736_s1    0    0    1          1   \n",
       "6          193    0      CORIELL_005187_s1    0    0    2          1   \n",
       "7          204    0      CORIELL_005874_s1    0    0    1          1   \n",
       "8          205    0      CORIELL_005948_s1    0    0    1          1   \n",
       "9          222    0      CORIELL_006457_s1    0    0    2          1   \n",
       "10         280    0      CORIELL_007000_s1    0    0    2          1   \n",
       "11         344    0      CORIELL_007332_s1    0    0    1          1   \n",
       "12         405    0      CORIELL_007450_s1    0    0    1          1   \n",
       "13         433    0      CORIELL_007492_s1    0    0    2          1   \n",
       "14         458    0      CORIELL_007529_s1    0    0    1          1   \n",
       "15         481    0      CORIELL_007562_s1    0    0    1          1   \n",
       "16         520    0      CORIELL_007619_s1    0    0    1          1   \n",
       "17         534    0      CORIELL_007635_s1    0    0    1          1   \n",
       "18         538    0      CORIELL_007641_s1    0    0    2          1   \n",
       "19         599    0      CORIELL_007722_s1    0    0    2          1   \n",
       "20         606    0      CORIELL_007733_s1    0    0    1          1   \n",
       "21         609    0      CORIELL_007736_s1    0    0    2          1   \n",
       "22         631    0      CORIELL_007802_s1    0    0    1          1   \n",
       "23         708    0      CORIELL_007917_s1    0    0    2          1   \n",
       "24         729    0      CORIELL_007956_s1    0    0    2          1   \n",
       "25         763    0      CORIELL_008004_s1    0    0    1          1   \n",
       "26         778    0      CORIELL_008023_s1    0    0    2          1   \n",
       "27         869    0      CORIELL_008157_s1    0    0    2          1   \n",
       "28         914    0      CORIELL_008216_s1    0    0    2          1   \n",
       "29         936    0      CORIELL_008249_s1    0    0    1          1   \n",
       "30         965    0      CORIELL_008287_s1    0    0    1          1   \n",
       "31         984    0      CORIELL_008318_s1    0    0    2          1   \n",
       "32        1006    0      CORIELL_008345_s1    0    0    1          1   \n",
       "33        1009    0      CORIELL_008349_s1    0    0    2          1   \n",
       "34        1018    0      CORIELL_008359_s1    0    0    2          1   \n",
       "35        1073    0      CORIELL_008445_s1    0    0    2          1   \n",
       "36        1102    0      CORIELL_008484_s1    0    0    1          1   \n",
       "37        1113    0      CORIELL_008501_s1    0    0    2          1   \n",
       "38        1165    0      CORIELL_008575_s1    0    0    2          1   \n",
       "\n",
       "    chr9:129134882:A:C_C  chr9:129147457:C:T_T  \n",
       "0                    0.0                   1.0  \n",
       "1                    0.0                   1.0  \n",
       "2                    0.0                   1.0  \n",
       "3                    0.0                   1.0  \n",
       "4                    0.0                   1.0  \n",
       "5                    0.0                   1.0  \n",
       "6                    0.0                   1.0  \n",
       "7                    0.0                   1.0  \n",
       "8                    0.0                   1.0  \n",
       "9                    0.0                   1.0  \n",
       "10                   0.0                   1.0  \n",
       "11                   0.0                   1.0  \n",
       "12                   0.0                   1.0  \n",
       "13                   0.0                   1.0  \n",
       "14                   0.0                   1.0  \n",
       "15                   0.0                   1.0  \n",
       "16                   0.0                   1.0  \n",
       "17                   0.0                   1.0  \n",
       "18                   0.0                   1.0  \n",
       "19                   0.0                   1.0  \n",
       "20                   0.0                   1.0  \n",
       "21                   0.0                   1.0  \n",
       "22                   0.0                   1.0  \n",
       "23                   0.0                   1.0  \n",
       "24                   0.0                   1.0  \n",
       "25                   0.0                   1.0  \n",
       "26                   0.0                   1.0  \n",
       "27                   0.0                   1.0  \n",
       "28                   0.0                   1.0  \n",
       "29                   0.0                   1.0  \n",
       "30                   0.0                   1.0  \n",
       "31                   0.0                   1.0  \n",
       "32                   0.0                   1.0  \n",
       "33                   0.0                   1.0  \n",
       "34                   0.0                   1.0  \n",
       "35                   0.0                   1.0  \n",
       "36                   0.0                   1.0  \n",
       "37                   0.0                   1.0  \n",
       "38                   0.0                   1.0  "
      ]
     },
     "execution_count": 156,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "temp = pd.read_csv('PP2A_het_controles_129147457_AAC.csv')\n",
    "temp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(39, 8)"
      ]
     },
     "execution_count": 152,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PP2A_het_controles_129147457_AAC.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                  GP2ID         GP2sampleID manifest_id phenotype pheno_for_qc\n",
      "    1: SYNAPS-KZ_000001 SYNAPS-KZ_000001_s1          m1   Control            1\n",
      "    2: SYNAPS-KZ_000002 SYNAPS-KZ_000002_s1          m1   Control            1\n",
      "    3: SYNAPS-KZ_000003 SYNAPS-KZ_000003_s1          m1   Control            1\n",
      "    4: SYNAPS-KZ_000004 SYNAPS-KZ_000004_s1          m1   Control            1\n",
      "    5: SYNAPS-KZ_000005 SYNAPS-KZ_000005_s1          m1   Control            1\n",
      "   ---                                                                        \n",
      "16853:      PPMI_000789      PPMI_000789_s1          m2     Other           -9\n",
      "16854:      PPMI_000781      PPMI_000781_s1          m2     Other           -9\n",
      "16855:      PPMI_000818      PPMI_000818_s1          m2     Other           -9\n",
      "16856:      PPMI_000841      PPMI_000841_s1          m2     Other           -9\n",
      "16857:      PPMI_000808      PPMI_000808_s1          m2     Other           -9\n",
      "                                  other_pheno sex_for_qc age age_of_onset\n",
      "    1:                                Control          2  64           NA\n",
      "    2:                                Control          1  53           NA\n",
      "    3:                                Control          2  59           NA\n",
      "    4:                                Control          1  57           NA\n",
      "    5:                                Control          1  59           NA\n",
      "   ---                                                                   \n",
      "16853:   Genetically-enriched-cohort Affected          2  NA           NA\n",
      "16854: Genetidally-enriched-cohort Unaffected          1  57           NA\n",
      "16855:   Genetically-enriched-cohort Affected          1  60           NA\n",
      "16856:   Genetically-enriched-cohort Affected          2  68           NA\n",
      "16857:   Genetically-enriched-cohort Affected          1  NA           NA\n",
      "       age_at_diagnosis age_at_death race_for_qc family_history_for_qc\n",
      "    1:               NA           NA       Asian          Not Reported\n",
      "    2:               NA           NA       Asian          Not Reported\n",
      "    3:               NA           NA       Asian          Not Reported\n",
      "    4:               NA           NA       Asian          Not Reported\n",
      "    5:               NA           NA       Asian          Not Reported\n",
      "   ---                                                                \n",
      "16853:               NA           NA       White          Not Reported\n",
      "16854:               NA           NA       White          Not Reported\n",
      "16855:               55           55       White          Not Reported\n",
      "16856:               57           57       White          Not Reported\n",
      "16857:               NA           NA       White          Not Reported\n",
      "       region_for_qc     study pruned  pruned_reason label related\n",
      "    1:           KAZ SYNAPS-KZ      0                  CAS      NA\n",
      "    2:           KAZ SYNAPS-KZ      0                  CAS      NA\n",
      "    3:           KAZ SYNAPS-KZ      0                  CAS      NA\n",
      "    4:           KAZ SYNAPS-KZ      0                  CAS      NA\n",
      "    5:           KAZ SYNAPS-KZ      0                  CAS      NA\n",
      "   ---                                                            \n",
      "16853:           ISR      PPMI      0                   AJ      NA\n",
      "16854:           ISR      PPMI      0                   AJ      NA\n",
      "16855:           ISR      PPMI      1 callrate_prune            NA\n",
      "16856:           ISR      PPMI      0                   AJ       1\n",
      "16857:           ISR      PPMI      0                   AJ      NA\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "data.table 1.14.6 using 2 threads (see ?getDTthreads).  Latest news: r-datatable.com\n",
       "In addition: Warning message:\n",
       "In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
       "  libraries ‘/home/jupyter/packages’, ‘/usr/lib/R/site-library’ contain no packages\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "setwd('/home/jupyter/PP2A_GP2/')\n",
    "library(data.table)\n",
    "covs <- fread(\"master_key_release3_final.csv\", header =T)\n",
    "covs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                  GP2ID                 IID manifest_id phenotype pheno_for_qc\n",
      "    1: SYNAPS-KZ_000001 SYNAPS-KZ_000001_s1          m1   Control            1\n",
      "    2: SYNAPS-KZ_000002 SYNAPS-KZ_000002_s1          m1   Control            1\n",
      "    3: SYNAPS-KZ_000003 SYNAPS-KZ_000003_s1          m1   Control            1\n",
      "    4: SYNAPS-KZ_000004 SYNAPS-KZ_000004_s1          m1   Control            1\n",
      "    5: SYNAPS-KZ_000005 SYNAPS-KZ_000005_s1          m1   Control            1\n",
      "   ---                                                                        \n",
      "16853:      PPMI_000789      PPMI_000789_s1          m2     Other           -9\n",
      "16854:      PPMI_000781      PPMI_000781_s1          m2     Other           -9\n",
      "16855:      PPMI_000818      PPMI_000818_s1          m2     Other           -9\n",
      "16856:      PPMI_000841      PPMI_000841_s1          m2     Other           -9\n",
      "16857:      PPMI_000808      PPMI_000808_s1          m2     Other           -9\n",
      "                                  other_pheno sex_for_qc age age_of_onset\n",
      "    1:                                Control          2  64           NA\n",
      "    2:                                Control          1  53           NA\n",
      "    3:                                Control          2  59           NA\n",
      "    4:                                Control          1  57           NA\n",
      "    5:                                Control          1  59           NA\n",
      "   ---                                                                   \n",
      "16853:   Genetically-enriched-cohort Affected          2  NA           NA\n",
      "16854: Genetidally-enriched-cohort Unaffected          1  57           NA\n",
      "16855:   Genetically-enriched-cohort Affected          1  60           NA\n",
      "16856:   Genetically-enriched-cohort Affected          2  68           NA\n",
      "16857:   Genetically-enriched-cohort Affected          1  NA           NA\n",
      "       age_at_diagnosis age_at_death race_for_qc family_history_for_qc\n",
      "    1:               NA           NA       Asian          Not Reported\n",
      "    2:               NA           NA       Asian          Not Reported\n",
      "    3:               NA           NA       Asian          Not Reported\n",
      "    4:               NA           NA       Asian          Not Reported\n",
      "    5:               NA           NA       Asian          Not Reported\n",
      "   ---                                                                \n",
      "16853:               NA           NA       White          Not Reported\n",
      "16854:               NA           NA       White          Not Reported\n",
      "16855:               55           55       White          Not Reported\n",
      "16856:               57           57       White          Not Reported\n",
      "16857:               NA           NA       White          Not Reported\n",
      "       region_for_qc     study pruned  pruned_reason label related\n",
      "    1:           KAZ SYNAPS-KZ      0                  CAS      NA\n",
      "    2:           KAZ SYNAPS-KZ      0                  CAS      NA\n",
      "    3:           KAZ SYNAPS-KZ      0                  CAS      NA\n",
      "    4:           KAZ SYNAPS-KZ      0                  CAS      NA\n",
      "    5:           KAZ SYNAPS-KZ      0                  CAS      NA\n",
      "   ---                                                            \n",
      "16853:           ISR      PPMI      0                   AJ      NA\n",
      "16854:           ISR      PPMI      0                   AJ      NA\n",
      "16855:           ISR      PPMI      1 callrate_prune            NA\n",
      "16856:           ISR      PPMI      0                   AJ       1\n",
      "16857:           ISR      PPMI      0                   AJ      NA\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "colnames(covs)[2]  <- \"IID\"  \n",
    "covs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                     IID              GP2ID manifest_id phenotype pheno_for_qc\n",
      "1: BLAACPD-UAB_000014_s1 BLAACPD-UAB_000014          m1        PD            2\n",
      "2:  BLAACPD-UC_000008_s1  BLAACPD-UC_000008          m1        PD            2\n",
      "3:     CORIELL_003677_s1     CORIELL_003677          m3        PD            2\n",
      "4:     CORIELL_004266_s1     CORIELL_004266          m3        PD            2\n",
      "5:     CORIELL_004346_s1     CORIELL_004346          m4        PD            2\n",
      "6:        PPMI_000087_s1        PPMI_000087          m1        PD            2\n",
      "   other_pheno sex_for_qc  age age_of_onset age_at_diagnosis age_at_death\n",
      "1:          PD          1   NA           NA               NA           NA\n",
      "2:          PD          1   NA           NA               NA           NA\n",
      "3:          PD          2 75.0           72               73           NA\n",
      "4:          PD          2 74.0           72               73           NA\n",
      "5:          PD          2 44.0           34               NA           NA\n",
      "6:          PD          2 42.8           NA               42           NA\n",
      "                 race_for_qc family_history_for_qc region_for_qc       study\n",
      "1: Black or African American          Not Reported           USA BLAACPD-UAB\n",
      "2: Black or African American          Not Reported           USA  BLAACPD-UC\n",
      "3: Black or African American                    No           USA     CORIELL\n",
      "4: Black or African American                    No           USA     CORIELL\n",
      "5:                     Other                    No           USA     CORIELL\n",
      "6:                   Unknown          Not Reported           USA        PPMI\n",
      "   pruned pruned_reason label related   V1 FID PAT MAT SEX PHENOTYPE\n",
      "1:      0            NA   AAC      NA   66   0   0   0   1         2\n",
      "2:      0            NA   AAC      NA   85   0   0   0   1         2\n",
      "3:      0            NA   AAC      NA  147   0   0   0   2         2\n",
      "4:      0            NA   AAC      NA  164   0   0   0   2         2\n",
      "5:      0            NA   AAC      NA  166   0   0   0   2         2\n",
      "6:      0            NA   AAC      NA 1196   0   0   0   2         2\n",
      "   chr9:129134882:A:C_C chr9:129147457:C:T_T\n",
      "1:                    0                    1\n",
      "2:                    0                    1\n",
      "3:                    0                    1\n",
      "4:                    0                    1\n",
      "5:                    0                    1\n",
      "6:                    0                    1\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "setwd('/home/jupyter/PP2A_GP2/')\n",
    "library(data.table)\n",
    "PP2A_het_casos_129147457_AAC <- fread(\"PP2A_het_casos_129147457_AAC.csv\", header =T)\n",
    "PP2A_het_casos_129147457_AAC\n",
    "temp <- merge(covs, PP2A_het_casos_129147457_AAC, by=\"IID\")\n",
    "head(temp)\n",
    "write.table(temp, file = \"PP2A_het_casos_129147457_AAC_extracted.txt\", quote = F, row.names = F, sep = \"\\t\")\n",
    "PP2A_het_casos_129147457_AAC_extracted <- fread(\"PP2A_het_casos_129147457_AAC_extracted.txt\", header =T)\n",
    "PP2A_het_casos_129147457_AAC_extracted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                      IID              GP2ID manifest_id phenotype pheno_for_qc\n",
      " 1:     BIOFIND_000173_s1     BIOFIND_000173          m1   Control            1\n",
      " 2: BLAACPD-UAB_000032_s1 BLAACPD-UAB_000032          m1   Control            1\n",
      " 3:  BLAACPD-UC_000021_s1  BLAACPD-UC_000021          m1   Control            1\n",
      " 4:     CORIELL_004672_s1     CORIELL_004672          m6   Control            1\n",
      " 5:     CORIELL_004696_s1     CORIELL_004696          m6   Control            1\n",
      " 6:     CORIELL_004736_s1     CORIELL_004736          m6   Control            1\n",
      " 7:     CORIELL_005187_s1     CORIELL_005187          m6   Control            1\n",
      " 8:     CORIELL_005874_s1     CORIELL_005874          m6   Control            1\n",
      " 9:     CORIELL_005948_s1     CORIELL_005948          m6   Control            1\n",
      "10:     CORIELL_006457_s1     CORIELL_006457          m6   Control            1\n",
      "11:     CORIELL_007000_s1     CORIELL_007000          m6   Control            1\n",
      "12:     CORIELL_007332_s1     CORIELL_007332          m6   Control            1\n",
      "13:     CORIELL_007450_s1     CORIELL_007450          m6   Control            1\n",
      "14:     CORIELL_007492_s1     CORIELL_007492          m6   Control            1\n",
      "15:     CORIELL_007529_s1     CORIELL_007529          m6   Control            1\n",
      "16:     CORIELL_007562_s1     CORIELL_007562          m6   Control            1\n",
      "17:     CORIELL_007619_s1     CORIELL_007619          m6   Control            1\n",
      "18:     CORIELL_007635_s1     CORIELL_007635          m6   Control            1\n",
      "19:     CORIELL_007641_s1     CORIELL_007641          m6   Control            1\n",
      "20:     CORIELL_007722_s1     CORIELL_007722          m6   Control            1\n",
      "21:     CORIELL_007733_s1     CORIELL_007733          m6   Control            1\n",
      "22:     CORIELL_007736_s1     CORIELL_007736          m6   Control            1\n",
      "23:     CORIELL_007802_s1     CORIELL_007802          m6   Control            1\n",
      "24:     CORIELL_007917_s1     CORIELL_007917          m6   Control            1\n",
      "25:     CORIELL_007956_s1     CORIELL_007956          m6   Control            1\n",
      "26:     CORIELL_008004_s1     CORIELL_008004          m6   Control            1\n",
      "27:     CORIELL_008023_s1     CORIELL_008023          m6   Control            1\n",
      "28:     CORIELL_008157_s1     CORIELL_008157          m6   Control            1\n",
      "29:     CORIELL_008216_s1     CORIELL_008216          m6   Control            1\n",
      "30:     CORIELL_008249_s1     CORIELL_008249          m6   Control            1\n",
      "31:     CORIELL_008287_s1     CORIELL_008287          m6   Control            1\n",
      "32:     CORIELL_008318_s1     CORIELL_008318          m6   Control            1\n",
      "33:     CORIELL_008345_s1     CORIELL_008345          m6   Control            1\n",
      "34:     CORIELL_008349_s1     CORIELL_008349          m6   Control            1\n",
      "35:     CORIELL_008359_s1     CORIELL_008359          m6   Control            1\n",
      "36:     CORIELL_008445_s1     CORIELL_008445          m6   Control            1\n",
      "37:     CORIELL_008484_s1     CORIELL_008484          m6   Control            1\n",
      "38:     CORIELL_008501_s1     CORIELL_008501          m6   Control            1\n",
      "39:     CORIELL_008575_s1     CORIELL_008575          m6   Control            1\n",
      "                      IID              GP2ID manifest_id phenotype pheno_for_qc\n",
      "    other_pheno sex_for_qc age age_of_onset age_at_diagnosis age_at_death\n",
      " 1:     Control          2  57           NA               NA           NA\n",
      " 2:     Control          2  NA           NA               NA           NA\n",
      " 3:     Control          2  NA           NA               NA           NA\n",
      " 4:     Control          2  71           NA               NA           NA\n",
      " 5:     Control          1  74           NA               NA           NA\n",
      " 6:     Control          1  64           NA               NA           NA\n",
      " 7:     Control          2  68           NA               NA           NA\n",
      " 8:     Control          1  69           NA               NA           NA\n",
      " 9:     Control          1  59           NA               NA           NA\n",
      "10:     Control          2  58           NA               NA           NA\n",
      "11:     Control          2  56           NA               NA           NA\n",
      "12:     Control          1  74           NA               NA           NA\n",
      "13:     Control          1  61           NA               NA           NA\n",
      "14:     Control          2  65           NA               NA           NA\n",
      "15:     Control          1  72           NA               NA           NA\n",
      "16:     Control          1  55           NA               NA           NA\n",
      "17:     Control          1  61           NA               NA           NA\n",
      "18:     Control          1  71           NA               NA           NA\n",
      "19:     Control          2  66           NA               NA           NA\n",
      "20:     Control          2  65           NA               NA           NA\n",
      "21:     Control          1  60           NA               NA           NA\n",
      "22:     Control          2  59           NA               NA           NA\n",
      "23:     Control          1  89           NA               NA           NA\n",
      "24:     Control          2  75           NA               NA           NA\n",
      "25:     Control          2  67           NA               NA           NA\n",
      "26:     Control          1  66           NA               NA           NA\n",
      "27:     Control          2  73           NA               NA           NA\n",
      "28:     Control          2  74           NA               NA           NA\n",
      "29:     Control          2  65           NA               NA           NA\n",
      "30:     Control          1  80           NA               NA           NA\n",
      "31:     Control          1  74           NA               NA           NA\n",
      "32:     Control          2  63           NA               NA           NA\n",
      "33:     Control          1  57           NA               NA           NA\n",
      "34:     Control          2  80           NA               NA           NA\n",
      "35:     Control          2  70           NA               NA           NA\n",
      "36:     Control          2  55           NA               NA           NA\n",
      "37:     Control          1  64           NA               NA           NA\n",
      "38:     Control          2  78           NA               NA           NA\n",
      "39:     Control          2  71           NA               NA           NA\n",
      "    other_pheno sex_for_qc age age_of_onset age_at_diagnosis age_at_death\n",
      "                  race_for_qc family_history_for_qc region_for_qc       study\n",
      " 1: Black or African American                    No           USA     BIOFIND\n",
      " 2: Black or African American          Not Reported           USA BLAACPD-UAB\n",
      " 3: Black or African American          Not Reported           USA  BLAACPD-UC\n",
      " 4: Black or African American          Not Reported           USA     CORIELL\n",
      " 5: Black or African American          Not Reported           USA     CORIELL\n",
      " 6: Black or African American          Not Reported           USA     CORIELL\n",
      " 7: Black or African American                    No           USA     CORIELL\n",
      " 8: Black or African American                    No           USA     CORIELL\n",
      " 9: Black or African American                    No           USA     CORIELL\n",
      "10: Black or African American                    No           USA     CORIELL\n",
      "11: Black or African American                    No           USA     CORIELL\n",
      "12: Black or African American          Not Reported           USA     CORIELL\n",
      "13: Black or African American          Not Reported           USA     CORIELL\n",
      "14: Black or African American          Not Reported           USA     CORIELL\n",
      "15: Black or African American          Not Reported           USA     CORIELL\n",
      "16: Black or African American          Not Reported           USA     CORIELL\n",
      "17: Black or African American          Not Reported           USA     CORIELL\n",
      "18: Black or African American          Not Reported           USA     CORIELL\n",
      "19: Black or African American          Not Reported           USA     CORIELL\n",
      "20: Black or African American          Not Reported           USA     CORIELL\n",
      "21: Black or African American          Not Reported           USA     CORIELL\n",
      "22: Black or African American          Not Reported           USA     CORIELL\n",
      "23: Black or African American          Not Reported           USA     CORIELL\n",
      "24: Black or African American          Not Reported           USA     CORIELL\n",
      "25: Black or African American          Not Reported           USA     CORIELL\n",
      "26: Black or African American          Not Reported           USA     CORIELL\n",
      "27: Black or African American          Not Reported           USA     CORIELL\n",
      "28: Black or African American                    No           USA     CORIELL\n",
      "29: Black or African American                    No           USA     CORIELL\n",
      "30: Black or African American                    No           USA     CORIELL\n",
      "31: Black or African American                    No           USA     CORIELL\n",
      "32: Black or African American                    No           USA     CORIELL\n",
      "33: Black or African American                    No           USA     CORIELL\n",
      "34: Black or African American                    No           USA     CORIELL\n",
      "35: Black or African American                    No           USA     CORIELL\n",
      "36: Black or African American                    No           USA     CORIELL\n",
      "37: Black or African American                    No           USA     CORIELL\n",
      "38: Black or African American                    No           USA     CORIELL\n",
      "39: Black or African American                    No           USA     CORIELL\n",
      "                  race_for_qc family_history_for_qc region_for_qc       study\n",
      "    pruned pruned_reason label related   V1 FID PAT MAT SEX PHENOTYPE\n",
      " 1:      0            NA   AAC      NA   22   0   0   0   2         1\n",
      " 2:      0            NA   AAC      NA   70   0   0   0   2         1\n",
      " 3:      0            NA   AAC      NA   95   0   0   0   2         1\n",
      " 4:      0            NA   AAC      NA  171   0   0   0   2         1\n",
      " 5:      0            NA   AAC       1  175   0   0   0   1         1\n",
      " 6:      0            NA   AAC      NA  181   0   0   0   1         1\n",
      " 7:      0            NA   AAC      NA  193   0   0   0   2         1\n",
      " 8:      0            NA   AAC      NA  204   0   0   0   1         1\n",
      " 9:      0            NA   AAC      NA  205   0   0   0   1         1\n",
      "10:      0            NA   AAC      NA  222   0   0   0   2         1\n",
      "11:      0            NA   AAC      NA  280   0   0   0   2         1\n",
      "12:      0            NA   AAC      NA  344   0   0   0   1         1\n",
      "13:      0            NA   AAC      NA  405   0   0   0   1         1\n",
      "14:      0            NA   AAC      NA  433   0   0   0   2         1\n",
      "15:      0            NA   AAC      NA  458   0   0   0   1         1\n",
      "16:      0            NA   AAC      NA  481   0   0   0   1         1\n",
      "17:      0            NA   AAC      NA  520   0   0   0   1         1\n",
      "18:      0            NA   AAC      NA  534   0   0   0   1         1\n",
      "19:      0            NA   AAC      NA  538   0   0   0   2         1\n",
      "20:      0            NA   AAC      NA  599   0   0   0   2         1\n",
      "21:      0            NA   AAC      NA  606   0   0   0   1         1\n",
      "22:      0            NA   AAC      NA  609   0   0   0   2         1\n",
      "23:      0            NA   AAC      NA  631   0   0   0   1         1\n",
      "24:      0            NA   AAC      NA  708   0   0   0   2         1\n",
      "25:      0            NA   AAC      NA  729   0   0   0   2         1\n",
      "26:      0            NA   AAC      NA  763   0   0   0   1         1\n",
      "27:      0            NA   AAC      NA  778   0   0   0   2         1\n",
      "28:      0            NA   AAC      NA  869   0   0   0   2         1\n",
      "29:      0            NA   AAC      NA  914   0   0   0   2         1\n",
      "30:      0            NA   AAC      NA  936   0   0   0   1         1\n",
      "31:      0            NA   AAC      NA  965   0   0   0   1         1\n",
      "32:      0            NA   AAC      NA  984   0   0   0   2         1\n",
      "33:      0            NA   AAC      NA 1006   0   0   0   1         1\n",
      "34:      0            NA   AAC      NA 1009   0   0   0   2         1\n",
      "35:      0            NA   AAC      NA 1018   0   0   0   2         1\n",
      "36:      0            NA   AAC      NA 1073   0   0   0   2         1\n",
      "37:      0            NA   AAC      NA 1102   0   0   0   1         1\n",
      "38:      0            NA   AAC      NA 1113   0   0   0   2         1\n",
      "39:      0            NA   AAC      NA 1165   0   0   0   2         1\n",
      "    pruned pruned_reason label related   V1 FID PAT MAT SEX PHENOTYPE\n",
      "    chr9:129134882:A:C_C chr9:129147457:C:T_T\n",
      " 1:                    0                    1\n",
      " 2:                    0                    1\n",
      " 3:                    0                    1\n",
      " 4:                    0                    1\n",
      " 5:                    0                    1\n",
      " 6:                    0                    1\n",
      " 7:                    0                    1\n",
      " 8:                    0                    1\n",
      " 9:                    0                    1\n",
      "10:                    0                    1\n",
      "11:                    0                    1\n",
      "12:                    0                    1\n",
      "13:                    0                    1\n",
      "14:                    0                    1\n",
      "15:                    0                    1\n",
      "16:                    0                    1\n",
      "17:                    0                    1\n",
      "18:                    0                    1\n",
      "19:                    0                    1\n",
      "20:                    0                    1\n",
      "21:                    0                    1\n",
      "22:                    0                    1\n",
      "23:                    0                    1\n",
      "24:                    0                    1\n",
      "25:                    0                    1\n",
      "26:                    0                    1\n",
      "27:                    0                    1\n",
      "28:                    0                    1\n",
      "29:                    0                    1\n",
      "30:                    0                    1\n",
      "31:                    0                    1\n",
      "32:                    0                    1\n",
      "33:                    0                    1\n",
      "34:                    0                    1\n",
      "35:                    0                    1\n",
      "36:                    0                    1\n",
      "37:                    0                    1\n",
      "38:                    0                    1\n",
      "39:                    0                    1\n",
      "    chr9:129134882:A:C_C chr9:129147457:C:T_T\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "setwd('/home/jupyter/PP2A_GP2/')\n",
    "library(data.table)\n",
    "PP2A_het_controles_129147457_AAC <- fread(\"PP2A_het_controles_129147457_AAC.csv\", header =T)\n",
    "PP2A_het_controles_129147457_AAC\n",
    "temp <- merge(covs, PP2A_het_controles_129147457_AAC, by=\"IID\")\n",
    "head(temp)\n",
    "write.table(temp, file = \"PP2A_het_controles_129147457_AAC_extracted.txt\", quote = F, row.names = F, sep = \"\\t\")\n",
    "PP2A_het_controles_129147457_AAC_extracted <- fread(\"PP2A_het_controles_129147457_AAC_extracted.txt\", header =T)\n",
    "PP2A_het_controles_129147457_AAC_extracted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 158,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129134882:A:C_C</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>0</td>\n",
       "      <td>BLAACPD-KPM_000016_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>213</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006341_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>418</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007472_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007591_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>675</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007875_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>678</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007878_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>774</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008019_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>819</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008083_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>876</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008166_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>890</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008184_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>911</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008211_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>999</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008335_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1059</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008415_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1133</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008531_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1164</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008573_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      FID                    IID  PAT  MAT  SEX  PHENOTYPE  \\\n",
       "30      0  BLAACPD-KPM_000016_s1    0    0    2          1   \n",
       "213     0      CORIELL_006341_s1    0    0    1          1   \n",
       "418     0      CORIELL_007472_s1    0    0    2          1   \n",
       "499     0      CORIELL_007591_s1    0    0    1          1   \n",
       "675     0      CORIELL_007875_s1    0    0    2          1   \n",
       "678     0      CORIELL_007878_s1    0    0    2          1   \n",
       "774     0      CORIELL_008019_s1    0    0    2          1   \n",
       "819     0      CORIELL_008083_s1    0    0    2          1   \n",
       "876     0      CORIELL_008166_s1    0    0    2          1   \n",
       "890     0      CORIELL_008184_s1    0    0    1          1   \n",
       "911     0      CORIELL_008211_s1    0    0    1          1   \n",
       "999     0      CORIELL_008335_s1    0    0    1          1   \n",
       "1059    0      CORIELL_008415_s1    0    0    1          1   \n",
       "1133    0      CORIELL_008531_s1    0    0    2          1   \n",
       "1164    0      CORIELL_008573_s1    0    0    1          1   \n",
       "\n",
       "      chr9:129134882:A:C_C  chr9:129147457:C:T_T  \n",
       "30                     1.0                   0.0  \n",
       "213                    1.0                   0.0  \n",
       "418                    1.0                   0.0  \n",
       "499                    1.0                   0.0  \n",
       "675                    1.0                   0.0  \n",
       "678                    1.0                   0.0  \n",
       "774                    1.0                   0.0  \n",
       "819                    1.0                   0.0  \n",
       "876                    1.0                   0.0  \n",
       "890                    1.0                   0.0  \n",
       "911                    1.0                   0.0  \n",
       "999                    1.0                   0.0  \n",
       "1059                   1.0                   0.0  \n",
       "1133                   1.0                   0.0  \n",
       "1164                   1.0                   0.0  "
      ]
     },
     "execution_count": 158,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Explore HETs for chr9:129134882:A:C_C in controls\n",
    "PP2A_het_controles_129134882_AAC = PP2A_AAC_recode[(PP2A_AAC_recode['chr9:129134882:A:C_C'] == 1) & (PP2A_AAC_recode['PHENOTYPE'] == 1)]\n",
    "PP2A_het_controles_129134882_AAC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "metadata": {},
   "outputs": [],
   "source": [
    "PP2A_het_controles_129134882_AAC.to_csv('PP2A_het_controles_129134882_AAC.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129134882:A:C_C</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>30</td>\n",
       "      <td>0</td>\n",
       "      <td>BLAACPD-KPM_000016_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>213</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006341_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>418</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007472_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>499</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007591_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>675</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007875_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>678</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007878_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>774</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008019_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>819</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008083_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>876</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008166_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>890</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008184_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>911</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008211_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>999</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008335_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>1059</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008415_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>1133</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008531_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>1164</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_008573_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Unnamed: 0  FID                    IID  PAT  MAT  SEX  PHENOTYPE  \\\n",
       "0           30    0  BLAACPD-KPM_000016_s1    0    0    2          1   \n",
       "1          213    0      CORIELL_006341_s1    0    0    1          1   \n",
       "2          418    0      CORIELL_007472_s1    0    0    2          1   \n",
       "3          499    0      CORIELL_007591_s1    0    0    1          1   \n",
       "4          675    0      CORIELL_007875_s1    0    0    2          1   \n",
       "5          678    0      CORIELL_007878_s1    0    0    2          1   \n",
       "6          774    0      CORIELL_008019_s1    0    0    2          1   \n",
       "7          819    0      CORIELL_008083_s1    0    0    2          1   \n",
       "8          876    0      CORIELL_008166_s1    0    0    2          1   \n",
       "9          890    0      CORIELL_008184_s1    0    0    1          1   \n",
       "10         911    0      CORIELL_008211_s1    0    0    1          1   \n",
       "11         999    0      CORIELL_008335_s1    0    0    1          1   \n",
       "12        1059    0      CORIELL_008415_s1    0    0    1          1   \n",
       "13        1133    0      CORIELL_008531_s1    0    0    2          1   \n",
       "14        1164    0      CORIELL_008573_s1    0    0    1          1   \n",
       "\n",
       "    chr9:129134882:A:C_C  chr9:129147457:C:T_T  \n",
       "0                    1.0                   0.0  \n",
       "1                    1.0                   0.0  \n",
       "2                    1.0                   0.0  \n",
       "3                    1.0                   0.0  \n",
       "4                    1.0                   0.0  \n",
       "5                    1.0                   0.0  \n",
       "6                    1.0                   0.0  \n",
       "7                    1.0                   0.0  \n",
       "8                    1.0                   0.0  \n",
       "9                    1.0                   0.0  \n",
       "10                   1.0                   0.0  \n",
       "11                   1.0                   0.0  \n",
       "12                   1.0                   0.0  \n",
       "13                   1.0                   0.0  \n",
       "14                   1.0                   0.0  "
      ]
     },
     "execution_count": 160,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "temp = pd.read_csv('PP2A_het_controles_129134882_AAC.csv')\n",
    "temp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                      IID              GP2ID manifest_id phenotype pheno_for_qc\n",
      " 1: BLAACPD-KPM_000016_s1 BLAACPD-KPM_000016          m1   Control            1\n",
      " 2:     CORIELL_006341_s1     CORIELL_006341          m6   Control            1\n",
      " 3:     CORIELL_007472_s1     CORIELL_007472          m6   Control            1\n",
      " 4:     CORIELL_007591_s1     CORIELL_007591          m6   Control            1\n",
      " 5:     CORIELL_007875_s1     CORIELL_007875          m6   Control            1\n",
      " 6:     CORIELL_007878_s1     CORIELL_007878          m6   Control            1\n",
      " 7:     CORIELL_008019_s1     CORIELL_008019          m6   Control            1\n",
      " 8:     CORIELL_008083_s1     CORIELL_008083          m6   Control            1\n",
      " 9:     CORIELL_008166_s1     CORIELL_008166          m6   Control            1\n",
      "10:     CORIELL_008184_s1     CORIELL_008184          m6   Control            1\n",
      "11:     CORIELL_008211_s1     CORIELL_008211          m6   Control            1\n",
      "12:     CORIELL_008335_s1     CORIELL_008335          m6   Control            1\n",
      "13:     CORIELL_008415_s1     CORIELL_008415          m6   Control            1\n",
      "14:     CORIELL_008531_s1     CORIELL_008531          m6   Control            1\n",
      "15:     CORIELL_008573_s1     CORIELL_008573          m6   Control            1\n",
      "    other_pheno sex_for_qc age age_of_onset age_at_diagnosis age_at_death\n",
      " 1:     Control          2  NA           NA               NA           NA\n",
      " 2:     Control          1  69           NA               NA           NA\n",
      " 3:     Control          2  67           NA               NA           NA\n",
      " 4:     Control          1  65           NA               NA           NA\n",
      " 5:     Control          2  57           NA               NA           NA\n",
      " 6:     Control          2  66           NA               NA           NA\n",
      " 7:     Control          2  65           NA               NA           NA\n",
      " 8:     Control          2  74           NA               NA           NA\n",
      " 9:     Control          2  58           NA               NA           NA\n",
      "10:     Control          1  79           NA               NA           NA\n",
      "11:     Control          1  67           NA               NA           NA\n",
      "12:     Control          1  56           NA               NA           NA\n",
      "13:     Control          1  57           NA               NA           NA\n",
      "14:     Control          2  90           NA               NA           NA\n",
      "15:     Control          1  74           NA               NA           NA\n",
      "                  race_for_qc family_history_for_qc region_for_qc       study\n",
      " 1: Black or African American          Not Reported           USA BLAACPD-KPM\n",
      " 2: Black or African American                    No           USA     CORIELL\n",
      " 3: Black or African American          Not Reported           USA     CORIELL\n",
      " 4: Black or African American          Not Reported           USA     CORIELL\n",
      " 5: Black or African American          Not Reported           USA     CORIELL\n",
      " 6: Black or African American          Not Reported           USA     CORIELL\n",
      " 7: Black or African American          Not Reported           USA     CORIELL\n",
      " 8: Black or African American          Not Reported           USA     CORIELL\n",
      " 9: Black or African American                    No           USA     CORIELL\n",
      "10: Black or African American                    No           USA     CORIELL\n",
      "11: Black or African American                    No           USA     CORIELL\n",
      "12: Black or African American                    No           USA     CORIELL\n",
      "13: Black or African American                    No           USA     CORIELL\n",
      "14: Black or African American                    No           USA     CORIELL\n",
      "15: Black or African American                    No           USA     CORIELL\n",
      "    pruned pruned_reason label related   V1 FID PAT MAT SEX PHENOTYPE\n",
      " 1:      0            NA   AAC      NA   30   0   0   0   2         1\n",
      " 2:      0            NA   AAC      NA  213   0   0   0   1         1\n",
      " 3:      0            NA   AAC      NA  418   0   0   0   2         1\n",
      " 4:      0            NA   AAC      NA  499   0   0   0   1         1\n",
      " 5:      0            NA   AAC      NA  675   0   0   0   2         1\n",
      " 6:      0            NA   AAC      NA  678   0   0   0   2         1\n",
      " 7:      0            NA   AAC      NA  774   0   0   0   2         1\n",
      " 8:      0            NA   AAC      NA  819   0   0   0   2         1\n",
      " 9:      0            NA   AAC      NA  876   0   0   0   2         1\n",
      "10:      0            NA   AAC      NA  890   0   0   0   1         1\n",
      "11:      0            NA   AAC      NA  911   0   0   0   1         1\n",
      "12:      0            NA   AAC      NA  999   0   0   0   1         1\n",
      "13:      0            NA   AAC      NA 1059   0   0   0   1         1\n",
      "14:      0            NA   AAC      NA 1133   0   0   0   2         1\n",
      "15:      0            NA   AAC      NA 1164   0   0   0   1         1\n",
      "    chr9:129134882:A:C_C chr9:129147457:C:T_T\n",
      " 1:                    1                    0\n",
      " 2:                    1                    0\n",
      " 3:                    1                    0\n",
      " 4:                    1                    0\n",
      " 5:                    1                    0\n",
      " 6:                    1                    0\n",
      " 7:                    1                    0\n",
      " 8:                    1                    0\n",
      " 9:                    1                    0\n",
      "10:                    1                    0\n",
      "11:                    1                    0\n",
      "12:                    1                    0\n",
      "13:                    1                    0\n",
      "14:                    1                    0\n",
      "15:                    1                    0\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "setwd('/home/jupyter/PP2A_GP2/')\n",
    "library(data.table)\n",
    "PP2A_het_controles_129134882_AAC <- fread(\"PP2A_het_controles_129134882_AAC.csv\", header =T)\n",
    "PP2A_het_controles_129134882_AAC\n",
    "temp <- merge(covs, PP2A_het_controles_129134882_AAC, by=\"IID\")\n",
    "head(temp)\n",
    "write.table(temp, file = \"PP2A_het_controles_129134882_AAC_extracted.txt\", quote = F, row.names = F, sep = \"\\t\")\n",
    "PP2A_het_controles_129134882_AAC_extracted <- fread(\"PP2A_het_controles_129134882_AAC_extracted.txt\", header =T)\n",
    "PP2A_het_controles_129134882_AAC_extracted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Save out results..!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -mu terra-02e7cd23 cp -r /home/jupyter/PP2A_GP2/ gs://fc-a2382a6f-59cf-4cf3-b70e-8cd83806d5e0\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/reduced_coding_nonsynonymous.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_controles_129147457_EUR_extracted.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AJ_rare_0.05.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/reduced_coding_nonsynonymous_AFR.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AAC.raw [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_CAS.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_EAS.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_SAS.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AAC.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AFR.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_casos_129129094_AJ_extracted.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AJ.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AAC_rare_0.05.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/reduced_coding_nonsynonymous_AMR.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AMR.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_MDE.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_MDE.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_rare_0.05.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AAC_release3.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_controles_129147457_AJ_extracted.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_hom_casos_129147457_AJ.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AJ_rare_0.05.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_ACC.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_CAS.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_casos_129129094_AJ.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_CAS_release3.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_SAS_rare_0.05.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AFR.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AFR.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_CAS.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_SAS.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AJ.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_CAS.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_MDE.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_EUR.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AJ_rare_0.05.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_EAS.assoc.adjusted [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AJ_release3.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AMR_rare_0.05.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AJ_release3.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_MDE.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_CAS_release3.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AMR.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_casos_129147457_AJ_extracted.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AAC_release3.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AJ.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AAC.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AFR_rare_0.05.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/reduced_coding_nonsynonymous_AJ.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AFR.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_hom_casos_129147457_AJ_extracted.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AJ.assoc [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AFR.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AFR.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_EAS.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_EAS_release3.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AJ.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_129147457_casos_EUR.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AFR_release3.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_MDE.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AMR_release3.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_SAS.raw [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_EUR_release3.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_MDE_release3.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/reduced_coding_nonsynonymous_CAS.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AMR.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_casos_129147457_AFR_extracted.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_casos_129147457_AAC_extracted.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_SAS.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AMR_rare_0.05.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_EUR.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AMR_release3.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AJ.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AAC.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AFR_rare_0.05.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AFR.raw [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AFR_rare_0.05.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AJ_release3.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/master_key_release3_final.csv [Content-Type=text/csv]...\n",
      "==> NOTE: You are uploading one or more large file(s), which would run          \n",
      "significantly faster if you enable parallel composite uploads. This\n",
      "feature can be enabled by editing the\n",
      "\"parallel_composite_upload_threshold\" value in your .boto\n",
      "configuration file. However, note that if you do this large files will\n",
      "be uploaded as `composite objects\n",
      "<https://cloud.google.com/storage/docs/composite-objects>`_,which\n",
      "means that any user who downloads such objects will need to have a\n",
      "compiled crcmod installed (see \"gsutil help crcmod\"). This is because\n",
      "without a compiled crcmod, computing checksums on composite objects is\n",
      "so slow that gsutil disables downloads of composite objects.\n",
      "\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_CAS_rare_0.05.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_CAS.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_MDE.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_MDE_release3.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_controles_129147457_AFR.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_EAS.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AFR.assoc [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AAC.assoc [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_CAS.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_CAS.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AFR_release3.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_MDE.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AMR.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/reduced_coding_nonsynonymous_EAS.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AMR.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_SAS.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AAC.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_EAS_release3.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_SAS_release3.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AFR_rare_0.05.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_SAS.assoc.adjusted [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AAC.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AAC.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_EUR.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AMR.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AAC_rare_0.05.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_CAS.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AAC_rare_0.05.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AAC_rare_0.05.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_CAS.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AMR.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_controles_129147457_AFR_extracted.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_CAS.raw [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_EAS.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_casos_129147457_AJ.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AFR_release3.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_rare_0.05.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AMR.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_SAS.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PODXL.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_controles_129134882_AFR.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_EAS.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_SAS_release3.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_rare_0.05.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_EAS.raw [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AJ.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AMR.assoc.adjusted [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AJ_release3.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/SEX.txt [Content-Type=text/plain]...      \n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AMR.raw [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_CAS_release3.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_CAS_release3.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_SAS.assoc [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_MDE.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_EUR_release3.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AFR.assoc.adjusted [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AMR_rare_0.05.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_EAS_release3.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_controles_129147457_AAC.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_CAS.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AFR.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_EAS.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AJ.assoc.adjusted [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AFR.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_EAS.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AJ.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AJ.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AMR.assoc [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AMR.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_129147457_controles_EUR.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_EAS.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_SAS_release3.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/reduced_coding_nonsynonymous_MDE.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_CAS.assoc [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PHENO.txt [Content-Type=text/plain]...    \n",
      "Copying file:///home/jupyter/PP2A_GP2/PTPA_het_129147457_controles_extracted.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_MDE.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AMR.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AAC.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_casos_129147457_AAC.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AAC_release3.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_EAS.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A.assoc [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AAC.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_EAS_release3.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_CAS.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AFR.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_EAS.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_controles_129134882_AAC.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AJ.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AAC.assoc.adjusted [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_MDE_rare_0.05.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AFR.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_MDE.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AJ.raw [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_SAS.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_controles_129129094_AJ.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_MDE.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_EUR.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_EUR.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_casos_129147457_EUR.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AJ.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/GP2_AJ_comphet.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_SAS.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AMR.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_MDE_release3.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/GP2_AMR_comphet.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_controles_129147457_EUR.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_SAS.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_SAS.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_SAS.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_EUR_release3.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_controles_129147457_AJ.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AFR_release3.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AAC.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AFR.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_MDE_release3.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AMR_release3.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AMR_release3.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A.raw [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_controles_129134882_AFR_extracted.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_SAS_release3.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A.assoc.adjusted [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_CAS.assoc.adjusted [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AJ_rare_0.05.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AAC.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PODXL.assoc [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_EAS.assoc [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_controles_129147457_AAC_extracted.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_EAS_rare_0.05.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_EUR_release3.pgen [Content-Type=application/octet-stream]...\n",
      "Resuming upload for file:///home/jupyter/PP2A_GP2/chr9_EUR_release3.pgen        \n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_controles_129134882_AAC_extracted.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AAC.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_SAS.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AJ.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/reduced_coding_nonsynonymous_SAS.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_EAS.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_casos_129147457_EUR_extracted.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AMR_rare_0.05.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_rare_0.05.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AAC_release3.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/reduced_coding_nonsynonymous_AAC.txt [Content-Type=text/plain]...\n",
      "\\ [233/233 files][  6.1 GiB/  6.1 GiB] 100% Done     0.0 B/s                    \n",
      "Operation completed over 233 objects/6.1 GiB.                                    \n"
     ]
    }
   ],
   "source": [
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR} {WORKSPACE_BUCKET}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.12"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {
    "height": "248.972px",
    "width": "263.062px"
   },
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "268.36px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
