{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# PP2A - Single gene analysis in GP2 data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Description\n",
    "\n",
    "Using \"individual level data\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 0. Getting Started\n",
    "\n",
    "- Loading Python libraries\n",
    "- Defining functions\n",
    "- Installing packages\n",
    "\n",
    "### 1. Copy data from workspace to cloud environment\n",
    "\n",
    "### 2. Extract PP2A\n",
    "\n",
    "### 3. Annotate PP2A variants\n",
    "\n",
    "### 4. Extract coding/non-syn variants\n",
    "\n",
    "### 5. Calculate frequency in cases versus controls\n",
    "\n",
    "### 6. Calculate frequency (homozygotes) in cases versus controls\n",
    "\n",
    "### 7. Save out results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Getting Started"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Loading Python libraries"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Defining functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "code_folding": []
   },
   "outputs": [],
   "source": [
    "# Use the os package to interact with the environment\n",
    "import os\n",
    "\n",
    "# Bring in Pandas for DatAMRame functionality\n",
    "import pandas as pd\n",
    "\n",
    "# Numpy for basics\n",
    "import numpy as np\n",
    "\n",
    "# Use StringIO for working with file contents\n",
    "from io import StringIO\n",
    "\n",
    "# Enable IPython to display matplotlib graphs\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline\n",
    "\n",
    "# Enable interaction with the FireCloud API\n",
    "from firecloud import api as fapi\n",
    "\n",
    "# Import the iPython HTML rendering for displaying links to Google Cloud Console\n",
    "from IPython.core.display import display, HTML\n",
    "\n",
    "# Import urllib modules for building URLs to Google Cloud Console\n",
    "import urllib.parse\n",
    "\n",
    "# BigQuery for querying data\n",
    "from google.cloud import bigquery\n",
    "\n",
    "#Import Sys\n",
    "import sys as sys"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "code_folding": [
     0
    ]
   },
   "outputs": [],
   "source": [
    "# Utility routine for printing a shell command before executing it\n",
    "def shell_do(command):\n",
    "    print(f'Executing: {command}', file=sys.stderr)\n",
    "    !$command\n",
    "    \n",
    "def shell_return(command):\n",
    "    print(f'Executing: {command}', file=sys.stderr)\n",
    "    output = !$command\n",
    "    return '\\n'.join(output)\n",
    "\n",
    "# Utility routine for printing a query before executing it\n",
    "def bq_query(query):\n",
    "    print(f'Executing: {query}', file=sys.stderr)\n",
    "    return pd.read_gbq(query, project_id=BILLING_PROJECT_ID, dialect='standard')\n",
    "\n",
    "# Utility routine for display a message and a link\n",
    "def display_html_link(description, link_text, url):\n",
    "    html = f'''\n",
    "    <p>\n",
    "    </p>\n",
    "    <p>\n",
    "    {description}\n",
    "    <a target=_blank href=\"{url}\">{link_text}</a>.\n",
    "    </p>\n",
    "    '''\n",
    "\n",
    "    display(HTML(html))\n",
    "\n",
    "# Utility routines for reading files from Google Cloud Storage\n",
    "def gcs_read_file(path):\n",
    "    \"\"\"Return the contents of a file in GCS\"\"\"\n",
    "    contents = !gsutil -u {BILLING_PROJECT_ID} cat {path}\n",
    "    return '\\n'.join(contents)\n",
    "    \n",
    "def gcs_read_csv(path, sep=None):\n",
    "    \"\"\"Return a DatAMRame from the contents of a delimited file in GCS\"\"\"\n",
    "    return pd.read_csv(StringIO(gcs_read_file(path)), sep=sep, engine='python')\n",
    "\n",
    "# Utility routine for displaying a message and link to Cloud Console\n",
    "def link_to_cloud_console_gcs(description, link_text, gcs_path):\n",
    "    url = '{}?{}'.format(\n",
    "        os.path.join('https://console.cloud.google.com/storage/browser',\n",
    "                     gcs_path.replace(\"gs://\",\"\")),\n",
    "        urllib.parse.urlencode({'userProject': BILLING_PROJECT_ID}))\n",
    "\n",
    "    display_html_link(description, link_text, url)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Set paths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "code_folding": [
     0
    ]
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Billing and Workspace\n",
      "Workspace Name: PP2A\n",
      "Billing Project: terra-02e7cd23\n",
      "Workspace Bucket, where you can upload and download data: gs://fc-a2382a6f-59cf-4cf3-b70e-8cd83806d5e0\n",
      "\n",
      "GP2 v3.0\n",
      "Path to GP2 v3.0 Clinical Data: gs://gp2tier2/release3_31102022/clinical_data\n",
      "Path to GP2 v3.0 Raw Genotype Data: gs://gp2tier2/release3_31102022/raw_genotypes\n",
      "Path to GP2 v3.0 Imputed Genotype Data: gs://gp2tier2/release3_31102022/imputed_genotypes\n",
      "AMP-PD v3.0\n",
      "Path to AMP-PD v3.0 Clinical Data: gs://amp-pd-data/releases/2022_v3release_1115/clinical\n",
      "Path to AMP-PD v3.0 WGS Data: gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink\n",
      "Path to AMP-PD v3.0 WGS Data: gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Set up billing project and data path variables\n",
    "BILLING_PROJECT_ID = os.environ['GOOGLE_PROJECT']\n",
    "WORKSPACE_NAMESPACE = os.environ['WORKSPACE_NAMESPACE']\n",
    "WORKSPACE_NAME = os.environ['WORKSPACE_NAME']\n",
    "WORKSPACE_BUCKET = os.environ['WORKSPACE_BUCKET']\n",
    "\n",
    "WORKSPACE_ATTRIBUTES = fapi.get_workspace(WORKSPACE_NAMESPACE, WORKSPACE_NAME).json().get('workspace',{}).get('attributes',{})\n",
    "\n",
    "## Print the information to check we are in the proper release and billing \n",
    "## This will be different for you, the user, depending on the billing project your workspace is on\n",
    "print('Billing and Workspace')\n",
    "print(f'Workspace Name: {WORKSPACE_NAME}')\n",
    "print(f'Billing Project: {BILLING_PROJECT_ID}')\n",
    "print(f'Workspace Bucket, where you can upload and download data: {WORKSPACE_BUCKET}')\n",
    "print('')\n",
    "\n",
    "\n",
    "## GP2 v3.0\n",
    "## Explicitly define release v3.0 path \n",
    "GP2_RELEASE_PATH = 'gs://gp2tier2/release3_31102022'\n",
    "GP2_CLINICAL_RELEASE_PATH = f'{GP2_RELEASE_PATH}/clinical_data'\n",
    "GP2_RAW_GENO_PATH = f'{GP2_RELEASE_PATH}/raw_genotypes'\n",
    "GP2_IMPUTED_GENO_PATH = f'{GP2_RELEASE_PATH}/imputed_genotypes'\n",
    "print('GP2 v3.0')\n",
    "print(f'Path to GP2 v3.0 Clinical Data: {GP2_CLINICAL_RELEASE_PATH}')\n",
    "print(f'Path to GP2 v3.0 Raw Genotype Data: {GP2_RAW_GENO_PATH}')\n",
    "print(f'Path to GP2 v3.0 Imputed Genotype Data: {GP2_IMPUTED_GENO_PATH}')\n",
    "\n",
    "\n",
    "## AMP-PD v3.0\n",
    "## Explicitly define release v3.0 path \n",
    "AMP_RELEASE_PATH = 'gs://amp-pd-data/releases/2022_v3release_1115'\n",
    "AMP_CLINICAL_RELEASE_PATH = f'{AMP_RELEASE_PATH}/clinical'\n",
    "\n",
    "AMP_WGS_RELEASE_PATH = 'gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS'\n",
    "AMP_WGS_RELEASE_PLINK_PATH = os.path.join(AMP_WGS_RELEASE_PATH, 'plink')\n",
    "AMP_WGS_RELEASE_PLINK_PFILES = os.path.join(AMP_WGS_RELEASE_PLINK_PATH, 'pfiles')\n",
    "\n",
    "print('AMP-PD v3.0')\n",
    "print(f'Path to AMP-PD v3.0 Clinical Data: {AMP_CLINICAL_RELEASE_PATH}')\n",
    "print(f'Path to AMP-PD v3.0 WGS Data: {AMP_WGS_RELEASE_PLINK_PATH}')\n",
    "print(f'Path to AMP-PD v3.0 WGS Data: {AMP_WGS_RELEASE_PLINK_PFILES}')\n",
    "print('')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Install packages"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Install Plink 1.9 and Plink 2.0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Plink1.9 is already installed in /home/jupyter/tools/\n",
      "Plink2 is already installed in /home/jupyter/tools/\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "mkdir -p ~/tools\n",
    "cd ~/tools\n",
    "\n",
    "if test -e /home/jupyter/tools/plink; then\n",
    "echo \"Plink1.9 is already installed in /home/jupyter/tools/\"\n",
    "\n",
    "else\n",
    "echo -e \"Downloading plink \\n    -------\"\n",
    "wget -N http://s3.amazonaws.com/plink1-assets/plink_linux_x86_64_20190304.zip \n",
    "unzip -o plink_linux_x86_64_20190304.zip\n",
    "echo -e \"\\n plink downloaded and unzipped in /home/jupyter/tools \\n \"\n",
    "\n",
    "fi\n",
    "\n",
    "\n",
    "if test -e /home/jupyter/tools/plink2; then\n",
    "echo \"Plink2 is already installed in /home/jupyter/tools/\"\n",
    "\n",
    "else\n",
    "echo -e \"Downloading plink2 \\n    -------\"\n",
    "wget -N https://s3.amazonaws.com/plink2-assets/alpha3/plink2_linux_avx2_20220603.zip\n",
    "unzip -o plink2_linux_avx2_20220603.zip\n",
    "echo -e \"\\n plink2 downloaded and unzipped in /home/jupyter/tools \\n \"\n",
    "\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annovar\n",
      "annovar.latest.tar.gz\n",
      "LICENSE\n",
      "plink\n",
      "plink2\n",
      "plink2_linux_avx2_20220603.zip\n",
      "plink_linux_x86_64_20190304.zip\n",
      "prettify\n",
      "toy.map\n",
      "toy.ped\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls /home/jupyter/tools/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Remote restrictions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "# chmod plink 1.9 \n",
    "chmod u+x /home/jupyter/tools/plink"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "# chmod plink 2.0\n",
    "chmod u+x /home/jupyter/tools/plink2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Install ANNOVAR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annovar is already installed in /home/jupyter/tools/\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Install ANNOVAR:\n",
    "# https://www.openbioinformatics.org/annovar/annovar_download_form.php\n",
    "\n",
    "if test -e /home/jupyter/tools/annovar; then\n",
    "\n",
    "echo \"annovar is already installed in /home/jupyter/tools/\"\n",
    "else\n",
    "echo \"annovar is not installed\"\n",
    "cd /home/jupyter/tools/\n",
    "\n",
    "wget http://www.openbioinformatics.org/annovar/download/0wgxR2rIVP/annovar.latest.tar.gz\n",
    "\n",
    "tar xvfz annovar.latest.tar.gz\n",
    "\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annovar\n",
      "annovar.latest.tar.gz\n",
      "LICENSE\n",
      "plink\n",
      "plink2\n",
      "plink2_linux_avx2_20220603.zip\n",
      "plink_linux_x86_64_20190304.zip\n",
      "prettify\n",
      "toy.map\n",
      "toy.ped\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls /home/jupyter/tools/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Install ANNOVAR: Download sources of annotation "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "NOTICE: Web-based checking to see whether ANNOVAR new version is available ... Done\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_refGene.txt.gz ... OK\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_refGeneMrna.fa.gz ... OK\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_refGeneVersion.txt.gz ... OK\n",
      "NOTICE: Uncompressing downloaded files\n",
      "NOTICE: Finished downloading annotation files for hg38 build version, with files saved at the 'humandb' directory\n",
      "NOTICE: Web-based checking to see whether ANNOVAR new version is available ... Done\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_clinvar_20140902.txt.gz ... OK\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_clinvar_20140902.txt.idx.gz ... OK\n",
      "NOTICE: Uncompressing downloaded files\n",
      "NOTICE: Finished downloading annotation files for hg38 build version, with files saved at the 'humandb' directory\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "cd /home/jupyter/tools/annovar/\n",
    "\n",
    "perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar refGene humandb/\n",
    "perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar clinvar_20140902 humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb cytoBand humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar ensGene humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar exac03 humandb/ \n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar avsnp147 humandb/ \n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar dbnsfp30a humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar gnomad211_genome humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar ljb26_all humandb/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annotate_variation.pl\n",
      "coding_change.pl\n",
      "convert2annovar.pl\n",
      "example\n",
      "humandb\n",
      "retrieve_seq_from_fasta.pl\n",
      "table_annovar.pl\n",
      "variants_reduction.pl\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls /home/jupyter/tools/annovar/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## AMRICAN"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Copy data from GP2 bucket to workspace"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Making a working directory\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: mkdir -p /home/jupyter/PP2A_GP2/\n"
     ]
    }
   ],
   "source": [
    "# Make a directory\n",
    "print(\"Making a working directory\")\n",
    "WORK_DIR = f'/home/jupyter/PP2A_GP2/'\n",
    "shell_do(f'mkdir -p {WORK_DIR}') # f' significa f-string - contiene expresiones para ejecutar el codigo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -u terra-02e7cd23 ls gs://gp2tier2/release3_31102022/imputed_genotypes\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gs://gp2tier2/release3_31102022/imputed_genotypes/AAC/\r\n",
      "gs://gp2tier2/release3_31102022/imputed_genotypes/AFR/\r\n",
      "gs://gp2tier2/release3_31102022/imputed_genotypes/AJ/\r\n",
      "gs://gp2tier2/release3_31102022/imputed_genotypes/AMR/\r\n",
      "gs://gp2tier2/release3_31102022/imputed_genotypes/CAS/\r\n",
      "gs://gp2tier2/release3_31102022/imputed_genotypes/EAS/\r\n",
      "gs://gp2tier2/release3_31102022/imputed_genotypes/EUR/\r\n",
      "gs://gp2tier2/release3_31102022/imputed_genotypes/MDE/\r\n",
      "gs://gp2tier2/release3_31102022/imputed_genotypes/SAS/\r\n"
     ]
    }
   ],
   "source": [
    "# Check directory where AMP-PD data is\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} ls {GP2_IMPUTED_GENO_PATH}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -u terra-02e7cd23 -m cp -r gs://gp2tier2/release3_31102022/imputed_genotypes/AMR/chr9* /home/jupyter/PP2A_GP2/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Copying gs://gp2tier2/release3_31102022/imputed_genotypes/AMR/chr9_AMR_release3.pgen...\n",
      "Copying gs://gp2tier2/release3_31102022/imputed_genotypes/AMR/chr9_AMR_release3.psam...\n",
      "Copying gs://gp2tier2/release3_31102022/imputed_genotypes/AMR/chr9_AMR_release3.log...\n",
      "Copying gs://gp2tier2/release3_31102022/imputed_genotypes/AMR/chr9_AMR_release3.pvar...\n",
      "- [4/4 files][163.2 MiB/163.2 MiB] 100% Done                                    \n",
      "Operation completed over 4 objects/163.2 MiB.                                    \n"
     ]
    }
   ],
   "source": [
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp -r {GP2_IMPUTED_GENO_PATH}/AMR/chr9* {WORK_DIR}')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Create a covariate file with GP2 data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -u terra-02e7cd23 ls gs://gp2tier2/release3_31102022/clinical_data\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gs://gp2tier2/release3_31102022/clinical_data/\r\n",
      "gs://gp2tier2/release3_31102022/clinical_data/master_key_release3_final.csv\r\n",
      "gs://gp2tier2/release3_31102022/clinical_data/release3_31102022_data_dictionary.csv\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -u terra-02e7cd23 -m cp -r gs://gp2tier2/release3_31102022/clinical_data/master_key_release3_final.csv /home/jupyter/PP2A_GP2/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Copying gs://gp2tier2/release3_31102022/clinical_data/master_key_release3_final.csv...\n",
      "/ [1/1 files][  1.6 MiB/  1.6 MiB] 100% Done                                    \n",
      "Operation completed over 1 objects/1.6 MiB.                                      \n"
     ]
    }
   ],
   "source": [
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} ls {GP2_CLINICAL_RELEASE_PATH}')\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp -r {GP2_CLINICAL_RELEASE_PATH}/master_key_release3_final.csv {WORK_DIR}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['GP2ID', 'GP2sampleID', 'manifest_id', 'phenotype', 'pheno_for_qc',\n",
       "       'other_pheno', 'sex_for_qc', 'age', 'age_of_onset', 'age_at_diagnosis',\n",
       "       'age_at_death', 'race_for_qc', 'family_history_for_qc', 'region_for_qc',\n",
       "       'study', 'pruned', 'pruned_reason', 'label', 'related'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cov = pd.read_csv(f'{WORK_DIR}/master_key_release3_final.csv')\n",
    "cov.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>SEX</th>\n",
       "      <th>AGE</th>\n",
       "      <th>PHENO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SYNAPS-KZ_000001_s1</td>\n",
       "      <td>SYNAPS-KZ_000001_s1</td>\n",
       "      <td>2</td>\n",
       "      <td>64.0</td>\n",
       "      <td>Control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SYNAPS-KZ_000002_s1</td>\n",
       "      <td>SYNAPS-KZ_000002_s1</td>\n",
       "      <td>1</td>\n",
       "      <td>53.0</td>\n",
       "      <td>Control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SYNAPS-KZ_000003_s1</td>\n",
       "      <td>SYNAPS-KZ_000003_s1</td>\n",
       "      <td>2</td>\n",
       "      <td>59.0</td>\n",
       "      <td>Control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SYNAPS-KZ_000004_s1</td>\n",
       "      <td>SYNAPS-KZ_000004_s1</td>\n",
       "      <td>1</td>\n",
       "      <td>57.0</td>\n",
       "      <td>Control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SYNAPS-KZ_000005_s1</td>\n",
       "      <td>SYNAPS-KZ_000005_s1</td>\n",
       "      <td>1</td>\n",
       "      <td>59.0</td>\n",
       "      <td>Control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16852</th>\n",
       "      <td>PPMI_000789_s1</td>\n",
       "      <td>PPMI_000789_s1</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16853</th>\n",
       "      <td>PPMI_000781_s1</td>\n",
       "      <td>PPMI_000781_s1</td>\n",
       "      <td>1</td>\n",
       "      <td>57.0</td>\n",
       "      <td>Other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16854</th>\n",
       "      <td>PPMI_000818_s1</td>\n",
       "      <td>PPMI_000818_s1</td>\n",
       "      <td>1</td>\n",
       "      <td>60.0</td>\n",
       "      <td>Other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16855</th>\n",
       "      <td>PPMI_000841_s1</td>\n",
       "      <td>PPMI_000841_s1</td>\n",
       "      <td>2</td>\n",
       "      <td>68.0</td>\n",
       "      <td>Other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16856</th>\n",
       "      <td>PPMI_000808_s1</td>\n",
       "      <td>PPMI_000808_s1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Other</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>16857 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                       FID                  IID  SEX   AGE    PHENO\n",
       "0      SYNAPS-KZ_000001_s1  SYNAPS-KZ_000001_s1    2  64.0  Control\n",
       "1      SYNAPS-KZ_000002_s1  SYNAPS-KZ_000002_s1    1  53.0  Control\n",
       "2      SYNAPS-KZ_000003_s1  SYNAPS-KZ_000003_s1    2  59.0  Control\n",
       "3      SYNAPS-KZ_000004_s1  SYNAPS-KZ_000004_s1    1  57.0  Control\n",
       "4      SYNAPS-KZ_000005_s1  SYNAPS-KZ_000005_s1    1  59.0  Control\n",
       "...                    ...                  ...  ...   ...      ...\n",
       "16852       PPMI_000789_s1       PPMI_000789_s1    2   NaN    Other\n",
       "16853       PPMI_000781_s1       PPMI_000781_s1    1  57.0    Other\n",
       "16854       PPMI_000818_s1       PPMI_000818_s1    1  60.0    Other\n",
       "16855       PPMI_000841_s1       PPMI_000841_s1    2  68.0    Other\n",
       "16856       PPMI_000808_s1       PPMI_000808_s1    1   NaN    Other\n",
       "\n",
       "[16857 rows x 5 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cov_reduced = cov[['GP2sampleID','GP2sampleID','sex_for_qc', 'age', 'phenotype']]\n",
    "cov_reduced.columns = ['FID','IID', 'SEX','AGE','PHENO']\n",
    "cov_reduced"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "conditions = [\n",
    "    (cov_reduced['PHENO'] == \"Case\"),\n",
    "    (cov_reduced['PHENO'] == \"Control\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.7/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  \n"
     ]
    }
   ],
   "source": [
    "choices = [2,1]\n",
    "cov_reduced['PHENOTYPE'] = np.select(conditions, choices, default=-9).astype(np.int64)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.7/site-packages/pandas/core/frame.py:4913: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  errors=errors,\n"
     ]
    }
   ],
   "source": [
    "cov_reduced.reset_index(inplace=True)\n",
    "cov_reduced.drop(columns=[\"index\"], inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "cov_reduced.drop(columns=['PHENO'], inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "sex = cov_reduced[['FID','IID','SEX']]\n",
    "sex.to_csv(f'{WORK_DIR}/SEX.txt',index=False, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "pheno = cov_reduced[['FID','IID','PHENOTYPE']]\n",
    "pheno.to_csv(f'{WORK_DIR}/PHENO.txt',index=False, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extract PP2A"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a3.3LM AVX2 Intel (3 Jun 2022)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2022 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to pheno_PP2A_AMR.log.\n",
      "Options in effect:\n",
      "  --chr 9\n",
      "  --from-bp 129110950\n",
      "  --make-bed\n",
      "  --out pheno_PP2A_AMR\n",
      "  --pfile chr9_AMR_release3\n",
      "  --to-bp 129148946\n",
      "\n",
      "Start time: Fri Jun  2 07:20:31 2023\n",
      "14998 MiB RAM detected; reserving 7499 MiB for main workspace.\n",
      "Using up to 4 compute threads.\n",
      "417 samples (186 females, 231 males; 417 founders) loaded from\n",
      "chr9_AMR_release3.psam.\n",
      "448914 variants loaded from chr9_AMR_release3.pvar.\n",
      "1 binary phenotype loaded (230 cases, 182 controls).\n",
      "178 variants remaining after main filters.\n",
      "Writing pheno_PP2A_AMR.fam ... done.\n",
      "Writing pheno_PP2A_AMR.bim ... done.\n",
      "Writing pheno_PP2A_AMR.bed ... 0%\b\bdone.\n",
      "End time: Fri Jun  2 07:20:31 2023\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "# PODXL: gene positions on hg38 (from https://useast.ensembl.org/index.html)\n",
    "/home/jupyter/tools/plink2 \\\n",
    "--pfile chr9_AMR_release3 \\\n",
    "--chr 9 \\\n",
    "--from-bp 129110950  \\\n",
    "--to-bp 129148946 \\\n",
    "--make-bed \\\n",
    "--out pheno_PP2A_AMR"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Visualize plink files bim, fam and bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9\tchr9:129110977:A:C\t0\t129110977\tC\tA\n",
      "9\tchr9:129111059:G:A\t0\t129111059\tA\tG\n",
      "9\tchr9:129111157:A:T\t0\t129111157\tT\tA\n",
      "9\tchr9:129111184:G:C\t0\t129111184\tC\tG\n",
      "9\tchr9:129111392:G:C\t0\t129111392\tC\tG\n",
      "9\tchr9:129111962:C:T\t0\t129111962\tT\tC\n",
      "9\tchr9:129112048:G:A\t0\t129112048\tA\tG\n",
      "9\tchr9:129112273:A:T\t0\t129112273\tT\tA\n",
      "9\tchr9:129112362:A:G\t0\t129112362\tG\tA\n",
      "9\tchr9:129112540:C:T\t0\t129112540\tT\tC\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Visualize bim file\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "head pheno_PP2A_AMR.bim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CORIELL_000189,CORIELL_000189_s1,m3,PD,2,PD,2,63.0,42.0,,,Unknown,No,USA,CORIELL,0,,AMR,\n",
      "UMD_000275,UMD_000275_s1,m1,PD,2,PD,1,,,,,Not Reported,No,USA,UMD,0,,AMR,\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "grep -e \"CORIELL_000189_s1\" -e \"UMD_000275_s1\" master_key_release3_final.csv > GP2_AMR_comphet.txt\n",
    "head GP2_AMR_comphet.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\tAPGS_000157_s1\t0\t0\t1\t2\n",
      "0\tAPGS_000551_s1\t0\t0\t1\t2\n",
      "0\tAPGS_000609_s1\t0\t0\t1\t2\n",
      "0\tAPGS_000724_s1\t0\t0\t1\t2\n",
      "0\tAPGS_000981_s1\t0\t0\t1\t2\n",
      "0\tAPGS_001003_s1\t0\t0\t2\t2\n",
      "0\tBCM_000007_s1\t0\t0\t1\t2\n",
      "0\tBCM_000020_s1\t0\t0\t1\t2\n",
      "0\tBCM_000054_s1\t0\t0\t2\t2\n",
      "0\tBCM_000127_s1\t0\t0\t1\t2\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Visualize fam file\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "head pheno_PP2A_AMR.fam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "l\u001b\u0001묺������>���������/Ϗ����������뺻���>���˿뿫������������\u000f��\u000f�����+���O���������*���ﯯ.��\u0000��������������������������������������������������������������������������������������������������������\u0003��������������������������������������������������������������������������������������������������������\u0003��������������������������������������������������������������������������������������������������������\u0003��������������������������������������������������������������������������������������������������������\u0003\u0000\u0000\u0000\u0000\u0000\u0000\u0002\u0000\u0000\u0000\u0002\u0000\u0000\u0000 \u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000�\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\b\u0000\u0000\u0002\u0000\u0003\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000 \u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0002\u0000\u0000\u0000\u0000\u0000\u0000\u0000\b\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\b\b\u0000(\b\u0000\u0002�\u0000\u0000\u0000\u0000\u0000\u0002\u0000\u0000\u0000\u0000\u0002\u0000��������������������������������������������������������������������������������������������������������\u0003�����*\n",
      "\b������8\u0002�\"�0�2���`�#�\n",
      "��8\b(��\u0003\n",
      "\u0002�\"(�\u000e��:�,���*���8�*\n",
      "���\f",
      "�\n",
      "�0*�\u000b",
      "((�#숸\b��(��\u000b",
      "��/����ꈢ�\"���(�:\u0003�����*\b\b������(\u0002�\"�0�2���`�#�\n",
      "��8\b(��\u0003\u0002\u0002� (�\u000e��:�,���*���8�*\n",
      "���\f",
      "�\n",
      "�0*�\u0003((�#숸\b��(��\u000b",
      "��\u000f����ꈢ�\"���(�8\u0003��������������������������������������������������������������������������������������������������������\u0003��������������������������������������������������������������������������������������������������������\u0003��������������������������������������������������������������������������������������������������������\u0003��������������������������������������������������������������������������������������������������������\u0003��������������������������������������������������������������������������������������������������������\u0003�������������������������������������������������������������������������������������������������������\u0002��������������������������������������������������������������������������������������������������������\u0003�����*\n",
      "\b������8\u0002�\"�0�2���`�#�\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Visualize fam file\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "head pheno_PP2A_AMR.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to pheno_PP2A_AMR.log.\n",
      "Options in effect:\n",
      "  --bfile pheno_PP2A_AMR\n",
      "  --out pheno_PP2A_AMR\n",
      "  --recode vcf-fid\n",
      "\n",
      "14998 MB RAM detected; reserving 7499 MB for main workspace.\n",
      "178 variants loaded from .bim file.\n",
      "417 people (231 males, 186 females) loaded from .fam.\n",
      "412 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 417 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.996915.\n",
      "178 variants and 417 people pass filters and QC.\n",
      "Among remaining phenotypes, 230 are cases and 182 are controls.  (5 phenotypes\n",
      "are missing.)\n",
      "--recode vcf-fid to pheno_PP2A_AMR.vcf ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "# Turn binary files into VCF\n",
    "/home/jupyter/tools/plink \\\n",
    "--bfile pheno_PP2A_AMR \\\n",
    "--recode vcf-fid \\\n",
    "--out pheno_PP2A_AMR"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Annotate PP2A variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[bgzip] can't create pheno_PP2A_AMR.vcf.gz: File exists\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "export PATH=$PATH:/home/jupyter/tools/rvtests/third/tabix-0.2.6/\n",
    "\n",
    "### Bgzip and Tabix\n",
    "bgzip pheno_PP2A_AMR.vcf\n",
    "\n",
    "tabix -f -p vcf pheno_PP2A_AMR.vcf.gz\n",
    "tabix -f -p vcf pheno_PP2A_AMR.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 pheno_PP2A_AMR.vcf.gz > pheno_PP2A_AMR.annovar.avinput>\n",
      "NOTICE: Finished reading 185 lines from VCF file\n",
      "NOTICE: A total of 178 locus in VCF file passed QC threshold, representing 163 SNPs (121 transitions and 42 transversions) and 15 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 67971 SNP genotypes (50457 transitions and 17514 transversions) and 6255 indels/substitutions for 417 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl pheno_PP2A_AMR.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile pheno_PP2A_AMR.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile pheno_PP2A_AMR.annovar.refGene -exonsort -nofirstcodondel pheno_PP2A_AMR.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to pheno_PP2A_AMR.annovar.refGene.variant_function, pheno_PP2A_AMR.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 178 unique variants in 178 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 6 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile pheno_PP2A_AMR.annovar pheno_PP2A_AMR.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to pheno_PP2A_AMR.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to pheno_PP2A_AMR.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 178 unique variants in 178 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 0\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to pheno_PP2A_AMR.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from pheno_PP2A_AMR.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to pheno_PP2A_AMR.annovar.hg38_multianno.vcf\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "### Annotate variants using ANNOVAR: https://annovar.openbioinformatics.org/en/latest/ \n",
    "perl /home/jupyter/tools/annovar/table_annovar.pl pheno_PP2A_AMR.vcf.gz /home/jupyter/tools/annovar/humandb/ -buildver hg38 \\\n",
    "-out pheno_PP2A_AMR.annovar \\\n",
    "-remove -protocol refGene,clinvar_20140902 \\\n",
    "-operation g,f \\\n",
    "--nopolish \\\n",
    "-nastring . \\\n",
    "-vcfinput"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extract coding and non-syn variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>Otherinfo420</th>\n",
       "      <th>Otherinfo421</th>\n",
       "      <th>Otherinfo422</th>\n",
       "      <th>Otherinfo423</th>\n",
       "      <th>Otherinfo424</th>\n",
       "      <th>Otherinfo425</th>\n",
       "      <th>Otherinfo426</th>\n",
       "      <th>Otherinfo427</th>\n",
       "      <th>Otherinfo428</th>\n",
       "      <th>Otherinfo429</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9</td>\n",
       "      <td>129110977</td>\n",
       "      <td>129110977</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>PTPA</td>\n",
       "      <td>NM_021131:c.-624A&gt;C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9</td>\n",
       "      <td>129111059</td>\n",
       "      <td>129111059</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>PTPA</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>9</td>\n",
       "      <td>129111157</td>\n",
       "      <td>129111157</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>PTPA</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>9</td>\n",
       "      <td>129111184</td>\n",
       "      <td>129111184</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>PTPA</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9</td>\n",
       "      <td>129111392</td>\n",
       "      <td>129111392</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>PTPA</td>\n",
       "      <td>NM_178001:c.-209G&gt;C;NM_001271832:c.-209G&gt;C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 440 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Chr      Start        End Ref Alt Func.refGene Gene.refGene  \\\n",
       "0    9  129110977  129110977   A   C         UTR5         PTPA   \n",
       "1    9  129111059  129111059   G   A     intronic         PTPA   \n",
       "2    9  129111157  129111157   A   T     intronic         PTPA   \n",
       "3    9  129111184  129111184   G   C     intronic         PTPA   \n",
       "4    9  129111392  129111392   G   C         UTR5         PTPA   \n",
       "\n",
       "                           GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "0                         NM_021131:c.-624A>C                  .   \n",
       "1                                           .                  .   \n",
       "2                                           .                  .   \n",
       "3                                           .                  .   \n",
       "4  NM_178001:c.-209G>C;NM_001271832:c.-209G>C                  .   \n",
       "\n",
       "  AAChange.refGene  ... Otherinfo420  Otherinfo421 Otherinfo422 Otherinfo423  \\\n",
       "0                .  ...          0/0           0/1          0/1          0/0   \n",
       "1                .  ...          0/0           0/0          0/0          0/0   \n",
       "2                .  ...          0/0           0/0          0/0          0/0   \n",
       "3                .  ...          0/0           0/0          0/0          0/0   \n",
       "4                .  ...          0/0           0/0          0/0          0/0   \n",
       "\n",
       "   Otherinfo424  Otherinfo425 Otherinfo426 Otherinfo427 Otherinfo428  \\\n",
       "0           0/1           0/0          0/1          1/1          0/0   \n",
       "1           0/0           0/0          0/0          0/0          0/0   \n",
       "2           0/0           0/0          0/0          0/0          0/0   \n",
       "3           0/0           0/0          0/0          0/0          0/0   \n",
       "4           0/0           0/0          0/0          0/0          0/0   \n",
       "\n",
       "  Otherinfo429  \n",
       "0          1/1  \n",
       "1          0/0  \n",
       "2          0/0  \n",
       "3          0/0  \n",
       "4          0/0  \n",
       "\n",
       "[5 rows x 440 columns]"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Visualize multianno file\n",
    "PP2A_AMR = pd.read_csv(f'{WORK_DIR}/pheno_PP2A_AMR.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "PP2A_AMR.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Chr             178\n",
       "Start           178\n",
       "End             178\n",
       "Ref             178\n",
       "Alt             178\n",
       "               ... \n",
       "Otherinfo425    178\n",
       "Otherinfo426    178\n",
       "Otherinfo427    178\n",
       "Otherinfo428    178\n",
       "Otherinfo429    178\n",
       "Length: 440, dtype: int64"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PP2A_AMR.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>Otherinfo420</th>\n",
       "      <th>Otherinfo421</th>\n",
       "      <th>Otherinfo422</th>\n",
       "      <th>Otherinfo423</th>\n",
       "      <th>Otherinfo424</th>\n",
       "      <th>Otherinfo425</th>\n",
       "      <th>Otherinfo426</th>\n",
       "      <th>Otherinfo427</th>\n",
       "      <th>Otherinfo428</th>\n",
       "      <th>Otherinfo429</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>9</td>\n",
       "      <td>129134852</td>\n",
       "      <td>129134852</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>PTPA</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>PTPA:NM_001271832:exon5:c.G431A:p.R144Q,PTPA:N...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170</th>\n",
       "      <td>9</td>\n",
       "      <td>129147457</td>\n",
       "      <td>129147457</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>PTPA</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>PTPA:NM_001271832:exon9:c.C878T:p.S293L,PTPA:N...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 440 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chr      Start        End Ref Alt Func.refGene Gene.refGene  \\\n",
       "108    9  129134852  129134852   G   A       exonic         PTPA   \n",
       "170    9  129147457  129147457   C   T       exonic         PTPA   \n",
       "\n",
       "    GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "108                  .  nonsynonymous SNV   \n",
       "170                  .  nonsynonymous SNV   \n",
       "\n",
       "                                      AAChange.refGene  ... Otherinfo420  \\\n",
       "108  PTPA:NM_001271832:exon5:c.G431A:p.R144Q,PTPA:N...  ...          0/0   \n",
       "170  PTPA:NM_001271832:exon9:c.C878T:p.S293L,PTPA:N...  ...          0/0   \n",
       "\n",
       "     Otherinfo421 Otherinfo422 Otherinfo423  Otherinfo424  Otherinfo425  \\\n",
       "108           0/0          0/0          0/0           0/0           0/0   \n",
       "170           0/0          0/1          0/0           0/0           0/0   \n",
       "\n",
       "    Otherinfo426 Otherinfo427 Otherinfo428 Otherinfo429  \n",
       "108          0/0          0/1          0/0          0/0  \n",
       "170          0/0          0/1          0/0          0/0  \n",
       "\n",
       "[2 rows x 440 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter exonic variants\n",
    "coding_AMR = PP2A_AMR[PP2A_AMR['Func.refGene'] == 'exonic']\n",
    "coding_AMR.count()\n",
    "coding_AMR.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "108    440\n",
       "170    440\n",
       "dtype: int64"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter exonic and non-syn vars\n",
    "coding_nonsynonymous_AMR = PP2A_AMR[(PP2A_AMR['Func.refGene'] == 'exonic') & (PP2A_AMR['ExonicFunc.refGene'] == 'nonsynonymous SNV')]\n",
    "coding_nonsynonymous_AMR.count(1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>Otherinfo420</th>\n",
       "      <th>Otherinfo421</th>\n",
       "      <th>Otherinfo422</th>\n",
       "      <th>Otherinfo423</th>\n",
       "      <th>Otherinfo424</th>\n",
       "      <th>Otherinfo425</th>\n",
       "      <th>Otherinfo426</th>\n",
       "      <th>Otherinfo427</th>\n",
       "      <th>Otherinfo428</th>\n",
       "      <th>Otherinfo429</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>9</td>\n",
       "      <td>129134852</td>\n",
       "      <td>129134852</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>PTPA</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>PTPA:NM_001271832:exon5:c.G431A:p.R144Q,PTPA:N...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 440 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chr      Start        End Ref Alt Func.refGene Gene.refGene  \\\n",
       "108    9  129134852  129134852   G   A       exonic         PTPA   \n",
       "\n",
       "    GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "108                  .  nonsynonymous SNV   \n",
       "\n",
       "                                      AAChange.refGene  ... Otherinfo420  \\\n",
       "108  PTPA:NM_001271832:exon5:c.G431A:p.R144Q,PTPA:N...  ...          0/0   \n",
       "\n",
       "     Otherinfo421 Otherinfo422 Otherinfo423  Otherinfo424  Otherinfo425  \\\n",
       "108           0/0          0/0          0/0           0/0           0/0   \n",
       "\n",
       "    Otherinfo426 Otherinfo427 Otherinfo428 Otherinfo429  \n",
       "108          0/0          0/1          0/0          0/0  \n",
       "\n",
       "[1 rows x 440 columns]"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding_nonsynonymous_AMR.head(1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "     Chr      Start        End Ref Alt Func.refGene Gene.refGene  \\\n",
      "108    9  129134852  129134852   G   A       exonic         PTPA   \n",
      "170    9  129147457  129147457   C   T       exonic         PTPA   \n",
      "\n",
      "    GeneDetail.refGene ExonicFunc.refGene  \\\n",
      "108                  .  nonsynonymous SNV   \n",
      "170                  .  nonsynonymous SNV   \n",
      "\n",
      "                                      AAChange.refGene  ... Otherinfo420  \\\n",
      "108  PTPA:NM_001271832:exon5:c.G431A:p.R144Q,PTPA:N...  ...          0/0   \n",
      "170  PTPA:NM_001271832:exon9:c.C878T:p.S293L,PTPA:N...  ...          0/0   \n",
      "\n",
      "     Otherinfo421 Otherinfo422 Otherinfo423  Otherinfo424  Otherinfo425  \\\n",
      "108           0/0          0/0          0/0           0/0           0/0   \n",
      "170           0/0          0/1          0/0           0/0           0/0   \n",
      "\n",
      "    Otherinfo426 Otherinfo427 Otherinfo428 Otherinfo429  \n",
      "108          0/0          0/1          0/0          0/0  \n",
      "170          0/0          0/1          0/0          0/0  \n",
      "\n",
      "[2 rows x 440 columns]\n"
     ]
    }
   ],
   "source": [
    "print(coding_nonsynonymous_AMR)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>Otherinfo420</th>\n",
       "      <th>Otherinfo421</th>\n",
       "      <th>Otherinfo422</th>\n",
       "      <th>Otherinfo423</th>\n",
       "      <th>Otherinfo424</th>\n",
       "      <th>Otherinfo425</th>\n",
       "      <th>Otherinfo426</th>\n",
       "      <th>Otherinfo427</th>\n",
       "      <th>Otherinfo428</th>\n",
       "      <th>Otherinfo429</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>9</td>\n",
       "      <td>129134852</td>\n",
       "      <td>129134852</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>PTPA</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>PTPA:NM_001271832:exon5:c.G431A:p.R144Q,PTPA:N...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170</th>\n",
       "      <td>9</td>\n",
       "      <td>129147457</td>\n",
       "      <td>129147457</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>PTPA</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>PTPA:NM_001271832:exon9:c.C878T:p.S293L,PTPA:N...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 440 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chr      Start        End Ref Alt Func.refGene Gene.refGene  \\\n",
       "108    9  129134852  129134852   G   A       exonic         PTPA   \n",
       "170    9  129147457  129147457   C   T       exonic         PTPA   \n",
       "\n",
       "    GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "108                  .  nonsynonymous SNV   \n",
       "170                  .  nonsynonymous SNV   \n",
       "\n",
       "                                      AAChange.refGene  ... Otherinfo420  \\\n",
       "108  PTPA:NM_001271832:exon5:c.G431A:p.R144Q,PTPA:N...  ...          0/0   \n",
       "170  PTPA:NM_001271832:exon9:c.C878T:p.S293L,PTPA:N...  ...          0/0   \n",
       "\n",
       "     Otherinfo421 Otherinfo422 Otherinfo423  Otherinfo424  Otherinfo425  \\\n",
       "108           0/0          0/0          0/0           0/0           0/0   \n",
       "170           0/0          0/1          0/0           0/0           0/0   \n",
       "\n",
       "    Otherinfo426 Otherinfo427 Otherinfo428 Otherinfo429  \n",
       "108          0/0          0/1          0/0          0/0  \n",
       "170          0/0          0/1          0/0          0/0  \n",
       "\n",
       "[2 rows x 440 columns]"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding_nonsynonymous_AMR.to_csv(f'{WORK_DIR}/coding_nonsynonymous.txt', sep = '\\t', index = False)\n",
    "coding_nonsynonymous_AMR.head(9)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Chr             167\n",
       "Start           167\n",
       "End             167\n",
       "Ref             167\n",
       "Alt             167\n",
       "               ... \n",
       "Otherinfo425    167\n",
       "Otherinfo426    167\n",
       "Otherinfo427    167\n",
       "Otherinfo428    167\n",
       "Otherinfo429    167\n",
       "Length: 440, dtype: int64"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter intronic variants\n",
    "intronic_AMR = PP2A_AMR[PP2A_AMR['Func.refGene'] == 'intronic']\n",
    "intronic_AMR.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Chr             7\n",
       "Start           7\n",
       "End             7\n",
       "Ref             7\n",
       "Alt             7\n",
       "               ..\n",
       "Otherinfo425    7\n",
       "Otherinfo426    7\n",
       "Otherinfo427    7\n",
       "Otherinfo428    7\n",
       "Otherinfo429    7\n",
       "Length: 440, dtype: int64"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter UTR3 variants\n",
    "UTR3_AMR = PP2A_AMR[PP2A_AMR['Func.refGene'] == ('UTR3')]\n",
    "UTR3_AMR.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Chr             2\n",
       "Start           2\n",
       "End             2\n",
       "Ref             2\n",
       "Alt             2\n",
       "               ..\n",
       "Otherinfo425    2\n",
       "Otherinfo426    2\n",
       "Otherinfo427    2\n",
       "Otherinfo428    2\n",
       "Otherinfo429    2\n",
       "Length: 440, dtype: int64"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter UTR5 variants\n",
    "UTR5_AMR = PP2A_AMR[PP2A_AMR['Func.refGene'] == ('UTR5')]\n",
    "UTR5_AMR.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Calculate freq - cases vs controls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "reduced_coding_nonsynonymous_AMR = coding_nonsynonymous_AMR[[\"Chr\", \"Start\", \"End\", \"Gene.refGene\"]]\n",
    "reduced_coding_nonsynonymous_AMR.to_csv(f'{WORK_DIR}/reduced_coding_nonsynonymous_AMR.txt', sep = '\\t', index = False, header= False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9\t129134852\t129134852\tPTPA\n",
      "9\t129147457\t129147457\tPTPA\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "head reduced_coding_nonsynonymous_AMR.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to coding_nonsynonymous_pheno_PP2A_AMR.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile pheno_PP2A_AMR\n",
      "  --ci 0.95\n",
      "  --extract range reduced_coding_nonsynonymous_AMR.txt\n",
      "  --out coding_nonsynonymous_pheno_PP2A_AMR\n",
      "\n",
      "14998 MB RAM detected; reserving 7499 MB for main workspace.\n",
      "178 variants loaded from .bim file.\n",
      "417 people (231 males, 186 females) loaded from .fam.\n",
      "412 phenotype values loaded from .fam.\n",
      "--extract range: 176 variants excluded.\n",
      "--extract range: 2 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 417 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.998801.\n",
      "2 variants and 417 people pass filters and QC.\n",
      "Among remaining phenotypes, 230 are cases and 182 are controls.  (5 phenotypes\n",
      "are missing.)\n",
      "Writing C/C --assoc report to coding_nonsynonymous_pheno_PP2A_AMR.assoc ...\n",
      "0%\b\bdone.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "/home/jupyter/tools/plink --bfile pheno_PP2A_AMR --extract range reduced_coding_nonsynonymous_AMR.txt --assoc --out coding_nonsynonymous_pheno_PP2A_AMR --allow-no-sex --ci 0.95"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to coding_nonsynonymous_AMR_rare_0.05.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile pheno_PP2A_AMR\n",
      "  --extract range reduced_coding_nonsynonymous_AMR.txt\n",
      "  --make-bed\n",
      "  --max-maf 0.05\n",
      "  --out coding_nonsynonymous_AMR_rare_0.05\n",
      "\n",
      "14998 MB RAM detected; reserving 7499 MB for main workspace.\n",
      "178 variants loaded from .bim file.\n",
      "417 people (231 males, 186 females) loaded from .fam.\n",
      "412 phenotype values loaded from .fam.\n",
      "--extract range: 176 variants excluded.\n",
      "--extract range: 2 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 417 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.998801.\n",
      "1 variant removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "1 variant and 417 people pass filters and QC.\n",
      "Among remaining phenotypes, 230 are cases and 182 are controls.  (5 phenotypes\n",
      "are missing.)\n",
      "--make-bed to coding_nonsynonymous_AMR_rare_0.05.bed +\n",
      "coding_nonsynonymous_AMR_rare_0.05.bim + coding_nonsynonymous_AMR_rare_0.05.fam\n",
      "... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "/home/jupyter/tools/plink --bfile pheno_PP2A_AMR --extract range reduced_coding_nonsynonymous_AMR.txt --make-bed --max-maf 0.05 --out coding_nonsynonymous_AMR_rare_0.05 --allow-no-sex"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP</th>\n",
       "      <th>BP</th>\n",
       "      <th>A1</th>\n",
       "      <th>F_A</th>\n",
       "      <th>F_U</th>\n",
       "      <th>A2</th>\n",
       "      <th>CHISQ</th>\n",
       "      <th>P</th>\n",
       "      <th>OR</th>\n",
       "      <th>SE</th>\n",
       "      <th>L95</th>\n",
       "      <th>U95</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:129134852:G:A</td>\n",
       "      <td>129134852</td>\n",
       "      <td>A</td>\n",
       "      <td>0.01739</td>\n",
       "      <td>0.005495</td>\n",
       "      <td>G</td>\n",
       "      <td>2.3990</td>\n",
       "      <td>0.1214</td>\n",
       "      <td>3.2040</td>\n",
       "      <td>0.7937</td>\n",
       "      <td>0.6761</td>\n",
       "      <td>15.180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:129147457:C:T</td>\n",
       "      <td>129147457</td>\n",
       "      <td>T</td>\n",
       "      <td>0.04565</td>\n",
       "      <td>0.058010</td>\n",
       "      <td>C</td>\n",
       "      <td>0.6382</td>\n",
       "      <td>0.4244</td>\n",
       "      <td>0.7768</td>\n",
       "      <td>0.3169</td>\n",
       "      <td>0.4174</td>\n",
       "      <td>1.446</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHR                 SNP         BP A1      F_A       F_U A2   CHISQ  \\\n",
       "0    9  chr9:129134852:G:A  129134852  A  0.01739  0.005495  G  2.3990   \n",
       "1    9  chr9:129147457:C:T  129147457  T  0.04565  0.058010  C  0.6382   \n",
       "\n",
       "        P      OR      SE     L95     U95  \n",
       "0  0.1214  3.2040  0.7937  0.6761  15.180  \n",
       "1  0.4244  0.7768  0.3169  0.4174   1.446  "
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PP2A_AMR_freq = pd.read_csv(f'{WORK_DIR}/coding_nonsynonymous_pheno_PP2A_AMR.assoc', delim_whitespace=True)\n",
    "PP2A_AMR_freq.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to coding_nonsynonymous_pheno_PP2A_AMR.log.\n",
      "Options in effect:\n",
      "  --adjust\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile pheno_PP2A_AMR\n",
      "  --extract range reduced_coding_nonsynonymous_AMR.txt\n",
      "  --out coding_nonsynonymous_pheno_PP2A_AMR\n",
      "\n",
      "14998 MB RAM detected; reserving 7499 MB for main workspace.\n",
      "178 variants loaded from .bim file.\n",
      "417 people (231 males, 186 females) loaded from .fam.\n",
      "412 phenotype values loaded from .fam.\n",
      "--extract range: 176 variants excluded.\n",
      "--extract range: 2 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 417 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.998801.\n",
      "2 variants and 417 people pass filters and QC.\n",
      "Among remaining phenotypes, 230 are cases and 182 are controls.  (5 phenotypes\n",
      "are missing.)\n",
      "Writing C/C --assoc report to coding_nonsynonymous_pheno_PP2A_AMR.assoc ...\n",
      "0%\b\bdone.\n",
      "--adjust: Genomic inflation est. lambda (based on median chisq) = 3.33021.\n",
      "0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b--adjust values (2 variants) written to\n",
      "coding_nonsynonymous_pheno_PP2A_AMR.assoc.adjusted .\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "/home/jupyter/tools/plink --bfile pheno_PP2A_AMR --extract range reduced_coding_nonsynonymous_AMR.txt --assoc --out coding_nonsynonymous_pheno_PP2A_AMR --allow-no-sex --adjust"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP</th>\n",
       "      <th>UNADJ</th>\n",
       "      <th>GC</th>\n",
       "      <th>BONF</th>\n",
       "      <th>HOLM</th>\n",
       "      <th>SIDAK_SS</th>\n",
       "      <th>SIDAK_SD</th>\n",
       "      <th>FDR_BH</th>\n",
       "      <th>FDR_BY</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:129134852:G:A</td>\n",
       "      <td>0.1214</td>\n",
       "      <td>0.3960</td>\n",
       "      <td>0.2428</td>\n",
       "      <td>0.2428</td>\n",
       "      <td>0.2281</td>\n",
       "      <td>0.2281</td>\n",
       "      <td>0.2428</td>\n",
       "      <td>0.3642</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:129147457:C:T</td>\n",
       "      <td>0.4244</td>\n",
       "      <td>0.6616</td>\n",
       "      <td>0.8487</td>\n",
       "      <td>0.4244</td>\n",
       "      <td>0.6686</td>\n",
       "      <td>0.4244</td>\n",
       "      <td>0.4244</td>\n",
       "      <td>0.6365</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHR                 SNP   UNADJ      GC    BONF    HOLM  SIDAK_SS  \\\n",
       "0    9  chr9:129134852:G:A  0.1214  0.3960  0.2428  0.2428    0.2281   \n",
       "1    9  chr9:129147457:C:T  0.4244  0.6616  0.8487  0.4244    0.6686   \n",
       "\n",
       "   SIDAK_SD  FDR_BH  FDR_BY  \n",
       "0    0.2281  0.2428  0.3642  \n",
       "1    0.4244  0.4244  0.6365  "
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PP2A_AMR_freq = pd.read_csv(f'{WORK_DIR}/coding_nonsynonymous_pheno_PP2A_AMR.assoc.adjusted', delim_whitespace=True)\n",
    "PP2A_AMR_freq.head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Calculate freq of HMZ in cases versus controls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to coding_nonsynonymous_pheno_PP2A_AMR.log.\n",
      "Options in effect:\n",
      "  --bfile pheno_PP2A_AMR\n",
      "  --extract range reduced_coding_nonsynonymous_AMR.txt\n",
      "  --out coding_nonsynonymous_pheno_PP2A_AMR\n",
      "  --recode A\n",
      "\n",
      "14998 MB RAM detected; reserving 7499 MB for main workspace.\n",
      "178 variants loaded from .bim file.\n",
      "417 people (231 males, 186 females) loaded from .fam.\n",
      "412 phenotype values loaded from .fam.\n",
      "--extract range: 176 variants excluded.\n",
      "--extract range: 2 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 417 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.998801.\n",
      "2 variants and 417 people pass filters and QC.\n",
      "Among remaining phenotypes, 230 are cases and 182 are controls.  (5 phenotypes\n",
      "are missing.)\n",
      "--recode A to coding_nonsynonymous_pheno_PP2A_AMR.raw ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "/home/jupyter/tools/plink --bfile pheno_PP2A_AMR --extract range reduced_coding_nonsynonymous_AMR.txt --recode A --out coding_nonsynonymous_pheno_PP2A_AMR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP</th>\n",
       "      <th>BP</th>\n",
       "      <th>A1</th>\n",
       "      <th>F_A</th>\n",
       "      <th>F_U</th>\n",
       "      <th>A2</th>\n",
       "      <th>CHISQ</th>\n",
       "      <th>P</th>\n",
       "      <th>OR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:129134852:G:A</td>\n",
       "      <td>129134852</td>\n",
       "      <td>A</td>\n",
       "      <td>0.01739</td>\n",
       "      <td>0.005495</td>\n",
       "      <td>G</td>\n",
       "      <td>2.3990</td>\n",
       "      <td>0.1214</td>\n",
       "      <td>3.2040</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:129147457:C:T</td>\n",
       "      <td>129147457</td>\n",
       "      <td>T</td>\n",
       "      <td>0.04565</td>\n",
       "      <td>0.058010</td>\n",
       "      <td>C</td>\n",
       "      <td>0.6382</td>\n",
       "      <td>0.4244</td>\n",
       "      <td>0.7768</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHR                 SNP         BP A1      F_A       F_U A2   CHISQ  \\\n",
       "0    9  chr9:129134852:G:A  129134852  A  0.01739  0.005495  G  2.3990   \n",
       "1    9  chr9:129147457:C:T  129147457  T  0.04565  0.058010  C  0.6382   \n",
       "\n",
       "        P      OR  \n",
       "0  0.1214  3.2040  \n",
       "1  0.4244  0.7768  "
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PP2A_AMR_freq = pd.read_csv(f'{WORK_DIR}/coding_nonsynonymous_pheno_PP2A_AMR.assoc', delim_whitespace=True)\n",
    "PP2A_AMR_freq.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129134852:G:A_A</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000157_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000551_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000609_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000724_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000981_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   FID             IID  PAT  MAT  SEX  PHENOTYPE  chr9:129134852:G:A_A  \\\n",
       "0    0  APGS_000157_s1    0    0    1          2                     0   \n",
       "1    0  APGS_000551_s1    0    0    1          2                     0   \n",
       "2    0  APGS_000609_s1    0    0    1          2                     0   \n",
       "3    0  APGS_000724_s1    0    0    1          2                     0   \n",
       "4    0  APGS_000981_s1    0    0    1          2                     0   \n",
       "\n",
       "   chr9:129147457:C:T_T  \n",
       "0                   0.0  \n",
       "1                   0.0  \n",
       "2                   0.0  \n",
       "3                   0.0  \n",
       "4                   1.0  "
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PP2A_AMR_recode = pd.read_csv(f'{WORK_DIR}/coding_nonsynonymous_pheno_PP2A_AMR.raw', delim_whitespace=True)\n",
    "PP2A_AMR_recode.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to coding_nonsynonymous_pheno_PP2A_AMR.log.\n",
      "Options in effect:\n",
      "  --bfile pheno_PP2A_AMR\n",
      "  --extract range reduced_coding_nonsynonymous_AMR.txt\n",
      "  --out coding_nonsynonymous_pheno_PP2A_AMR\n",
      "  --recode A\n",
      "\n",
      "14998 MB RAM detected; reserving 7499 MB for main workspace.\n",
      "178 variants loaded from .bim file.\n",
      "417 people (231 males, 186 females) loaded from .fam.\n",
      "412 phenotype values loaded from .fam.\n",
      "--extract range: 176 variants excluded.\n",
      "--extract range: 2 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 417 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.998801.\n",
      "2 variants and 417 people pass filters and QC.\n",
      "Among remaining phenotypes, 230 are cases and 182 are controls.  (5 phenotypes\n",
      "are missing.)\n",
      "--recode A to coding_nonsynonymous_pheno_PP2A_AMR.raw ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "/home/jupyter/tools/plink --bfile  pheno_PP2A_AMR --extract range reduced_coding_nonsynonymous_AMR.txt --recode A --out coding_nonsynonymous_pheno_PP2A_AMR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129134852:G:A_A</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [FID, IID, PAT, MAT, SEX, PHENOTYPE, chr9:129134852:G:A_A, chr9:129147457:C:T_T]\n",
       "Index: []"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Explore HMZs for chr9:129147457:C:T_T in cases\n",
    "PP2A_hom_casos_AMR = PP2A_AMR_recode[(PP2A_AMR_recode['chr9:129147457:C:T_T'] == 2) & (PP2A_AMR_recode['PHENOTYPE'] == 2)]\n",
    "PP2A_hom_casos_AMR.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(0, 8)"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PP2A_hom_casos_AMR.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129134852:G:A_A</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [FID, IID, PAT, MAT, SEX, PHENOTYPE, chr9:129134852:G:A_A, chr9:129147457:C:T_T]\n",
       "Index: []"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Explore HMZs for chr9:129147457:C:T_T in controls\n",
    "PP2A_hom_controles_AMR = PP2A_AMR_recode[(PP2A_AMR_recode['chr9:129147457:C:T_T'] == 2) & (PP2A_AMR_recode['PHENOTYPE'] == 1)]\n",
    "PP2A_hom_controles_AMR.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(0, 8)"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PP2A_hom_controles_AMR.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129134852:G:A_A</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [FID, IID, PAT, MAT, SEX, PHENOTYPE, chr9:129134852:G:A_A, chr9:129147457:C:T_T]\n",
       "Index: []"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Explore HMZs for chr9:129134852:G:A_A in cases\n",
    "PP2A_hom_casos_AMR = PP2A_AMR_recode[(PP2A_AMR_recode['chr9:129134852:G:A_A'] == 2) & (PP2A_AMR_recode['PHENOTYPE'] == 2)]\n",
    "PP2A_hom_casos_AMR.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129134852:G:A_A</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [FID, IID, PAT, MAT, SEX, PHENOTYPE, chr9:129134852:G:A_A, chr9:129147457:C:T_T]\n",
       "Index: []"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Explore HMZs for chr9:129134852:G:A_A in controls\n",
    "PP2A_hom_controles_AMR = PP2A_AMR_recode[(PP2A_AMR_recode['chr9:129134852:G:A_A'] == 2) & (PP2A_AMR_recode['PHENOTYPE'] == 1)]\n",
    "PP2A_hom_controles_AMR.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(50, 8)\n"
     ]
    }
   ],
   "source": [
    "# Identifying individuals het for the alternative allele\n",
    "het = PP2A_AMR_recode[(PP2A_AMR_recode['chr9:129147457:C:T_T'] == 1) | (PP2A_AMR_recode['chr9:129134852:G:A_A'] == 1)]\n",
    "het = het.dropna()\n",
    "print(het.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129134852:G:A_A</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000981_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000054_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000127_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000136_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000180_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    FID             IID  PAT  MAT  SEX  PHENOTYPE  chr9:129134852:G:A_A  \\\n",
       "4     0  APGS_000981_s1    0    0    1          2                     0   \n",
       "8     0   BCM_000054_s1    0    0    2          2                     0   \n",
       "9     0   BCM_000127_s1    0    0    1          2                     0   \n",
       "11    0   BCM_000136_s1    0    0    1          2                     0   \n",
       "15    0   BCM_000180_s1    0    0    2          1                     0   \n",
       "\n",
       "    chr9:129147457:C:T_T  \n",
       "4                    1.0  \n",
       "8                    1.0  \n",
       "9                    1.0  \n",
       "11                   1.0  \n",
       "15                   1.0  "
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "het.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1.0    47\n",
       "2.0     3\n",
       "Name: compound, dtype: int64"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "het['compound'] = het['chr9:129147457:C:T_T'] + het['chr9:129134852:G:A_A']\n",
    "het['compound'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(2, 9)\n"
     ]
    }
   ],
   "source": [
    "comp_het_info_cases = het[(het['PHENOTYPE'] == 2) & (het['compound'] == 2)]\n",
    "print(comp_het_info_cases.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1, 9)\n"
     ]
    }
   ],
   "source": [
    "comp_het_info_control = het[(het['PHENOTYPE'] == 1) & (het['compound'] == 2)]\n",
    "print(comp_het_info_control.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129134852:G:A_A</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "      <th>compound</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_000189_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>414</th>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000275_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     FID                IID  PAT  MAT  SEX  PHENOTYPE  chr9:129134852:G:A_A  \\\n",
       "90     0  CORIELL_000189_s1    0    0    2          2                     1   \n",
       "414    0      UMD_000275_s1    0    0    1          2                     1   \n",
       "\n",
       "     chr9:129147457:C:T_T  compound  \n",
       "90                    1.0       2.0  \n",
       "414                   1.0       2.0  "
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "comp_het_info_cases.to_csv('/home/jupyter/PTPA/comp_het_info_cases.txt', sep = '\\t', index=True)\n",
    "comp_het_info_cases.head(54)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129134852:G:A_A</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "      <th>compound</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>270</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006462_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     FID                IID  PAT  MAT  SEX  PHENOTYPE  chr9:129134852:G:A_A  \\\n",
       "270    0  CORIELL_006462_s1    0    0    1          1                     1   \n",
       "\n",
       "     chr9:129147457:C:T_T  compound  \n",
       "270                   1.0       2.0  "
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "comp_het_info_control.to_csv('/home/jupyter/PTPA/comp_het_info_control.txt', sep = '\\t', index=True)\n",
    "comp_het_info_control.head(54)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129134852:G:A_A</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000981_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000054_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000127_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000136_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000206_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000214_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_000189_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_001342_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_001771_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>147</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_002148_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>176</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_003627_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>183</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_003890_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>187</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_004101_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>190</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_004134_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>206</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_004339_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>213</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_004348_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>214</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_004349_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>235</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_004421_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>380</th>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_000883_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>405</th>\n",
       "      <td>0</td>\n",
       "      <td>PPMI_000507_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>414</th>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000275_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     FID                IID  PAT  MAT  SEX  PHENOTYPE  chr9:129134852:G:A_A  \\\n",
       "4      0     APGS_000981_s1    0    0    1          2                     0   \n",
       "8      0      BCM_000054_s1    0    0    2          2                     0   \n",
       "9      0      BCM_000127_s1    0    0    1          2                     0   \n",
       "11     0      BCM_000136_s1    0    0    1          2                     0   \n",
       "19     0      BCM_000206_s1    0    0    1          2                     0   \n",
       "20     0      BCM_000214_s1    0    0    1          2                     0   \n",
       "90     0  CORIELL_000189_s1    0    0    2          2                     1   \n",
       "130    0  CORIELL_001342_s1    0    0    1          2                     0   \n",
       "143    0  CORIELL_001771_s1    0    0    2          2                     0   \n",
       "147    0  CORIELL_002148_s1    0    0    2          2                     0   \n",
       "176    0  CORIELL_003627_s1    0    0    1          2                     0   \n",
       "183    0  CORIELL_003890_s1    0    0    2          2                     0   \n",
       "187    0  CORIELL_004101_s1    0    0    1          2                     0   \n",
       "190    0  CORIELL_004134_s1    0    0    1          2                     0   \n",
       "206    0  CORIELL_004339_s1    0    0    2          2                     0   \n",
       "213    0  CORIELL_004348_s1    0    0    2          2                     0   \n",
       "214    0  CORIELL_004349_s1    0    0    1          2                     0   \n",
       "235    0  CORIELL_004421_s1    0    0    1          2                     0   \n",
       "380    0     PAGE_000883_s1    0    0    2          2                     0   \n",
       "405    0     PPMI_000507_s1    0    0    2          2                     0   \n",
       "414    0      UMD_000275_s1    0    0    1          2                     1   \n",
       "\n",
       "     chr9:129147457:C:T_T  \n",
       "4                     1.0  \n",
       "8                     1.0  \n",
       "9                     1.0  \n",
       "11                    1.0  \n",
       "19                    1.0  \n",
       "20                    1.0  \n",
       "90                    1.0  \n",
       "130                   1.0  \n",
       "143                   1.0  \n",
       "147                   1.0  \n",
       "176                   1.0  \n",
       "183                   1.0  \n",
       "187                   1.0  \n",
       "190                   1.0  \n",
       "206                   1.0  \n",
       "213                   1.0  \n",
       "214                   1.0  \n",
       "235                   1.0  \n",
       "380                   1.0  \n",
       "405                   1.0  \n",
       "414                   1.0  "
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Explore HETs for chr9:129147457:C:T_T in cases\n",
    "PP2A_het_casos_129147457_AMR = PP2A_AMR_recode[(PP2A_AMR_recode['chr9:129147457:C:T_T'] == 1) & (PP2A_AMR_recode['PHENOTYPE'] == 2)]\n",
    "PP2A_het_casos_129147457_AMR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129134852:G:A_A</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000180_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000641_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>255</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_005578_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006462_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006464_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>283</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006511_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>297</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006609_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>299</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006634_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006647_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>313</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006662_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>314</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006663_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>336</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006692_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>344</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006701_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>352</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006730_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>356</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007003_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>359</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007370_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>362</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007429_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>364</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007590_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>385</th>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_001290_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>388</th>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_001582_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>398</th>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_002782_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     FID                IID  PAT  MAT  SEX  PHENOTYPE  chr9:129134852:G:A_A  \\\n",
       "15     0      BCM_000180_s1    0    0    2          1                     0   \n",
       "66     0      BCM_000641_s1    0    0    2          1                     0   \n",
       "255    0  CORIELL_005578_s1    0    0    1          1                     0   \n",
       "270    0  CORIELL_006462_s1    0    0    1          1                     1   \n",
       "271    0  CORIELL_006464_s1    0    0    1          1                     0   \n",
       "283    0  CORIELL_006511_s1    0    0    1          1                     0   \n",
       "297    0  CORIELL_006609_s1    0    0    1          1                     0   \n",
       "299    0  CORIELL_006634_s1    0    0    2          1                     0   \n",
       "306    0  CORIELL_006647_s1    0    0    1          1                     0   \n",
       "313    0  CORIELL_006662_s1    0    0    2          1                     0   \n",
       "314    0  CORIELL_006663_s1    0    0    2          1                     0   \n",
       "336    0  CORIELL_006692_s1    0    0    1          1                     0   \n",
       "344    0  CORIELL_006701_s1    0    0    1          1                     0   \n",
       "352    0  CORIELL_006730_s1    0    0    2          1                     0   \n",
       "356    0  CORIELL_007003_s1    0    0    2          1                     0   \n",
       "359    0  CORIELL_007370_s1    0    0    2          1                     0   \n",
       "362    0  CORIELL_007429_s1    0    0    2          1                     0   \n",
       "364    0  CORIELL_007590_s1    0    0    1          1                     0   \n",
       "385    0     PAGE_001290_s1    0    0    1          1                     0   \n",
       "388    0     PAGE_001582_s1    0    0    1          1                     0   \n",
       "398    0     PAGE_002782_s1    0    0    1          1                     0   \n",
       "\n",
       "     chr9:129147457:C:T_T  \n",
       "15                    1.0  \n",
       "66                    1.0  \n",
       "255                   1.0  \n",
       "270                   1.0  \n",
       "271                   1.0  \n",
       "283                   1.0  \n",
       "297                   1.0  \n",
       "299                   1.0  \n",
       "306                   1.0  \n",
       "313                   1.0  \n",
       "314                   1.0  \n",
       "336                   1.0  \n",
       "344                   1.0  \n",
       "352                   1.0  \n",
       "356                   1.0  \n",
       "359                   1.0  \n",
       "362                   1.0  \n",
       "364                   1.0  \n",
       "385                   1.0  \n",
       "388                   1.0  \n",
       "398                   1.0  "
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Explore HETs for chr9:129147457:C:T_T in controls\n",
    "PP2A_het_controles_129147457_AMR = PP2A_AMR_recode[(PP2A_AMR_recode['chr9:129147457:C:T_T'] == 1) & (PP2A_AMR_recode['PHENOTYPE'] == 1)]\n",
    "PP2A_het_controles_129147457_AMR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [],
   "source": [
    "PP2A_het_casos_129147457_AMR.to_csv('PP2A_het_casos_129147457_AMR.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129134852:G:A_A</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000981_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000054_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000127_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000136_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000206_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>20</td>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000214_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>90</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_000189_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>130</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_001342_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>143</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_001771_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>147</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_002148_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>176</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_003627_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>183</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_003890_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>187</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_004101_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>190</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_004134_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>206</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_004339_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>213</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_004348_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>214</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_004349_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>235</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_004421_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>380</td>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_000883_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>405</td>\n",
       "      <td>0</td>\n",
       "      <td>PPMI_000507_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>414</td>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000275_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Unnamed: 0  FID                IID  PAT  MAT  SEX  PHENOTYPE  \\\n",
       "0            4    0     APGS_000981_s1    0    0    1          2   \n",
       "1            8    0      BCM_000054_s1    0    0    2          2   \n",
       "2            9    0      BCM_000127_s1    0    0    1          2   \n",
       "3           11    0      BCM_000136_s1    0    0    1          2   \n",
       "4           19    0      BCM_000206_s1    0    0    1          2   \n",
       "5           20    0      BCM_000214_s1    0    0    1          2   \n",
       "6           90    0  CORIELL_000189_s1    0    0    2          2   \n",
       "7          130    0  CORIELL_001342_s1    0    0    1          2   \n",
       "8          143    0  CORIELL_001771_s1    0    0    2          2   \n",
       "9          147    0  CORIELL_002148_s1    0    0    2          2   \n",
       "10         176    0  CORIELL_003627_s1    0    0    1          2   \n",
       "11         183    0  CORIELL_003890_s1    0    0    2          2   \n",
       "12         187    0  CORIELL_004101_s1    0    0    1          2   \n",
       "13         190    0  CORIELL_004134_s1    0    0    1          2   \n",
       "14         206    0  CORIELL_004339_s1    0    0    2          2   \n",
       "15         213    0  CORIELL_004348_s1    0    0    2          2   \n",
       "16         214    0  CORIELL_004349_s1    0    0    1          2   \n",
       "17         235    0  CORIELL_004421_s1    0    0    1          2   \n",
       "18         380    0     PAGE_000883_s1    0    0    2          2   \n",
       "19         405    0     PPMI_000507_s1    0    0    2          2   \n",
       "20         414    0      UMD_000275_s1    0    0    1          2   \n",
       "\n",
       "    chr9:129134852:G:A_A  chr9:129147457:C:T_T  \n",
       "0                      0                   1.0  \n",
       "1                      0                   1.0  \n",
       "2                      0                   1.0  \n",
       "3                      0                   1.0  \n",
       "4                      0                   1.0  \n",
       "5                      0                   1.0  \n",
       "6                      1                   1.0  \n",
       "7                      0                   1.0  \n",
       "8                      0                   1.0  \n",
       "9                      0                   1.0  \n",
       "10                     0                   1.0  \n",
       "11                     0                   1.0  \n",
       "12                     0                   1.0  \n",
       "13                     0                   1.0  \n",
       "14                     0                   1.0  \n",
       "15                     0                   1.0  \n",
       "16                     0                   1.0  \n",
       "17                     0                   1.0  \n",
       "18                     0                   1.0  \n",
       "19                     0                   1.0  \n",
       "20                     1                   1.0  "
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "temp = pd.read_csv('PP2A_het_casos_129147457_AMR.csv')\n",
    "temp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [],
   "source": [
    "PP2A_het_controles_129147457_AMR.to_csv('PP2A_het_controles_129147457_AMR.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129134852:G:A_A</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000180_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>66</td>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000641_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>255</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_005578_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>270</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006462_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>271</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006464_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>283</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006511_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>297</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006609_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>299</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006634_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>306</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006647_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>313</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006662_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>314</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006663_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>336</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006692_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>344</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006701_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>352</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006730_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>356</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007003_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>359</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007370_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>362</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007429_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>364</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007590_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>385</td>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_001290_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>388</td>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_001582_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>398</td>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_002782_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Unnamed: 0  FID                IID  PAT  MAT  SEX  PHENOTYPE  \\\n",
       "0           15    0      BCM_000180_s1    0    0    2          1   \n",
       "1           66    0      BCM_000641_s1    0    0    2          1   \n",
       "2          255    0  CORIELL_005578_s1    0    0    1          1   \n",
       "3          270    0  CORIELL_006462_s1    0    0    1          1   \n",
       "4          271    0  CORIELL_006464_s1    0    0    1          1   \n",
       "5          283    0  CORIELL_006511_s1    0    0    1          1   \n",
       "6          297    0  CORIELL_006609_s1    0    0    1          1   \n",
       "7          299    0  CORIELL_006634_s1    0    0    2          1   \n",
       "8          306    0  CORIELL_006647_s1    0    0    1          1   \n",
       "9          313    0  CORIELL_006662_s1    0    0    2          1   \n",
       "10         314    0  CORIELL_006663_s1    0    0    2          1   \n",
       "11         336    0  CORIELL_006692_s1    0    0    1          1   \n",
       "12         344    0  CORIELL_006701_s1    0    0    1          1   \n",
       "13         352    0  CORIELL_006730_s1    0    0    2          1   \n",
       "14         356    0  CORIELL_007003_s1    0    0    2          1   \n",
       "15         359    0  CORIELL_007370_s1    0    0    2          1   \n",
       "16         362    0  CORIELL_007429_s1    0    0    2          1   \n",
       "17         364    0  CORIELL_007590_s1    0    0    1          1   \n",
       "18         385    0     PAGE_001290_s1    0    0    1          1   \n",
       "19         388    0     PAGE_001582_s1    0    0    1          1   \n",
       "20         398    0     PAGE_002782_s1    0    0    1          1   \n",
       "\n",
       "    chr9:129134852:G:A_A  chr9:129147457:C:T_T  \n",
       "0                      0                   1.0  \n",
       "1                      0                   1.0  \n",
       "2                      0                   1.0  \n",
       "3                      1                   1.0  \n",
       "4                      0                   1.0  \n",
       "5                      0                   1.0  \n",
       "6                      0                   1.0  \n",
       "7                      0                   1.0  \n",
       "8                      0                   1.0  \n",
       "9                      0                   1.0  \n",
       "10                     0                   1.0  \n",
       "11                     0                   1.0  \n",
       "12                     0                   1.0  \n",
       "13                     0                   1.0  \n",
       "14                     0                   1.0  \n",
       "15                     0                   1.0  \n",
       "16                     0                   1.0  \n",
       "17                     0                   1.0  \n",
       "18                     0                   1.0  \n",
       "19                     0                   1.0  \n",
       "20                     0                   1.0  "
      ]
     },
     "execution_count": 98,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "temp = pd.read_csv('PP2A_het_controles_129147457_AMR.csv')\n",
    "temp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129134852:G:A_A</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_000189_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_000589_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>124</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_001146_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>163</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_003334_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>195</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_004207_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>198</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_004245_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>205</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_004338_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>414</th>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000275_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     FID                IID  PAT  MAT  SEX  PHENOTYPE  chr9:129134852:G:A_A  \\\n",
       "90     0  CORIELL_000189_s1    0    0    2          2                     1   \n",
       "107    0  CORIELL_000589_s1    0    0    2          2                     1   \n",
       "124    0  CORIELL_001146_s1    0    0    1          2                     1   \n",
       "163    0  CORIELL_003334_s1    0    0    2          2                     1   \n",
       "195    0  CORIELL_004207_s1    0    0    2          2                     1   \n",
       "198    0  CORIELL_004245_s1    0    0    2          2                     1   \n",
       "205    0  CORIELL_004338_s1    0    0    2          2                     1   \n",
       "414    0      UMD_000275_s1    0    0    1          2                     1   \n",
       "\n",
       "     chr9:129147457:C:T_T  \n",
       "90                    1.0  \n",
       "107                   0.0  \n",
       "124                   0.0  \n",
       "163                   0.0  \n",
       "195                   0.0  \n",
       "198                   0.0  \n",
       "205                   0.0  \n",
       "414                   1.0  "
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Explore HETs for chr9:129134852:G:A_A in cases\n",
    "PP2A_het_casos_129134852_AMR = PP2A_AMR_recode[(PP2A_AMR_recode['chr9:129134852:G:A_A'] == 1) & (PP2A_AMR_recode['PHENOTYPE'] == 2)]\n",
    "PP2A_het_casos_129134852_AMR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129134852:G:A_A</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000529_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006462_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     FID                IID  PAT  MAT  SEX  PHENOTYPE  chr9:129134852:G:A_A  \\\n",
       "47     0      BCM_000529_s1    0    0    2          1                     1   \n",
       "270    0  CORIELL_006462_s1    0    0    1          1                     1   \n",
       "\n",
       "     chr9:129147457:C:T_T  \n",
       "47                    0.0  \n",
       "270                   1.0  "
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Explore HETs for chr9:129134852:G:A_A in controles\n",
    "PP2A_het_controles_129134852_AMR = PP2A_AMR_recode[(PP2A_AMR_recode['chr9:129134852:G:A_A'] == 1) & (PP2A_AMR_recode['PHENOTYPE'] == 1)]\n",
    "PP2A_het_controles_129134852_AMR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [],
   "source": [
    "PP2A_het_casos_129134852_AMR.to_csv('PP2A_het_casos_129134852_AMR.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129134852:G:A_A</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>90</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_000189_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>107</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_000589_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>124</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_001146_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>163</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_003334_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>195</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_004207_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>198</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_004245_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>205</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_004338_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>414</td>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000275_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Unnamed: 0  FID                IID  PAT  MAT  SEX  PHENOTYPE  \\\n",
       "0          90    0  CORIELL_000189_s1    0    0    2          2   \n",
       "1         107    0  CORIELL_000589_s1    0    0    2          2   \n",
       "2         124    0  CORIELL_001146_s1    0    0    1          2   \n",
       "3         163    0  CORIELL_003334_s1    0    0    2          2   \n",
       "4         195    0  CORIELL_004207_s1    0    0    2          2   \n",
       "5         198    0  CORIELL_004245_s1    0    0    2          2   \n",
       "6         205    0  CORIELL_004338_s1    0    0    2          2   \n",
       "7         414    0      UMD_000275_s1    0    0    1          2   \n",
       "\n",
       "   chr9:129134852:G:A_A  chr9:129147457:C:T_T  \n",
       "0                     1                   1.0  \n",
       "1                     1                   0.0  \n",
       "2                     1                   0.0  \n",
       "3                     1                   0.0  \n",
       "4                     1                   0.0  \n",
       "5                     1                   0.0  \n",
       "6                     1                   0.0  \n",
       "7                     1                   1.0  "
      ]
     },
     "execution_count": 88,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "temp = pd.read_csv('PP2A_het_casos_129134852_AMR.csv')\n",
    "temp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [],
   "source": [
    "PP2A_het_controles_129134852_AMR.to_csv('PP2A_het_controles_129134852_AMR.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129134852:G:A_A</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>47</td>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000529_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>270</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006462_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Unnamed: 0  FID                IID  PAT  MAT  SEX  PHENOTYPE  \\\n",
       "0          47    0      BCM_000529_s1    0    0    2          1   \n",
       "1         270    0  CORIELL_006462_s1    0    0    1          1   \n",
       "\n",
       "   chr9:129134852:G:A_A  chr9:129147457:C:T_T  \n",
       "0                     1                   0.0  \n",
       "1                     1                   1.0  "
      ]
     },
     "execution_count": 93,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "temp = pd.read_csv('PP2A_het_controles_129134852_AMR.csv')\n",
    "temp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: rpy2 in /home/jupyter/.local/lib/python3.7/site-packages (3.5.12)\n",
      "Requirement already satisfied: cffi>=1.10.0 in /opt/conda/lib/python3.7/site-packages (from rpy2) (1.15.0)\n",
      "Requirement already satisfied: jinja2 in /opt/conda/lib/python3.7/site-packages (from rpy2) (3.1.2)\n",
      "Requirement already satisfied: pytz in /opt/conda/lib/python3.7/site-packages (from rpy2) (2022.6)\n",
      "Requirement already satisfied: tzlocal<5.0 in /home/jupyter/.local/lib/python3.7/site-packages (from rpy2) (4.3)\n",
      "Requirement already satisfied: typing-extensions in /opt/conda/lib/python3.7/site-packages (from rpy2) (4.1.1)\n",
      "Requirement already satisfied: pycparser in /opt/conda/lib/python3.7/site-packages (from cffi>=1.10.0->rpy2) (2.21)\n",
      "Requirement already satisfied: pytz-deprecation-shim in /home/jupyter/.local/lib/python3.7/site-packages (from tzlocal<5.0->rpy2) (0.1.0.post0)\n",
      "Requirement already satisfied: backports.zoneinfo in /home/jupyter/.local/lib/python3.7/site-packages (from tzlocal<5.0->rpy2) (0.2.1)\n",
      "Requirement already satisfied: MarkupSafe>=2.0 in /opt/conda/lib/python3.7/site-packages (from jinja2->rpy2) (2.0.1)\n",
      "Requirement already satisfied: tzdata in /home/jupyter/.local/lib/python3.7/site-packages (from pytz-deprecation-shim->tzlocal<5.0->rpy2) (2023.3)\n"
     ]
    }
   ],
   "source": [
    "!pip install rpy2\n",
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The rpy2.ipython extension is already loaded. To reload it, use:\n",
      "  %reload_ext rpy2.ipython\n"
     ]
    }
   ],
   "source": [
    "import rpy2\n",
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                  GP2ID         GP2sampleID manifest_id phenotype pheno_for_qc\n",
      "    1: SYNAPS-KZ_000001 SYNAPS-KZ_000001_s1          m1   Control            1\n",
      "    2: SYNAPS-KZ_000002 SYNAPS-KZ_000002_s1          m1   Control            1\n",
      "    3: SYNAPS-KZ_000003 SYNAPS-KZ_000003_s1          m1   Control            1\n",
      "    4: SYNAPS-KZ_000004 SYNAPS-KZ_000004_s1          m1   Control            1\n",
      "    5: SYNAPS-KZ_000005 SYNAPS-KZ_000005_s1          m1   Control            1\n",
      "   ---                                                                        \n",
      "16853:      PPMI_000789      PPMI_000789_s1          m2     Other           -9\n",
      "16854:      PPMI_000781      PPMI_000781_s1          m2     Other           -9\n",
      "16855:      PPMI_000818      PPMI_000818_s1          m2     Other           -9\n",
      "16856:      PPMI_000841      PPMI_000841_s1          m2     Other           -9\n",
      "16857:      PPMI_000808      PPMI_000808_s1          m2     Other           -9\n",
      "                                  other_pheno sex_for_qc age age_of_onset\n",
      "    1:                                Control          2  64           NA\n",
      "    2:                                Control          1  53           NA\n",
      "    3:                                Control          2  59           NA\n",
      "    4:                                Control          1  57           NA\n",
      "    5:                                Control          1  59           NA\n",
      "   ---                                                                   \n",
      "16853:   Genetically-enriched-cohort Affected          2  NA           NA\n",
      "16854: Genetidally-enriched-cohort Unaffected          1  57           NA\n",
      "16855:   Genetically-enriched-cohort Affected          1  60           NA\n",
      "16856:   Genetically-enriched-cohort Affected          2  68           NA\n",
      "16857:   Genetically-enriched-cohort Affected          1  NA           NA\n",
      "       age_at_diagnosis age_at_death race_for_qc family_history_for_qc\n",
      "    1:               NA           NA       Asian          Not Reported\n",
      "    2:               NA           NA       Asian          Not Reported\n",
      "    3:               NA           NA       Asian          Not Reported\n",
      "    4:               NA           NA       Asian          Not Reported\n",
      "    5:               NA           NA       Asian          Not Reported\n",
      "   ---                                                                \n",
      "16853:               NA           NA       White          Not Reported\n",
      "16854:               NA           NA       White          Not Reported\n",
      "16855:               55           55       White          Not Reported\n",
      "16856:               57           57       White          Not Reported\n",
      "16857:               NA           NA       White          Not Reported\n",
      "       region_for_qc     study pruned  pruned_reason label related\n",
      "    1:           KAZ SYNAPS-KZ      0                  CAS      NA\n",
      "    2:           KAZ SYNAPS-KZ      0                  CAS      NA\n",
      "    3:           KAZ SYNAPS-KZ      0                  CAS      NA\n",
      "    4:           KAZ SYNAPS-KZ      0                  CAS      NA\n",
      "    5:           KAZ SYNAPS-KZ      0                  CAS      NA\n",
      "   ---                                                            \n",
      "16853:           ISR      PPMI      0                   AJ      NA\n",
      "16854:           ISR      PPMI      0                   AJ      NA\n",
      "16855:           ISR      PPMI      1 callrate_prune            NA\n",
      "16856:           ISR      PPMI      0                   AJ       1\n",
      "16857:           ISR      PPMI      0                   AJ      NA\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "data.table 1.14.6 using 2 threads (see ?getDTthreads).  Latest news: r-datatable.com\n",
       "In addition: Warning message:\n",
       "In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
       "  libraries ‘/home/jupyter/packages’, ‘/usr/lib/R/site-library’ contain no packages\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "setwd('/home/jupyter/PP2A_GP2/')\n",
    "library(data.table)\n",
    "covs <- fread(\"master_key_release3_final.csv\", header =T)\n",
    "covs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                  GP2ID                 IID manifest_id phenotype pheno_for_qc\n",
      "    1: SYNAPS-KZ_000001 SYNAPS-KZ_000001_s1          m1   Control            1\n",
      "    2: SYNAPS-KZ_000002 SYNAPS-KZ_000002_s1          m1   Control            1\n",
      "    3: SYNAPS-KZ_000003 SYNAPS-KZ_000003_s1          m1   Control            1\n",
      "    4: SYNAPS-KZ_000004 SYNAPS-KZ_000004_s1          m1   Control            1\n",
      "    5: SYNAPS-KZ_000005 SYNAPS-KZ_000005_s1          m1   Control            1\n",
      "   ---                                                                        \n",
      "16853:      PPMI_000789      PPMI_000789_s1          m2     Other           -9\n",
      "16854:      PPMI_000781      PPMI_000781_s1          m2     Other           -9\n",
      "16855:      PPMI_000818      PPMI_000818_s1          m2     Other           -9\n",
      "16856:      PPMI_000841      PPMI_000841_s1          m2     Other           -9\n",
      "16857:      PPMI_000808      PPMI_000808_s1          m2     Other           -9\n",
      "                                  other_pheno sex_for_qc age age_of_onset\n",
      "    1:                                Control          2  64           NA\n",
      "    2:                                Control          1  53           NA\n",
      "    3:                                Control          2  59           NA\n",
      "    4:                                Control          1  57           NA\n",
      "    5:                                Control          1  59           NA\n",
      "   ---                                                                   \n",
      "16853:   Genetically-enriched-cohort Affected          2  NA           NA\n",
      "16854: Genetidally-enriched-cohort Unaffected          1  57           NA\n",
      "16855:   Genetically-enriched-cohort Affected          1  60           NA\n",
      "16856:   Genetically-enriched-cohort Affected          2  68           NA\n",
      "16857:   Genetically-enriched-cohort Affected          1  NA           NA\n",
      "       age_at_diagnosis age_at_death race_for_qc family_history_for_qc\n",
      "    1:               NA           NA       Asian          Not Reported\n",
      "    2:               NA           NA       Asian          Not Reported\n",
      "    3:               NA           NA       Asian          Not Reported\n",
      "    4:               NA           NA       Asian          Not Reported\n",
      "    5:               NA           NA       Asian          Not Reported\n",
      "   ---                                                                \n",
      "16853:               NA           NA       White          Not Reported\n",
      "16854:               NA           NA       White          Not Reported\n",
      "16855:               55           55       White          Not Reported\n",
      "16856:               57           57       White          Not Reported\n",
      "16857:               NA           NA       White          Not Reported\n",
      "       region_for_qc     study pruned  pruned_reason label related\n",
      "    1:           KAZ SYNAPS-KZ      0                  CAS      NA\n",
      "    2:           KAZ SYNAPS-KZ      0                  CAS      NA\n",
      "    3:           KAZ SYNAPS-KZ      0                  CAS      NA\n",
      "    4:           KAZ SYNAPS-KZ      0                  CAS      NA\n",
      "    5:           KAZ SYNAPS-KZ      0                  CAS      NA\n",
      "   ---                                                            \n",
      "16853:           ISR      PPMI      0                   AJ      NA\n",
      "16854:           ISR      PPMI      0                   AJ      NA\n",
      "16855:           ISR      PPMI      1 callrate_prune            NA\n",
      "16856:           ISR      PPMI      0                   AJ       1\n",
      "16857:           ISR      PPMI      0                   AJ      NA\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "colnames(covs)[2]  <- \"IID\"  \n",
    "covs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                  IID          GP2ID manifest_id phenotype pheno_for_qc\n",
      " 1:    APGS_000981_s1    APGS_000981          m1        PD            2\n",
      " 2:     BCM_000054_s1     BCM_000054          m1        PD            2\n",
      " 3:     BCM_000127_s1     BCM_000127          m1        PD            2\n",
      " 4:     BCM_000136_s1     BCM_000136          m1        PD            2\n",
      " 5:     BCM_000206_s1     BCM_000206          m1        PD            2\n",
      " 6:     BCM_000214_s1     BCM_000214          m1        PD            2\n",
      " 7: CORIELL_000189_s1 CORIELL_000189          m3        PD            2\n",
      " 8: CORIELL_001342_s1 CORIELL_001342          m3        PD            2\n",
      " 9: CORIELL_001771_s1 CORIELL_001771          m3        PD            2\n",
      "10: CORIELL_002148_s1 CORIELL_002148          m3        PD            2\n",
      "11: CORIELL_003627_s1 CORIELL_003627          m3        PD            2\n",
      "12: CORIELL_003890_s1 CORIELL_003890          m3        PD            2\n",
      "13: CORIELL_004101_s1 CORIELL_004101          m3        PD            2\n",
      "14: CORIELL_004134_s1 CORIELL_004134          m3        PD            2\n",
      "15: CORIELL_004339_s1 CORIELL_004339          m4        PD            2\n",
      "16: CORIELL_004348_s1 CORIELL_004348          m4        PD            2\n",
      "17: CORIELL_004349_s1 CORIELL_004349          m4        PD            2\n",
      "18: CORIELL_004421_s1 CORIELL_004421          m4        PD            2\n",
      "19:    PAGE_000883_s1    PAGE_000883          m1        PD            2\n",
      "20:    PPMI_000507_s1    PPMI_000507          m1        PD            2\n",
      "21:     UMD_000275_s1     UMD_000275          m1        PD            2\n",
      "                  IID          GP2ID manifest_id phenotype pheno_for_qc\n",
      "    other_pheno sex_for_qc      age age_of_onset age_at_diagnosis age_at_death\n",
      " 1:          PD          1       NA           NA               NA           NA\n",
      " 2:          PD          2       NA           NA               NA           NA\n",
      " 3:          PD          1       NA           NA               NA           NA\n",
      " 4:          PD          1       NA           NA               NA           NA\n",
      " 5:          PD          1       NA           NA               NA           NA\n",
      " 6:          PD          1       NA           NA               NA           NA\n",
      " 7:          PD          2 63.00000     42.00000               NA           NA\n",
      " 8:          PD          1 60.00000     51.00000               NA           NA\n",
      " 9:          PD          2 59.00000     52.00000               NA           NA\n",
      "10:          PD          2 57.00000     44.00000         44.00000           NA\n",
      "11:          PD          1 78.00000     75.00000         76.00000           NA\n",
      "12:          PD          2 38.00000     37.00000         38.00000           NA\n",
      "13:          PD          1 54.00000     48.00000         48.00000           NA\n",
      "14:          PD          1 50.00000     50.00000         50.00000           NA\n",
      "15:          PD          2 66.00000     40.00000         41.00000           NA\n",
      "16:          PD          2 57.00000     46.00000         46.00000           NA\n",
      "17:          PD          1 51.00000     45.00000         45.00000           NA\n",
      "18:          PD          1 44.00000     41.00000         41.00000           NA\n",
      "19:          PD          2 81.16632     76.41667         76.41667           NA\n",
      "20:          PD          2 33.70000           NA         32.00000           NA\n",
      "21:          PD          1       NA           NA               NA           NA\n",
      "    other_pheno sex_for_qc      age age_of_onset age_at_diagnosis age_at_death\n",
      "     race_for_qc family_history_for_qc region_for_qc   study pruned\n",
      " 1: Not Reported          Not Reported           AUS    APGS      0\n",
      " 2:      Unknown                    No           USA     BCM      0\n",
      " 3:        White                    No           USA     BCM      0\n",
      " 4:      Unknown                    No           USA     BCM      0\n",
      " 5:      Unknown                    No           USA     BCM      0\n",
      " 6:      Unknown                    No           USA     BCM      0\n",
      " 7:      Unknown                    No           USA CORIELL      0\n",
      " 8:      Unknown                    No           USA CORIELL      0\n",
      " 9:        White                    No           USA CORIELL      0\n",
      "10:        Other                    No           USA CORIELL      0\n",
      "11:        White                    No           USA CORIELL      0\n",
      "12:        White                    No           USA CORIELL      0\n",
      "13:        White                    No           USA CORIELL      0\n",
      "14:        Other                   Yes           USA CORIELL      0\n",
      "15:        Other                    No           USA CORIELL      0\n",
      "16:        Other                    No           USA CORIELL      0\n",
      "17:        Other                    No           USA CORIELL      0\n",
      "18:        Other                   Yes           USA CORIELL      0\n",
      "19:        Other                   Yes           USA    PAGE      0\n",
      "20:        Other          Not Reported           USA    PPMI      0\n",
      "21: Not Reported                    No           USA     UMD      0\n",
      "     race_for_qc family_history_for_qc region_for_qc   study pruned\n",
      "    pruned_reason label related  V1 FID PAT MAT SEX PHENOTYPE\n",
      " 1:            NA   AMR      NA   4   0   0   0   1         2\n",
      " 2:            NA   AMR      NA   8   0   0   0   2         2\n",
      " 3:            NA   AMR      NA   9   0   0   0   1         2\n",
      " 4:            NA   AMR      NA  11   0   0   0   1         2\n",
      " 5:            NA   AMR      NA  19   0   0   0   1         2\n",
      " 6:            NA   AMR      NA  20   0   0   0   1         2\n",
      " 7:            NA   AMR      NA  90   0   0   0   2         2\n",
      " 8:            NA   AMR      NA 130   0   0   0   1         2\n",
      " 9:            NA   AMR      NA 143   0   0   0   2         2\n",
      "10:            NA   AMR      NA 147   0   0   0   2         2\n",
      "11:            NA   AMR      NA 176   0   0   0   1         2\n",
      "12:            NA   AMR      NA 183   0   0   0   2         2\n",
      "13:            NA   AMR      NA 187   0   0   0   1         2\n",
      "14:            NA   AMR      NA 190   0   0   0   1         2\n",
      "15:            NA   AMR      NA 206   0   0   0   2         2\n",
      "16:            NA   AMR      NA 213   0   0   0   2         2\n",
      "17:            NA   AMR      NA 214   0   0   0   1         2\n",
      "18:            NA   AMR      NA 235   0   0   0   1         2\n",
      "19:            NA   AMR      NA 380   0   0   0   2         2\n",
      "20:            NA   AMR      NA 405   0   0   0   2         2\n",
      "21:            NA   AMR      NA 414   0   0   0   1         2\n",
      "    pruned_reason label related  V1 FID PAT MAT SEX PHENOTYPE\n",
      "    chr9:129134852:G:A_A chr9:129147457:C:T_T\n",
      " 1:                    0                    1\n",
      " 2:                    0                    1\n",
      " 3:                    0                    1\n",
      " 4:                    0                    1\n",
      " 5:                    0                    1\n",
      " 6:                    0                    1\n",
      " 7:                    1                    1\n",
      " 8:                    0                    1\n",
      " 9:                    0                    1\n",
      "10:                    0                    1\n",
      "11:                    0                    1\n",
      "12:                    0                    1\n",
      "13:                    0                    1\n",
      "14:                    0                    1\n",
      "15:                    0                    1\n",
      "16:                    0                    1\n",
      "17:                    0                    1\n",
      "18:                    0                    1\n",
      "19:                    0                    1\n",
      "20:                    0                    1\n",
      "21:                    1                    1\n",
      "    chr9:129134852:G:A_A chr9:129147457:C:T_T\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "setwd('/home/jupyter/PP2A_GP2/')\n",
    "library(data.table)\n",
    "PP2A_het_casos_129147457_AMR <- fread(\"PP2A_het_casos_129147457_AMR.csv\", header =T)\n",
    "PP2A_het_casos_129147457_AMR\n",
    "temp <- merge(covs, PP2A_het_casos_129147457_AMR, by=\"IID\")\n",
    "head(temp)\n",
    "write.table(temp, file = \"PP2A_het_casos_129147457_AMR_extracted.txt\", quote = F, row.names = F, sep = \"\\t\")\n",
    "PP2A_het_casos_129147457_AMR_extracted <- fread(\"PP2A_het_casos_129147457_AMR_extracted.txt\", header =T)\n",
    "PP2A_het_casos_129147457_AMR_extracted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                  IID          GP2ID manifest_id phenotype pheno_for_qc\n",
      " 1:     BCM_000180_s1     BCM_000180          m1   Control            1\n",
      " 2:     BCM_000641_s1     BCM_000641          m1   Control            1\n",
      " 3: CORIELL_005578_s1 CORIELL_005578          m6   Control            1\n",
      " 4: CORIELL_006462_s1 CORIELL_006462          m6   Control            1\n",
      " 5: CORIELL_006464_s1 CORIELL_006464          m6   Control            1\n",
      " 6: CORIELL_006511_s1 CORIELL_006511          m6   Control            1\n",
      " 7: CORIELL_006609_s1 CORIELL_006609          m6   Control            1\n",
      " 8: CORIELL_006634_s1 CORIELL_006634          m6   Control            1\n",
      " 9: CORIELL_006647_s1 CORIELL_006647          m6   Control            1\n",
      "10: CORIELL_006662_s1 CORIELL_006662          m6   Control            1\n",
      "11: CORIELL_006663_s1 CORIELL_006663          m6   Control            1\n",
      "12: CORIELL_006692_s1 CORIELL_006692          m6   Control            1\n",
      "13: CORIELL_006701_s1 CORIELL_006701          m6   Control            1\n",
      "14: CORIELL_006730_s1 CORIELL_006730          m6   Control            1\n",
      "15: CORIELL_007003_s1 CORIELL_007003          m6   Control            1\n",
      "16: CORIELL_007370_s1 CORIELL_007370          m6   Control            1\n",
      "17: CORIELL_007429_s1 CORIELL_007429          m6   Control            1\n",
      "18: CORIELL_007590_s1 CORIELL_007590          m6   Control            1\n",
      "19:    PAGE_001290_s1    PAGE_001290          m1   Control            1\n",
      "20:    PAGE_001582_s1    PAGE_001582          m1   Control            1\n",
      "21:    PAGE_002782_s1    PAGE_002782          m1   Control            1\n",
      "                  IID          GP2ID manifest_id phenotype pheno_for_qc\n",
      "    other_pheno sex_for_qc age age_of_onset age_at_diagnosis age_at_death\n",
      " 1:     Control          2  NA           NA               NA           NA\n",
      " 2:     Control          2  NA           NA               NA           NA\n",
      " 3:     Control          1  72           NA               NA           NA\n",
      " 4:     Control          1  59           NA               NA           NA\n",
      " 5:     Control          1  60           NA               NA           NA\n",
      " 6:     Control          1  75           NA               NA           NA\n",
      " 7:     Control          1  61           NA               NA           NA\n",
      " 8:     Control          2  55           NA               NA           NA\n",
      " 9:     Control          1  62           NA               NA           NA\n",
      "10:     Control          2  77           NA               NA           NA\n",
      "11:     Control          2  56           NA               NA           NA\n",
      "12:     Control          1  59           NA               NA           NA\n",
      "13:     Control          1  55           NA               NA           NA\n",
      "14:     Control          2  58           NA               NA           NA\n",
      "15:     Control          2  71           NA               NA           NA\n",
      "16:     Control          2  78           NA               NA           NA\n",
      "17:     Control          2  57           NA               NA           NA\n",
      "18:     Control          1  78           NA               NA           NA\n",
      "19:     Control          1  NA           NA               NA           NA\n",
      "20:     Control          1  NA           NA               NA           NA\n",
      "21:     Control          1  NA           NA               NA           NA\n",
      "    other_pheno sex_for_qc age age_of_onset age_at_diagnosis age_at_death\n",
      "                  race_for_qc family_history_for_qc region_for_qc   study\n",
      " 1:                     White                    No           USA     BCM\n",
      " 2:                   Unknown                    No           USA     BCM\n",
      " 3:                     Other                    No           USA CORIELL\n",
      " 4:                     Other                    No           USA CORIELL\n",
      " 5:                     Other                    No           USA CORIELL\n",
      " 6:                     White                    No           USA CORIELL\n",
      " 7:                     White                    No           USA CORIELL\n",
      " 8:                     White                    No           USA CORIELL\n",
      " 9:                     White                    No           USA CORIELL\n",
      "10:                     White                    No           USA CORIELL\n",
      "11:                     White                    No           USA CORIELL\n",
      "12:                     White                    No           USA CORIELL\n",
      "13:                     White                    No           USA CORIELL\n",
      "14:                     White                    No           USA CORIELL\n",
      "15:                     White                    No           USA CORIELL\n",
      "16: Black or African American          Not Reported           USA CORIELL\n",
      "17: Black or African American          Not Reported           USA CORIELL\n",
      "18: Black or African American          Not Reported           USA CORIELL\n",
      "19:                     Other          Not Reported           USA    PAGE\n",
      "20:                     Other          Not Reported           USA    PAGE\n",
      "21:                     Other          Not Reported           USA    PAGE\n",
      "                  race_for_qc family_history_for_qc region_for_qc   study\n",
      "    pruned pruned_reason label related  V1 FID PAT MAT SEX PHENOTYPE\n",
      " 1:      0            NA   AMR      NA  15   0   0   0   2         1\n",
      " 2:      0            NA   AMR      NA  66   0   0   0   2         1\n",
      " 3:      0            NA   AMR      NA 255   0   0   0   1         1\n",
      " 4:      0            NA   AMR      NA 270   0   0   0   1         1\n",
      " 5:      0            NA   AMR      NA 271   0   0   0   1         1\n",
      " 6:      0            NA   AMR      NA 283   0   0   0   1         1\n",
      " 7:      0            NA   AMR      NA 297   0   0   0   1         1\n",
      " 8:      0            NA   AMR      NA 299   0   0   0   2         1\n",
      " 9:      0            NA   AMR      NA 306   0   0   0   1         1\n",
      "10:      0            NA   AMR      NA 313   0   0   0   2         1\n",
      "11:      0            NA   AMR       1 314   0   0   0   2         1\n",
      "12:      0            NA   AMR      NA 336   0   0   0   1         1\n",
      "13:      0            NA   AMR      NA 344   0   0   0   1         1\n",
      "14:      0            NA   AMR      NA 352   0   0   0   2         1\n",
      "15:      0            NA   AMR      NA 356   0   0   0   2         1\n",
      "16:      0            NA   AMR      NA 359   0   0   0   2         1\n",
      "17:      0            NA   AMR      NA 362   0   0   0   2         1\n",
      "18:      0            NA   AMR      NA 364   0   0   0   1         1\n",
      "19:      0            NA   AMR      NA 385   0   0   0   1         1\n",
      "20:      0            NA   AMR      NA 388   0   0   0   1         1\n",
      "21:      0            NA   AMR      NA 398   0   0   0   1         1\n",
      "    pruned pruned_reason label related  V1 FID PAT MAT SEX PHENOTYPE\n",
      "    chr9:129134852:G:A_A chr9:129147457:C:T_T\n",
      " 1:                    0                    1\n",
      " 2:                    0                    1\n",
      " 3:                    0                    1\n",
      " 4:                    1                    1\n",
      " 5:                    0                    1\n",
      " 6:                    0                    1\n",
      " 7:                    0                    1\n",
      " 8:                    0                    1\n",
      " 9:                    0                    1\n",
      "10:                    0                    1\n",
      "11:                    0                    1\n",
      "12:                    0                    1\n",
      "13:                    0                    1\n",
      "14:                    0                    1\n",
      "15:                    0                    1\n",
      "16:                    0                    1\n",
      "17:                    0                    1\n",
      "18:                    0                    1\n",
      "19:                    0                    1\n",
      "20:                    0                    1\n",
      "21:                    0                    1\n",
      "    chr9:129134852:G:A_A chr9:129147457:C:T_T\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "setwd('/home/jupyter/PP2A_GP2/')\n",
    "library(data.table)\n",
    "PP2A_het_controles_129147457_AMR <- fread(\"PP2A_het_controles_129147457_AMR.csv\", header =T)\n",
    "PP2A_het_controles_129147457_AMR\n",
    "temp <- merge(covs, PP2A_het_controles_129147457_AMR, by=\"IID\")\n",
    "head(temp)\n",
    "write.table(temp, file = \"PP2A_het_controles_129147457_AMR_extracted.txt\", quote = F, row.names = F, sep = \"\\t\")\n",
    "PP2A_het_controles_129147457_AMR_extracted <- fread(\"PP2A_het_controles_129147457_AMR_extracted.txt\", header =T)\n",
    "PP2A_het_controles_129147457_AMR_extracted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                 IID          GP2ID manifest_id phenotype pheno_for_qc\n",
      "1: CORIELL_000189_s1 CORIELL_000189          m3        PD            2\n",
      "2: CORIELL_000589_s1 CORIELL_000589          m3        PD            2\n",
      "3: CORIELL_001146_s1 CORIELL_001146          m3        PD            2\n",
      "4: CORIELL_003334_s1 CORIELL_003334          m3        PD            2\n",
      "5: CORIELL_004207_s1 CORIELL_004207          m3        PD            2\n",
      "6: CORIELL_004245_s1 CORIELL_004245          m3        PD            2\n",
      "7: CORIELL_004338_s1 CORIELL_004338          m4        PD            2\n",
      "8:     UMD_000275_s1     UMD_000275          m1        PD            2\n",
      "   other_pheno sex_for_qc age age_of_onset age_at_diagnosis age_at_death\n",
      "1:          PD          2  63           42               NA           NA\n",
      "2:          PD          2  42           38               NA           NA\n",
      "3:          PD          1  63           60               NA           NA\n",
      "4:          PD          2  60           52               52           NA\n",
      "5:          PD          2  49           44               44           NA\n",
      "6:          PD          2  39           34               39           NA\n",
      "7:          PD          2  43           35               38           NA\n",
      "8:          PD          1  NA           NA               NA           NA\n",
      "    race_for_qc family_history_for_qc region_for_qc   study pruned\n",
      "1:      Unknown                    No           USA CORIELL      0\n",
      "2:        White                    No           USA CORIELL      0\n",
      "3:        White                    No           USA CORIELL      0\n",
      "4:        White                    No           USA CORIELL      0\n",
      "5: Multi-racial                    No           USA CORIELL      0\n",
      "6:        Other                    No           USA CORIELL      0\n",
      "7:        Other                    No           USA CORIELL      0\n",
      "8: Not Reported                    No           USA     UMD      0\n",
      "   pruned_reason label related  V1 FID PAT MAT SEX PHENOTYPE\n",
      "1:            NA   AMR      NA  90   0   0   0   2         2\n",
      "2:            NA   AMR      NA 107   0   0   0   2         2\n",
      "3:            NA   AMR      NA 124   0   0   0   1         2\n",
      "4:            NA   AMR      NA 163   0   0   0   2         2\n",
      "5:            NA   AMR      NA 195   0   0   0   2         2\n",
      "6:            NA   AMR      NA 198   0   0   0   2         2\n",
      "7:            NA   AMR      NA 205   0   0   0   2         2\n",
      "8:            NA   AMR      NA 414   0   0   0   1         2\n",
      "   chr9:129134852:G:A_A chr9:129147457:C:T_T\n",
      "1:                    1                    1\n",
      "2:                    1                    0\n",
      "3:                    1                    0\n",
      "4:                    1                    0\n",
      "5:                    1                    0\n",
      "6:                    1                    0\n",
      "7:                    1                    0\n",
      "8:                    1                    1\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "setwd('/home/jupyter/PP2A_GP2/')\n",
    "library(data.table)\n",
    "PP2A_het_casos_129134852_AMR <- fread(\"PP2A_het_casos_129134852_AMR.csv\", header =T)\n",
    "PP2A_het_casos_129134852_AMR\n",
    "temp <- merge(covs, PP2A_het_casos_129134852_AMR, by=\"IID\")\n",
    "head(temp)\n",
    "write.table(temp, file = \"PP2A_het_casos_129134852_AMR_extracted.txt\", quote = F, row.names = F, sep = \"\\t\")\n",
    "PP2A_het_casos_129134852_AMR_extracted <- fread(\"PP2A_het_casos_129134852_AMR_extracted.txt\", header =T)\n",
    "PP2A_het_casos_129134852_AMR_extracted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                 IID          GP2ID manifest_id phenotype pheno_for_qc\n",
      "1:     BCM_000529_s1     BCM_000529          m1   Control            1\n",
      "2: CORIELL_006462_s1 CORIELL_006462          m6   Control            1\n",
      "   other_pheno sex_for_qc age age_of_onset age_at_diagnosis age_at_death\n",
      "1:     Control          2  NA           NA               NA           NA\n",
      "2:     Control          1  59           NA               NA           NA\n",
      "   race_for_qc family_history_for_qc region_for_qc   study pruned pruned_reason\n",
      "1:     Unknown                    No           USA     BCM      0            NA\n",
      "2:       Other                    No           USA CORIELL      0            NA\n",
      "   label related  V1 FID PAT MAT SEX PHENOTYPE chr9:129134852:G:A_A\n",
      "1:   AMR      NA  47   0   0   0   2         1                    1\n",
      "2:   AMR      NA 270   0   0   0   1         1                    1\n",
      "   chr9:129147457:C:T_T\n",
      "1:                    0\n",
      "2:                    1\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "setwd('/home/jupyter/PP2A_GP2/')\n",
    "library(data.table)\n",
    "PP2A_het_controles_129134852_AMR <- fread(\"PP2A_het_controles_129134852_AMR.csv\", header =T)\n",
    "PP2A_het_controles_129134852_AMR\n",
    "temp <- merge(covs, PP2A_het_controles_129134852_AMR, by=\"IID\")\n",
    "head(temp)\n",
    "write.table(temp, file = \"PP2A_het_controles_129134852_AMR_extracted.txt\", quote = F, row.names = F, sep = \"\\t\")\n",
    "PP2A_het_controles_129134852_AMR_extracted <- fread(\"PP2A_het_controles_129134852_AMR_extracted.txt\", header =T)\n",
    "PP2A_het_controles_129134852_AMR_extracted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Save out results..!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -mu terra-02e7cd23 cp -r /home/jupyter/PP2A_GP2/ gs://fc-a2382a6f-59cf-4cf3-b70e-8cd83806d5e0\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Copying file:///home/jupyter/PP2A_GP2/reduced_coding_nonsynonymous.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_controles_129147457_EUR_extracted.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AJ_rare_0.05.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_CAS.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AFR.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AMR.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/reduced_coding_nonsynonymous_AFR.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/reduced_coding_nonsynonymous_AMR.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AAC.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AJ.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_rare_0.05.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_EAS.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AAC_rare_0.05.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_MDE.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_MDE.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_casos_129129094_AJ_extracted.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AAC.raw [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_SAS.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AAC_release3.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_controles_129147457_AJ_extracted.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_hom_casos_129147457_AJ.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_controles_129147457_AMR_extracted.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AJ_rare_0.05.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_CAS.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_ACC.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_casos_129147457_AFR.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_casos_129129094_AJ.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_casos_129134852_AMR.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_SAS_rare_0.05.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_CAS_release3.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AFR.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_CAS.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AFR.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_SAS.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AJ.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_CAS.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_casos_129134852_AMR_extracted.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_MDE.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_EUR.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AJ_rare_0.05.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_EAS.assoc.adjusted [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AJ_release3.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AMR_rare_0.05.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_MDE.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AJ_release3.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_CAS_release3.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AAC_release3.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AMR.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_casos_129147457_AJ_extracted.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AJ.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AAC.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/reduced_coding_nonsynonymous_AJ.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_hom_casos_129147457_AJ_extracted.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AFR_rare_0.05.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AFR.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AJ.assoc [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AFR.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AFR.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_EAS.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_EAS_release3.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_129147457_casos_EUR.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_MDE.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AFR_release3.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AMR_release3.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AJ.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_EUR_release3.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_casos_129147457_AMR.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_SAS.raw [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_MDE_release3.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/reduced_coding_nonsynonymous_CAS.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_SAS.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AMR.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_casos_129147457_AFR_extracted.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_EUR.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_casos_129147457_AAC_extracted.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AAC.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AMR_rare_0.05.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AJ.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AMR_release3.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AFR_rare_0.05.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AFR.raw [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AJ_release3.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AFR_rare_0.05.bim [Content-Type=application/octet-stream]...\n",
      "==> NOTE: You are uploading one or more large file(s), which would run          \n",
      "significantly faster if you enable parallel composite uploads. This\n",
      "feature can be enabled by editing the\n",
      "\"parallel_composite_upload_threshold\" value in your .boto\n",
      "configuration file. However, note that if you do this large files will\n",
      "be uploaded as `composite objects\n",
      "<https://cloud.google.com/storage/docs/composite-objects>`_,which\n",
      "means that any user who downloads such objects will need to have a\n",
      "compiled crcmod installed (see \"gsutil help crcmod\"). This is because\n",
      "without a compiled crcmod, computing checksums on composite objects is\n",
      "so slow that gsutil disables downloads of composite objects.\n",
      "\n",
      "Copying file:///home/jupyter/PP2A_GP2/master_key_release3_final.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_CAS.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_MDE.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_MDE_release3.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_CAS_rare_0.05.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_controles_129147457_AFR.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_EAS.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AAC.assoc [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AFR.assoc [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_CAS.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_CAS.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AFR_release3.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_MDE.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AMR.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AMR.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AAC.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/reduced_coding_nonsynonymous_EAS.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_SAS.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_EAS_release3.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_SAS_release3.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AFR_rare_0.05.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_SAS.assoc.adjusted [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AAC.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AAC.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_EUR.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AMR.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AAC_rare_0.05.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AAC_rare_0.05.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_CAS.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AAC_rare_0.05.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_CAS.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AMR.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_CAS.raw [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_controles_129147457_AFR_extracted.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_EAS.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_casos_129147457_AJ.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AFR_release3.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_rare_0.05.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_casos_129147457_AMR_extracted.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AMR.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_SAS.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PODXL.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_controles_129134882_AFR.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_EAS.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_SAS_release3.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_rare_0.05.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AJ.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_EAS.raw [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AMR.assoc.adjusted [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AJ_release3.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/SEX.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_CAS_release3.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AMR.raw [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_CAS_release3.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_controles_129134852_AMR.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_SAS.assoc [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_MDE.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_controles_129147457_AMR.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_EUR_release3.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_EAS_release3.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AFR.assoc.adjusted [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AMR_rare_0.05.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_CAS.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_controles_129147457_AAC.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AFR.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AJ.assoc.adjusted [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AFR.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_EAS.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AJ.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_EAS.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AJ.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AMR.assoc [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AMR.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_129147457_controles_EUR.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_EAS.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_controles_129134852_AMR_extracted.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_SAS_release3.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_CAS.assoc [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/reduced_coding_nonsynonymous_MDE.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PHENO.txt [Content-Type=text/plain]...    \n",
      "Copying file:///home/jupyter/PP2A_GP2/PTPA_het_129147457_controles_extracted.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_casos_129147457_AAC.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AMR.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_MDE.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AAC_release3.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AAC.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_EAS.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A.assoc [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_EAS_release3.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AAC.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_CAS.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AFR.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_MDE_rare_0.05.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AAC.assoc.adjusted [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_EAS.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_controles_129134882_AAC.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AJ.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AFR.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AJ.raw [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_MDE.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_SAS.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_controles_129129094_AJ.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_MDE.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_casos_129147457_EUR.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_EUR.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_EUR.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AJ.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/GP2_AJ_comphet.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_SAS.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_MDE_release3.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AMR.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_SAS.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/GP2_AMR_comphet.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_SAS.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_controles_129147457_EUR.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_SAS.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_EUR_release3.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_controles_129147457_AJ.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AFR_release3.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AFR.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AAC.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AMR_release3.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_MDE_release3.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AMR_release3.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_SAS_release3.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_CAS.assoc.adjusted [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A.assoc.adjusted [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_controles_129134882_AFR_extracted.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A.raw [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AJ_rare_0.05.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AAC.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PODXL.assoc [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_EAS.assoc [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_controles_129147457_AAC_extracted.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_EUR_release3.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_EAS_rare_0.05.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_controles_129134882_AAC_extracted.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AAC.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_SAS.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_EAS.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AJ.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/reduced_coding_nonsynonymous_SAS.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_casos_129147457_EUR_extracted.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AMR_rare_0.05.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_rare_0.05.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AAC_release3.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/reduced_coding_nonsynonymous_AAC.txt [Content-Type=text/plain]...\n",
      "\\ [242/242 files][  6.1 GiB/  6.1 GiB] 100% Done  90.9 MiB/s ETA 00:00:00       \n",
      "Operation completed over 242 objects/6.1 GiB.                                    \n"
     ]
    }
   ],
   "source": [
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR} {WORKSPACE_BUCKET}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.12"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {
    "height": "248.972px",
    "width": "263.062px"
   },
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "268.372px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
