{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# PP2A - Single gene analysis in GP2 data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Description\n",
    "\n",
    "Using \"individual level data\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 0. Getting Started\n",
    "\n",
    "- Loading Python libraries\n",
    "- Defining functions\n",
    "- Installing packages\n",
    "\n",
    "### 1. Copy data from workspace to cloud environment\n",
    "\n",
    "### 2. Extract PP2A\n",
    "\n",
    "### 3. Annotate PP2A variants\n",
    "\n",
    "### 4. Extract coding/non-syn variants\n",
    "\n",
    "### 5. Calculate frequency in cases versus controls\n",
    "\n",
    "### 6. Calculate frequency (homozygotes) in cases versus controls\n",
    "\n",
    "### 7. Save out results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Getting Started"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Loading Python libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Use the os package to interact with the environment\n",
    "import os\n",
    "\n",
    "# Bring in Pandas for Dataframe functionality\n",
    "import pandas as pd\n",
    "\n",
    "# Numpy for basics\n",
    "import numpy as np\n",
    "\n",
    "# Use StringIO for working with file contents\n",
    "from io import StringIO\n",
    "\n",
    "# Enable IPython to display matplotlib graphs\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline\n",
    "\n",
    "# Enable interaction with the FireCloud API\n",
    "from firecloud import api as fapi\n",
    "\n",
    "# Import the iPython HTML rendering for displaying links to Google Cloud Console\n",
    "from IPython.core.display import display, HTML\n",
    "\n",
    "# Import urllib modules for building URLs to Google Cloud Console\n",
    "import urllib.parse\n",
    "\n",
    "# BigQuery for querying data\n",
    "from google.cloud import bigquery\n",
    "\n",
    "#Import Sys\n",
    "import sys as sys"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Defining functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Utility routine for printing a shell command before executing it\n",
    "def shell_do(command):\n",
    "    print(f'Executing: {command}', file=sys.stderr)\n",
    "    !$command\n",
    "    \n",
    "def shell_return(command):\n",
    "    print(f'Executing: {command}', file=sys.stderr)\n",
    "    output = !$command\n",
    "    return '\\n'.join(output)\n",
    "\n",
    "# Utility routine for printing a query before executing it\n",
    "def bq_query(query):\n",
    "    print(f'Executing: {query}', file=sys.stderr)\n",
    "    return pd.read_gbq(query, project_id=BILLING_PROJECT_ID, dialect='standard')\n",
    "\n",
    "# Utility routine for display a message and a link\n",
    "def display_html_link(description, link_text, url):\n",
    "    html = f'''\n",
    "    <p>\n",
    "    </p>\n",
    "    <p>\n",
    "    {description}\n",
    "    <a target=_blank href=\"{url}\">{link_text}</a>.\n",
    "    </p>\n",
    "    '''\n",
    "\n",
    "    display(HTML(html))\n",
    "\n",
    "# Utility routines for reading files from Google Cloud Storage\n",
    "def gcs_read_file(path):\n",
    "    \"\"\"Return the contents of a file in GCS\"\"\"\n",
    "    contents = !gsutil -u {BILLING_PROJECT_ID} cat {path}\n",
    "    return '\\n'.join(contents)\n",
    "    \n",
    "def gcs_read_csv(path, sep=None):\n",
    "    \"\"\"Return a DataFrame from the contents of a delimited file in GCS\"\"\"\n",
    "    return pd.read_csv(StringIO(gcs_read_file(path)), sep=sep, engine='python')\n",
    "\n",
    "# Utility routine for displaying a message and link to Cloud Console\n",
    "def link_to_cloud_console_gcs(description, link_text, gcs_path):\n",
    "    url = '{}?{}'.format(\n",
    "        os.path.join('https://console.cloud.google.com/storage/browser',\n",
    "                     gcs_path.replace(\"gs://\",\"\")),\n",
    "        urllib.parse.urlencode({'userProject': BILLING_PROJECT_ID}))\n",
    "\n",
    "    display_html_link(description, link_text, url)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Set paths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Billing and Workspace\n",
      "Workspace Name: PP2A\n",
      "Billing Project: terra-02e7cd23\n",
      "Workspace Bucket, where you can upload and download data: gs://fc-a2382a6f-59cf-4cf3-b70e-8cd83806d5e0\n",
      "\n",
      "GP2 v3.0\n",
      "Path to GP2 v3.0 Clinical Data: gs://gp2tier2/release3_31102022/clinical_data\n",
      "Path to GP2 v3.0 Raw Genotype Data: gs://gp2tier2/release3_31102022/raw_genotypes\n",
      "Path to GP2 v3.0 Imputed Genotype Data: gs://gp2tier2/release3_31102022/imputed_genotypes\n",
      "AMP-PD v3.0\n",
      "Path to AMP-PD v3.0 Clinical Data: gs://amp-pd-data/releases/2022_v3release_1115/clinical\n",
      "Path to AMP-PD v3.0 WGS Data: gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink\n",
      "Path to AMP-PD v3.0 WGS Data: gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Set up billing project and data path variables\n",
    "BILLING_PROJECT_ID = os.environ['GOOGLE_PROJECT']\n",
    "WORKSPACE_NAMESPACE = os.environ['WORKSPACE_NAMESPACE']\n",
    "WORKSPACE_NAME = os.environ['WORKSPACE_NAME']\n",
    "WORKSPACE_BUCKET = os.environ['WORKSPACE_BUCKET']\n",
    "\n",
    "WORKSPACE_ATTRIBUTES = fapi.get_workspace(WORKSPACE_NAMESPACE, WORKSPACE_NAME).json().get('workspace',{}).get('attributes',{})\n",
    "\n",
    "## Print the information to check we are in the proper release and billing \n",
    "## This will be different for you, the user, depending on the billing project your workspace is on\n",
    "print('Billing and Workspace')\n",
    "print(f'Workspace Name: {WORKSPACE_NAME}')\n",
    "print(f'Billing Project: {BILLING_PROJECT_ID}')\n",
    "print(f'Workspace Bucket, where you can upload and download data: {WORKSPACE_BUCKET}')\n",
    "print('')\n",
    "\n",
    "\n",
    "## GP2 v3.0\n",
    "## Explicitly define release v3.0 path \n",
    "GP2_RELEASE_PATH = 'gs://gp2tier2/release3_31102022'\n",
    "GP2_CLINICAL_RELEASE_PATH = f'{GP2_RELEASE_PATH}/clinical_data'\n",
    "GP2_RAW_GENO_PATH = f'{GP2_RELEASE_PATH}/raw_genotypes'\n",
    "GP2_IMPUTED_GENO_PATH = f'{GP2_RELEASE_PATH}/imputed_genotypes'\n",
    "print('GP2 v3.0')\n",
    "print(f'Path to GP2 v3.0 Clinical Data: {GP2_CLINICAL_RELEASE_PATH}')\n",
    "print(f'Path to GP2 v3.0 Raw Genotype Data: {GP2_RAW_GENO_PATH}')\n",
    "print(f'Path to GP2 v3.0 Imputed Genotype Data: {GP2_IMPUTED_GENO_PATH}')\n",
    "\n",
    "\n",
    "## AMP-PD v3.0\n",
    "## Explicitly define release v3.0 path \n",
    "AMP_RELEASE_PATH = 'gs://amp-pd-data/releases/2022_v3release_1115'\n",
    "AMP_CLINICAL_RELEASE_PATH = f'{AMP_RELEASE_PATH}/clinical'\n",
    "\n",
    "AMP_WGS_RELEASE_PATH = 'gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS'\n",
    "AMP_WGS_RELEASE_PLINK_PATH = os.path.join(AMP_WGS_RELEASE_PATH, 'plink')\n",
    "AMP_WGS_RELEASE_PLINK_PFILES = os.path.join(AMP_WGS_RELEASE_PLINK_PATH, 'pfiles')\n",
    "\n",
    "print('AMP-PD v3.0')\n",
    "print(f'Path to AMP-PD v3.0 Clinical Data: {AMP_CLINICAL_RELEASE_PATH}')\n",
    "print(f'Path to AMP-PD v3.0 WGS Data: {AMP_WGS_RELEASE_PLINK_PATH}')\n",
    "print(f'Path to AMP-PD v3.0 WGS Data: {AMP_WGS_RELEASE_PLINK_PFILES}')\n",
    "print('')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Install packages"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Install Plink 1.9 and Plink 2.0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Plink1.9 is already installed in /home/jupyter/tools/\n",
      "Plink2 is already installed in /home/jupyter/tools/\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "mkdir -p ~/tools\n",
    "cd ~/tools\n",
    "\n",
    "if test -e /home/jupyter/tools/plink; then\n",
    "echo \"Plink1.9 is already installed in /home/jupyter/tools/\"\n",
    "\n",
    "else\n",
    "echo -e \"Downloading plink \\n    -------\"\n",
    "wget -N http://s3.amazonaws.com/plink1-assets/plink_linux_x86_64_20190304.zip \n",
    "unzip -o plink_linux_x86_64_20190304.zip\n",
    "echo -e \"\\n plink downloaded and unzipped in /home/jupyter/tools \\n \"\n",
    "\n",
    "fi\n",
    "\n",
    "\n",
    "if test -e /home/jupyter/tools/plink2; then\n",
    "echo \"Plink2 is already installed in /home/jupyter/tools/\"\n",
    "\n",
    "else\n",
    "echo -e \"Downloading plink2 \\n    -------\"\n",
    "wget -N https://s3.amazonaws.com/plink2-assets/alpha3/plink2_linux_avx2_20220603.zip\n",
    "unzip -o plink2_linux_avx2_20220603.zip\n",
    "echo -e \"\\n plink2 downloaded and unzipped in /home/jupyter/tools \\n \"\n",
    "\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annovar\n",
      "annovar.latest.tar.gz\n",
      "LICENSE\n",
      "plink\n",
      "plink2\n",
      "plink2_linux_avx2_20220603.zip\n",
      "plink_linux_x86_64_20190304.zip\n",
      "prettify\n",
      "toy.map\n",
      "toy.ped\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls /home/jupyter/tools/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Remote restrictions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "# chmod plink 1.9 \n",
    "chmod u+x /home/jupyter/tools/plink"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "# chmod plink 2.0\n",
    "chmod u+x /home/jupyter/tools/plink2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Install ANNOVAR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annovar is already installed in /home/jupyter/tools/\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Install ANNOVAR:\n",
    "# https://www.openbioinformatics.org/annovar/annovar_download_form.php\n",
    "\n",
    "if test -e /home/jupyter/tools/annovar; then\n",
    "\n",
    "echo \"annovar is already installed in /home/jupyter/tools/\"\n",
    "else\n",
    "echo \"annovar is not installed\"\n",
    "cd /home/jupyter/tools/\n",
    "\n",
    "wget http://www.openbioinformatics.org/annovar/download/0wgxR2rIVP/annovar.latest.tar.gz\n",
    "\n",
    "tar xvfz annovar.latest.tar.gz\n",
    "\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annovar\n",
      "annovar.latest.tar.gz\n",
      "LICENSE\n",
      "plink\n",
      "plink2\n",
      "plink2_linux_avx2_20220603.zip\n",
      "plink_linux_x86_64_20190304.zip\n",
      "prettify\n",
      "toy.map\n",
      "toy.ped\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls /home/jupyter/tools/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Install ANNOVAR: Download sources of annotation "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "NOTICE: Web-based checking to see whether ANNOVAR new version is available ... Done\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_refGene.txt.gz ... OK\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_refGeneMrna.fa.gz ... OK\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_refGeneVersion.txt.gz ... OK\n",
      "NOTICE: Uncompressing downloaded files\n",
      "NOTICE: Finished downloading annotation files for hg38 build version, with files saved at the 'humandb' directory\n",
      "NOTICE: Web-based checking to see whether ANNOVAR new version is available ... Done\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_clinvar_20140902.txt.gz ... OK\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_clinvar_20140902.txt.idx.gz ... OK\n",
      "NOTICE: Uncompressing downloaded files\n",
      "NOTICE: Finished downloading annotation files for hg38 build version, with files saved at the 'humandb' directory\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "cd /home/jupyter/tools/annovar/\n",
    "\n",
    "perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar refGene humandb/\n",
    "perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar clinvar_20140902 humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb cytoBand humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar ensGene humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar exac03 humandb/ \n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar avsnp147 humandb/ \n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar dbnsfp30a humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar gnomad211_genome humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar ljb26_all humandb/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annotate_variation.pl\n",
      "coding_change.pl\n",
      "convert2annovar.pl\n",
      "example\n",
      "humandb\n",
      "retrieve_seq_from_fasta.pl\n",
      "table_annovar.pl\n",
      "variants_reduction.pl\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls /home/jupyter/tools/annovar/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Copy data from GP2 bucket to workspace"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Making a working directory\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: mkdir -p /home/jupyter/PP2A_GP2/\n"
     ]
    }
   ],
   "source": [
    "# Make a directory\n",
    "print(\"Making a working directory\")\n",
    "WORK_DIR = f'/home/jupyter/PP2A_GP2/'\n",
    "shell_do(f'mkdir -p {WORK_DIR}') # f' significa f-string - contiene expresiones para ejecutar el codigo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -u terra-02e7cd23 ls gs://gp2tier2/release3_31102022/imputed_genotypes\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gs://gp2tier2/release3_31102022/imputed_genotypes/AAC/\r\n",
      "gs://gp2tier2/release3_31102022/imputed_genotypes/AFR/\r\n",
      "gs://gp2tier2/release3_31102022/imputed_genotypes/AJ/\r\n",
      "gs://gp2tier2/release3_31102022/imputed_genotypes/AMR/\r\n",
      "gs://gp2tier2/release3_31102022/imputed_genotypes/CAS/\r\n",
      "gs://gp2tier2/release3_31102022/imputed_genotypes/EAS/\r\n",
      "gs://gp2tier2/release3_31102022/imputed_genotypes/EUR/\r\n",
      "gs://gp2tier2/release3_31102022/imputed_genotypes/MDE/\r\n",
      "gs://gp2tier2/release3_31102022/imputed_genotypes/SAS/\r\n"
     ]
    }
   ],
   "source": [
    "# Check directory where AMP-PD data is\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} ls {GP2_IMPUTED_GENO_PATH}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -u terra-02e7cd23 -m cp -r gs://gp2tier2/release3_31102022/imputed_genotypes/EUR/chr9* /home/jupyter/PP2A_GP2/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Copying gs://gp2tier2/release3_31102022/imputed_genotypes/EUR/chr9_EUR_release3.psam...\n",
      "Copying gs://gp2tier2/release3_31102022/imputed_genotypes/EUR/chr9_EUR_release3.log...\n",
      "Copying gs://gp2tier2/release3_31102022/imputed_genotypes/EUR/chr9_EUR_release3.pgen...\n",
      "Copying gs://gp2tier2/release3_31102022/imputed_genotypes/EUR/chr9_EUR_release3.pvar...\n",
      "\\ [4/4 files][  4.4 GiB/  4.4 GiB] 100% Done  38.5 MiB/s ETA 00:00:00           \n",
      "Operation completed over 4 objects/4.4 GiB.                                      \n"
     ]
    }
   ],
   "source": [
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp -r {GP2_IMPUTED_GENO_PATH}/EUR/chr9* {WORK_DIR}')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Create a covariate file with GP2 data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -u terra-02e7cd23 ls gs://gp2tier2/release3_31102022/clinical_data\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gs://gp2tier2/release3_31102022/clinical_data/\r\n",
      "gs://gp2tier2/release3_31102022/clinical_data/master_key_release3_final.csv\r\n",
      "gs://gp2tier2/release3_31102022/clinical_data/release3_31102022_data_dictionary.csv\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -u terra-02e7cd23 -m cp -r gs://gp2tier2/release3_31102022/clinical_data/master_key_release3_final.csv /home/jupyter/PP2A_GP2/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Copying gs://gp2tier2/release3_31102022/clinical_data/master_key_release3_final.csv...\n",
      "/ [1/1 files][  1.6 MiB/  1.6 MiB] 100% Done                                    \n",
      "Operation completed over 1 objects/1.6 MiB.                                      \n"
     ]
    }
   ],
   "source": [
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} ls {GP2_CLINICAL_RELEASE_PATH}')\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp -r {GP2_CLINICAL_RELEASE_PATH}/master_key_release3_final.csv {WORK_DIR}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['GP2ID', 'GP2sampleID', 'manifest_id', 'phenotype', 'pheno_for_qc',\n",
       "       'other_pheno', 'sex_for_qc', 'age', 'age_of_onset', 'age_at_diagnosis',\n",
       "       'age_at_death', 'race_for_qc', 'family_history_for_qc', 'region_for_qc',\n",
       "       'study', 'pruned', 'pruned_reason', 'label', 'related'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cov = pd.read_csv(f'{WORK_DIR}/master_key_release3_final.csv')\n",
    "cov.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>SEX</th>\n",
       "      <th>AGE</th>\n",
       "      <th>PHENO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SYNAPS-KZ_000001_s1</td>\n",
       "      <td>SYNAPS-KZ_000001_s1</td>\n",
       "      <td>2</td>\n",
       "      <td>64.0</td>\n",
       "      <td>Control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SYNAPS-KZ_000002_s1</td>\n",
       "      <td>SYNAPS-KZ_000002_s1</td>\n",
       "      <td>1</td>\n",
       "      <td>53.0</td>\n",
       "      <td>Control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SYNAPS-KZ_000003_s1</td>\n",
       "      <td>SYNAPS-KZ_000003_s1</td>\n",
       "      <td>2</td>\n",
       "      <td>59.0</td>\n",
       "      <td>Control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SYNAPS-KZ_000004_s1</td>\n",
       "      <td>SYNAPS-KZ_000004_s1</td>\n",
       "      <td>1</td>\n",
       "      <td>57.0</td>\n",
       "      <td>Control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SYNAPS-KZ_000005_s1</td>\n",
       "      <td>SYNAPS-KZ_000005_s1</td>\n",
       "      <td>1</td>\n",
       "      <td>59.0</td>\n",
       "      <td>Control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16852</th>\n",
       "      <td>PPMI_000789_s1</td>\n",
       "      <td>PPMI_000789_s1</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16853</th>\n",
       "      <td>PPMI_000781_s1</td>\n",
       "      <td>PPMI_000781_s1</td>\n",
       "      <td>1</td>\n",
       "      <td>57.0</td>\n",
       "      <td>Other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16854</th>\n",
       "      <td>PPMI_000818_s1</td>\n",
       "      <td>PPMI_000818_s1</td>\n",
       "      <td>1</td>\n",
       "      <td>60.0</td>\n",
       "      <td>Other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16855</th>\n",
       "      <td>PPMI_000841_s1</td>\n",
       "      <td>PPMI_000841_s1</td>\n",
       "      <td>2</td>\n",
       "      <td>68.0</td>\n",
       "      <td>Other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16856</th>\n",
       "      <td>PPMI_000808_s1</td>\n",
       "      <td>PPMI_000808_s1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Other</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>16857 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                       FID                  IID  SEX   AGE    PHENO\n",
       "0      SYNAPS-KZ_000001_s1  SYNAPS-KZ_000001_s1    2  64.0  Control\n",
       "1      SYNAPS-KZ_000002_s1  SYNAPS-KZ_000002_s1    1  53.0  Control\n",
       "2      SYNAPS-KZ_000003_s1  SYNAPS-KZ_000003_s1    2  59.0  Control\n",
       "3      SYNAPS-KZ_000004_s1  SYNAPS-KZ_000004_s1    1  57.0  Control\n",
       "4      SYNAPS-KZ_000005_s1  SYNAPS-KZ_000005_s1    1  59.0  Control\n",
       "...                    ...                  ...  ...   ...      ...\n",
       "16852       PPMI_000789_s1       PPMI_000789_s1    2   NaN    Other\n",
       "16853       PPMI_000781_s1       PPMI_000781_s1    1  57.0    Other\n",
       "16854       PPMI_000818_s1       PPMI_000818_s1    1  60.0    Other\n",
       "16855       PPMI_000841_s1       PPMI_000841_s1    2  68.0    Other\n",
       "16856       PPMI_000808_s1       PPMI_000808_s1    1   NaN    Other\n",
       "\n",
       "[16857 rows x 5 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cov_reduced = cov[['GP2sampleID','GP2sampleID','sex_for_qc', 'age', 'phenotype']]\n",
    "cov_reduced.columns = ['FID','IID', 'SEX','AGE','PHENO']\n",
    "cov_reduced"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "conditions = [(cov_reduced['PHENO'] == \"Case\"),\n",
    "    (cov_reduced['PHENO'] == \"Control\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.7/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  \n"
     ]
    }
   ],
   "source": [
    "choices = [2,1]\n",
    "cov_reduced['PHENOTYPE'] = np.select(conditions, choices, default=-9).astype(np.int64)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### cov_reduced.reset_index(inplace=True)\n",
    "cov_reduced.drop(columns=[\"index\"], inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.7/site-packages/pandas/core/frame.py:4913: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  errors=errors,\n"
     ]
    }
   ],
   "source": [
    "cov_reduced.drop(columns=['PHENO'], inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "sex = cov_reduced[['FID','IID','SEX']]\n",
    "sex.to_csv(f'{WORK_DIR}/SEX.txt',index=False, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "pheno = cov_reduced[['FID','IID','PHENOTYPE']]\n",
    "pheno.to_csv(f'{WORK_DIR}/PHENO.txt',index=False, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extract PP2A"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a3.3LM AVX2 Intel (3 Jun 2022)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2022 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to pheno_PP2A_EUR.log.\n",
      "Options in effect:\n",
      "  --chr 9\n",
      "  --from-bp 129110950\n",
      "  --make-bed\n",
      "  --out pheno_PP2A_EUR\n",
      "  --pfile chr9_EUR_release3\n",
      "  --to-bp 129148946\n",
      "\n",
      "Start time: Thu Jun  1 14:03:35 2023\n",
      "14998 MiB RAM detected; reserving 7499 MiB for main workspace.\n",
      "Using up to 4 compute threads.\n",
      "11002 samples (4271 females, 6731 males; 11002 founders) loaded from\n",
      "chr9_EUR_release3.psam.\n",
      "859295 variants loaded from chr9_EUR_release3.pvar.\n",
      "1 binary phenotype loaded (6852 cases, 3875 controls).\n",
      "328 variants remaining after main filters.\n",
      "Writing pheno_PP2A_EUR.fam ... done.\n",
      "Writing pheno_PP2A_EUR.bim ... done.\n",
      "Writing pheno_PP2A_EUR.bed ... 0%\b\bdone.\n",
      "End time: Thu Jun  1 14:03:36 2023\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "# PODXL: gene positions on hg38 (from https://useast.ensembl.org/index.html)\n",
    "/home/jupyter/tools/plink2 \\\n",
    "--pfile chr9_EUR_release3 \\\n",
    "--chr 9 \\\n",
    "--from-bp 129110950  \\\n",
    "--to-bp 129148946 \\\n",
    "--make-bed \\\n",
    "--out pheno_PP2A_EUR"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Visualize plink files bim, fam and bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9\tchr9:129110955:G:C\t0\t129110955\tC\tG\n",
      "9\tchr9:129110977:A:C\t0\t129110977\tC\tA\n",
      "9\tchr9:129111059:G:A\t0\t129111059\tA\tG\n",
      "9\tchr9:129111157:A:T\t0\t129111157\tT\tA\n",
      "9\tchr9:129111178:T:C\t0\t129111178\tC\tT\n",
      "9\tchr9:129111184:G:C\t0\t129111184\tC\tG\n",
      "9\tchr9:129111323:C:G\t0\t129111323\tG\tC\n",
      "9\tchr9:129111355:C:G\t0\t129111355\tG\tC\n",
      "9\tchr9:129111392:G:C\t0\t129111392\tC\tG\n",
      "9\tchr9:129111663:T:C\t0\t129111663\tC\tT\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Visualize bim file\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "head pheno_PP2A_EUR.bim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\tAPGS_000001_s1\t0\t0\t1\t2\n",
      "0\tAPGS_000002_s1\t0\t0\t1\t2\n",
      "0\tAPGS_000003_s1\t0\t0\t1\t2\n",
      "0\tAPGS_000005_s1\t0\t0\t1\t2\n",
      "0\tAPGS_000006_s1\t0\t0\t1\t2\n",
      "0\tAPGS_000007_s1\t0\t0\t1\t2\n",
      "0\tAPGS_000009_s1\t0\t0\t1\t2\n",
      "0\tAPGS_000010_s1\t0\t0\t1\t2\n",
      "0\tAPGS_000012_s1\t0\t0\t2\t2\n",
      "0\tAPGS_000013_s1\t0\t0\t1\t2\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Visualize fam file\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "head pheno_PP2A_EUR.fam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "l\u001b\u0001��������������������������߻�����������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������/��������ߺ����{�����������������������������������;���������������������������������������������/��������ߺ����{�����������������������������������;�������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������/��������ߺ����{�����������������������������������;�������������������������������������������������������������������������������������������������������\u0000\u0000\u0000\u0000\u0000\u0004\u0000\u0000\u0000\u0000�\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0010\u0002 \u0000\u0000\u0004\u0000\u0000\u0000\u0000\f",
      "\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0002\u0000�\u0010\u0000\b\u0000\u0000\u0000@\u0000\u0000\u0000\b\u0000\u0000\u0000\u0000\b\u0000\u0000\u0000\u0000\u0000@\u0000\u0000\u0000\u0001\u0000\u0000\u0002\u0000\u0000\b\u0010\u0000\u0000\u0000\u0000\u0000\u0000@\u0000\u0002\u0000\u0000\u0000\u0000\u0000�������������������������������������������/��������ߺ����{�����������������������������������;��������������������������������������������������������������������������������������������������������������������������w�������������������������������������������������{���������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������\u0000\u0000\u0002\u0000\u0000\u0000\u0000�\u0000\b�\"��\u0002\u0000\u0000\n",
      "\u0000\u0000 \u0000\u0002\u0000\u0000\b\u0000\u0000\u0000\u0000� \u0000\u0000�\b\u0000\u0000\u0002\f",
      "\u0000\b\u0000\u0002�\u0000\u0000�\b0\u0000\u0002\u0000�  \b��\u0002H\u0000�\u0000\b�\u0000\u0002��\u0002\b\u0000� \b��\u0000�\u0002\u0000\"\b0(\b\u0000\u0000\u0000\u00000\u0002�\u0003ʀ\b\u0000\u0000 *���������������/�����������������������������������o�����������/���������������*����������x���h�����z8���������+o￟��:���ϫ������������?�������+���\u000f������,�����n>���;����.�����;����������������������������������������o����������������������������������������������������������������x���h�����z8���������+o￟��:���ϫ������������?������.+�{�\u000f������,�����n>���;����.����);��������������������������������������������������������������������������������������������������������*���������������/�����������������������������������o����������ﯯ��������������*�����������������������~������㿿����������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������/��������ߺ����{�����������������������������������;����K'+���������+J��\u0006.뻢����`ڪ�����*<��\u000e�;���*��,�Ꞔ�޺����j�����\"�������������*Ȫ����������b���;ʪ������������������������������������������������������������������������������������������������������\u0000\u0000\u0000\u0000\u0000\u0004\u0000\u0000\u0000\u0000�\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0010\u0002\u0000\u0000\u0000\u0004\u0000\u0000\u0000\u0000\f",
      "\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0002\u0000�\u0010\u0000\u0000\u0000\u0000\u0000@\u0000\u0000\u0000\b\u0000\u0000\u0000\u0000\b\u0000\u0000\u0000\u0000\u0000@\u0000\u0000\u0000\u0001\u0000\u0000\u0002\u0000\u0000\b\u0010\u0000\u0000\u0000\u0000\u0000\u0000@\u0000\u0002\u0000\u0000\u0000\u0000\u0000����������������������������������������������������������������������������������������������������K3����������J��\u0006��������b�������������;�g����.����������������������������������������j������������������������������������������������������������������������������������������������������������������������������������������������������������������������{�����������������������������������������x���h�����{8���������/￟��:���ϫ������������?���������������������n�����;����������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������\u0000\u0000\u0000\u0000\u0000\u0004\u0000\u0000\u0000\u0000�\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0010\u0002\u0000\u0000\u0000\u0004\u0000\u0000\u0000\u0000\f",
      "\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0002\u0000�\u0010\u0000\u0000\u0000\u0000\u0000@\u0000\u0000\u0000\b\u0000\u0000\u0000\u0000\b\u0000\u0000\u0000\u0000\u0000@\u0000\u0000\u0000\u0001\u0000\u0000\u0002\u0000\u0000\b\u0010\u0000\u0000\u0000\u0000\u0000\u0000@\u0000\u0002\u0000\u0000\u0000\u0000\u0000�����������������������������������������������������������������������������������������������������K3����������J��\u0006��������b�������������;�g����.����������������������������������������j�������\u0000\u0000\u0000\u0000\u0000\u0004\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0010\u0002\u0000\u0000\u0000\u0004\u0000\u0000\u0000\u0000\f",
      "\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0002\u0000�\u0010\u0000\u0000\u0000\u0000\u0000@\u0000\u0000\u0000\b\u0000\u0000\u0000\u0000\b\u0000\u0000\u0000\u0000\u0000@\u0000\u0000\u0000\u0001\u0000\u0000\u0000\u0000\u0000\b\u0010\u0000\u0000\u0000\u0000\u0000\u0000@\u0000\u0002\u0000\u0000\u0000\u0000\u0000�~���h�����{8���������/o���������﻿�����������?�����������������שׂ����n�����;����������������������������������������������������������������������������������������������������������������������������������������������������������������/��������ߺ����{�����������������������������������;����K3����������J��\u0006��������b�������������;�g����.����������������������������������������j��������x���h�����z8���������+o￟��:���ϫ������������?�������+���\u000f������,�����n>���;��������;���*���������������/�����������������������������������o����������ﯯ��������������*����������������������������������������������������������������������������������������������������������������������������~������㿿������������������������������������������������������������������������������K#+���������+J��\u0006.뻢����`ꪚ����*<��\u000e�;���*��,�Ꞔ�ߺ����j�����\"�������������*Ȫ����������b���;ʪ��x���h�����z8���������+o￟��:���ϫ������������?�������+���\u000f������,�����n>���;��������;���*���������������/������������������������������������o����������ﯿ��������������*����������x���h�����z8���������+o￟��:���ϫ������������?�������+���\u000f������,�����n>���;��������;����x���h�����z8���������+o￟��:���ϫ������������?�������+���\u000f������,�����n>���;��������;�����������������~������㿿�����������������������������������������������������������������������������������������������������������������������������������������{�����������������������������������������x���h�����z8���������+o￟��:���ϫ������������?�������+���\u000f������,�����n>���;��������;����x���h�����z8���������+o￟��:���ϫ������������?�������+���\u000f������,�����n>���;��������;�������ʷ��������������꫿�������������>����.������֯ߺ����{�����;�����������������������������;������������������������������������������/������ޯߺ����{�����������������������������������;���������������������������������������������������������������������������������������������������������x���h�����z8���������+o￟��:���ϫ������������?�������+���\u000f������,�����n>���;��������;����o��������������/������������������������g�������������������������������������������������~��������x���h�����{8���������/o￟��:���ϫ������������?���������������������n�����;����������������x���h�����z8���������+o￟��:���ϫ������������?�������+���\u000f������,�����n>���;��������;����x���h�����z8���������+o￟��:���ϫ������������?�������+���\u000f������,�����n>���;��������;���*���������������/������������������������������������o����������ﯯ��������������*������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������K#+���������+J��\u0006.뻢����`ꪚ����*<��\u000e�;���*��,�Ꞔ�ߺ����j�����\"�������������*Ȫ����������b���;ʪ��\u0003\n",
      "\u0002\u0004\f",
      "(\u0000��0�\n",
      "�(�\u0000(�*\u0002(��\u0000��+\u001a3�(\u0002�*\f",
      "\b\u000b",
      "\f",
      "����/\u0000��(�+\u000b",
      "���\u0000��(��,\n",
      "��� ��,� ���l\bŰ�\u0000 *.��\u0010\u001c",
      "�� \u0003�b��\n",
      "\"��(������������������������������������������������������������������������������������������������������K#+���������+J��\u0006.뻢����`ꪚ����*<��\u000e�;���*��,�Ꞔ�ߺ����j�����\"�������������*Ȫ����������b���;ʪ��K3����������J��\u0006��������b�������������;�g����.����������������������������������������j�������������������������������������������������/������ޯߺ����{�����������������������������������;���������������������������������������������/������ޯߺ����{�����������������������������������;��������������������������������������������������������������������������������������������������������\u0000\u0000\u0000\u0000\u0000\u0004\u0000\u0000\u0000\u0000�\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0010\u0002\u0000\u0000\u0000\u0004\u0000\u0000\u0000\u0000\f",
      "\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0002\u0000�\u0010\u0000\u0000\u0000\u0000\u0000@\u0000\u0000\u0000\b\u0000\u0000\u0000\u0000\u0004\u0000\u0000\u0000\u0000\u0000@\u0000\u0000\u0000\u0000\u0000\u0000\u0002\u0000\u0000\b\u0010\u0000\u0000\u0000\u0000\u0000\u0000@\u0000\u0002\u0000\u0000\u0000\u0000\u0000�x���h�����z8���������+o￟��:���ϫ������������?�������+���\u000f������,�����n>���;��������;������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������\u0000\u0000\u0000\u0000\u0000\u0004\u0000\u0000\u0000\u0000�\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0010\u0002\u0000\u0000\u0000\u0004\u0000\u0000\u0000\u0000\f",
      "\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0002\u0000�\u0010\u0000\u0000\u0000\u0000\u0000@\u0000\u0000\u0000\b\u0000\u0000\u0000\u0000\u0004\u0000\u0000\u0000\u0000\u0000@\u0000\u0000\u0000\u0000\u0000\u0000\u0002\u0000\u0000\b\u0010\u0000\u0000\u0000\u0000\u0000\u0000@\u0000\u0002\u0000\u0000\u0000\u0000\u0000�����������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������x���h�����z8���������+o￟��:���ϫ������������?�������+���\u000f������,�����n>���;��������;����x���h�����z8���������+o￟��:���ϫ������������?�������+���\u000f������,�����n>���;��������;�������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������\u0000\u0003\n",
      "\u0002\u0004\f",
      "(\u0000��0�\n",
      "�(�\u0000(�*\u0002(��\u0000��*\u001a3�(\u0002�\"\f",
      "\b\u000b",
      "\f",
      "�\"��/\u0000��(�*\u000b",
      "���\u0000��(��,\n",
      "���\u0000��,� ���l\bư�\u0000\u0000*.��\u0010\f",
      "�� \u0003�b��\n",
      "\"��(*��������������/����뫺��:��Ϋ����������������������뾪������������>�����꺪��*+������������������������������������������o����������������������������������������������������������������x���h�����z8���������+o￟��:���ϫ������������?�������+���\u000f������,�����n>���;��������;����o��������������/������������������������g�������������������������������������������������~��������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������o��������������������������������������������������/������ޯߺ����{�����������������������������������;�������������������������������������������������������������������������������������������������������������������������������������������������������������������{����������������������������������������������������������������������������������������������������������������������������������������������x���h�����z8���������+o￟��:���ϫ������������?�������+���\u000f������,�����n>���;��������;����K3����������J��\u0006��������b�������������;�g����.����������������������������������������j��������\u0000\u0003\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Visualize fam file\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "head pheno_PP2A.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to pheno_PP2A_EUR.log.\n",
      "Options in effect:\n",
      "  --bfile pheno_PP2A_EUR\n",
      "  --out pheno_PP2A_EUR\n",
      "  --recode vcf-fid\n",
      "\n",
      "14998 MB RAM detected; reserving 7499 MB for main workspace.\n",
      "328 variants loaded from .bim file.\n",
      "11002 people (6731 males, 4271 females) loaded from .fam.\n",
      "10727 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 11002 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.997237.\n",
      "328 variants and 11002 people pass filters and QC.\n",
      "Among remaining phenotypes, 6852 are cases and 3875 are controls.  (275\n",
      "phenotypes are missing.)\n",
      "--recode vcf-fid to pheno_PP2A_EUR.vcf ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "# Turn binary files into VCF\n",
    "/home/jupyter/tools/plink \\\n",
    "--bfile pheno_PP2A_EUR \\\n",
    "--recode vcf-fid \\\n",
    "--out pheno_PP2A_EUR"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Annotate PP2A variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[bgzip] can't create pheno_PP2A.vcf.gz: File exists\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "export PATH=$PATH:/home/jupyter/tools/rvtests/third/tabix-0.2.6/\n",
    "\n",
    "### Bgzip and Tabix\n",
    "bgzip pheno_PP2A.vcf\n",
    "\n",
    "tabix -f -p vcf pheno_PP2A.vcf.gz\n",
    "tabix -f -p vcf pheno_PP2A.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 pheno_PP2A.vcf.gz > pheno_PP2A.annovar.avinput>\n",
      "NOTICE: Finished reading 335 lines from VCF file\n",
      "NOTICE: A total of 328 locus in VCF file passed QC threshold, representing 303 SNPs (226 transitions and 77 transversions) and 25 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 3333606 SNP genotypes (2486452 transitions and 847154 transversions) and 275050 indels/substitutions for 11002 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl pheno_PP2A.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile pheno_PP2A.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile pheno_PP2A.annovar.refGene -exonsort -nofirstcodondel pheno_PP2A.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to pheno_PP2A.annovar.refGene.variant_function, pheno_PP2A.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 328 unique variants in 328 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 6 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile pheno_PP2A.annovar pheno_PP2A.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to pheno_PP2A.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to pheno_PP2A.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 328 unique variants in 328 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 0\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to pheno_PP2A.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from pheno_PP2A.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to pheno_PP2A.annovar.hg38_multianno.vcf\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "### Annotate variants using ANNOVAR: https://annovar.openbioinformatics.org/en/latest/ \n",
    "perl /home/jupyter/tools/annovar/table_annovar.pl pheno_PP2A.vcf.gz /home/jupyter/tools/annovar/humandb/ -buildver hg38 \\\n",
    "-out pheno_PP2A.annovar \\\n",
    "-remove -protocol refGene,clinvar_20140902 \\\n",
    "-operation g,f \\\n",
    "--nopolish \\\n",
    "-nastring . \\\n",
    "-vcfinput"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extract coding and non-syn variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>Otherinfo11005</th>\n",
       "      <th>Otherinfo11006</th>\n",
       "      <th>Otherinfo11007</th>\n",
       "      <th>Otherinfo11008</th>\n",
       "      <th>Otherinfo11009</th>\n",
       "      <th>Otherinfo11010</th>\n",
       "      <th>Otherinfo11011</th>\n",
       "      <th>Otherinfo11012</th>\n",
       "      <th>Otherinfo11013</th>\n",
       "      <th>Otherinfo11014</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9</td>\n",
       "      <td>129110955</td>\n",
       "      <td>129110955</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>PTPA</td>\n",
       "      <td>NM_021131:c.-646G&gt;C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9</td>\n",
       "      <td>129110977</td>\n",
       "      <td>129110977</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>PTPA</td>\n",
       "      <td>NM_021131:c.-624A&gt;C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>9</td>\n",
       "      <td>129111059</td>\n",
       "      <td>129111059</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>PTPA</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>9</td>\n",
       "      <td>129111157</td>\n",
       "      <td>129111157</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>PTPA</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9</td>\n",
       "      <td>129111178</td>\n",
       "      <td>129111178</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>PTPA</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 11025 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Chr      Start        End Ref Alt Func.refGene Gene.refGene  \\\n",
       "0    9  129110955  129110955   G   C         UTR5         PTPA   \n",
       "1    9  129110977  129110977   A   C         UTR5         PTPA   \n",
       "2    9  129111059  129111059   G   A     intronic         PTPA   \n",
       "3    9  129111157  129111157   A   T     intronic         PTPA   \n",
       "4    9  129111178  129111178   T   C     intronic         PTPA   \n",
       "\n",
       "    GeneDetail.refGene ExonicFunc.refGene AAChange.refGene  ...  \\\n",
       "0  NM_021131:c.-646G>C                  .                .  ...   \n",
       "1  NM_021131:c.-624A>C                  .                .  ...   \n",
       "2                    .                  .                .  ...   \n",
       "3                    .                  .                .  ...   \n",
       "4                    .                  .                .  ...   \n",
       "\n",
       "  Otherinfo11005  Otherinfo11006 Otherinfo11007 Otherinfo11008  \\\n",
       "0            0/0             0/0            0/0            0/0   \n",
       "1            1/1             0/0            0/0            0/0   \n",
       "2            0/0             0/0            0/0            0/0   \n",
       "3            0/0             0/0            0/0            0/0   \n",
       "4            0/0             0/0            0/0            0/0   \n",
       "\n",
       "   Otherinfo11009  Otherinfo11010 Otherinfo11011 Otherinfo11012  \\\n",
       "0             0/0             0/0            0/0            0/0   \n",
       "1             0/1             0/0            0/0            0/0   \n",
       "2             0/0             0/0            0/0            0/0   \n",
       "3             0/0             0/0            0/0            0/0   \n",
       "4             0/0             0/0            0/0            0/0   \n",
       "\n",
       "  Otherinfo11013 Otherinfo11014  \n",
       "0            0/0            0/0  \n",
       "1            0/0            0/1  \n",
       "2            0/0            0/0  \n",
       "3            0/0            0/0  \n",
       "4            0/0            0/0  \n",
       "\n",
       "[5 rows x 11025 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Visualize multianno file\n",
    "PP2A = pd.read_csv(f'{WORK_DIR}/pheno_PP2A.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "PP2A.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Chr               4\n",
       "Start             4\n",
       "End               4\n",
       "Ref               4\n",
       "Alt               4\n",
       "                 ..\n",
       "Otherinfo11010    4\n",
       "Otherinfo11011    4\n",
       "Otherinfo11012    4\n",
       "Otherinfo11013    4\n",
       "Otherinfo11014    4\n",
       "Length: 11025, dtype: int64"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter exonic variants\n",
    "coding = PP2A[PP2A['Func.refGene'] == 'exonic']\n",
    "coding.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>Otherinfo11005</th>\n",
       "      <th>Otherinfo11006</th>\n",
       "      <th>Otherinfo11007</th>\n",
       "      <th>Otherinfo11008</th>\n",
       "      <th>Otherinfo11009</th>\n",
       "      <th>Otherinfo11010</th>\n",
       "      <th>Otherinfo11011</th>\n",
       "      <th>Otherinfo11012</th>\n",
       "      <th>Otherinfo11013</th>\n",
       "      <th>Otherinfo11014</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>9</td>\n",
       "      <td>129127987</td>\n",
       "      <td>129127987</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>PTPA</td>\n",
       "      <td>.</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>PTPA:NM_178001:exon4:c.A240G:p.E80E</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>9</td>\n",
       "      <td>129128008</td>\n",
       "      <td>129128008</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>PTPA</td>\n",
       "      <td>.</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>PTPA:NM_178001:exon4:c.G261A:p.Q87Q</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>260</th>\n",
       "      <td>9</td>\n",
       "      <td>129142498</td>\n",
       "      <td>129142498</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>PTPA</td>\n",
       "      <td>.</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>PTPA:NM_001271832:exon8:c.C753T:p.A251A,PTPA:N...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>312</th>\n",
       "      <td>9</td>\n",
       "      <td>129147457</td>\n",
       "      <td>129147457</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>PTPA</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>PTPA:NM_001271832:exon9:c.C878T:p.S293L,PTPA:N...</td>\n",
       "      <td>...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4 rows × 11025 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chr      Start        End Ref Alt Func.refGene Gene.refGene  \\\n",
       "139    9  129127987  129127987   A   G       exonic         PTPA   \n",
       "140    9  129128008  129128008   G   A       exonic         PTPA   \n",
       "260    9  129142498  129142498   C   T       exonic         PTPA   \n",
       "312    9  129147457  129147457   C   T       exonic         PTPA   \n",
       "\n",
       "    GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "139                  .     synonymous SNV   \n",
       "140                  .     synonymous SNV   \n",
       "260                  .     synonymous SNV   \n",
       "312                  .  nonsynonymous SNV   \n",
       "\n",
       "                                      AAChange.refGene  ... Otherinfo11005  \\\n",
       "139                PTPA:NM_178001:exon4:c.A240G:p.E80E  ...            0/0   \n",
       "140                PTPA:NM_178001:exon4:c.G261A:p.Q87Q  ...            0/0   \n",
       "260  PTPA:NM_001271832:exon8:c.C753T:p.A251A,PTPA:N...  ...            0/0   \n",
       "312  PTPA:NM_001271832:exon9:c.C878T:p.S293L,PTPA:N...  ...            1/1   \n",
       "\n",
       "     Otherinfo11006 Otherinfo11007 Otherinfo11008  Otherinfo11009  \\\n",
       "139             0/0            0/0            0/0             0/0   \n",
       "140             0/0            0/0            0/0             0/0   \n",
       "260             0/0            0/0            0/0             0/0   \n",
       "312             0/0            0/0            0/0             0/0   \n",
       "\n",
       "     Otherinfo11010 Otherinfo11011 Otherinfo11012 Otherinfo11013  \\\n",
       "139             0/0            0/0            0/0            0/0   \n",
       "140             0/0            0/0            0/0            0/0   \n",
       "260             0/0            0/0            0/0            0/0   \n",
       "312             0/0            0/0            0/0            0/0   \n",
       "\n",
       "    Otherinfo11014  \n",
       "139            0/0  \n",
       "140            0/0  \n",
       "260            0/0  \n",
       "312            0/0  \n",
       "\n",
       "[4 rows x 11025 columns]"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Chr               304\n",
       "Start             304\n",
       "End               304\n",
       "Ref               304\n",
       "Alt               304\n",
       "                 ... \n",
       "Otherinfo11010    304\n",
       "Otherinfo11011    304\n",
       "Otherinfo11012    304\n",
       "Otherinfo11013    304\n",
       "Otherinfo11014    304\n",
       "Length: 11025, dtype: int64"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding = PP2A[PP2A['Func.refGene'] == 'intronic']\n",
    "coding.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Chr               5\n",
       "Start             5\n",
       "End               5\n",
       "Ref               5\n",
       "Alt               5\n",
       "                 ..\n",
       "Otherinfo11010    5\n",
       "Otherinfo11011    5\n",
       "Otherinfo11012    5\n",
       "Otherinfo11013    5\n",
       "Otherinfo11014    5\n",
       "Length: 11025, dtype: int64"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding = PP2A[PP2A['Func.refGene'] == 'UTR5']\n",
    "coding.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Chr               15\n",
       "Start             15\n",
       "End               15\n",
       "Ref               15\n",
       "Alt               15\n",
       "                  ..\n",
       "Otherinfo11010    15\n",
       "Otherinfo11011    15\n",
       "Otherinfo11012    15\n",
       "Otherinfo11013    15\n",
       "Otherinfo11014    15\n",
       "Length: 11025, dtype: int64"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding = PP2A[PP2A['Func.refGene'] == 'UTR3']\n",
    "coding.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Chr               1\n",
       "Start             1\n",
       "End               1\n",
       "Ref               1\n",
       "Alt               1\n",
       "                 ..\n",
       "Otherinfo11010    1\n",
       "Otherinfo11011    1\n",
       "Otherinfo11012    1\n",
       "Otherinfo11013    1\n",
       "Otherinfo11014    1\n",
       "Length: 11025, dtype: int64"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter exonic and non-syn vars\n",
    "coding_nonsynonymous = PP2A[(PP2A['Func.refGene'] == 'exonic') & (PP2A['ExonicFunc.refGene'] == 'nonsynonymous SNV')]\n",
    "coding_nonsynonymous.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>Otherinfo11005</th>\n",
       "      <th>Otherinfo11006</th>\n",
       "      <th>Otherinfo11007</th>\n",
       "      <th>Otherinfo11008</th>\n",
       "      <th>Otherinfo11009</th>\n",
       "      <th>Otherinfo11010</th>\n",
       "      <th>Otherinfo11011</th>\n",
       "      <th>Otherinfo11012</th>\n",
       "      <th>Otherinfo11013</th>\n",
       "      <th>Otherinfo11014</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>312</th>\n",
       "      <td>9</td>\n",
       "      <td>129147457</td>\n",
       "      <td>129147457</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>PTPA</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>PTPA:NM_001271832:exon9:c.C878T:p.S293L,PTPA:N...</td>\n",
       "      <td>...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 11025 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chr      Start        End Ref Alt Func.refGene Gene.refGene  \\\n",
       "312    9  129147457  129147457   C   T       exonic         PTPA   \n",
       "\n",
       "    GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "312                  .  nonsynonymous SNV   \n",
       "\n",
       "                                      AAChange.refGene  ... Otherinfo11005  \\\n",
       "312  PTPA:NM_001271832:exon9:c.C878T:p.S293L,PTPA:N...  ...            1/1   \n",
       "\n",
       "     Otherinfo11006 Otherinfo11007 Otherinfo11008  Otherinfo11009  \\\n",
       "312             0/0            0/0            0/0             0/0   \n",
       "\n",
       "     Otherinfo11010 Otherinfo11011 Otherinfo11012 Otherinfo11013  \\\n",
       "312             0/0            0/0            0/0            0/0   \n",
       "\n",
       "    Otherinfo11014  \n",
       "312            0/0  \n",
       "\n",
       "[1 rows x 11025 columns]"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding_nonsynonymous.head(1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "     Chr      Start        End Ref Alt Func.refGene Gene.refGene  \\\n",
      "312    9  129147457  129147457   C   T       exonic         PTPA   \n",
      "\n",
      "    GeneDetail.refGene ExonicFunc.refGene  \\\n",
      "312                  .  nonsynonymous SNV   \n",
      "\n",
      "                                      AAChange.refGene  ... Otherinfo11005  \\\n",
      "312  PTPA:NM_001271832:exon9:c.C878T:p.S293L,PTPA:N...  ...            1/1   \n",
      "\n",
      "     Otherinfo11006 Otherinfo11007 Otherinfo11008  Otherinfo11009  \\\n",
      "312             0/0            0/0            0/0             0/0   \n",
      "\n",
      "     Otherinfo11010 Otherinfo11011 Otherinfo11012 Otherinfo11013  \\\n",
      "312             0/0            0/0            0/0            0/0   \n",
      "\n",
      "    Otherinfo11014  \n",
      "312            0/0  \n",
      "\n",
      "[1 rows x 11025 columns]\n"
     ]
    }
   ],
   "source": [
    "print(coding_nonsynonymous)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>Otherinfo11005</th>\n",
       "      <th>Otherinfo11006</th>\n",
       "      <th>Otherinfo11007</th>\n",
       "      <th>Otherinfo11008</th>\n",
       "      <th>Otherinfo11009</th>\n",
       "      <th>Otherinfo11010</th>\n",
       "      <th>Otherinfo11011</th>\n",
       "      <th>Otherinfo11012</th>\n",
       "      <th>Otherinfo11013</th>\n",
       "      <th>Otherinfo11014</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>312</th>\n",
       "      <td>9</td>\n",
       "      <td>129147457</td>\n",
       "      <td>129147457</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>PTPA</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>PTPA:NM_001271832:exon9:c.C878T:p.S293L,PTPA:N...</td>\n",
       "      <td>...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 11025 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chr      Start        End Ref Alt Func.refGene Gene.refGene  \\\n",
       "312    9  129147457  129147457   C   T       exonic         PTPA   \n",
       "\n",
       "    GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "312                  .  nonsynonymous SNV   \n",
       "\n",
       "                                      AAChange.refGene  ... Otherinfo11005  \\\n",
       "312  PTPA:NM_001271832:exon9:c.C878T:p.S293L,PTPA:N...  ...            1/1   \n",
       "\n",
       "     Otherinfo11006 Otherinfo11007 Otherinfo11008  Otherinfo11009  \\\n",
       "312             0/0            0/0            0/0             0/0   \n",
       "\n",
       "     Otherinfo11010 Otherinfo11011 Otherinfo11012 Otherinfo11013  \\\n",
       "312             0/0            0/0            0/0            0/0   \n",
       "\n",
       "    Otherinfo11014  \n",
       "312            0/0  \n",
       "\n",
       "[1 rows x 11025 columns]"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding_nonsynonymous.to_csv(f'{WORK_DIR}/coding_nonsynonymous.txt', sep = '\\t', index = False)\n",
    "coding_nonsynonymous.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Calculate freq of coding and non-syn vars in cases versus controls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "reduced_coding_nonsynonymous = coding_nonsynonymous[[\"Chr\", \"Start\", \"End\", \"Gene.refGene\"]]\n",
    "reduced_coding_nonsynonymous.to_csv(f'{WORK_DIR}/reduced_coding_nonsynonymous.txt', sep = '\\t', index = False, header= False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9\t129147457\t129147457\tPTPA\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "head reduced_coding_nonsynonymous.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to coding_nonsynonymous_pheno_PP2A.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile pheno_PP2A_EUR\n",
      "  --ci 0.95\n",
      "  --extract range reduced_coding_nonsynonymous.txt\n",
      "  --out coding_nonsynonymous_pheno_PP2A\n",
      "\n",
      "14998 MB RAM detected; reserving 7499 MB for main workspace.\n",
      "328 variants loaded from .bim file.\n",
      "11002 people (6731 males, 4271 females) loaded from .fam.\n",
      "10727 phenotype values loaded from .fam.\n",
      "--extract range: 327 variants excluded.\n",
      "--extract range: 1 variant remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 11002 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.981912.\n",
      "1 variant and 11002 people pass filters and QC.\n",
      "Among remaining phenotypes, 6852 are cases and 3875 are controls.  (275\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to coding_nonsynonymous_pheno_PP2A.assoc ...\n",
      "0%\b\bdone.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "/home/jupyter/tools/plink --bfile pheno_PP2A_EUR --extract range reduced_coding_nonsynonymous.txt --assoc --out coding_nonsynonymous_pheno_PP2A --allow-no-sex --ci 0.95"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP</th>\n",
       "      <th>BP</th>\n",
       "      <th>A1</th>\n",
       "      <th>F_A</th>\n",
       "      <th>F_U</th>\n",
       "      <th>A2</th>\n",
       "      <th>CHISQ</th>\n",
       "      <th>P</th>\n",
       "      <th>OR</th>\n",
       "      <th>SE</th>\n",
       "      <th>L95</th>\n",
       "      <th>U95</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:129147457:C:T</td>\n",
       "      <td>129147457</td>\n",
       "      <td>T</td>\n",
       "      <td>0.07145</td>\n",
       "      <td>0.07838</td>\n",
       "      <td>C</td>\n",
       "      <td>3.408</td>\n",
       "      <td>0.06489</td>\n",
       "      <td>0.9048</td>\n",
       "      <td>0.05422</td>\n",
       "      <td>0.8136</td>\n",
       "      <td>1.006</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHR                 SNP         BP A1      F_A      F_U A2  CHISQ        P  \\\n",
       "0    9  chr9:129147457:C:T  129147457  T  0.07145  0.07838  C  3.408  0.06489   \n",
       "\n",
       "       OR       SE     L95    U95  \n",
       "0  0.9048  0.05422  0.8136  1.006  "
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PP2A_freq = pd.read_csv(f'{WORK_DIR}/coding_nonsynonymous_pheno_PP2A.assoc', delim_whitespace=True)\n",
    "PP2A_freq.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to coding_nonsynonymous_rare_0.05.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile pheno_PP2A_EUR\n",
      "  --extract range reduced_coding_nonsynonymous.txt\n",
      "  --make-bed\n",
      "  --max-maf 0.05\n",
      "  --out coding_nonsynonymous_rare_0.05\n",
      "\n",
      "14998 MB RAM detected; reserving 7499 MB for main workspace.\n",
      "328 variants loaded from .bim file.\n",
      "11002 people (6731 males, 4271 females) loaded from .fam.\n",
      "10727 phenotype values loaded from .fam.\n",
      "--extract range: 327 variants excluded.\n",
      "--extract range: 1 variant remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 11002 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.981912.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Error: All variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n"
     ]
    },
    {
     "ename": "CalledProcessError",
     "evalue": "Command 'b\"\\nWORK_DIR='/home/jupyter/PP2A_GP2/'\\ncd $WORK_DIR\\n\\n/home/jupyter/tools/plink --bfile pheno_PP2A_EUR --extract range reduced_coding_nonsynonymous.txt --make-bed --max-maf 0.05 --out coding_nonsynonymous_rare_0.05 --allow-no-sex\\n\"' returned non-zero exit status 13.",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mCalledProcessError\u001b[0m                        Traceback (most recent call last)",
      "\u001b[0;32m/tmp/ipykernel_116/2547657113.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mget_ipython\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrun_cell_magic\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'bash'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m''\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"\\nWORK_DIR='/home/jupyter/PP2A_GP2/'\\ncd $WORK_DIR\\n\\n/home/jupyter/tools/plink --bfile pheno_PP2A_EUR --extract range reduced_coding_nonsynonymous.txt --make-bed --max-maf 0.05 --out coding_nonsynonymous_rare_0.05 --allow-no-sex\\n\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m/opt/conda/lib/python3.7/site-packages/IPython/core/interactiveshell.py\u001b[0m in \u001b[0;36mrun_cell_magic\u001b[0;34m(self, magic_name, line, cell)\u001b[0m\n\u001b[1;32m   2460\u001b[0m             \u001b[0;32mwith\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbuiltin_trap\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2461\u001b[0m                 \u001b[0margs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mmagic_arg_s\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2462\u001b[0;31m                 \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfn\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2463\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2464\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/conda/lib/python3.7/site-packages/IPython/core/magics/script.py\u001b[0m in \u001b[0;36mnamed_script_magic\u001b[0;34m(line, cell)\u001b[0m\n\u001b[1;32m    140\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    141\u001b[0m                 \u001b[0mline\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mscript\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 142\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshebang\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mline\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    143\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    144\u001b[0m         \u001b[0;31m# write a basic docstring:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/conda/lib/python3.7/site-packages/decorator.py\u001b[0m in \u001b[0;36mfun\u001b[0;34m(*args, **kw)\u001b[0m\n\u001b[1;32m    230\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mkwsyntax\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    231\u001b[0m                 \u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkw\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfix\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkw\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msig\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 232\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mcaller\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfunc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mextras\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0margs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkw\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    233\u001b[0m     \u001b[0mfun\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__name__\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfunc\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__name__\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    234\u001b[0m     \u001b[0mfun\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__doc__\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfunc\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__doc__\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/conda/lib/python3.7/site-packages/IPython/core/magic.py\u001b[0m in \u001b[0;36m<lambda>\u001b[0;34m(f, *a, **k)\u001b[0m\n\u001b[1;32m    185\u001b[0m     \u001b[0;31m# but it's overkill for just that one bit of state.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    186\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mmagic_deco\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 187\u001b[0;31m         \u001b[0mcall\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mlambda\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0ma\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mk\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0ma\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mk\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    188\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    189\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mcallable\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/conda/lib/python3.7/site-packages/IPython/core/magics/script.py\u001b[0m in \u001b[0;36mshebang\u001b[0;34m(self, line, cell)\u001b[0m\n\u001b[1;32m    243\u001b[0m             \u001b[0msys\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstderr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mflush\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    244\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0margs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mraise_error\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreturncode\u001b[0m\u001b[0;34m!=\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 245\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mCalledProcessError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreturncode\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0moutput\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mout\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstderr\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0merr\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    246\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    247\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_run_script\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mp\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mto_close\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mCalledProcessError\u001b[0m: Command 'b\"\\nWORK_DIR='/home/jupyter/PP2A_GP2/'\\ncd $WORK_DIR\\n\\n/home/jupyter/tools/plink --bfile pheno_PP2A_EUR --extract range reduced_coding_nonsynonymous.txt --make-bed --max-maf 0.05 --out coding_nonsynonymous_rare_0.05 --allow-no-sex\\n\"' returned non-zero exit status 13."
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "/home/jupyter/tools/plink --bfile pheno_PP2A_EUR --extract range reduced_coding_nonsynonymous.txt --make-bed --max-maf 0.05 --out coding_nonsynonymous_rare_0.05 --allow-no-sex"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP</th>\n",
       "      <th>UNADJ</th>\n",
       "      <th>GC</th>\n",
       "      <th>BONF</th>\n",
       "      <th>HOLM</th>\n",
       "      <th>SIDAK_SS</th>\n",
       "      <th>SIDAK_SD</th>\n",
       "      <th>FDR_BH</th>\n",
       "      <th>FDR_BY</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:129147457:C:T</td>\n",
       "      <td>0.06489</td>\n",
       "      <td>0.4995</td>\n",
       "      <td>0.06489</td>\n",
       "      <td>0.06489</td>\n",
       "      <td>0.06489</td>\n",
       "      <td>0.06489</td>\n",
       "      <td>0.06489</td>\n",
       "      <td>0.06489</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHR                 SNP    UNADJ      GC     BONF     HOLM  SIDAK_SS  \\\n",
       "0    9  chr9:129147457:C:T  0.06489  0.4995  0.06489  0.06489   0.06489   \n",
       "\n",
       "   SIDAK_SD   FDR_BH   FDR_BY  \n",
       "0   0.06489  0.06489  0.06489  "
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PP2A_freq = pd.read_csv(f'{WORK_DIR}/coding_nonsynonymous_pheno_PP2A.assoc.adjusted', delim_whitespace=True)\n",
    "PP2A_freq.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to coding_nonsynonymous_pheno_PP2A.log.\n",
      "Options in effect:\n",
      "  --adjust\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile pheno_PP2A_EUR\n",
      "  --extract range reduced_coding_nonsynonymous.txt\n",
      "  --out coding_nonsynonymous_pheno_PP2A\n",
      "\n",
      "14998 MB RAM detected; reserving 7499 MB for main workspace.\n",
      "328 variants loaded from .bim file.\n",
      "11002 people (6731 males, 4271 females) loaded from .fam.\n",
      "10727 phenotype values loaded from .fam.\n",
      "--extract range: 327 variants excluded.\n",
      "--extract range: 1 variant remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 11002 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.981912.\n",
      "1 variant and 11002 people pass filters and QC.\n",
      "Among remaining phenotypes, 6852 are cases and 3875 are controls.  (275\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to coding_nonsynonymous_pheno_PP2A.assoc ...\n",
      "0%\b\bdone.\n",
      "--adjust: Genomic inflation est. lambda (based on median chisq) = 7.47339.\n",
      "0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b--adjust values (1 variant) written to\n",
      "coding_nonsynonymous_pheno_PP2A.assoc.adjusted .\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "/home/jupyter/tools/plink --bfile pheno_PP2A_EUR --extract range reduced_coding_nonsynonymous.txt --assoc --out coding_nonsynonymous_pheno_PP2A --allow-no-sex --adjust"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP</th>\n",
       "      <th>UNADJ</th>\n",
       "      <th>GC</th>\n",
       "      <th>BONF</th>\n",
       "      <th>HOLM</th>\n",
       "      <th>SIDAK_SS</th>\n",
       "      <th>SIDAK_SD</th>\n",
       "      <th>FDR_BH</th>\n",
       "      <th>FDR_BY</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:129147457:C:T</td>\n",
       "      <td>0.06489</td>\n",
       "      <td>0.4995</td>\n",
       "      <td>0.06489</td>\n",
       "      <td>0.06489</td>\n",
       "      <td>0.06489</td>\n",
       "      <td>0.06489</td>\n",
       "      <td>0.06489</td>\n",
       "      <td>0.06489</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHR                 SNP    UNADJ      GC     BONF     HOLM  SIDAK_SS  \\\n",
       "0    9  chr9:129147457:C:T  0.06489  0.4995  0.06489  0.06489   0.06489   \n",
       "\n",
       "   SIDAK_SD   FDR_BH   FDR_BY  \n",
       "0   0.06489  0.06489  0.06489  "
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PP2A_freq = pd.read_csv(f'{WORK_DIR}/coding_nonsynonymous_pheno_PP2A.assoc.adjusted', delim_whitespace=True)\n",
    "PP2A_freq.head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Calculate freq of HMZ in cases versus controls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to coding_nonsynonymous_pheno_PP2A.log.\n",
      "Options in effect:\n",
      "  --bfile pheno_PP2A_EUR\n",
      "  --extract range reduced_coding_nonsynonymous.txt\n",
      "  --out coding_nonsynonymous_pheno_PP2A\n",
      "  --recode A\n",
      "\n",
      "14998 MB RAM detected; reserving 7499 MB for main workspace.\n",
      "328 variants loaded from .bim file.\n",
      "11002 people (6731 males, 4271 females) loaded from .fam.\n",
      "10727 phenotype values loaded from .fam.\n",
      "--extract range: 327 variants excluded.\n",
      "--extract range: 1 variant remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 11002 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.981912.\n",
      "1 variant and 11002 people pass filters and QC.\n",
      "Among remaining phenotypes, 6852 are cases and 3875 are controls.  (275\n",
      "phenotypes are missing.)\n",
      "--recode A to coding_nonsynonymous_pheno_PP2A.raw ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "/home/jupyter/tools/plink --bfile pheno_PP2A_EUR --extract range reduced_coding_nonsynonymous.txt --recode A --out coding_nonsynonymous_pheno_PP2A"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000001_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000002_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000003_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000005_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000006_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   FID             IID  PAT  MAT  SEX  PHENOTYPE  chr9:129147457:C:T_T\n",
       "0    0  APGS_000001_s1    0    0    1          2                   0.0\n",
       "1    0  APGS_000002_s1    0    0    1          2                   0.0\n",
       "2    0  APGS_000003_s1    0    0    1          2                   0.0\n",
       "3    0  APGS_000005_s1    0    0    1          2                   0.0\n",
       "4    0  APGS_000006_s1    0    0    1          2                   0.0"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PP2A_recode = pd.read_csv(f'{WORK_DIR}/coding_nonsynonymous_pheno_PP2A.raw', delim_whitespace=True)\n",
    "PP2A_recode.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>500</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000651_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>596</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000771_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1157</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_001455_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1322</th>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000181_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1327</th>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000189_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2269</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_000366_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2445</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_000565_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2779</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_000971_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2793</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_000992_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2949</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_001185_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2980</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_001219_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3093</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_001360_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3154</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_001432_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3707</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_002154_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3785</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_002265_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4121</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_002691_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4294</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_002901_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4632</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_003320_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4815</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_003551_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5013</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_003795_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5046</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_003835_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5050</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_003842_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5157</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_003991_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5212</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_004064_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5291</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_004166_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5322</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_004213_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5426</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_004400_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7611</th>\n",
       "      <td>0</td>\n",
       "      <td>KOC_000005_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8099</th>\n",
       "      <td>0</td>\n",
       "      <td>MDGAP-QSBB_000077_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8108</th>\n",
       "      <td>0</td>\n",
       "      <td>MDGAP-QSBB_000086_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8244</th>\n",
       "      <td>0</td>\n",
       "      <td>NZP3_000119_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8699</th>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_000340_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10205</th>\n",
       "      <td>0</td>\n",
       "      <td>PPMI_000009_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10410</th>\n",
       "      <td>0</td>\n",
       "      <td>PPMI_000509_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10850</th>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000281_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10992</th>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000470_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       FID                   IID  PAT  MAT  SEX  PHENOTYPE  \\\n",
       "500      0        APGS_000651_s1    0    0    1          2   \n",
       "596      0        APGS_000771_s1    0    0    1          2   \n",
       "1157     0        APGS_001455_s1    0    0    2          2   \n",
       "1322     0         BCM_000181_s1    0    0    1          2   \n",
       "1327     0         BCM_000189_s1    0    0    2          2   \n",
       "2269     0     CORIELL_000366_s1    0    0    2          2   \n",
       "2445     0     CORIELL_000565_s1    0    0    1          2   \n",
       "2779     0     CORIELL_000971_s1    0    0    2          2   \n",
       "2793     0     CORIELL_000992_s1    0    0    1          2   \n",
       "2949     0     CORIELL_001185_s1    0    0    2          2   \n",
       "2980     0     CORIELL_001219_s1    0    0    1          2   \n",
       "3093     0     CORIELL_001360_s1    0    0    1          2   \n",
       "3154     0     CORIELL_001432_s1    0    0    1          2   \n",
       "3707     0     CORIELL_002154_s1    0    0    1          2   \n",
       "3785     0     CORIELL_002265_s1    0    0    1          2   \n",
       "4121     0     CORIELL_002691_s1    0    0    1          2   \n",
       "4294     0     CORIELL_002901_s1    0    0    2          2   \n",
       "4632     0     CORIELL_003320_s1    0    0    1          2   \n",
       "4815     0     CORIELL_003551_s1    0    0    2          2   \n",
       "5013     0     CORIELL_003795_s1    0    0    1          2   \n",
       "5046     0     CORIELL_003835_s1    0    0    1          2   \n",
       "5050     0     CORIELL_003842_s1    0    0    1          2   \n",
       "5157     0     CORIELL_003991_s1    0    0    1          2   \n",
       "5212     0     CORIELL_004064_s1    0    0    1          2   \n",
       "5291     0     CORIELL_004166_s1    0    0    1          2   \n",
       "5322     0     CORIELL_004213_s1    0    0    2          2   \n",
       "5426     0     CORIELL_004400_s1    0    0    1          2   \n",
       "7611     0         KOC_000005_s1    0    0    2          2   \n",
       "8099     0  MDGAP-QSBB_000077_s1    0    0    1          2   \n",
       "8108     0  MDGAP-QSBB_000086_s1    0    0    1          2   \n",
       "8244     0        NZP3_000119_s1    0    0    2          2   \n",
       "8699     0        PAGE_000340_s1    0    0    2          2   \n",
       "10205    0        PPMI_000009_s1    0    0    2          2   \n",
       "10410    0        PPMI_000509_s1    0    0    1          2   \n",
       "10850    0         UMD_000281_s1    0    0    2          2   \n",
       "10992    0         UMD_000470_s1    0    0    2          2   \n",
       "\n",
       "       chr9:129147457:C:T_T  \n",
       "500                     2.0  \n",
       "596                     2.0  \n",
       "1157                    2.0  \n",
       "1322                    2.0  \n",
       "1327                    2.0  \n",
       "2269                    2.0  \n",
       "2445                    2.0  \n",
       "2779                    2.0  \n",
       "2793                    2.0  \n",
       "2949                    2.0  \n",
       "2980                    2.0  \n",
       "3093                    2.0  \n",
       "3154                    2.0  \n",
       "3707                    2.0  \n",
       "3785                    2.0  \n",
       "4121                    2.0  \n",
       "4294                    2.0  \n",
       "4632                    2.0  \n",
       "4815                    2.0  \n",
       "5013                    2.0  \n",
       "5046                    2.0  \n",
       "5050                    2.0  \n",
       "5157                    2.0  \n",
       "5212                    2.0  \n",
       "5291                    2.0  \n",
       "5322                    2.0  \n",
       "5426                    2.0  \n",
       "7611                    2.0  \n",
       "8099                    2.0  \n",
       "8108                    2.0  \n",
       "8244                    2.0  \n",
       "8699                    2.0  \n",
       "10205                   2.0  \n",
       "10410                   2.0  \n",
       "10850                   2.0  \n",
       "10992                   2.0  "
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Explore HMZs for chr9:129147457:C:T_T in cases\n",
    "PP2A_hom_casos = PP2A_recode[(PP2A_recode['chr9:129147457:C:T_T'] == 2) & (PP2A_recode['PHENOTYPE'] == 2)]\n",
    "PP2A_hom_casos.head(36)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(36, 7)"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PP2A_hom_casos.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1661</th>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000623_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5545</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_004565_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5571</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_004595_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5665</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_004703_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5693</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_004748_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5698</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_004754_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5954</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_005055_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6408</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_005582_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6436</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_005612_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6668</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_005880_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6969</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006276_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7075</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006416_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7446</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007038_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7752</th>\n",
       "      <td>0</td>\n",
       "      <td>LCC_000130_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7852</th>\n",
       "      <td>0</td>\n",
       "      <td>LCC_000241_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8291</th>\n",
       "      <td>0</td>\n",
       "      <td>NZP3_000168_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8465</th>\n",
       "      <td>0</td>\n",
       "      <td>NZP3_000358_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8576</th>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_000115_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8984</th>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_000878_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9476</th>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_001699_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9479</th>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_001706_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9516</th>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_001770_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9551</th>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_001824_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9768</th>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_002155_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10431</th>\n",
       "      <td>0</td>\n",
       "      <td>PPMI_000537_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10732</th>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000129_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       FID                IID  PAT  MAT  SEX  PHENOTYPE  chr9:129147457:C:T_T\n",
       "1661     0      BCM_000623_s1    0    0    2          1                   2.0\n",
       "5545     0  CORIELL_004565_s1    0    0    2          1                   2.0\n",
       "5571     0  CORIELL_004595_s1    0    0    2          1                   2.0\n",
       "5665     0  CORIELL_004703_s1    0    0    2          1                   2.0\n",
       "5693     0  CORIELL_004748_s1    0    0    2          1                   2.0\n",
       "5698     0  CORIELL_004754_s1    0    0    1          1                   2.0\n",
       "5954     0  CORIELL_005055_s1    0    0    1          1                   2.0\n",
       "6408     0  CORIELL_005582_s1    0    0    2          1                   2.0\n",
       "6436     0  CORIELL_005612_s1    0    0    2          1                   2.0\n",
       "6668     0  CORIELL_005880_s1    0    0    2          1                   2.0\n",
       "6969     0  CORIELL_006276_s1    0    0    2          1                   2.0\n",
       "7075     0  CORIELL_006416_s1    0    0    1          1                   2.0\n",
       "7446     0  CORIELL_007038_s1    0    0    1          1                   2.0\n",
       "7752     0      LCC_000130_s1    0    0    2          1                   2.0\n",
       "7852     0      LCC_000241_s1    0    0    2          1                   2.0\n",
       "8291     0     NZP3_000168_s1    0    0    1          1                   2.0\n",
       "8465     0     NZP3_000358_s1    0    0    2          1                   2.0\n",
       "8576     0     PAGE_000115_s1    0    0    1          1                   2.0\n",
       "8984     0     PAGE_000878_s1    0    0    2          1                   2.0\n",
       "9476     0     PAGE_001699_s1    0    0    1          1                   2.0\n",
       "9479     0     PAGE_001706_s1    0    0    1          1                   2.0\n",
       "9516     0     PAGE_001770_s1    0    0    1          1                   2.0\n",
       "9551     0     PAGE_001824_s1    0    0    1          1                   2.0\n",
       "9768     0     PAGE_002155_s1    0    0    1          1                   2.0\n",
       "10431    0     PPMI_000537_s1    0    0    1          1                   2.0\n",
       "10732    0      UMD_000129_s1    0    0    2          1                   2.0"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Explore HMZs for chr9:129147457:C:T_T in controls\n",
    "PP2A_hom_controles = PP2A_recode[(PP2A_recode['chr9:129147457:C:T_T'] == 2) & (PP2A_recode['PHENOTYPE'] == 1)]\n",
    "PP2A_hom_controles.head(36)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(26, 7)"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PP2A_hom_controles.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000007_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000018_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000024_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000027_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000029_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10981</th>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000456_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10984</th>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000460_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10985</th>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000461_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10987</th>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000464_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10989</th>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000466_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>890 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       FID             IID  PAT  MAT  SEX  PHENOTYPE  chr9:129147457:C:T_T\n",
       "5        0  APGS_000007_s1    0    0    1          2                   1.0\n",
       "13       0  APGS_000018_s1    0    0    1          2                   1.0\n",
       "17       0  APGS_000024_s1    0    0    2          2                   1.0\n",
       "20       0  APGS_000027_s1    0    0    1          2                   1.0\n",
       "22       0  APGS_000029_s1    0    0    1          2                   1.0\n",
       "...    ...             ...  ...  ...  ...        ...                   ...\n",
       "10981    0   UMD_000456_s1    0    0    1          2                   1.0\n",
       "10984    0   UMD_000460_s1    0    0    1          2                   1.0\n",
       "10985    0   UMD_000461_s1    0    0    1          2                   1.0\n",
       "10987    0   UMD_000464_s1    0    0    1          2                   1.0\n",
       "10989    0   UMD_000466_s1    0    0    1          2                   1.0\n",
       "\n",
       "[890 rows x 7 columns]"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Explore HETs for chr9:129147457:C:T_T in cases\n",
    "PP2A_het_casos_129147457_EUR = PP2A_recode[(PP2A_recode['chr9:129147457:C:T_T'] == 1) & (PP2A_recode['PHENOTYPE'] == 2)]\n",
    "PP2A_het_casos_129147457_EUR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1204</th>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000035_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1261</th>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000099_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1302</th>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000149_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1326</th>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000186_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1360</th>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000237_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10827</th>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000249_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10905</th>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000352_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10911</th>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000359_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10929</th>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000384_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10979</th>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000454_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>544 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       FID            IID  PAT  MAT  SEX  PHENOTYPE  chr9:129147457:C:T_T\n",
       "1204     0  BCM_000035_s1    0    0    2          1                   1.0\n",
       "1261     0  BCM_000099_s1    0    0    2          1                   1.0\n",
       "1302     0  BCM_000149_s1    0    0    2          1                   1.0\n",
       "1326     0  BCM_000186_s1    0    0    1          1                   1.0\n",
       "1360     0  BCM_000237_s1    0    0    2          1                   1.0\n",
       "...    ...            ...  ...  ...  ...        ...                   ...\n",
       "10827    0  UMD_000249_s1    0    0    2          1                   1.0\n",
       "10905    0  UMD_000352_s1    0    0    2          1                   1.0\n",
       "10911    0  UMD_000359_s1    0    0    2          1                   1.0\n",
       "10929    0  UMD_000384_s1    0    0    2          1                   1.0\n",
       "10979    0  UMD_000454_s1    0    0    1          1                   1.0\n",
       "\n",
       "[544 rows x 7 columns]"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Explore HETs for chr9:129147457:C:T_T in controls\n",
    "PP2A_het_controles_129147457_EUR = PP2A_recode[(PP2A_recode['chr9:129147457:C:T_T'] == 1) & (PP2A_recode['PHENOTYPE'] == 1)]\n",
    "PP2A_het_controles_129147457_EUR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: rpy2 in /home/jupyter/.local/lib/python3.7/site-packages (3.5.12)\n",
      "Requirement already satisfied: cffi>=1.10.0 in /opt/conda/lib/python3.7/site-packages (from rpy2) (1.15.0)\n",
      "Requirement already satisfied: jinja2 in /opt/conda/lib/python3.7/site-packages (from rpy2) (3.1.2)\n",
      "Requirement already satisfied: pytz in /opt/conda/lib/python3.7/site-packages (from rpy2) (2022.6)\n",
      "Requirement already satisfied: tzlocal<5.0 in /home/jupyter/.local/lib/python3.7/site-packages (from rpy2) (4.3)\n",
      "Requirement already satisfied: typing-extensions in /opt/conda/lib/python3.7/site-packages (from rpy2) (4.1.1)\n",
      "Requirement already satisfied: pycparser in /opt/conda/lib/python3.7/site-packages (from cffi>=1.10.0->rpy2) (2.21)\n",
      "Requirement already satisfied: pytz-deprecation-shim in /home/jupyter/.local/lib/python3.7/site-packages (from tzlocal<5.0->rpy2) (0.1.0.post0)\n",
      "Requirement already satisfied: backports.zoneinfo in /home/jupyter/.local/lib/python3.7/site-packages (from tzlocal<5.0->rpy2) (0.2.1)\n",
      "Requirement already satisfied: MarkupSafe>=2.0 in /opt/conda/lib/python3.7/site-packages (from jinja2->rpy2) (2.0.1)\n",
      "Requirement already satisfied: tzdata in /home/jupyter/.local/lib/python3.7/site-packages (from pytz-deprecation-shim->tzlocal<5.0->rpy2) (2023.3)\n"
     ]
    }
   ],
   "source": [
    "!pip install rpy2\n",
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The rpy2.ipython extension is already loaded. To reload it, use:\n",
      "  %reload_ext rpy2.ipython\n"
     ]
    }
   ],
   "source": [
    "import rpy2\n",
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "PP2A_het_casos_129147457_EUR.to_csv('PP2A_het_casos_129147457_EUR.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000007_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000018_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>17</td>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000024_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>20</td>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000027_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000029_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>885</th>\n",
       "      <td>10981</td>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000456_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>886</th>\n",
       "      <td>10984</td>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000460_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>887</th>\n",
       "      <td>10985</td>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000461_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>888</th>\n",
       "      <td>10987</td>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000464_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>889</th>\n",
       "      <td>10989</td>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000466_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>890 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Unnamed: 0  FID             IID  PAT  MAT  SEX  PHENOTYPE  \\\n",
       "0             5    0  APGS_000007_s1    0    0    1          2   \n",
       "1            13    0  APGS_000018_s1    0    0    1          2   \n",
       "2            17    0  APGS_000024_s1    0    0    2          2   \n",
       "3            20    0  APGS_000027_s1    0    0    1          2   \n",
       "4            22    0  APGS_000029_s1    0    0    1          2   \n",
       "..          ...  ...             ...  ...  ...  ...        ...   \n",
       "885       10981    0   UMD_000456_s1    0    0    1          2   \n",
       "886       10984    0   UMD_000460_s1    0    0    1          2   \n",
       "887       10985    0   UMD_000461_s1    0    0    1          2   \n",
       "888       10987    0   UMD_000464_s1    0    0    1          2   \n",
       "889       10989    0   UMD_000466_s1    0    0    1          2   \n",
       "\n",
       "     chr9:129147457:C:T_T  \n",
       "0                     1.0  \n",
       "1                     1.0  \n",
       "2                     1.0  \n",
       "3                     1.0  \n",
       "4                     1.0  \n",
       "..                    ...  \n",
       "885                   1.0  \n",
       "886                   1.0  \n",
       "887                   1.0  \n",
       "888                   1.0  \n",
       "889                   1.0  \n",
       "\n",
       "[890 rows x 8 columns]"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "temp = pd.read_csv('PP2A_het_casos_129147457_EUR.csv')\n",
    "temp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "PP2A_het_controles_129147457_EUR.to_csv('PP2A_het_controles_129147457_EUR.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1204</td>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000035_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1261</td>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000099_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1302</td>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000149_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1326</td>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000186_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1360</td>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000237_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>539</th>\n",
       "      <td>10827</td>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000249_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>540</th>\n",
       "      <td>10905</td>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000352_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>541</th>\n",
       "      <td>10911</td>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000359_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>542</th>\n",
       "      <td>10929</td>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000384_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>543</th>\n",
       "      <td>10979</td>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000454_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>544 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Unnamed: 0  FID            IID  PAT  MAT  SEX  PHENOTYPE  \\\n",
       "0          1204    0  BCM_000035_s1    0    0    2          1   \n",
       "1          1261    0  BCM_000099_s1    0    0    2          1   \n",
       "2          1302    0  BCM_000149_s1    0    0    2          1   \n",
       "3          1326    0  BCM_000186_s1    0    0    1          1   \n",
       "4          1360    0  BCM_000237_s1    0    0    2          1   \n",
       "..          ...  ...            ...  ...  ...  ...        ...   \n",
       "539       10827    0  UMD_000249_s1    0    0    2          1   \n",
       "540       10905    0  UMD_000352_s1    0    0    2          1   \n",
       "541       10911    0  UMD_000359_s1    0    0    2          1   \n",
       "542       10929    0  UMD_000384_s1    0    0    2          1   \n",
       "543       10979    0  UMD_000454_s1    0    0    1          1   \n",
       "\n",
       "     chr9:129147457:C:T_T  \n",
       "0                     1.0  \n",
       "1                     1.0  \n",
       "2                     1.0  \n",
       "3                     1.0  \n",
       "4                     1.0  \n",
       "..                    ...  \n",
       "539                   1.0  \n",
       "540                   1.0  \n",
       "541                   1.0  \n",
       "542                   1.0  \n",
       "543                   1.0  \n",
       "\n",
       "[544 rows x 8 columns]"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "temp = pd.read_csv('PP2A_het_controles_129147457_EUR.csv')\n",
    "temp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                  GP2ID         GP2sampleID manifest_id phenotype pheno_for_qc\n",
      "    1: SYNAPS-KZ_000001 SYNAPS-KZ_000001_s1          m1   Control            1\n",
      "    2: SYNAPS-KZ_000002 SYNAPS-KZ_000002_s1          m1   Control            1\n",
      "    3: SYNAPS-KZ_000003 SYNAPS-KZ_000003_s1          m1   Control            1\n",
      "    4: SYNAPS-KZ_000004 SYNAPS-KZ_000004_s1          m1   Control            1\n",
      "    5: SYNAPS-KZ_000005 SYNAPS-KZ_000005_s1          m1   Control            1\n",
      "   ---                                                                        \n",
      "16853:      PPMI_000789      PPMI_000789_s1          m2     Other           -9\n",
      "16854:      PPMI_000781      PPMI_000781_s1          m2     Other           -9\n",
      "16855:      PPMI_000818      PPMI_000818_s1          m2     Other           -9\n",
      "16856:      PPMI_000841      PPMI_000841_s1          m2     Other           -9\n",
      "16857:      PPMI_000808      PPMI_000808_s1          m2     Other           -9\n",
      "                                  other_pheno sex_for_qc age age_of_onset\n",
      "    1:                                Control          2  64           NA\n",
      "    2:                                Control          1  53           NA\n",
      "    3:                                Control          2  59           NA\n",
      "    4:                                Control          1  57           NA\n",
      "    5:                                Control          1  59           NA\n",
      "   ---                                                                   \n",
      "16853:   Genetically-enriched-cohort Affected          2  NA           NA\n",
      "16854: Genetidally-enriched-cohort Unaffected          1  57           NA\n",
      "16855:   Genetically-enriched-cohort Affected          1  60           NA\n",
      "16856:   Genetically-enriched-cohort Affected          2  68           NA\n",
      "16857:   Genetically-enriched-cohort Affected          1  NA           NA\n",
      "       age_at_diagnosis age_at_death race_for_qc family_history_for_qc\n",
      "    1:               NA           NA       Asian          Not Reported\n",
      "    2:               NA           NA       Asian          Not Reported\n",
      "    3:               NA           NA       Asian          Not Reported\n",
      "    4:               NA           NA       Asian          Not Reported\n",
      "    5:               NA           NA       Asian          Not Reported\n",
      "   ---                                                                \n",
      "16853:               NA           NA       White          Not Reported\n",
      "16854:               NA           NA       White          Not Reported\n",
      "16855:               55           55       White          Not Reported\n",
      "16856:               57           57       White          Not Reported\n",
      "16857:               NA           NA       White          Not Reported\n",
      "       region_for_qc     study pruned  pruned_reason label related\n",
      "    1:           KAZ SYNAPS-KZ      0                  CAS      NA\n",
      "    2:           KAZ SYNAPS-KZ      0                  CAS      NA\n",
      "    3:           KAZ SYNAPS-KZ      0                  CAS      NA\n",
      "    4:           KAZ SYNAPS-KZ      0                  CAS      NA\n",
      "    5:           KAZ SYNAPS-KZ      0                  CAS      NA\n",
      "   ---                                                            \n",
      "16853:           ISR      PPMI      0                   AJ      NA\n",
      "16854:           ISR      PPMI      0                   AJ      NA\n",
      "16855:           ISR      PPMI      1 callrate_prune            NA\n",
      "16856:           ISR      PPMI      0                   AJ       1\n",
      "16857:           ISR      PPMI      0                   AJ      NA\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "setwd('/home/jupyter/PP2A_GP2/')\n",
    "library(data.table)\n",
    "covs <- fread(\"master_key_release3_final.csv\", header =T)\n",
    "covs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                  GP2ID                 IID manifest_id phenotype pheno_for_qc\n",
      "    1: SYNAPS-KZ_000001 SYNAPS-KZ_000001_s1          m1   Control            1\n",
      "    2: SYNAPS-KZ_000002 SYNAPS-KZ_000002_s1          m1   Control            1\n",
      "    3: SYNAPS-KZ_000003 SYNAPS-KZ_000003_s1          m1   Control            1\n",
      "    4: SYNAPS-KZ_000004 SYNAPS-KZ_000004_s1          m1   Control            1\n",
      "    5: SYNAPS-KZ_000005 SYNAPS-KZ_000005_s1          m1   Control            1\n",
      "   ---                                                                        \n",
      "16853:      PPMI_000789      PPMI_000789_s1          m2     Other           -9\n",
      "16854:      PPMI_000781      PPMI_000781_s1          m2     Other           -9\n",
      "16855:      PPMI_000818      PPMI_000818_s1          m2     Other           -9\n",
      "16856:      PPMI_000841      PPMI_000841_s1          m2     Other           -9\n",
      "16857:      PPMI_000808      PPMI_000808_s1          m2     Other           -9\n",
      "                                  other_pheno sex_for_qc age age_of_onset\n",
      "    1:                                Control          2  64           NA\n",
      "    2:                                Control          1  53           NA\n",
      "    3:                                Control          2  59           NA\n",
      "    4:                                Control          1  57           NA\n",
      "    5:                                Control          1  59           NA\n",
      "   ---                                                                   \n",
      "16853:   Genetically-enriched-cohort Affected          2  NA           NA\n",
      "16854: Genetidally-enriched-cohort Unaffected          1  57           NA\n",
      "16855:   Genetically-enriched-cohort Affected          1  60           NA\n",
      "16856:   Genetically-enriched-cohort Affected          2  68           NA\n",
      "16857:   Genetically-enriched-cohort Affected          1  NA           NA\n",
      "       age_at_diagnosis age_at_death race_for_qc family_history_for_qc\n",
      "    1:               NA           NA       Asian          Not Reported\n",
      "    2:               NA           NA       Asian          Not Reported\n",
      "    3:               NA           NA       Asian          Not Reported\n",
      "    4:               NA           NA       Asian          Not Reported\n",
      "    5:               NA           NA       Asian          Not Reported\n",
      "   ---                                                                \n",
      "16853:               NA           NA       White          Not Reported\n",
      "16854:               NA           NA       White          Not Reported\n",
      "16855:               55           55       White          Not Reported\n",
      "16856:               57           57       White          Not Reported\n",
      "16857:               NA           NA       White          Not Reported\n",
      "       region_for_qc     study pruned  pruned_reason label related\n",
      "    1:           KAZ SYNAPS-KZ      0                  CAS      NA\n",
      "    2:           KAZ SYNAPS-KZ      0                  CAS      NA\n",
      "    3:           KAZ SYNAPS-KZ      0                  CAS      NA\n",
      "    4:           KAZ SYNAPS-KZ      0                  CAS      NA\n",
      "    5:           KAZ SYNAPS-KZ      0                  CAS      NA\n",
      "   ---                                                            \n",
      "16853:           ISR      PPMI      0                   AJ      NA\n",
      "16854:           ISR      PPMI      0                   AJ      NA\n",
      "16855:           ISR      PPMI      1 callrate_prune            NA\n",
      "16856:           ISR      PPMI      0                   AJ       1\n",
      "16857:           ISR      PPMI      0                   AJ      NA\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "colnames(covs)[2]  <- \"IID\"  \n",
    "covs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                IID       GP2ID manifest_id phenotype pheno_for_qc other_pheno\n",
      "  1: APGS_000007_s1 APGS_000007          m1        PD            2          PD\n",
      "  2: APGS_000018_s1 APGS_000018          m1        PD            2          PD\n",
      "  3: APGS_000024_s1 APGS_000024          m1        PD            2          PD\n",
      "  4: APGS_000027_s1 APGS_000027          m1        PD            2          PD\n",
      "  5: APGS_000029_s1 APGS_000029          m1        PD            2          PD\n",
      " ---                                                                          \n",
      "886:  UMD_000456_s1  UMD_000456          m1        PD            2          PD\n",
      "887:  UMD_000460_s1  UMD_000460          m1        PD            2          PD\n",
      "888:  UMD_000461_s1  UMD_000461          m1        PD            2          PD\n",
      "889:  UMD_000464_s1  UMD_000464          m1        PD            2          PD\n",
      "890:  UMD_000466_s1  UMD_000466          m1        PD            2          PD\n",
      "     sex_for_qc age age_of_onset age_at_diagnosis age_at_death  race_for_qc\n",
      "  1:          1  NA           NA               NA           NA Not Reported\n",
      "  2:          1  NA           NA               NA           NA Not Reported\n",
      "  3:          2  NA           NA               NA           NA Not Reported\n",
      "  4:          1  NA           NA               NA           NA Not Reported\n",
      "  5:          1  NA           NA               NA           NA Not Reported\n",
      " ---                                                                       \n",
      "886:          1  NA           NA               NA           NA        White\n",
      "887:          1  NA           NA               NA           NA        White\n",
      "888:          1  NA           NA               NA           NA        White\n",
      "889:          1  NA           NA               NA           NA        White\n",
      "890:          1  NA           NA               NA           NA        White\n",
      "     family_history_for_qc region_for_qc study pruned pruned_reason label\n",
      "  1:          Not Reported           AUS  APGS      0            NA   EUR\n",
      "  2:          Not Reported           AUS  APGS      0            NA   EUR\n",
      "  3:          Not Reported           AUS  APGS      0            NA   EUR\n",
      "  4:          Not Reported           AUS  APGS      0            NA   EUR\n",
      "  5:          Not Reported           AUS  APGS      0            NA   EUR\n",
      " ---                                                                     \n",
      "886:                   Yes           USA   UMD      0            NA   EUR\n",
      "887:                    No           USA   UMD      0            NA   EUR\n",
      "888:                    No           USA   UMD      0            NA   EUR\n",
      "889:                    No           USA   UMD      0            NA   EUR\n",
      "890:                    No           USA   UMD      0            NA   EUR\n",
      "     related    V1 FID PAT MAT SEX PHENOTYPE chr9:129147457:C:T_T\n",
      "  1:      NA     5   0   0   0   1         2                    1\n",
      "  2:      NA    13   0   0   0   1         2                    1\n",
      "  3:      NA    17   0   0   0   2         2                    1\n",
      "  4:      NA    20   0   0   0   1         2                    1\n",
      "  5:      NA    22   0   0   0   1         2                    1\n",
      " ---                                                             \n",
      "886:      NA 10981   0   0   0   1         2                    1\n",
      "887:      NA 10984   0   0   0   1         2                    1\n",
      "888:      NA 10985   0   0   0   1         2                    1\n",
      "889:      NA 10987   0   0   0   1         2                    1\n",
      "890:      NA 10989   0   0   0   1         2                    1\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "setwd('/home/jupyter/PP2A_GP2/')\n",
    "library(data.table)\n",
    "PP2A_het_casos_129147457_EUR <- fread(\"PP2A_het_casos_129147457_EUR.csv\", header =T)\n",
    "PP2A_het_casos_129147457_EUR\n",
    "temp <- merge(covs, PP2A_het_casos_129147457_EUR, by=\"IID\")\n",
    "head(temp)\n",
    "write.table(temp, file = \"PP2A_het_casos_129147457_EUR_extracted.txt\", quote = F, row.names = F, sep = \"\\t\")\n",
    "PP2A_het_casos_129147457_EUR_extracted <- fread(\"PP2A_het_casos_129147457_EUR_extracted.txt\", header =T)\n",
    "PP2A_het_casos_129147457_EUR_extracted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "               IID      GP2ID manifest_id phenotype pheno_for_qc other_pheno\n",
      "  1: BCM_000035_s1 BCM_000035          m1   Control            1     Control\n",
      "  2: BCM_000099_s1 BCM_000099          m1   Control            1     Control\n",
      "  3: BCM_000149_s1 BCM_000149          m1   Control            1     Control\n",
      "  4: BCM_000186_s1 BCM_000186          m1   Control            1     Control\n",
      "  5: BCM_000237_s1 BCM_000237          m1   Control            1     Control\n",
      " ---                                                                        \n",
      "540: UMD_000249_s1 UMD_000249          m1   Control            1     Control\n",
      "541: UMD_000352_s1 UMD_000352          m1   Control            1     Control\n",
      "542: UMD_000359_s1 UMD_000359          m1   Control            1     Control\n",
      "543: UMD_000384_s1 UMD_000384          m1   Control            1     Control\n",
      "544: UMD_000454_s1 UMD_000454          m1   Control            1     Control\n",
      "     sex_for_qc age age_of_onset age_at_diagnosis age_at_death  race_for_qc\n",
      "  1:          2  NA           NA               NA           NA        White\n",
      "  2:          2  NA           NA               NA           NA        White\n",
      "  3:          2  NA           NA               NA           NA Multi-racial\n",
      "  4:          1  NA           NA               NA           NA        White\n",
      "  5:          2  NA           NA               NA           NA        White\n",
      " ---                                                                       \n",
      "540:          2  NA           NA               NA           NA        White\n",
      "541:          2  NA           NA               NA           NA        White\n",
      "542:          2  NA           NA               NA           NA        White\n",
      "543:          2  NA           NA               NA           NA        White\n",
      "544:          1  NA           NA               NA           NA        White\n",
      "     family_history_for_qc region_for_qc study pruned pruned_reason label\n",
      "  1:                    No           USA   BCM      0            NA   EUR\n",
      "  2:                    No           USA   BCM      0            NA   EUR\n",
      "  3:                    No           USA   BCM      0            NA   EUR\n",
      "  4:                    No           USA   BCM      0            NA   EUR\n",
      "  5:                   Yes           USA   BCM      0            NA   EUR\n",
      " ---                                                                     \n",
      "540:                    No           USA   UMD      0            NA   EUR\n",
      "541:                    No           USA   UMD      0            NA   EUR\n",
      "542:                    No           USA   UMD      0            NA   EUR\n",
      "543:                    No           USA   UMD      0            NA   EUR\n",
      "544:                    No           USA   UMD      0            NA   EUR\n",
      "     related    V1 FID PAT MAT SEX PHENOTYPE chr9:129147457:C:T_T\n",
      "  1:      NA  1204   0   0   0   2         1                    1\n",
      "  2:      NA  1261   0   0   0   2         1                    1\n",
      "  3:      NA  1302   0   0   0   2         1                    1\n",
      "  4:      NA  1326   0   0   0   1         1                    1\n",
      "  5:      NA  1360   0   0   0   2         1                    1\n",
      " ---                                                             \n",
      "540:      NA 10827   0   0   0   2         1                    1\n",
      "541:      NA 10905   0   0   0   2         1                    1\n",
      "542:      NA 10911   0   0   0   2         1                    1\n",
      "543:      NA 10929   0   0   0   2         1                    1\n",
      "544:      NA 10979   0   0   0   1         1                    1\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "setwd('/home/jupyter/PP2A_GP2/')\n",
    "library(data.table)\n",
    "PP2A_het_controles_129147457_EUR <- fread(\"PP2A_het_controles_129147457_EUR.csv\", header =T)\n",
    "PP2A_het_controles_129147457_EUR\n",
    "temp <- merge(covs, PP2A_het_controles_129147457_EUR, by=\"IID\")\n",
    "head(temp)\n",
    "write.table(temp, file = \"PP2A_het_controles_129147457_EUR_extracted.txt\", quote = F, row.names = F, sep = \"\\t\")\n",
    "PP2A_het_controles_129147457_EUR_extracted <- fread(\"PP2A_het_controles_129147457_EUR_extracted.txt\", header =T)\n",
    "PP2A_het_controles_129147457_EUR_extracted"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Save out results..!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -mu terra-02e7cd23 cp -r /home/jupyter/PP2A_GP2/ gs://fc-a2382a6f-59cf-4cf3-b70e-8cd83806d5e0\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Copying file:///home/jupyter/PP2A_GP2/reduced_coding_nonsynonymous.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_controles_129147457_EUR_extracted.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AJ_rare_0.05.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_CAS.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AFR.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AAC_rare_0.05.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_EAS.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AJ.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/reduced_coding_nonsynonymous_AFR.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AMR.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AAC.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AAC.raw [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_SAS.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/reduced_coding_nonsynonymous_AMR.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_MDE.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_casos_129129094_AJ_extracted.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_rare_0.05.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_MDE.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AAC_release3.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_controles_129147457_AJ_extracted.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_hom_casos_129147457_AJ.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AJ_rare_0.05.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_ACC.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_casos_129129094_AJ.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_CAS.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_CAS_release3.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_SAS_rare_0.05.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AFR.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AFR.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_CAS.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_SAS.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AJ.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_CAS.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_MDE.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_EUR.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_EAS.assoc.adjusted [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AJ_rare_0.05.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AJ_release3.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AMR_rare_0.05.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_MDE.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AJ_release3.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AAC_release3.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AMR.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AJ.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_casos_129147457_AJ_extracted.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_CAS_release3.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AAC.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_hom_casos_129147457_AJ_extracted.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/reduced_coding_nonsynonymous_AJ.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AFR_rare_0.05.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AFR.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AJ.assoc [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AFR.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AFR.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_EAS.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_EAS_release3.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_129147457_casos_EUR.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AJ.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_MDE.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AFR_release3.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AMR_release3.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_EUR_release3.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_SAS.raw [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/reduced_coding_nonsynonymous_CAS.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_MDE_release3.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AMR.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_EUR.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_SAS.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AAC.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AMR_rare_0.05.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AJ.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AMR_release3.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AFR_rare_0.05.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AFR.raw [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AFR_rare_0.05.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AJ_release3.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/master_key_release3_final.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_CAS.log [Content-Type=application/octet-stream]...\n",
      "==> NOTE: You are uploading one or more large file(s), which would run          \n",
      "significantly faster if you enable parallel composite uploads. This\n",
      "feature can be enabled by editing the\n",
      "\"parallel_composite_upload_threshold\" value in your .boto\n",
      "configuration file. However, note that if you do this large files will\n",
      "be uploaded as `composite objects\n",
      "<https://cloud.google.com/storage/docs/composite-objects>`_,which\n",
      "means that any user who downloads such objects will need to have a\n",
      "compiled crcmod installed (see \"gsutil help crcmod\"). This is because\n",
      "without a compiled crcmod, computing checksums on composite objects is\n",
      "so slow that gsutil disables downloads of composite objects.\n",
      "\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_CAS_rare_0.05.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_MDE_release3.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_MDE.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_EAS.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AAC.assoc [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AFR.assoc [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_CAS.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_CAS.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AFR_release3.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_MDE.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AMR.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AAC.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/reduced_coding_nonsynonymous_EAS.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AMR.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_SAS.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_EAS_release3.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_SAS_release3.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AFR_rare_0.05.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_SAS.assoc.adjusted [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AAC.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AAC.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_EUR.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AAC_rare_0.05.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AMR.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_CAS.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AAC_rare_0.05.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AAC_rare_0.05.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_CAS.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AMR.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_CAS.raw [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_EAS.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_casos_129147457_AJ.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AFR_release3.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AMR.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_rare_0.05.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_SAS.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_EAS.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PODXL.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_SAS_release3.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_rare_0.05.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AJ.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_EAS.raw [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AMR.assoc.adjusted [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AJ_release3.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_CAS_release3.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/SEX.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AMR.raw [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_CAS_release3.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_SAS.assoc [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_MDE.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_EUR_release3.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_EAS_release3.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AFR.assoc.adjusted [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AMR_rare_0.05.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_CAS.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AJ.assoc.adjusted [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AFR.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AFR.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_EAS.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_EAS.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AJ.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AJ.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AMR.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AMR.assoc [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_129147457_controles_EUR.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_EAS.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_SAS_release3.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/reduced_coding_nonsynonymous_MDE.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_CAS.assoc [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PHENO.txt [Content-Type=text/plain]...    \n",
      "Copying file:///home/jupyter/PP2A_GP2/PTPA_het_129147457_controles_extracted.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_MDE.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AMR.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AAC_release3.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AAC.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_EAS.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A.assoc [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AAC.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_CAS.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AFR.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_EAS_release3.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_EAS.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_MDE_rare_0.05.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AAC.assoc.adjusted [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AFR.vcf.gz [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AJ.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AJ.raw [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_MDE.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_SAS.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_controles_129129094_AJ.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_MDE.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_casos_129147457_EUR.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_EUR.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_EUR.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AJ.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_SAS.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/GP2_AJ_comphet.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_MDE_release3.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AMR.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_SAS.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_controles_129147457_EUR.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_SAS.fam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_controles_129147457_AJ.csv [Content-Type=text/csv]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/GP2_AMR_comphet.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_SAS.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_EUR_release3.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AFR_release3.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AFR.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AAC.annovar.hg38_multianno.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_MDE_release3.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AMR_release3.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AMR_release3.psam [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A.raw [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_SAS_release3.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_CAS.assoc.adjusted [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A.assoc.adjusted [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AJ_rare_0.05.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_AAC.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PODXL.assoc [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_pheno_PP2A_EAS.assoc [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A.bim [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_EUR_release3.pgen [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_EAS_rare_0.05.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AAC.annovar.hg38_multianno.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_SAS.annovar.avinput [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_AJ.vcf.gz.tbi [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/pheno_PP2A_EAS.vcf [Content-Type=text/vcard]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/reduced_coding_nonsynonymous_SAS.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/PP2A_het_casos_129147457_EUR_extracted.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_AMR_rare_0.05.bed [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/coding_nonsynonymous_rare_0.05.log [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/chr9_AAC_release3.pvar [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/PP2A_GP2/reduced_coding_nonsynonymous_AAC.txt [Content-Type=text/plain]...\n",
      "/ [222/222 files][  6.1 GiB/  6.1 GiB] 100% Done  84.0 MiB/s ETA 00:00:00       \n",
      "Operation completed over 222 objects/6.1 GiB.                                    \n"
     ]
    }
   ],
   "source": [
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR} {WORKSPACE_BUCKET}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.12"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {
    "height": "251px",
    "width": "261px"
   },
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
