{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# PP2A - Single gene analysis in GP2 data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Description\n",
    "\n",
    "Using \"individual level data\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 0. Getting Started\n",
    "\n",
    "- Loading Python libraries\n",
    "- Defining functions\n",
    "- Installing packages\n",
    "\n",
    "### 1. Copy data from workspace to cloud environment\n",
    "\n",
    "### 2. Extract PP2A\n",
    "\n",
    "### 3. Annotate PP2A variants\n",
    "\n",
    "### 4. Extract coding/non-syn variants\n",
    "\n",
    "### 5. Calculate frequency in cases versus controls\n",
    "\n",
    "### 6. Calculate frequency (homozygotes) in cases versus controls\n",
    "\n",
    "### 7. Save out results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Getting Started"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Loading Python libraries"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Defining functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "code_folding": []
   },
   "outputs": [],
   "source": [
    "# Use the os package to interact with the environment\n",
    "import os\n",
    "\n",
    "# Bring in Pandas for Dataframe functionality\n",
    "import pandas as pd\n",
    "\n",
    "# Numpy for basics\n",
    "import numpy as np\n",
    "\n",
    "# Use StringIO for working with file contents\n",
    "from io import StringIO\n",
    "\n",
    "# Enable IPython to display matplotlib graphs\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline\n",
    "\n",
    "# Enable interaction with the FireCloud API\n",
    "from firecloud import api as fapi\n",
    "\n",
    "# Import the iPython HTML rendering for displaying links to Google Cloud Console\n",
    "from IPython.core.display import display, HTML\n",
    "\n",
    "# Import urllib modules for building URLs to Google Cloud Console\n",
    "import urllib.parse\n",
    "\n",
    "# BigQuery for querying data\n",
    "from google.cloud import bigquery\n",
    "\n",
    "#Import Sys\n",
    "import sys as sys"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "code_folding": [
     0
    ]
   },
   "outputs": [],
   "source": [
    "# Utility routine for printing a shell command before executing it\n",
    "def shell_do(command):\n",
    "    print(f'Executing: {command}', file=sys.stderr)\n",
    "    !$command\n",
    "    \n",
    "def shell_return(command):\n",
    "    print(f'Executing: {command}', file=sys.stderr)\n",
    "    output = !$command\n",
    "    return '\\n'.join(output)\n",
    "\n",
    "# Utility routine for printing a query before executing it\n",
    "def bq_query(query):\n",
    "    print(f'Executing: {query}', file=sys.stderr)\n",
    "    return pd.read_gbq(query, project_id=BILLING_PROJECT_ID, dialect='standard')\n",
    "\n",
    "# Utility routine for display a message and a link\n",
    "def display_html_link(description, link_text, url):\n",
    "    html = f'''\n",
    "    <p>\n",
    "    </p>\n",
    "    <p>\n",
    "    {description}\n",
    "    <a target=_blank href=\"{url}\">{link_text}</a>.\n",
    "    </p>\n",
    "    '''\n",
    "\n",
    "    display(HTML(html))\n",
    "\n",
    "# Utility routines for reading files from Google Cloud Storage\n",
    "def gcs_read_file(path):\n",
    "    \"\"\"Return the contents of a file in GCS\"\"\"\n",
    "    contents = !gsutil -u {BILLING_PROJECT_ID} cat {path}\n",
    "    return '\\n'.join(contents)\n",
    "    \n",
    "def gcs_read_csv(path, sep=None):\n",
    "    \"\"\"Return a DataFrame from the contents of a delimited file in GCS\"\"\"\n",
    "    return pd.read_csv(StringIO(gcs_read_file(path)), sep=sep, engine='python')\n",
    "\n",
    "# Utility routine for displaying a message and link to Cloud Console\n",
    "def link_to_cloud_console_gcs(description, link_text, gcs_path):\n",
    "    url = '{}?{}'.format(\n",
    "        os.path.join('https://console.cloud.google.com/storage/browser',\n",
    "                     gcs_path.replace(\"gs://\",\"\")),\n",
    "        urllib.parse.urlencode({'userProject': BILLING_PROJECT_ID}))\n",
    "\n",
    "    display_html_link(description, link_text, url)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Set paths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "code_folding": [
     0
    ]
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Billing and Workspace\n",
      "Workspace Name: PP2A\n",
      "Billing Project: terra-02e7cd23\n",
      "Workspace Bucket, where you can upload and download data: gs://fc-a2382a6f-59cf-4cf3-b70e-8cd83806d5e0\n",
      "\n",
      "GP2 v3.0\n",
      "Path to GP2 v3.0 Clinical Data: gs://gp2tier2/release3_31102022/clinical_data\n",
      "Path to GP2 v3.0 Raw Genotype Data: gs://gp2tier2/release3_31102022/raw_genotypes\n",
      "Path to GP2 v3.0 Imputed Genotype Data: gs://gp2tier2/release3_31102022/imputed_genotypes\n",
      "AMP-PD v3.0\n",
      "Path to AMP-PD v3.0 Clinical Data: gs://amp-pd-data/releases/2022_v3release_1115/clinical\n",
      "Path to AMP-PD v3.0 WGS Data: gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink\n",
      "Path to AMP-PD v3.0 WGS Data: gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS/plink/pfiles\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Set up billing project and data path variables\n",
    "BILLING_PROJECT_ID = os.environ['GOOGLE_PROJECT']\n",
    "WORKSPACE_NAMESPACE = os.environ['WORKSPACE_NAMESPACE']\n",
    "WORKSPACE_NAME = os.environ['WORKSPACE_NAME']\n",
    "WORKSPACE_BUCKET = os.environ['WORKSPACE_BUCKET']\n",
    "\n",
    "WORKSPACE_ATTRIBUTES = fapi.get_workspace(WORKSPACE_NAMESPACE, WORKSPACE_NAME).json().get('workspace',{}).get('attributes',{})\n",
    "\n",
    "## Print the information to check we are in the proper release and billing \n",
    "## This will be different for you, the user, depending on the billing project your workspace is on\n",
    "print('Billing and Workspace')\n",
    "print(f'Workspace Name: {WORKSPACE_NAME}')\n",
    "print(f'Billing Project: {BILLING_PROJECT_ID}')\n",
    "print(f'Workspace Bucket, where you can upload and download data: {WORKSPACE_BUCKET}')\n",
    "print('')\n",
    "\n",
    "\n",
    "## GP2 v3.0\n",
    "## Explicitly define release v3.0 path \n",
    "GP2_RELEASE_PATH = 'gs://gp2tier2/release3_31102022'\n",
    "GP2_CLINICAL_RELEASE_PATH = f'{GP2_RELEASE_PATH}/clinical_data'\n",
    "GP2_RAW_GENO_PATH = f'{GP2_RELEASE_PATH}/raw_genotypes'\n",
    "GP2_IMPUTED_GENO_PATH = f'{GP2_RELEASE_PATH}/imputed_genotypes'\n",
    "print('GP2 v3.0')\n",
    "print(f'Path to GP2 v3.0 Clinical Data: {GP2_CLINICAL_RELEASE_PATH}')\n",
    "print(f'Path to GP2 v3.0 Raw Genotype Data: {GP2_RAW_GENO_PATH}')\n",
    "print(f'Path to GP2 v3.0 Imputed Genotype Data: {GP2_IMPUTED_GENO_PATH}')\n",
    "\n",
    "\n",
    "## AMP-PD v3.0\n",
    "## Explicitly define release v3.0 path \n",
    "AMP_RELEASE_PATH = 'gs://amp-pd-data/releases/2022_v3release_1115'\n",
    "AMP_CLINICAL_RELEASE_PATH = f'{AMP_RELEASE_PATH}/clinical'\n",
    "\n",
    "AMP_WGS_RELEASE_PATH = 'gs://amp-pd-genomics/releases/2022_v3release_1115/wgs-WB-DWGS'\n",
    "AMP_WGS_RELEASE_PLINK_PATH = os.path.join(AMP_WGS_RELEASE_PATH, 'plink')\n",
    "AMP_WGS_RELEASE_PLINK_PFILES = os.path.join(AMP_WGS_RELEASE_PLINK_PATH, 'pfiles')\n",
    "\n",
    "print('AMP-PD v3.0')\n",
    "print(f'Path to AMP-PD v3.0 Clinical Data: {AMP_CLINICAL_RELEASE_PATH}')\n",
    "print(f'Path to AMP-PD v3.0 WGS Data: {AMP_WGS_RELEASE_PLINK_PATH}')\n",
    "print(f'Path to AMP-PD v3.0 WGS Data: {AMP_WGS_RELEASE_PLINK_PFILES}')\n",
    "print('')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Install packages"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Install Plink 1.9 and Plink 2.0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Plink1.9 is already installed in /home/jupyter/tools/\n",
      "Plink2 is already installed in /home/jupyter/tools/\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "mkdir -p ~/tools\n",
    "cd ~/tools\n",
    "\n",
    "if test -e /home/jupyter/tools/plink; then\n",
    "echo \"Plink1.9 is already installed in /home/jupyter/tools/\"\n",
    "\n",
    "else\n",
    "echo -e \"Downloading plink \\n    -------\"\n",
    "wget -N http://s3.amazonaws.com/plink1-assets/plink_linux_x86_64_20190304.zip \n",
    "unzip -o plink_linux_x86_64_20190304.zip\n",
    "echo -e \"\\n plink downloaded and unzipped in /home/jupyter/tools \\n \"\n",
    "\n",
    "fi\n",
    "\n",
    "\n",
    "if test -e /home/jupyter/tools/plink2; then\n",
    "echo \"Plink2 is already installed in /home/jupyter/tools/\"\n",
    "\n",
    "else\n",
    "echo -e \"Downloading plink2 \\n    -------\"\n",
    "wget -N https://s3.amazonaws.com/plink2-assets/alpha3/plink2_linux_avx2_20220603.zip\n",
    "unzip -o plink2_linux_avx2_20220603.zip\n",
    "echo -e \"\\n plink2 downloaded and unzipped in /home/jupyter/tools \\n \"\n",
    "\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annovar\n",
      "annovar.latest.tar.gz\n",
      "LICENSE\n",
      "plink\n",
      "plink2\n",
      "plink2_linux_avx2_20220603.zip\n",
      "plink_linux_x86_64_20190304.zip\n",
      "prettify\n",
      "toy.map\n",
      "toy.ped\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls /home/jupyter/tools/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Remote restrictions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "# chmod plink 1.9 \n",
    "chmod u+x /home/jupyter/tools/plink"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "# chmod plink 2.0\n",
    "chmod u+x /home/jupyter/tools/plink2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Install ANNOVAR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annovar is already installed in /home/jupyter/tools/\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Install ANNOVAR:\n",
    "# https://www.openbioinformatics.org/annovar/annovar_download_form.php\n",
    "\n",
    "if test -e /home/jupyter/tools/annovar; then\n",
    "\n",
    "echo \"annovar is already installed in /home/jupyter/tools/\"\n",
    "else\n",
    "echo \"annovar is not installed\"\n",
    "cd /home/jupyter/tools/\n",
    "\n",
    "wget http://www.openbioinformatics.org/annovar/download/0wgxR2rIVP/annovar.latest.tar.gz\n",
    "\n",
    "tar xvfz annovar.latest.tar.gz\n",
    "\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annovar\n",
      "annovar.latest.tar.gz\n",
      "LICENSE\n",
      "plink\n",
      "plink2\n",
      "plink2_linux_avx2_20220603.zip\n",
      "plink_linux_x86_64_20190304.zip\n",
      "prettify\n",
      "toy.map\n",
      "toy.ped\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls /home/jupyter/tools/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Install ANNOVAR: Download sources of annotation "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "NOTICE: Web-based checking to see whether ANNOVAR new version is available ... Done\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_refGene.txt.gz ... OK\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_refGeneMrna.fa.gz ... OK\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_refGeneVersion.txt.gz ... OK\n",
      "NOTICE: Uncompressing downloaded files\n",
      "NOTICE: Finished downloading annotation files for hg38 build version, with files saved at the 'humandb' directory\n",
      "NOTICE: Web-based checking to see whether ANNOVAR new version is available ... Done\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_clinvar_20140902.txt.gz ... OK\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_clinvar_20140902.txt.idx.gz ... OK\n",
      "NOTICE: Uncompressing downloaded files\n",
      "NOTICE: Finished downloading annotation files for hg38 build version, with files saved at the 'humandb' directory\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "cd /home/jupyter/tools/annovar/\n",
    "\n",
    "perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar refGene humandb/\n",
    "perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar clinvar_20140902 humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb cytoBand humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar ensGene humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar exac03 humandb/ \n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar avsnp147 humandb/ \n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar dbnsfp30a humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar gnomad211_genome humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar ljb26_all humandb/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annotate_variation.pl\n",
      "coding_change.pl\n",
      "convert2annovar.pl\n",
      "example\n",
      "humandb\n",
      "retrieve_seq_from_fasta.pl\n",
      "table_annovar.pl\n",
      "variants_reduction.pl\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls /home/jupyter/tools/annovar/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## AJ"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Copy data from GP2 bucket to workspace"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Making a working directory\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: mkdir -p /home/jupyter/PP2A_GP2/\n"
     ]
    }
   ],
   "source": [
    "# Make a directory\n",
    "print(\"Making a working directory\")\n",
    "WORK_DIR = f'/home/jupyter/PP2A_GP2/'\n",
    "shell_do(f'mkdir -p {WORK_DIR}') # f' significa f-string - contiene expresiones para ejecutar el codigo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -u terra-02e7cd23 ls gs://gp2tier2/release3_31102022/imputed_genotypes\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gs://gp2tier2/release3_31102022/imputed_genotypes/AAC/\r\n",
      "gs://gp2tier2/release3_31102022/imputed_genotypes/AFR/\r\n",
      "gs://gp2tier2/release3_31102022/imputed_genotypes/AJ/\r\n",
      "gs://gp2tier2/release3_31102022/imputed_genotypes/AMR/\r\n",
      "gs://gp2tier2/release3_31102022/imputed_genotypes/CAS/\r\n",
      "gs://gp2tier2/release3_31102022/imputed_genotypes/EAS/\r\n",
      "gs://gp2tier2/release3_31102022/imputed_genotypes/EUR/\r\n",
      "gs://gp2tier2/release3_31102022/imputed_genotypes/MDE/\r\n",
      "gs://gp2tier2/release3_31102022/imputed_genotypes/SAS/\r\n"
     ]
    }
   ],
   "source": [
    "# Check directory where AMP-PD data is\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} ls {GP2_IMPUTED_GENO_PATH}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -u terra-02e7cd23 -m cp -r gs://gp2tier2/release3_31102022/imputed_genotypes/AJ/chr9* /home/jupyter/PP2A_GP2/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Copying gs://gp2tier2/release3_31102022/imputed_genotypes/AJ/chr9_AJ_release3.psam...\n",
      "Copying gs://gp2tier2/release3_31102022/imputed_genotypes/AJ/chr9_AJ_release3.pgen...\n",
      "Copying gs://gp2tier2/release3_31102022/imputed_genotypes/AJ/chr9_AJ_release3.pvar...\n",
      "Copying gs://gp2tier2/release3_31102022/imputed_genotypes/AJ/chr9_AJ_release3.log...\n",
      "| [4/4 files][300.0 MiB/300.0 MiB] 100% Done                                    \n",
      "Operation completed over 4 objects/300.0 MiB.                                    \n"
     ]
    }
   ],
   "source": [
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp -r {GP2_IMPUTED_GENO_PATH}/AJ/chr9* {WORK_DIR}')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Create a covariate file with GP2 data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -u terra-02e7cd23 ls gs://gp2tier2/release3_31102022/clinical_data\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gs://gp2tier2/release3_31102022/clinical_data/\r\n",
      "gs://gp2tier2/release3_31102022/clinical_data/master_key_release3_final.csv\r\n",
      "gs://gp2tier2/release3_31102022/clinical_data/release3_31102022_data_dictionary.csv\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -u terra-02e7cd23 -m cp -r gs://gp2tier2/release3_31102022/clinical_data/master_key_release3_final.csv /home/jupyter/PP2A_GP2/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Copying gs://gp2tier2/release3_31102022/clinical_data/master_key_release3_final.csv...\n",
      "/ [1/1 files][  1.6 MiB/  1.6 MiB] 100% Done                                    \n",
      "Operation completed over 1 objects/1.6 MiB.                                      \n"
     ]
    }
   ],
   "source": [
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} ls {GP2_CLINICAL_RELEASE_PATH}')\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp -r {GP2_CLINICAL_RELEASE_PATH}/master_key_release3_final.csv {WORK_DIR}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['GP2ID', 'GP2sampleID', 'manifest_id', 'phenotype', 'pheno_for_qc',\n",
       "       'other_pheno', 'sex_for_qc', 'age', 'age_of_onset', 'age_at_diagnosis',\n",
       "       'age_at_death', 'race_for_qc', 'family_history_for_qc', 'region_for_qc',\n",
       "       'study', 'pruned', 'pruned_reason', 'label', 'related'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cov = pd.read_csv(f'{WORK_DIR}/master_key_release3_final.csv')\n",
    "cov.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>SEX</th>\n",
       "      <th>AGE</th>\n",
       "      <th>PHENO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SYNAPS-KZ_000001_s1</td>\n",
       "      <td>SYNAPS-KZ_000001_s1</td>\n",
       "      <td>2</td>\n",
       "      <td>64.0</td>\n",
       "      <td>Control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SYNAPS-KZ_000002_s1</td>\n",
       "      <td>SYNAPS-KZ_000002_s1</td>\n",
       "      <td>1</td>\n",
       "      <td>53.0</td>\n",
       "      <td>Control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SYNAPS-KZ_000003_s1</td>\n",
       "      <td>SYNAPS-KZ_000003_s1</td>\n",
       "      <td>2</td>\n",
       "      <td>59.0</td>\n",
       "      <td>Control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SYNAPS-KZ_000004_s1</td>\n",
       "      <td>SYNAPS-KZ_000004_s1</td>\n",
       "      <td>1</td>\n",
       "      <td>57.0</td>\n",
       "      <td>Control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SYNAPS-KZ_000005_s1</td>\n",
       "      <td>SYNAPS-KZ_000005_s1</td>\n",
       "      <td>1</td>\n",
       "      <td>59.0</td>\n",
       "      <td>Control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16852</th>\n",
       "      <td>PPMI_000789_s1</td>\n",
       "      <td>PPMI_000789_s1</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16853</th>\n",
       "      <td>PPMI_000781_s1</td>\n",
       "      <td>PPMI_000781_s1</td>\n",
       "      <td>1</td>\n",
       "      <td>57.0</td>\n",
       "      <td>Other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16854</th>\n",
       "      <td>PPMI_000818_s1</td>\n",
       "      <td>PPMI_000818_s1</td>\n",
       "      <td>1</td>\n",
       "      <td>60.0</td>\n",
       "      <td>Other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16855</th>\n",
       "      <td>PPMI_000841_s1</td>\n",
       "      <td>PPMI_000841_s1</td>\n",
       "      <td>2</td>\n",
       "      <td>68.0</td>\n",
       "      <td>Other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16856</th>\n",
       "      <td>PPMI_000808_s1</td>\n",
       "      <td>PPMI_000808_s1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Other</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>16857 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                       FID                  IID  SEX   AGE    PHENO\n",
       "0      SYNAPS-KZ_000001_s1  SYNAPS-KZ_000001_s1    2  64.0  Control\n",
       "1      SYNAPS-KZ_000002_s1  SYNAPS-KZ_000002_s1    1  53.0  Control\n",
       "2      SYNAPS-KZ_000003_s1  SYNAPS-KZ_000003_s1    2  59.0  Control\n",
       "3      SYNAPS-KZ_000004_s1  SYNAPS-KZ_000004_s1    1  57.0  Control\n",
       "4      SYNAPS-KZ_000005_s1  SYNAPS-KZ_000005_s1    1  59.0  Control\n",
       "...                    ...                  ...  ...   ...      ...\n",
       "16852       PPMI_000789_s1       PPMI_000789_s1    2   NaN    Other\n",
       "16853       PPMI_000781_s1       PPMI_000781_s1    1  57.0    Other\n",
       "16854       PPMI_000818_s1       PPMI_000818_s1    1  60.0    Other\n",
       "16855       PPMI_000841_s1       PPMI_000841_s1    2  68.0    Other\n",
       "16856       PPMI_000808_s1       PPMI_000808_s1    1   NaN    Other\n",
       "\n",
       "[16857 rows x 5 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cov_reduced = cov[['GP2sampleID','GP2sampleID','sex_for_qc', 'age', 'phenotype']]\n",
    "cov_reduced.columns = ['FID','IID', 'SEX','AGE','PHENO']\n",
    "cov_reduced"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "conditions = [\n",
    "    (cov_reduced['PHENO'] == \"Case\"),\n",
    "    (cov_reduced['PHENO'] == \"Control\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.7/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  \n"
     ]
    }
   ],
   "source": [
    "choices = [2,1]\n",
    "cov_reduced['PHENOTYPE'] = np.select(conditions, choices, default=-9).astype(np.int64)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.7/site-packages/pandas/core/frame.py:4913: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  errors=errors,\n"
     ]
    }
   ],
   "source": [
    "cov_reduced.reset_index(inplace=True)\n",
    "cov_reduced.drop(columns=[\"index\"], inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "cov_reduced.drop(columns=['PHENO'], inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "sex = cov_reduced[['FID','IID','SEX']]\n",
    "sex.to_csv(f'{WORK_DIR}/SEX.txt',index=False, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "pheno = cov_reduced[['FID','IID','PHENOTYPE']]\n",
    "pheno.to_csv(f'{WORK_DIR}/PHENO.txt',index=False, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extract PP2A"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a3.3LM AVX2 Intel (3 Jun 2022)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2022 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to pheno_PP2A_AJ.log.\n",
      "Options in effect:\n",
      "  --chr 9\n",
      "  --from-bp 129110950\n",
      "  --make-bed\n",
      "  --out pheno_PP2A_AJ\n",
      "  --pfile chr9_AJ_release3\n",
      "  --to-bp 129148946\n",
      "\n",
      "Start time: Sun Jun  4 21:13:31 2023\n",
      "14998 MiB RAM detected; reserving 7499 MiB for main workspace.\n",
      "Using up to 4 compute threads.\n",
      "1285 samples (477 females, 808 males; 1285 founders) loaded from\n",
      "chr9_AJ_release3.psam.\n",
      "524194 variants loaded from chr9_AJ_release3.pvar.\n",
      "1 binary phenotype loaded (600 cases, 323 controls).\n",
      "177 variants remaining after main filters.\n",
      "Writing pheno_PP2A_AJ.fam ... done.\n",
      "Writing pheno_PP2A_AJ.bim ... done.\n",
      "Writing pheno_PP2A_AJ.bed ... 0%\b\bdone.\n",
      "End time: Sun Jun  4 21:13:31 2023\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "# PODXL: gene positions on hg38 (from https://useast.ensembl.org/index.html)\n",
    "/home/jupyter/tools/plink2 \\\n",
    "--pfile chr9_AJ_release3 \\\n",
    "--chr 9 \\\n",
    "--from-bp 129110950  \\\n",
    "--to-bp 129148946 \\\n",
    "--make-bed \\\n",
    "--out pheno_PP2A_AJ"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CORIELL_003933,CORIELL_003933_s1,m3,PD,2,PD,2,78.0,60.0,,,White,Yes,USA,CORIELL,0,,AJ,\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "grep -e \"CORIELL_003933_s1\" master_key_release3_final.csv > GP2_AJ_comphet.txt\n",
    "head GP2_AJ_comphet.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Visualize plink files bim, fam and bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9\tchr9:129110977:A:C\t0\t129110977\tC\tA\n",
      "9\tchr9:129111323:C:G\t0\t129111323\tG\tC\n",
      "9\tchr9:129111384:C:A\t0\t129111384\tA\tC\n",
      "9\tchr9:129111646:GC:G\t0\t129111646\tG\tGC\n",
      "9\tchr9:129112273:A:T\t0\t129112273\tT\tA\n",
      "9\tchr9:129112362:A:G\t0\t129112362\tG\tA\n",
      "9\tchr9:129112974:T:C\t0\t129112974\tC\tT\n",
      "9\tchr9:129113056:T:TA\t0\t129113056\tTA\tT\n",
      "9\tchr9:129113634:GTGGCACACGTC:G\t0\t129113634\tG\tGTGGCACACGTC\n",
      "9\tchr9:129113698:A:T\t0\t129113698\tT\tA\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Visualize bim file\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "head pheno_PP2A_AJ.bim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\tAPGS_000021_s1\t0\t0\t1\t2\n",
      "0\tAPGS_000475_s1\t0\t0\t1\t2\n",
      "0\tAPGS_000486_s1\t0\t0\t1\t2\n",
      "0\tAPGS_000646_s1\t0\t0\t2\t2\n",
      "0\tAPGS_000808_s1\t0\t0\t1\t2\n",
      "0\tAPGS_001017_s1\t0\t0\t1\t2\n",
      "0\tAPGS_001059_s1\t0\t0\t1\t2\n",
      "0\tAPGS_001066_s1\t0\t0\t2\t2\n",
      "0\tAPGS_001426_s1\t0\t0\t2\t2\n",
      "0\tAPGS_001474_s1\t0\t0\t2\t2\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Visualize fam file\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "head pheno_PP2A_AJ.fam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "l\u001b\u0001�������>�����:/��+��������ꫮ����/�;�����������+���������<��������+����������ꪫ������ί������������ξ����������Ͽ������ﾸ\u000e쏺�����������������?�ﺿ�������:�����������?�����Ϻ����������.�����������������������������>����?���������?���+��������.>��������?꿻�������8�Ϊ꾿�����ʈ��������캯������ﺪ������\u0003���������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������\u0003���������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������\u0003��������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������\u0003\u0000�\u0002\"\u000e�\u0000��,�\b��*���\u0003舨���� #*��\u0002*�:(�(Ȉ\u0000;* \u000b",
      "\u0002.\b�*耊�2\b*\n",
      "��à\n",
      "\n",
      "0\"\u0000� ��\u0002��\f",
      "((�\u000e(\"\u0002*��\b\u0000���\u00002�\u0002�<\n",
      "\n",
      "�\b���; �2�\b�\b\u0002���\"�\b\u00000��(\u0003\b�\n",
      " ���+��\b\b�\u0000\b����\n",
      "��������� ��\u0000\u000e�\b\u0000(#�� \"\u0000\u0002\b��\u0002\"\u0002*��:8\b�0�� \u000e \u0000��(�\b��\u0000�\b��8\u0002\u0000���\b��\u0002� \b\n",
      "*\b�\u0002��;*\u0000¢\u0002\u0000\u000b",
      "�\u0000(\"�*�\u0000\u000e\u0000�*�����\u0002\f",
      "\u0000\b��\u0002\u0002�� \n",
      "�(�\b\u000b",
      "\u0002\u0002�*��\b\b;(\b��ˈ2�*��\u0002\u0002\b�*�:�\u0002*\u0000\u0000 � �2\u0000#��\u0000� \b#.� ��*��\b\b\u0000\u0000\u0000�\u0002\"\u000e�\u0000.�,�\b��*���\u0003舨���� #*��\u0002*�:(�(Ȉ\u0000;* \u000b",
      "\u0002.\b�*耊�2\b*\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Visualize fam file\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "head pheno_PP2A_AJ.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to pheno_PP2A_AJ.log.\n",
      "Options in effect:\n",
      "  --bfile pheno_PP2A_AJ\n",
      "  --out pheno_PP2A_AJ\n",
      "  --recode vcf-fid\n",
      "\n",
      "14998 MB RAM detected; reserving 7499 MB for main workspace.\n",
      "177 variants loaded from .bim file.\n",
      "1285 people (808 males, 477 females) loaded from .fam.\n",
      "923 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1285 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.998958.\n",
      "177 variants and 1285 people pass filters and QC.\n",
      "Among remaining phenotypes, 600 are cases and 323 are controls.  (362\n",
      "phenotypes are missing.)\n",
      "--recode vcf-fid to pheno_PP2A_AJ.vcf ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "# Turn binary files into VCF\n",
    "/home/jupyter/tools/plink \\\n",
    "--bfile pheno_PP2A_AJ \\\n",
    "--recode vcf-fid \\\n",
    "--out pheno_PP2A_AJ"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Annotate PP2A variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[bgzip] can't create pheno_PP2A_AJ.vcf.gz: File exists\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "export PATH=$PATH:/home/jupyter/tools/rvtests/third/tabix-0.2.6/\n",
    "\n",
    "### Bgzip and Tabix\n",
    "bgzip pheno_PP2A_AJ.vcf\n",
    "\n",
    "tabix -f -p vcf pheno_PP2A_AJ.vcf.gz\n",
    "tabix -f -p vcf pheno_PP2A_AJ.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 pheno_PP2A_AJ.vcf.gz > pheno_PP2A_AJ.annovar.avinput>\n",
      "NOTICE: Finished reading 184 lines from VCF file\n",
      "NOTICE: A total of 177 locus in VCF file passed QC threshold, representing 161 SNPs (118 transitions and 43 transversions) and 16 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 206885 SNP genotypes (151630 transitions and 55255 transversions) and 20560 indels/substitutions for 1285 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl pheno_PP2A_AJ.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile pheno_PP2A_AJ.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile pheno_PP2A_AJ.annovar.refGene -exonsort -nofirstcodondel pheno_PP2A_AJ.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to pheno_PP2A_AJ.annovar.refGene.variant_function, pheno_PP2A_AJ.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 177 unique variants in 177 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 6 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile pheno_PP2A_AJ.annovar pheno_PP2A_AJ.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to pheno_PP2A_AJ.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to pheno_PP2A_AJ.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 177 unique variants in 177 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 0\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to pheno_PP2A_AJ.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from pheno_PP2A_AJ.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to pheno_PP2A_AJ.annovar.hg38_multianno.vcf\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "### Annotate variants using ANNOVAR: https://annovar.openbioinformatics.org/en/latest/ \n",
    "perl /home/jupyter/tools/annovar/table_annovar.pl pheno_PP2A_AJ.vcf.gz /home/jupyter/tools/annovar/humandb/ -buildver hg38 \\\n",
    "-out pheno_PP2A_AJ.annovar \\\n",
    "-remove -protocol refGene,clinvar_20140902 \\\n",
    "-operation g,f \\\n",
    "--nopolish \\\n",
    "-nastring . \\\n",
    "-vcfinput"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extract coding and non-syn variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>Otherinfo1288</th>\n",
       "      <th>Otherinfo1289</th>\n",
       "      <th>Otherinfo1290</th>\n",
       "      <th>Otherinfo1291</th>\n",
       "      <th>Otherinfo1292</th>\n",
       "      <th>Otherinfo1293</th>\n",
       "      <th>Otherinfo1294</th>\n",
       "      <th>Otherinfo1295</th>\n",
       "      <th>Otherinfo1296</th>\n",
       "      <th>Otherinfo1297</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9</td>\n",
       "      <td>129110977</td>\n",
       "      <td>129110977</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>PTPA</td>\n",
       "      <td>NM_021131:c.-624A&gt;C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9</td>\n",
       "      <td>129111323</td>\n",
       "      <td>129111323</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>PTPA</td>\n",
       "      <td>NM_178001:c.-278C&gt;G;NM_001271832:c.-278C&gt;G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>9</td>\n",
       "      <td>129111384</td>\n",
       "      <td>129111384</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>PTPA</td>\n",
       "      <td>NM_178001:c.-217C&gt;A;NM_001271832:c.-217C&gt;A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>9</td>\n",
       "      <td>129111647</td>\n",
       "      <td>129111647</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>intronic</td>\n",
       "      <td>PTPA</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9</td>\n",
       "      <td>129112273</td>\n",
       "      <td>129112273</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>PTPA</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 1308 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Chr      Start        End Ref Alt Func.refGene Gene.refGene  \\\n",
       "0    9  129110977  129110977   A   C         UTR5         PTPA   \n",
       "1    9  129111323  129111323   C   G         UTR5         PTPA   \n",
       "2    9  129111384  129111384   C   A         UTR5         PTPA   \n",
       "3    9  129111647  129111647   C   -     intronic         PTPA   \n",
       "4    9  129112273  129112273   T   A     intronic         PTPA   \n",
       "\n",
       "                           GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "0                         NM_021131:c.-624A>C                  .   \n",
       "1  NM_178001:c.-278C>G;NM_001271832:c.-278C>G                  .   \n",
       "2  NM_178001:c.-217C>A;NM_001271832:c.-217C>A                  .   \n",
       "3                                           .                  .   \n",
       "4                                           .                  .   \n",
       "\n",
       "  AAChange.refGene  ... Otherinfo1288  Otherinfo1289 Otherinfo1290  \\\n",
       "0                .  ...           0/0            0/0           0/1   \n",
       "1                .  ...           0/0            0/0           0/0   \n",
       "2                .  ...           0/0            0/0           0/0   \n",
       "3                .  ...           0/0            0/1           0/0   \n",
       "4                .  ...           0/0            0/0           0/1   \n",
       "\n",
       "  Otherinfo1291  Otherinfo1292  Otherinfo1293 Otherinfo1294 Otherinfo1295  \\\n",
       "0           0/0            0/0            0/0           0/0           0/0   \n",
       "1           0/0            0/0            0/0           0/0           0/0   \n",
       "2           0/0            0/0            0/0           0/0           0/0   \n",
       "3           0/0            0/0            0/0           0/0           0/0   \n",
       "4           0/0            0/0            0/0           0/0           0/0   \n",
       "\n",
       "  Otherinfo1296 Otherinfo1297  \n",
       "0           0/0           0/0  \n",
       "1           0/0           0/0  \n",
       "2           0/0           0/0  \n",
       "3           0/0           0/0  \n",
       "4           0/0           0/0  \n",
       "\n",
       "[5 rows x 1308 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Visualize multianno file\n",
    "PP2A_AJ = pd.read_csv(f'{WORK_DIR}/pheno_PP2A_AJ.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "PP2A_AJ.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Chr              177\n",
       "Start            177\n",
       "End              177\n",
       "Ref              177\n",
       "Alt              177\n",
       "                ... \n",
       "Otherinfo1293    177\n",
       "Otherinfo1294    177\n",
       "Otherinfo1295    177\n",
       "Otherinfo1296    177\n",
       "Otherinfo1297    177\n",
       "Length: 1308, dtype: int64"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PP2A_AJ.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>Otherinfo1288</th>\n",
       "      <th>Otherinfo1289</th>\n",
       "      <th>Otherinfo1290</th>\n",
       "      <th>Otherinfo1291</th>\n",
       "      <th>Otherinfo1292</th>\n",
       "      <th>Otherinfo1293</th>\n",
       "      <th>Otherinfo1294</th>\n",
       "      <th>Otherinfo1295</th>\n",
       "      <th>Otherinfo1296</th>\n",
       "      <th>Otherinfo1297</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>9</td>\n",
       "      <td>129129094</td>\n",
       "      <td>129129094</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>PTPA</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>PTPA:NM_001271832:exon3:c.A239G:p.Y80C,PTPA:NM...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>144</th>\n",
       "      <td>9</td>\n",
       "      <td>129142498</td>\n",
       "      <td>129142498</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>PTPA</td>\n",
       "      <td>.</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>PTPA:NM_001271832:exon8:c.C753T:p.A251A,PTPA:N...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>168</th>\n",
       "      <td>9</td>\n",
       "      <td>129147457</td>\n",
       "      <td>129147457</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>PTPA</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>PTPA:NM_001271832:exon9:c.C878T:p.S293L,PTPA:N...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows × 1308 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chr      Start        End Ref Alt Func.refGene Gene.refGene  \\\n",
       "79     9  129129094  129129094   A   G       exonic         PTPA   \n",
       "144    9  129142498  129142498   C   T       exonic         PTPA   \n",
       "168    9  129147457  129147457   C   T       exonic         PTPA   \n",
       "\n",
       "    GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "79                   .  nonsynonymous SNV   \n",
       "144                  .     synonymous SNV   \n",
       "168                  .  nonsynonymous SNV   \n",
       "\n",
       "                                      AAChange.refGene  ... Otherinfo1288  \\\n",
       "79   PTPA:NM_001271832:exon3:c.A239G:p.Y80C,PTPA:NM...  ...           0/0   \n",
       "144  PTPA:NM_001271832:exon8:c.C753T:p.A251A,PTPA:N...  ...           0/0   \n",
       "168  PTPA:NM_001271832:exon9:c.C878T:p.S293L,PTPA:N...  ...           0/0   \n",
       "\n",
       "     Otherinfo1289 Otherinfo1290 Otherinfo1291  Otherinfo1292  Otherinfo1293  \\\n",
       "79             0/0           0/0           0/0            0/0            0/0   \n",
       "144            0/0           0/0           0/0            0/0            0/0   \n",
       "168            0/0           0/1           0/0            0/0            0/0   \n",
       "\n",
       "    Otherinfo1294 Otherinfo1295 Otherinfo1296 Otherinfo1297  \n",
       "79            0/0           0/0           0/0           0/0  \n",
       "144           0/0           0/0           0/0           0/0  \n",
       "168           0/0           0/0           0/0           0/0  \n",
       "\n",
       "[3 rows x 1308 columns]"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter exonic variants\n",
    "coding_AJ = PP2A_AJ[PP2A_AJ['Func.refGene'] == 'exonic']\n",
    "coding_AJ.count()\n",
    "coding_AJ.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "79     1308\n",
       "168    1308\n",
       "dtype: int64"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter exonic and non-syn vars\n",
    "coding_nonsynonymous_AJ = PP2A_AJ[(PP2A_AJ['Func.refGene'] == 'exonic') & (PP2A_AJ['ExonicFunc.refGene'] == 'nonsynonymous SNV')]\n",
    "coding_nonsynonymous_AJ.count(1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>Otherinfo1288</th>\n",
       "      <th>Otherinfo1289</th>\n",
       "      <th>Otherinfo1290</th>\n",
       "      <th>Otherinfo1291</th>\n",
       "      <th>Otherinfo1292</th>\n",
       "      <th>Otherinfo1293</th>\n",
       "      <th>Otherinfo1294</th>\n",
       "      <th>Otherinfo1295</th>\n",
       "      <th>Otherinfo1296</th>\n",
       "      <th>Otherinfo1297</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>9</td>\n",
       "      <td>129129094</td>\n",
       "      <td>129129094</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>PTPA</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>PTPA:NM_001271832:exon3:c.A239G:p.Y80C,PTPA:NM...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 1308 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Chr      Start        End Ref Alt Func.refGene Gene.refGene  \\\n",
       "79    9  129129094  129129094   A   G       exonic         PTPA   \n",
       "\n",
       "   GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "79                  .  nonsynonymous SNV   \n",
       "\n",
       "                                     AAChange.refGene  ... Otherinfo1288  \\\n",
       "79  PTPA:NM_001271832:exon3:c.A239G:p.Y80C,PTPA:NM...  ...           0/0   \n",
       "\n",
       "    Otherinfo1289 Otherinfo1290 Otherinfo1291  Otherinfo1292  Otherinfo1293  \\\n",
       "79            0/0           0/0           0/0            0/0            0/0   \n",
       "\n",
       "   Otherinfo1294 Otherinfo1295 Otherinfo1296 Otherinfo1297  \n",
       "79           0/0           0/0           0/0           0/0  \n",
       "\n",
       "[1 rows x 1308 columns]"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding_nonsynonymous_AJ.head(1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "     Chr      Start        End Ref Alt Func.refGene Gene.refGene  \\\n",
      "79     9  129129094  129129094   A   G       exonic         PTPA   \n",
      "168    9  129147457  129147457   C   T       exonic         PTPA   \n",
      "\n",
      "    GeneDetail.refGene ExonicFunc.refGene  \\\n",
      "79                   .  nonsynonymous SNV   \n",
      "168                  .  nonsynonymous SNV   \n",
      "\n",
      "                                      AAChange.refGene  ... Otherinfo1288  \\\n",
      "79   PTPA:NM_001271832:exon3:c.A239G:p.Y80C,PTPA:NM...  ...           0/0   \n",
      "168  PTPA:NM_001271832:exon9:c.C878T:p.S293L,PTPA:N...  ...           0/0   \n",
      "\n",
      "     Otherinfo1289 Otherinfo1290 Otherinfo1291  Otherinfo1292  Otherinfo1293  \\\n",
      "79             0/0           0/0           0/0            0/0            0/0   \n",
      "168            0/0           0/1           0/0            0/0            0/0   \n",
      "\n",
      "    Otherinfo1294 Otherinfo1295 Otherinfo1296 Otherinfo1297  \n",
      "79            0/0           0/0           0/0           0/0  \n",
      "168           0/0           0/0           0/0           0/0  \n",
      "\n",
      "[2 rows x 1308 columns]\n"
     ]
    }
   ],
   "source": [
    "print(coding_nonsynonymous_AJ)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>Otherinfo1288</th>\n",
       "      <th>Otherinfo1289</th>\n",
       "      <th>Otherinfo1290</th>\n",
       "      <th>Otherinfo1291</th>\n",
       "      <th>Otherinfo1292</th>\n",
       "      <th>Otherinfo1293</th>\n",
       "      <th>Otherinfo1294</th>\n",
       "      <th>Otherinfo1295</th>\n",
       "      <th>Otherinfo1296</th>\n",
       "      <th>Otherinfo1297</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>9</td>\n",
       "      <td>129129094</td>\n",
       "      <td>129129094</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>PTPA</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>PTPA:NM_001271832:exon3:c.A239G:p.Y80C,PTPA:NM...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>168</th>\n",
       "      <td>9</td>\n",
       "      <td>129147457</td>\n",
       "      <td>129147457</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>PTPA</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>PTPA:NM_001271832:exon9:c.C878T:p.S293L,PTPA:N...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 1308 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chr      Start        End Ref Alt Func.refGene Gene.refGene  \\\n",
       "79     9  129129094  129129094   A   G       exonic         PTPA   \n",
       "168    9  129147457  129147457   C   T       exonic         PTPA   \n",
       "\n",
       "    GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "79                   .  nonsynonymous SNV   \n",
       "168                  .  nonsynonymous SNV   \n",
       "\n",
       "                                      AAChange.refGene  ... Otherinfo1288  \\\n",
       "79   PTPA:NM_001271832:exon3:c.A239G:p.Y80C,PTPA:NM...  ...           0/0   \n",
       "168  PTPA:NM_001271832:exon9:c.C878T:p.S293L,PTPA:N...  ...           0/0   \n",
       "\n",
       "     Otherinfo1289 Otherinfo1290 Otherinfo1291  Otherinfo1292  Otherinfo1293  \\\n",
       "79             0/0           0/0           0/0            0/0            0/0   \n",
       "168            0/0           0/1           0/0            0/0            0/0   \n",
       "\n",
       "    Otherinfo1294 Otherinfo1295 Otherinfo1296 Otherinfo1297  \n",
       "79            0/0           0/0           0/0           0/0  \n",
       "168           0/0           0/0           0/0           0/0  \n",
       "\n",
       "[2 rows x 1308 columns]"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding_nonsynonymous_AJ.to_csv(f'{WORK_DIR}/coding_nonsynonymous.txt', sep = '\\t', index = False)\n",
    "coding_nonsynonymous_AJ.head(9)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Chr              163\n",
       "Start            163\n",
       "End              163\n",
       "Ref              163\n",
       "Alt              163\n",
       "                ... \n",
       "Otherinfo1293    163\n",
       "Otherinfo1294    163\n",
       "Otherinfo1295    163\n",
       "Otherinfo1296    163\n",
       "Otherinfo1297    163\n",
       "Length: 1308, dtype: int64"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter intronic variants\n",
    "intronic_AJ = PP2A_AJ[PP2A_AJ['Func.refGene'] == 'intronic']\n",
    "intronic_AJ.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Chr              8\n",
       "Start            8\n",
       "End              8\n",
       "Ref              8\n",
       "Alt              8\n",
       "                ..\n",
       "Otherinfo1293    8\n",
       "Otherinfo1294    8\n",
       "Otherinfo1295    8\n",
       "Otherinfo1296    8\n",
       "Otherinfo1297    8\n",
       "Length: 1308, dtype: int64"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter UTR3 variants\n",
    "UTR3_AJ = PP2A_AJ[PP2A_AJ['Func.refGene'] == ('UTR3')]\n",
    "UTR3_AJ.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Chr              3\n",
       "Start            3\n",
       "End              3\n",
       "Ref              3\n",
       "Alt              3\n",
       "                ..\n",
       "Otherinfo1293    3\n",
       "Otherinfo1294    3\n",
       "Otherinfo1295    3\n",
       "Otherinfo1296    3\n",
       "Otherinfo1297    3\n",
       "Length: 1308, dtype: int64"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter UTR5 variants\n",
    "UTR5_AJ = PP2A_AJ[PP2A_AJ['Func.refGene'] == ('UTR5')]\n",
    "UTR5_AJ.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Calculate freq - cases vs controls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "reduced_coding_nonsynonymous_AJ = coding_nonsynonymous_AJ[[\"Chr\", \"Start\", \"End\", \"Gene.refGene\"]]\n",
    "reduced_coding_nonsynonymous_AJ.to_csv(f'{WORK_DIR}/reduced_coding_nonsynonymous_AJ.txt', sep = '\\t', index = False, header= False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9\t129129094\t129129094\tPTPA\n",
      "9\t129147457\t129147457\tPTPA\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "head reduced_coding_nonsynonymous_AJ.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to coding_nonsynonymous_pheno_PP2A_AJ.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile pheno_PP2A_AJ\n",
      "  --ci 0.95\n",
      "  --extract range reduced_coding_nonsynonymous_AJ.txt\n",
      "  --out coding_nonsynonymous_pheno_PP2A_AJ\n",
      "\n",
      "14998 MB RAM detected; reserving 7499 MB for main workspace.\n",
      "177 variants loaded from .bim file.\n",
      "1285 people (808 males, 477 females) loaded from .fam.\n",
      "923 phenotype values loaded from .fam.\n",
      "--extract range: 175 variants excluded.\n",
      "--extract range: 2 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1285 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.999222.\n",
      "2 variants and 1285 people pass filters and QC.\n",
      "Among remaining phenotypes, 600 are cases and 323 are controls.  (362\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to coding_nonsynonymous_pheno_PP2A_AJ.assoc ...\n",
      "0%\b\bdone.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "/home/jupyter/tools/plink --bfile pheno_PP2A_AJ --extract range reduced_coding_nonsynonymous_AJ.txt --assoc --out coding_nonsynonymous_pheno_PP2A_AJ --allow-no-sex --ci 0.95"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to coding_nonsynonymous_AJ_rare_0.05.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile pheno_PP2A_AJ\n",
      "  --extract range reduced_coding_nonsynonymous_AJ.txt\n",
      "  --make-bed\n",
      "  --max-maf 0.05\n",
      "  --out coding_nonsynonymous_AJ_rare_0.05\n",
      "\n",
      "14998 MB RAM detected; reserving 7499 MB for main workspace.\n",
      "177 variants loaded from .bim file.\n",
      "1285 people (808 males, 477 females) loaded from .fam.\n",
      "923 phenotype values loaded from .fam.\n",
      "--extract range: 175 variants excluded.\n",
      "--extract range: 2 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1285 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.999222.\n",
      "1 variant removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "1 variant and 1285 people pass filters and QC.\n",
      "Among remaining phenotypes, 600 are cases and 323 are controls.  (362\n",
      "phenotypes are missing.)\n",
      "--make-bed to coding_nonsynonymous_AJ_rare_0.05.bed +\n",
      "coding_nonsynonymous_AJ_rare_0.05.bim + coding_nonsynonymous_AJ_rare_0.05.fam\n",
      "... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "/home/jupyter/tools/plink --bfile pheno_PP2A_AJ --extract range reduced_coding_nonsynonymous_AJ.txt --make-bed --max-maf 0.05 --out coding_nonsynonymous_AJ_rare_0.05 --allow-no-sex"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP</th>\n",
       "      <th>BP</th>\n",
       "      <th>A1</th>\n",
       "      <th>F_A</th>\n",
       "      <th>F_U</th>\n",
       "      <th>A2</th>\n",
       "      <th>CHISQ</th>\n",
       "      <th>P</th>\n",
       "      <th>OR</th>\n",
       "      <th>SE</th>\n",
       "      <th>L95</th>\n",
       "      <th>U95</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:129129094:A:G</td>\n",
       "      <td>129129094</td>\n",
       "      <td>G</td>\n",
       "      <td>0.0075</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>A</td>\n",
       "      <td>4.8690</td>\n",
       "      <td>0.02735</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:129147457:C:T</td>\n",
       "      <td>129147457</td>\n",
       "      <td>T</td>\n",
       "      <td>0.0576</td>\n",
       "      <td>0.06677</td>\n",
       "      <td>C</td>\n",
       "      <td>0.6173</td>\n",
       "      <td>0.43200</td>\n",
       "      <td>0.8542</td>\n",
       "      <td>0.2007</td>\n",
       "      <td>0.5764</td>\n",
       "      <td>1.266</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHR                 SNP         BP A1     F_A      F_U A2   CHISQ        P  \\\n",
       "0    9  chr9:129129094:A:G  129129094  G  0.0075  0.00000  A  4.8690  0.02735   \n",
       "1    9  chr9:129147457:C:T  129147457  T  0.0576  0.06677  C  0.6173  0.43200   \n",
       "\n",
       "       OR      SE     L95    U95  \n",
       "0     NaN     NaN     NaN    NaN  \n",
       "1  0.8542  0.2007  0.5764  1.266  "
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PP2A_AJ_freq = pd.read_csv(f'{WORK_DIR}/coding_nonsynonymous_pheno_PP2A_AJ.assoc', delim_whitespace=True)\n",
    "PP2A_AJ_freq.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to coding_nonsynonymous_pheno_PP2A_AJ.log.\n",
      "Options in effect:\n",
      "  --adjust\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile pheno_PP2A_AJ\n",
      "  --extract range reduced_coding_nonsynonymous_AJ.txt\n",
      "  --out coding_nonsynonymous_pheno_PP2A_AJ\n",
      "\n",
      "14998 MB RAM detected; reserving 7499 MB for main workspace.\n",
      "177 variants loaded from .bim file.\n",
      "1285 people (808 males, 477 females) loaded from .fam.\n",
      "923 phenotype values loaded from .fam.\n",
      "--extract range: 175 variants excluded.\n",
      "--extract range: 2 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1285 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.999222.\n",
      "2 variants and 1285 people pass filters and QC.\n",
      "Among remaining phenotypes, 600 are cases and 323 are controls.  (362\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to coding_nonsynonymous_pheno_PP2A_AJ.assoc ...\n",
      "0%\b\bdone.\n",
      "--adjust: Genomic inflation est. lambda (based on median chisq) = 6.0154.\n",
      "0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b--adjust values (2 variants) written to\n",
      "coding_nonsynonymous_pheno_PP2A_AJ.assoc.adjusted .\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "/home/jupyter/tools/plink --bfile pheno_PP2A_AJ --extract range reduced_coding_nonsynonymous_AJ.txt --assoc --out coding_nonsynonymous_pheno_PP2A_AJ --allow-no-sex --adjust"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP</th>\n",
       "      <th>UNADJ</th>\n",
       "      <th>GC</th>\n",
       "      <th>BONF</th>\n",
       "      <th>HOLM</th>\n",
       "      <th>SIDAK_SS</th>\n",
       "      <th>SIDAK_SD</th>\n",
       "      <th>FDR_BH</th>\n",
       "      <th>FDR_BY</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:129129094:A:G</td>\n",
       "      <td>0.02735</td>\n",
       "      <td>0.3683</td>\n",
       "      <td>0.0547</td>\n",
       "      <td>0.0547</td>\n",
       "      <td>0.05395</td>\n",
       "      <td>0.05395</td>\n",
       "      <td>0.0547</td>\n",
       "      <td>0.08204</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:129147457:C:T</td>\n",
       "      <td>0.43200</td>\n",
       "      <td>0.7487</td>\n",
       "      <td>0.8641</td>\n",
       "      <td>0.4320</td>\n",
       "      <td>0.67740</td>\n",
       "      <td>0.43200</td>\n",
       "      <td>0.4320</td>\n",
       "      <td>0.64810</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHR                 SNP    UNADJ      GC    BONF    HOLM  SIDAK_SS  \\\n",
       "0    9  chr9:129129094:A:G  0.02735  0.3683  0.0547  0.0547   0.05395   \n",
       "1    9  chr9:129147457:C:T  0.43200  0.7487  0.8641  0.4320   0.67740   \n",
       "\n",
       "   SIDAK_SD  FDR_BH   FDR_BY  \n",
       "0   0.05395  0.0547  0.08204  \n",
       "1   0.43200  0.4320  0.64810  "
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PP2A_AJ_freq = pd.read_csv(f'{WORK_DIR}/coding_nonsynonymous_pheno_PP2A_AJ.assoc.adjusted', delim_whitespace=True)\n",
    "PP2A_AJ_freq.head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Calculate freq of HMZ in cases versus controls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to coding_nonsynonymous_pheno_PP2A_AJ.log.\n",
      "Options in effect:\n",
      "  --bfile pheno_PP2A_AJ\n",
      "  --extract range reduced_coding_nonsynonymous_AJ.txt\n",
      "  --out coding_nonsynonymous_pheno_PP2A_AJ\n",
      "  --recode A\n",
      "\n",
      "14998 MB RAM detected; reserving 7499 MB for main workspace.\n",
      "177 variants loaded from .bim file.\n",
      "1285 people (808 males, 477 females) loaded from .fam.\n",
      "923 phenotype values loaded from .fam.\n",
      "--extract range: 175 variants excluded.\n",
      "--extract range: 2 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1285 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.999222.\n",
      "2 variants and 1285 people pass filters and QC.\n",
      "Among remaining phenotypes, 600 are cases and 323 are controls.  (362\n",
      "phenotypes are missing.)\n",
      "--recode A to coding_nonsynonymous_pheno_PP2A_AJ.raw ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "/home/jupyter/tools/plink --bfile pheno_PP2A_AJ --extract range reduced_coding_nonsynonymous_AJ.txt --recode A --out coding_nonsynonymous_pheno_PP2A_AJ"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP</th>\n",
       "      <th>BP</th>\n",
       "      <th>A1</th>\n",
       "      <th>F_A</th>\n",
       "      <th>F_U</th>\n",
       "      <th>A2</th>\n",
       "      <th>CHISQ</th>\n",
       "      <th>P</th>\n",
       "      <th>OR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:129129094:A:G</td>\n",
       "      <td>129129094</td>\n",
       "      <td>G</td>\n",
       "      <td>0.0075</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>A</td>\n",
       "      <td>4.8690</td>\n",
       "      <td>0.02735</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:129147457:C:T</td>\n",
       "      <td>129147457</td>\n",
       "      <td>T</td>\n",
       "      <td>0.0576</td>\n",
       "      <td>0.06677</td>\n",
       "      <td>C</td>\n",
       "      <td>0.6173</td>\n",
       "      <td>0.43200</td>\n",
       "      <td>0.8542</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHR                 SNP         BP A1     F_A      F_U A2   CHISQ        P  \\\n",
       "0    9  chr9:129129094:A:G  129129094  G  0.0075  0.00000  A  4.8690  0.02735   \n",
       "1    9  chr9:129147457:C:T  129147457  T  0.0576  0.06677  C  0.6173  0.43200   \n",
       "\n",
       "       OR  \n",
       "0     NaN  \n",
       "1  0.8542  "
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PP2A_AJ_freq = pd.read_csv(f'{WORK_DIR}/coding_nonsynonymous_pheno_PP2A_AJ.assoc', delim_whitespace=True)\n",
    "PP2A_AJ_freq.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129129094:A:G_G</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000021_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000475_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000486_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000646_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0</td>\n",
       "      <td>APGS_000808_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   FID             IID  PAT  MAT  SEX  PHENOTYPE  chr9:129129094:A:G_G  \\\n",
       "0    0  APGS_000021_s1    0    0    1          2                     0   \n",
       "1    0  APGS_000475_s1    0    0    1          2                     0   \n",
       "2    0  APGS_000486_s1    0    0    1          2                     0   \n",
       "3    0  APGS_000646_s1    0    0    2          2                     0   \n",
       "4    0  APGS_000808_s1    0    0    1          2                     0   \n",
       "\n",
       "   chr9:129147457:C:T_T  \n",
       "0                   0.0  \n",
       "1                   0.0  \n",
       "2                   0.0  \n",
       "3                   0.0  \n",
       "4                   0.0  "
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PP2A_AJ_recode = pd.read_csv(f'{WORK_DIR}/coding_nonsynonymous_pheno_PP2A_AJ.raw', delim_whitespace=True)\n",
    "PP2A_AJ_recode.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to coding_nonsynonymous_pheno_PP2A_AJ.log.\n",
      "Options in effect:\n",
      "  --bfile pheno_PP2A_AJ\n",
      "  --extract range reduced_coding_nonsynonymous_AJ.txt\n",
      "  --out coding_nonsynonymous_pheno_PP2A_AJ\n",
      "  --recode A\n",
      "\n",
      "14998 MB RAM detected; reserving 7499 MB for main workspace.\n",
      "177 variants loaded from .bim file.\n",
      "1285 people (808 males, 477 females) loaded from .fam.\n",
      "923 phenotype values loaded from .fam.\n",
      "--extract range: 175 variants excluded.\n",
      "--extract range: 2 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1285 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.999222.\n",
      "2 variants and 1285 people pass filters and QC.\n",
      "Among remaining phenotypes, 600 are cases and 323 are controls.  (362\n",
      "phenotypes are missing.)\n",
      "--recode A to coding_nonsynonymous_pheno_PP2A_AJ.raw ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "\n",
    "/home/jupyter/tools/plink --bfile  pheno_PP2A_AJ --extract range reduced_coding_nonsynonymous_AJ.txt --recode A --out coding_nonsynonymous_pheno_PP2A_AJ"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129129094:A:G_G</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>0</td>\n",
       "      <td>BIOFIND_000043_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>176</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_001619_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     FID                IID  PAT  MAT  SEX  PHENOTYPE  chr9:129129094:A:G_G  \\\n",
       "35     0  BIOFIND_000043_s1    0    0    1          2                     0   \n",
       "176    0  CORIELL_001619_s1    0    0    2          2                     0   \n",
       "\n",
       "     chr9:129147457:C:T_T  \n",
       "35                    2.0  \n",
       "176                   2.0  "
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Explore HMZs for chr9:129147457:C:T_T in cases\n",
    "PP2A_hom_casos_129147457_AJ = PP2A_AJ_recode[(PP2A_AJ_recode['chr9:129147457:C:T_T'] == 2) & (PP2A_AJ_recode['PHENOTYPE'] == 2)]\n",
    "PP2A_hom_casos_129147457_AJ.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(2, 8)"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PP2A_hom_casos_129147457_AJ.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "PP2A_hom_casos_129147457_AJ.to_csv('PP2A_hom_casos_129147457_AJ.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129129094:A:G_G</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>530</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_005579_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     FID                IID  PAT  MAT  SEX  PHENOTYPE  chr9:129129094:A:G_G  \\\n",
       "530    0  CORIELL_005579_s1    0    0    1          1                     0   \n",
       "\n",
       "     chr9:129147457:C:T_T  \n",
       "530                   2.0  "
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Explore HMZs for chr9:129147457:C:T_T in controls\n",
    "PP2A_hom_controles_129147457_AJ = PP2A_AJ_recode[(PP2A_AJ_recode['chr9:129147457:C:T_T'] == 2) & (PP2A_AJ_recode['PHENOTYPE'] == 1)]\n",
    "PP2A_hom_controles_129147457_AJ.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1, 8)"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PP2A_hom_controles_129147457_AJ.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129129094:A:G_G</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [FID, IID, PAT, MAT, SEX, PHENOTYPE, chr9:129129094:A:G_G, chr9:129147457:C:T_T]\n",
       "Index: []"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Explore HMZs for chr9:129129094:A:G_G in cases\n",
    "PP2A_hom_casos_129129094_AJ = PP2A_AJ_recode[(PP2A_AJ_recode['chr9:129129094:A:G_G'] == 2) & (PP2A_AJ_recode['PHENOTYPE'] == 2)]\n",
    "PP2A_hom_casos_129129094_AJ.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(0, 8)"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PP2A_hom_casos_129129094_AJ.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129129094:A:G_G</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [FID, IID, PAT, MAT, SEX, PHENOTYPE, chr9:129129094:A:G_G, chr9:129147457:C:T_T]\n",
       "Index: []"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Explore HMZs for chr9:129129094:A:G_G in controls\n",
    "PP2A_hom_controles_129129094_AJ = PP2A_AJ_recode[(PP2A_AJ_recode['chr9:129129094:A:G_G'] == 2) & (PP2A_AJ_recode['PHENOTYPE'] == 1)]\n",
    "PP2A_hom_controles_129129094_AJ.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(0, 8)"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PP2A_hom_controles_129129094_AJ.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(161, 8)\n"
     ]
    }
   ],
   "source": [
    "# Identifying individuals het for the alternative allele\n",
    "het = PP2A_AJ_recode[(PP2A_AJ_recode['chr9:129147457:C:T_T'] == 1) | (PP2A_AJ_recode['chr9:129129094:A:G_G'] == 1)]\n",
    "het = het.dropna()\n",
    "print(het.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129129094:A:G_G</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000051_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000158_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>0</td>\n",
       "      <td>BIOFIND_000049_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>0</td>\n",
       "      <td>BIOFIND_000116_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>0</td>\n",
       "      <td>BIOFIND_000126_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1257</th>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000252_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1267</th>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000346_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1268</th>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000364_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1273</th>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000386_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1277</th>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000410_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>161 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      FID                IID  PAT  MAT  SEX  PHENOTYPE  chr9:129129094:A:G_G  \\\n",
       "12      0      BCM_000051_s1    0    0    1          2                     0   \n",
       "14      0      BCM_000158_s1    0    0    1          2                     0   \n",
       "37      0  BIOFIND_000049_s1    0    0    1          2                     0   \n",
       "43      0  BIOFIND_000116_s1    0    0    2          1                     0   \n",
       "45      0  BIOFIND_000126_s1    0    0    1          1                     0   \n",
       "...   ...                ...  ...  ...  ...        ...                   ...   \n",
       "1257    0      UMD_000252_s1    0    0    2          2                     0   \n",
       "1267    0      UMD_000346_s1    0    0    1          2                     0   \n",
       "1268    0      UMD_000364_s1    0    0    1          2                     0   \n",
       "1273    0      UMD_000386_s1    0    0    2          1                     0   \n",
       "1277    0      UMD_000410_s1    0    0    1          2                     0   \n",
       "\n",
       "      chr9:129147457:C:T_T  \n",
       "12                     1.0  \n",
       "14                     1.0  \n",
       "37                     1.0  \n",
       "43                     1.0  \n",
       "45                     1.0  \n",
       "...                    ...  \n",
       "1257                   1.0  \n",
       "1267                   1.0  \n",
       "1268                   1.0  \n",
       "1273                   1.0  \n",
       "1277                   1.0  \n",
       "\n",
       "[161 rows x 8 columns]"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "het.head (161)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1.0    158\n",
       "2.0      3\n",
       "Name: compound, dtype: int64"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "het['compound'] = het['chr9:129147457:C:T_T'] + het['chr9:129129094:A:G_G']\n",
    "het['compound'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1, 9)\n"
     ]
    }
   ],
   "source": [
    "comp_het_info_cases = het[(het['PHENOTYPE'] == 2) & (het['compound'] == 2)]\n",
    "print(comp_het_info_cases.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(0, 9)\n"
     ]
    }
   ],
   "source": [
    "comp_het_info_control = het[(het['PHENOTYPE'] == 1) & (het['compound'] == 2)]\n",
    "print(comp_het_info_control.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129129094:A:G_G</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "      <th>compound</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>421</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_003933_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     FID                IID  PAT  MAT  SEX  PHENOTYPE  chr9:129129094:A:G_G  \\\n",
       "421    0  CORIELL_003933_s1    0    0    2          2                     1   \n",
       "\n",
       "     chr9:129147457:C:T_T  compound  \n",
       "421                   1.0       2.0  "
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "comp_het_info_cases.to_csv('/home/jupyter/PTPA/comp_het_info_cases.txt', sep = '\\t', index=True)\n",
    "comp_het_info_cases.head(54)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129129094:A:G_G</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "      <th>compound</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [FID, IID, PAT, MAT, SEX, PHENOTYPE, chr9:129129094:A:G_G, chr9:129147457:C:T_T, compound]\n",
       "Index: []"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "comp_het_info_control.to_csv('/home/jupyter/PTPA/comp_het_info_control.txt', sep = '\\t', index=True)\n",
    "comp_het_info_control.head(54)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129129094:A:G_G</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000051_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000158_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>0</td>\n",
       "      <td>BIOFIND_000049_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>0</td>\n",
       "      <td>BIOFIND_000136_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>0</td>\n",
       "      <td>BIOFIND_000205_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1241</th>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000047_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1257</th>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000252_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1267</th>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000346_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1268</th>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000364_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1277</th>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000410_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>65 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      FID                IID  PAT  MAT  SEX  PHENOTYPE  chr9:129129094:A:G_G  \\\n",
       "12      0      BCM_000051_s1    0    0    1          2                     0   \n",
       "14      0      BCM_000158_s1    0    0    1          2                     0   \n",
       "37      0  BIOFIND_000049_s1    0    0    1          2                     0   \n",
       "48      0  BIOFIND_000136_s1    0    0    2          2                     0   \n",
       "56      0  BIOFIND_000205_s1    0    0    1          2                     0   \n",
       "...   ...                ...  ...  ...  ...        ...                   ...   \n",
       "1241    0      UMD_000047_s1    0    0    1          2                     0   \n",
       "1257    0      UMD_000252_s1    0    0    2          2                     0   \n",
       "1267    0      UMD_000346_s1    0    0    1          2                     0   \n",
       "1268    0      UMD_000364_s1    0    0    1          2                     0   \n",
       "1277    0      UMD_000410_s1    0    0    1          2                     0   \n",
       "\n",
       "      chr9:129147457:C:T_T  \n",
       "12                     1.0  \n",
       "14                     1.0  \n",
       "37                     1.0  \n",
       "48                     1.0  \n",
       "56                     1.0  \n",
       "...                    ...  \n",
       "1241                   1.0  \n",
       "1257                   1.0  \n",
       "1267                   1.0  \n",
       "1268                   1.0  \n",
       "1277                   1.0  \n",
       "\n",
       "[65 rows x 8 columns]"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Explore HETs for chr9:129147457:C:T_T in cases\n",
    "PP2A_het_casos_129147457_AJ = PP2A_AJ_recode[(PP2A_AJ_recode['chr9:129147457:C:T_T'] == 1) & (PP2A_AJ_recode['PHENOTYPE'] == 2)]\n",
    "PP2A_het_casos_129147457_AJ"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(65, 8)"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PP2A_het_casos_129147457_AJ.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129129094:A:G_G</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>0</td>\n",
       "      <td>BIOFIND_000116_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>0</td>\n",
       "      <td>BIOFIND_000126_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>484</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_004515_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>485</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_004521_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>487</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_004628_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>501</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_005008_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>517</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_005259_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>520</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_005327_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>523</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_005425_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>532</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_005730_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>533</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_005747_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>557</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_005951_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>571</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006063_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>573</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006079_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>586</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006173_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>590</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006211_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>601</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006409_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>602</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006443_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>603</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006451_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>607</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006835_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>628</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007129_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>648</th>\n",
       "      <td>0</td>\n",
       "      <td>LCC_000015_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>664</th>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_000030_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>711</th>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_000860_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>721</th>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_001040_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>744</th>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_001439_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>763</th>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_001773_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>766</th>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_001814_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>769</th>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_001871_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>780</th>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_002073_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>787</th>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_002134_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>804</th>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_002324_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>807</th>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_002355_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>812</th>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_002407_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>824</th>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_002574_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>833</th>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_002680_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>839</th>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_002754_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>843</th>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_002787_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>857</th>\n",
       "      <td>0</td>\n",
       "      <td>PPMI_000079_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1104</th>\n",
       "      <td>0</td>\n",
       "      <td>PPMI_000674_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1273</th>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000386_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      FID                IID  PAT  MAT  SEX  PHENOTYPE  chr9:129129094:A:G_G  \\\n",
       "43      0  BIOFIND_000116_s1    0    0    2          1                     0   \n",
       "45      0  BIOFIND_000126_s1    0    0    1          1                     0   \n",
       "484     0  CORIELL_004515_s1    0    0    2          1                     0   \n",
       "485     0  CORIELL_004521_s1    0    0    2          1                     0   \n",
       "487     0  CORIELL_004628_s1    0    0    1          1                     0   \n",
       "501     0  CORIELL_005008_s1    0    0    1          1                     0   \n",
       "517     0  CORIELL_005259_s1    0    0    2          1                     0   \n",
       "520     0  CORIELL_005327_s1    0    0    1          1                     0   \n",
       "523     0  CORIELL_005425_s1    0    0    2          1                     0   \n",
       "532     0  CORIELL_005730_s1    0    0    2          1                     0   \n",
       "533     0  CORIELL_005747_s1    0    0    1          1                     0   \n",
       "557     0  CORIELL_005951_s1    0    0    2          1                     0   \n",
       "571     0  CORIELL_006063_s1    0    0    2          1                     0   \n",
       "573     0  CORIELL_006079_s1    0    0    1          1                     0   \n",
       "586     0  CORIELL_006173_s1    0    0    1          1                     0   \n",
       "590     0  CORIELL_006211_s1    0    0    2          1                     0   \n",
       "601     0  CORIELL_006409_s1    0    0    2          1                     0   \n",
       "602     0  CORIELL_006443_s1    0    0    1          1                     0   \n",
       "603     0  CORIELL_006451_s1    0    0    1          1                     0   \n",
       "607     0  CORIELL_006835_s1    0    0    1          1                     0   \n",
       "628     0  CORIELL_007129_s1    0    0    1          1                     0   \n",
       "648     0      LCC_000015_s1    0    0    2          1                     0   \n",
       "664     0     PAGE_000030_s1    0    0    2          1                     0   \n",
       "711     0     PAGE_000860_s1    0    0    1          1                     0   \n",
       "721     0     PAGE_001040_s1    0    0    1          1                     0   \n",
       "744     0     PAGE_001439_s1    0    0    1          1                     0   \n",
       "763     0     PAGE_001773_s1    0    0    1          1                     0   \n",
       "766     0     PAGE_001814_s1    0    0    1          1                     0   \n",
       "769     0     PAGE_001871_s1    0    0    1          1                     0   \n",
       "780     0     PAGE_002073_s1    0    0    1          1                     0   \n",
       "787     0     PAGE_002134_s1    0    0    2          1                     0   \n",
       "804     0     PAGE_002324_s1    0    0    1          1                     0   \n",
       "807     0     PAGE_002355_s1    0    0    1          1                     0   \n",
       "812     0     PAGE_002407_s1    0    0    2          1                     0   \n",
       "824     0     PAGE_002574_s1    0    0    2          1                     0   \n",
       "833     0     PAGE_002680_s1    0    0    2          1                     0   \n",
       "839     0     PAGE_002754_s1    0    0    1          1                     0   \n",
       "843     0     PAGE_002787_s1    0    0    1          1                     0   \n",
       "857     0     PPMI_000079_s1    0    0    2          1                     0   \n",
       "1104    0     PPMI_000674_s1    0    0    1          1                     0   \n",
       "1273    0      UMD_000386_s1    0    0    2          1                     0   \n",
       "\n",
       "      chr9:129147457:C:T_T  \n",
       "43                     1.0  \n",
       "45                     1.0  \n",
       "484                    1.0  \n",
       "485                    1.0  \n",
       "487                    1.0  \n",
       "501                    1.0  \n",
       "517                    1.0  \n",
       "520                    1.0  \n",
       "523                    1.0  \n",
       "532                    1.0  \n",
       "533                    1.0  \n",
       "557                    1.0  \n",
       "571                    1.0  \n",
       "573                    1.0  \n",
       "586                    1.0  \n",
       "590                    1.0  \n",
       "601                    1.0  \n",
       "602                    1.0  \n",
       "603                    1.0  \n",
       "607                    1.0  \n",
       "628                    1.0  \n",
       "648                    1.0  \n",
       "664                    1.0  \n",
       "711                    1.0  \n",
       "721                    1.0  \n",
       "744                    1.0  \n",
       "763                    1.0  \n",
       "766                    1.0  \n",
       "769                    1.0  \n",
       "780                    1.0  \n",
       "787                    1.0  \n",
       "804                    1.0  \n",
       "807                    1.0  \n",
       "812                    1.0  \n",
       "824                    1.0  \n",
       "833                    1.0  \n",
       "839                    1.0  \n",
       "843                    1.0  \n",
       "857                    1.0  \n",
       "1104                   1.0  \n",
       "1273                   1.0  "
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Explore HETs for chr9:129147457:C:T_T in controles\n",
    "PP2A_het_controles_129147457_AJ = PP2A_AJ_recode[(PP2A_AJ_recode['chr9:129147457:C:T_T'] == 1) & (PP2A_AJ_recode['PHENOTYPE'] == 1)]\n",
    "PP2A_het_controles_129147457_AJ"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(41, 8)"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PP2A_het_controles_129147457_AJ.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129129094:A:G_G</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_000035_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_000138_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>137</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_001140_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>224</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_001997_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>301</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_002302_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>421</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_003933_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>437</th>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_004167_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>718</th>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_001019_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1252</th>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000157_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      FID                IID  PAT  MAT  SEX  PHENOTYPE  chr9:129129094:A:G_G  \\\n",
       "60      0  CORIELL_000035_s1    0    0    2          2                     1   \n",
       "67      0  CORIELL_000138_s1    0    0    1          2                     1   \n",
       "137     0  CORIELL_001140_s1    0    0    1          2                     1   \n",
       "224     0  CORIELL_001997_s1    0    0    1          2                     1   \n",
       "301     0  CORIELL_002302_s1    0    0    1          2                     1   \n",
       "421     0  CORIELL_003933_s1    0    0    2          2                     1   \n",
       "437     0  CORIELL_004167_s1    0    0    2          2                     1   \n",
       "718     0     PAGE_001019_s1    0    0    1          2                     1   \n",
       "1252    0      UMD_000157_s1    0    0    1          2                     1   \n",
       "\n",
       "      chr9:129147457:C:T_T  \n",
       "60                     0.0  \n",
       "67                     0.0  \n",
       "137                    0.0  \n",
       "224                    0.0  \n",
       "301                    0.0  \n",
       "421                    1.0  \n",
       "437                    0.0  \n",
       "718                    0.0  \n",
       "1252                   0.0  "
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Explore HETs for chr9:129129094:A:G_G in cases\n",
    "PP2A_het_casos_129129094_AJ = PP2A_AJ_recode[(PP2A_AJ_recode['chr9:129129094:A:G_G'] == 1) & (PP2A_AJ_recode['PHENOTYPE'] == 2)]\n",
    "PP2A_het_casos_129129094_AJ.head(600)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(9, 8)"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PP2A_het_casos_129129094_AJ.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129129094:A:G_G</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [FID, IID, PAT, MAT, SEX, PHENOTYPE, chr9:129129094:A:G_G, chr9:129147457:C:T_T]\n",
       "Index: []"
      ]
     },
     "execution_count": 100,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Explore HETs for chr9:129129094:A:G_G in controles\n",
    "PP2A_het_controles_129129094_AJ = PP2A_AJ_recode[(PP2A_AJ_recode['chr9:129129094:A:G_G'] == 1) & (PP2A_AJ_recode['PHENOTYPE'] == 1)]\n",
    "PP2A_het_controles_129129094_AJ"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(0, 8)"
      ]
     },
     "execution_count": 78,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PP2A_het_controles_129129094_AJ.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: rpy2 in /home/jupyter/.local/lib/python3.7/site-packages (3.5.12)\n",
      "Requirement already satisfied: cffi>=1.10.0 in /opt/conda/lib/python3.7/site-packages (from rpy2) (1.15.0)\n",
      "Requirement already satisfied: jinja2 in /opt/conda/lib/python3.7/site-packages (from rpy2) (3.1.2)\n",
      "Requirement already satisfied: pytz in /opt/conda/lib/python3.7/site-packages (from rpy2) (2022.6)\n",
      "Requirement already satisfied: tzlocal<5.0 in /home/jupyter/.local/lib/python3.7/site-packages (from rpy2) (4.3)\n",
      "Requirement already satisfied: typing-extensions in /opt/conda/lib/python3.7/site-packages (from rpy2) (4.1.1)\n",
      "Requirement already satisfied: pycparser in /opt/conda/lib/python3.7/site-packages (from cffi>=1.10.0->rpy2) (2.21)\n",
      "Requirement already satisfied: pytz-deprecation-shim in /home/jupyter/.local/lib/python3.7/site-packages (from tzlocal<5.0->rpy2) (0.1.0.post0)\n",
      "Requirement already satisfied: backports.zoneinfo in /home/jupyter/.local/lib/python3.7/site-packages (from tzlocal<5.0->rpy2) (0.2.1)\n",
      "Requirement already satisfied: MarkupSafe>=2.0 in /opt/conda/lib/python3.7/site-packages (from jinja2->rpy2) (2.0.1)\n",
      "Requirement already satisfied: tzdata in /home/jupyter/.local/lib/python3.7/site-packages (from pytz-deprecation-shim->tzlocal<5.0->rpy2) (2023.3)\n"
     ]
    }
   ],
   "source": [
    "!pip install rpy2\n",
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: pip in /home/jupyter/.local/lib/python3.7/site-packages (23.1.2)\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "pip install --upgrade pip"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The rpy2.ipython extension is already loaded. To reload it, use:\n",
      "  %reload_ext rpy2.ipython\n"
     ]
    }
   ],
   "source": [
    "import rpy2\n",
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [],
   "source": [
    "PP2A_het_casos_129147457_AJ.to_csv('PP2A_het_casos_129147457_AJ.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129129094:A:G_G</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000051_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14</td>\n",
       "      <td>0</td>\n",
       "      <td>BCM_000158_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>37</td>\n",
       "      <td>0</td>\n",
       "      <td>BIOFIND_000049_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>48</td>\n",
       "      <td>0</td>\n",
       "      <td>BIOFIND_000136_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>56</td>\n",
       "      <td>0</td>\n",
       "      <td>BIOFIND_000205_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>1241</td>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000047_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>1257</td>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000252_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>1267</td>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000346_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>1268</td>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000364_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>1277</td>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000410_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>65 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Unnamed: 0  FID                IID  PAT  MAT  SEX  PHENOTYPE  \\\n",
       "0           12    0      BCM_000051_s1    0    0    1          2   \n",
       "1           14    0      BCM_000158_s1    0    0    1          2   \n",
       "2           37    0  BIOFIND_000049_s1    0    0    1          2   \n",
       "3           48    0  BIOFIND_000136_s1    0    0    2          2   \n",
       "4           56    0  BIOFIND_000205_s1    0    0    1          2   \n",
       "..         ...  ...                ...  ...  ...  ...        ...   \n",
       "60        1241    0      UMD_000047_s1    0    0    1          2   \n",
       "61        1257    0      UMD_000252_s1    0    0    2          2   \n",
       "62        1267    0      UMD_000346_s1    0    0    1          2   \n",
       "63        1268    0      UMD_000364_s1    0    0    1          2   \n",
       "64        1277    0      UMD_000410_s1    0    0    1          2   \n",
       "\n",
       "    chr9:129129094:A:G_G  chr9:129147457:C:T_T  \n",
       "0                      0                   1.0  \n",
       "1                      0                   1.0  \n",
       "2                      0                   1.0  \n",
       "3                      0                   1.0  \n",
       "4                      0                   1.0  \n",
       "..                   ...                   ...  \n",
       "60                     0                   1.0  \n",
       "61                     0                   1.0  \n",
       "62                     0                   1.0  \n",
       "63                     0                   1.0  \n",
       "64                     0                   1.0  \n",
       "\n",
       "[65 rows x 9 columns]"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "temp = pd.read_csv('PP2A_het_casos_129147457_AJ.csv')\n",
    "temp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [],
   "source": [
    "PP2A_het_controles_129147457_AJ.to_csv('PP2A_het_controles_129147457_AJ.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129129094:A:G_G</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>43</td>\n",
       "      <td>0</td>\n",
       "      <td>BIOFIND_000116_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>45</td>\n",
       "      <td>0</td>\n",
       "      <td>BIOFIND_000126_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>484</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_004515_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>485</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_004521_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>487</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_004628_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>501</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_005008_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>517</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_005259_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>520</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_005327_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>523</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_005425_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>532</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_005730_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>533</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_005747_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>557</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_005951_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>571</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006063_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>573</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006079_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>586</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006173_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>590</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006211_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>601</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006409_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>602</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006443_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>603</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006451_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>607</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_006835_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>628</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_007129_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>648</td>\n",
       "      <td>0</td>\n",
       "      <td>LCC_000015_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>664</td>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_000030_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>711</td>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_000860_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>721</td>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_001040_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>744</td>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_001439_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>763</td>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_001773_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>766</td>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_001814_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>769</td>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_001871_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>780</td>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_002073_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>787</td>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_002134_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>804</td>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_002324_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>807</td>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_002355_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>812</td>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_002407_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>824</td>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_002574_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>833</td>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_002680_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>839</td>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_002754_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>843</td>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_002787_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>857</td>\n",
       "      <td>0</td>\n",
       "      <td>PPMI_000079_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>1104</td>\n",
       "      <td>0</td>\n",
       "      <td>PPMI_000674_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>1273</td>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000386_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Unnamed: 0  FID                IID  PAT  MAT  SEX  PHENOTYPE  \\\n",
       "0           43    0  BIOFIND_000116_s1    0    0    2          1   \n",
       "1           45    0  BIOFIND_000126_s1    0    0    1          1   \n",
       "2          484    0  CORIELL_004515_s1    0    0    2          1   \n",
       "3          485    0  CORIELL_004521_s1    0    0    2          1   \n",
       "4          487    0  CORIELL_004628_s1    0    0    1          1   \n",
       "5          501    0  CORIELL_005008_s1    0    0    1          1   \n",
       "6          517    0  CORIELL_005259_s1    0    0    2          1   \n",
       "7          520    0  CORIELL_005327_s1    0    0    1          1   \n",
       "8          523    0  CORIELL_005425_s1    0    0    2          1   \n",
       "9          532    0  CORIELL_005730_s1    0    0    2          1   \n",
       "10         533    0  CORIELL_005747_s1    0    0    1          1   \n",
       "11         557    0  CORIELL_005951_s1    0    0    2          1   \n",
       "12         571    0  CORIELL_006063_s1    0    0    2          1   \n",
       "13         573    0  CORIELL_006079_s1    0    0    1          1   \n",
       "14         586    0  CORIELL_006173_s1    0    0    1          1   \n",
       "15         590    0  CORIELL_006211_s1    0    0    2          1   \n",
       "16         601    0  CORIELL_006409_s1    0    0    2          1   \n",
       "17         602    0  CORIELL_006443_s1    0    0    1          1   \n",
       "18         603    0  CORIELL_006451_s1    0    0    1          1   \n",
       "19         607    0  CORIELL_006835_s1    0    0    1          1   \n",
       "20         628    0  CORIELL_007129_s1    0    0    1          1   \n",
       "21         648    0      LCC_000015_s1    0    0    2          1   \n",
       "22         664    0     PAGE_000030_s1    0    0    2          1   \n",
       "23         711    0     PAGE_000860_s1    0    0    1          1   \n",
       "24         721    0     PAGE_001040_s1    0    0    1          1   \n",
       "25         744    0     PAGE_001439_s1    0    0    1          1   \n",
       "26         763    0     PAGE_001773_s1    0    0    1          1   \n",
       "27         766    0     PAGE_001814_s1    0    0    1          1   \n",
       "28         769    0     PAGE_001871_s1    0    0    1          1   \n",
       "29         780    0     PAGE_002073_s1    0    0    1          1   \n",
       "30         787    0     PAGE_002134_s1    0    0    2          1   \n",
       "31         804    0     PAGE_002324_s1    0    0    1          1   \n",
       "32         807    0     PAGE_002355_s1    0    0    1          1   \n",
       "33         812    0     PAGE_002407_s1    0    0    2          1   \n",
       "34         824    0     PAGE_002574_s1    0    0    2          1   \n",
       "35         833    0     PAGE_002680_s1    0    0    2          1   \n",
       "36         839    0     PAGE_002754_s1    0    0    1          1   \n",
       "37         843    0     PAGE_002787_s1    0    0    1          1   \n",
       "38         857    0     PPMI_000079_s1    0    0    2          1   \n",
       "39        1104    0     PPMI_000674_s1    0    0    1          1   \n",
       "40        1273    0      UMD_000386_s1    0    0    2          1   \n",
       "\n",
       "    chr9:129129094:A:G_G  chr9:129147457:C:T_T  \n",
       "0                      0                   1.0  \n",
       "1                      0                   1.0  \n",
       "2                      0                   1.0  \n",
       "3                      0                   1.0  \n",
       "4                      0                   1.0  \n",
       "5                      0                   1.0  \n",
       "6                      0                   1.0  \n",
       "7                      0                   1.0  \n",
       "8                      0                   1.0  \n",
       "9                      0                   1.0  \n",
       "10                     0                   1.0  \n",
       "11                     0                   1.0  \n",
       "12                     0                   1.0  \n",
       "13                     0                   1.0  \n",
       "14                     0                   1.0  \n",
       "15                     0                   1.0  \n",
       "16                     0                   1.0  \n",
       "17                     0                   1.0  \n",
       "18                     0                   1.0  \n",
       "19                     0                   1.0  \n",
       "20                     0                   1.0  \n",
       "21                     0                   1.0  \n",
       "22                     0                   1.0  \n",
       "23                     0                   1.0  \n",
       "24                     0                   1.0  \n",
       "25                     0                   1.0  \n",
       "26                     0                   1.0  \n",
       "27                     0                   1.0  \n",
       "28                     0                   1.0  \n",
       "29                     0                   1.0  \n",
       "30                     0                   1.0  \n",
       "31                     0                   1.0  \n",
       "32                     0                   1.0  \n",
       "33                     0                   1.0  \n",
       "34                     0                   1.0  \n",
       "35                     0                   1.0  \n",
       "36                     0                   1.0  \n",
       "37                     0                   1.0  \n",
       "38                     0                   1.0  \n",
       "39                     0                   1.0  \n",
       "40                     0                   1.0  "
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "temp = pd.read_csv('PP2A_het_controles_129147457_AJ.csv')\n",
    "temp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [],
   "source": [
    "PP2A_het_casos_129129094_AJ.to_csv('PP2A_het_casos_129129094_AJ.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129129094:A:G_G</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>60</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_000035_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>67</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_000138_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>137</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_001140_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>224</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_001997_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>301</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_002302_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>421</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_003933_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>437</td>\n",
       "      <td>0</td>\n",
       "      <td>CORIELL_004167_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>718</td>\n",
       "      <td>0</td>\n",
       "      <td>PAGE_001019_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>1252</td>\n",
       "      <td>0</td>\n",
       "      <td>UMD_000157_s1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Unnamed: 0  FID                IID  PAT  MAT  SEX  PHENOTYPE  \\\n",
       "0          60    0  CORIELL_000035_s1    0    0    2          2   \n",
       "1          67    0  CORIELL_000138_s1    0    0    1          2   \n",
       "2         137    0  CORIELL_001140_s1    0    0    1          2   \n",
       "3         224    0  CORIELL_001997_s1    0    0    1          2   \n",
       "4         301    0  CORIELL_002302_s1    0    0    1          2   \n",
       "5         421    0  CORIELL_003933_s1    0    0    2          2   \n",
       "6         437    0  CORIELL_004167_s1    0    0    2          2   \n",
       "7         718    0     PAGE_001019_s1    0    0    1          2   \n",
       "8        1252    0      UMD_000157_s1    0    0    1          2   \n",
       "\n",
       "   chr9:129129094:A:G_G  chr9:129147457:C:T_T  \n",
       "0                     1                   0.0  \n",
       "1                     1                   0.0  \n",
       "2                     1                   0.0  \n",
       "3                     1                   0.0  \n",
       "4                     1                   0.0  \n",
       "5                     1                   1.0  \n",
       "6                     1                   0.0  \n",
       "7                     1                   0.0  \n",
       "8                     1                   0.0  "
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "temp = pd.read_csv('PP2A_het_casos_129129094_AJ.csv')\n",
    "temp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [],
   "source": [
    "PP2A_het_controles_129129094_AJ.to_csv('PP2A_het_controles_129129094_AJ.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>PAT</th>\n",
       "      <th>MAT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>PHENOTYPE</th>\n",
       "      <th>chr9:129129094:A:G_G</th>\n",
       "      <th>chr9:129147457:C:T_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [Unnamed: 0, FID, IID, PAT, MAT, SEX, PHENOTYPE, chr9:129129094:A:G_G, chr9:129147457:C:T_T]\n",
       "Index: []"
      ]
     },
     "execution_count": 89,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "temp = pd.read_csv('PP2A_het_controles_129129094_AJ.csv')\n",
    "temp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GP2ID,GP2sampleID,manifest_id,phenotype,pheno_for_qc,other_pheno,sex_for_qc,age,age_of_onset,age_at_diagnosis,age_at_death,race_for_qc,family_history_for_qc,region_for_qc,study,pruned,pruned_reason,label,related\n",
      "SYNAPS-KZ_000001,SYNAPS-KZ_000001_s1,m1,Control,1,Control,2,64.0,,,,Asian,Not Reported,KAZ,SYNAPS-KZ,0,,CAS,\n",
      "SYNAPS-KZ_000002,SYNAPS-KZ_000002_s1,m1,Control,1,Control,1,53.0,,,,Asian,Not Reported,KAZ,SYNAPS-KZ,0,,CAS,\n",
      "SYNAPS-KZ_000003,SYNAPS-KZ_000003_s1,m1,Control,1,Control,2,59.0,,,,Asian,Not Reported,KAZ,SYNAPS-KZ,0,,CAS,\n",
      "SYNAPS-KZ_000004,SYNAPS-KZ_000004_s1,m1,Control,1,Control,1,57.0,,,,Asian,Not Reported,KAZ,SYNAPS-KZ,0,,CAS,\n",
      "SYNAPS-KZ_000005,SYNAPS-KZ_000005_s1,m1,Control,1,Control,1,59.0,,,,Asian,Not Reported,KAZ,SYNAPS-KZ,0,,CAS,\n",
      "SYNAPS-KZ_000006,SYNAPS-KZ_000006_s1,m1,Control,1,Control,1,53.0,,,,Asian,Not Reported,KAZ,SYNAPS-KZ,0,,CAS,\n",
      "SYNAPS-KZ_000007,SYNAPS-KZ_000007_s1,m1,Control,1,Control,1,56.0,,,,Asian,Not Reported,KAZ,SYNAPS-KZ,1,sex_prune,,\n",
      "SYNAPS-KZ_000008,SYNAPS-KZ_000008_s1,m1,Control,1,Control,2,48.0,,,,Asian,Not Reported,KAZ,SYNAPS-KZ,0,,CAS,\n",
      "SYNAPS-KZ_000009,SYNAPS-KZ_000009_s1,m1,Control,1,Control,2,56.0,,,,Asian,Not Reported,KAZ,SYNAPS-KZ,0,,CAS,\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "WORK_DIR='/home/jupyter/PP2A_GP2/'\n",
    "cd $WORK_DIR\n",
    "head master_key_release3_final.csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                  GP2ID         GP2sampleID manifest_id phenotype pheno_for_qc\n",
      "    1: SYNAPS-KZ_000001 SYNAPS-KZ_000001_s1          m1   Control            1\n",
      "    2: SYNAPS-KZ_000002 SYNAPS-KZ_000002_s1          m1   Control            1\n",
      "    3: SYNAPS-KZ_000003 SYNAPS-KZ_000003_s1          m1   Control            1\n",
      "    4: SYNAPS-KZ_000004 SYNAPS-KZ_000004_s1          m1   Control            1\n",
      "    5: SYNAPS-KZ_000005 SYNAPS-KZ_000005_s1          m1   Control            1\n",
      "   ---                                                                        \n",
      "16853:      PPMI_000789      PPMI_000789_s1          m2     Other           -9\n",
      "16854:      PPMI_000781      PPMI_000781_s1          m2     Other           -9\n",
      "16855:      PPMI_000818      PPMI_000818_s1          m2     Other           -9\n",
      "16856:      PPMI_000841      PPMI_000841_s1          m2     Other           -9\n",
      "16857:      PPMI_000808      PPMI_000808_s1          m2     Other           -9\n",
      "                                  other_pheno sex_for_qc age age_of_onset\n",
      "    1:                                Control          2  64           NA\n",
      "    2:                                Control          1  53           NA\n",
      "    3:                                Control          2  59           NA\n",
      "    4:                                Control          1  57           NA\n",
      "    5:                                Control          1  59           NA\n",
      "   ---                                                                   \n",
      "16853:   Genetically-enriched-cohort Affected          2  NA           NA\n",
      "16854: Genetidally-enriched-cohort Unaffected          1  57           NA\n",
      "16855:   Genetically-enriched-cohort Affected          1  60           NA\n",
      "16856:   Genetically-enriched-cohort Affected          2  68           NA\n",
      "16857:   Genetically-enriched-cohort Affected          1  NA           NA\n",
      "       age_at_diagnosis age_at_death race_for_qc family_history_for_qc\n",
      "    1:               NA           NA       Asian          Not Reported\n",
      "    2:               NA           NA       Asian          Not Reported\n",
      "    3:               NA           NA       Asian          Not Reported\n",
      "    4:               NA           NA       Asian          Not Reported\n",
      "    5:               NA           NA       Asian          Not Reported\n",
      "   ---                                                                \n",
      "16853:               NA           NA       White          Not Reported\n",
      "16854:               NA           NA       White          Not Reported\n",
      "16855:               55           55       White          Not Reported\n",
      "16856:               57           57       White          Not Reported\n",
      "16857:               NA           NA       White          Not Reported\n",
      "       region_for_qc     study pruned  pruned_reason label related\n",
      "    1:           KAZ SYNAPS-KZ      0                  CAS      NA\n",
      "    2:           KAZ SYNAPS-KZ      0                  CAS      NA\n",
      "    3:           KAZ SYNAPS-KZ      0                  CAS      NA\n",
      "    4:           KAZ SYNAPS-KZ      0                  CAS      NA\n",
      "    5:           KAZ SYNAPS-KZ      0                  CAS      NA\n",
      "   ---                                                            \n",
      "16853:           ISR      PPMI      0                   AJ      NA\n",
      "16854:           ISR      PPMI      0                   AJ      NA\n",
      "16855:           ISR      PPMI      1 callrate_prune            NA\n",
      "16856:           ISR      PPMI      0                   AJ       1\n",
      "16857:           ISR      PPMI      0                   AJ      NA\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "data.table 1.14.6 using 2 threads (see ?getDTthreads).  Latest news: r-datatable.com\n",
       "In addition: Warning message:\n",
       "In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
       "  libraries ‘/home/jupyter/packages’, ‘/usr/lib/R/site-library’ contain no packages\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "setwd('/home/jupyter/PP2A_GP2/')\n",
    "library(data.table)\n",
    "covs <- fread(\"master_key_release3_final.csv\", header =T)\n",
    "covs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                  GP2ID                 IID manifest_id phenotype pheno_for_qc\n",
      "    1: SYNAPS-KZ_000001 SYNAPS-KZ_000001_s1          m1   Control            1\n",
      "    2: SYNAPS-KZ_000002 SYNAPS-KZ_000002_s1          m1   Control            1\n",
      "    3: SYNAPS-KZ_000003 SYNAPS-KZ_000003_s1          m1   Control            1\n",
      "    4: SYNAPS-KZ_000004 SYNAPS-KZ_000004_s1          m1   Control            1\n",
      "    5: SYNAPS-KZ_000005 SYNAPS-KZ_000005_s1          m1   Control            1\n",
      "   ---                                                                        \n",
      "16853:      PPMI_000789      PPMI_000789_s1          m2     Other           -9\n",
      "16854:      PPMI_000781      PPMI_000781_s1          m2     Other           -9\n",
      "16855:      PPMI_000818      PPMI_000818_s1          m2     Other           -9\n",
      "16856:      PPMI_000841      PPMI_000841_s1          m2     Other           -9\n",
      "16857:      PPMI_000808      PPMI_000808_s1          m2     Other           -9\n",
      "                                  other_pheno sex_for_qc age age_of_onset\n",
      "    1:                                Control          2  64           NA\n",
      "    2:                                Control          1  53           NA\n",
      "    3:                                Control          2  59           NA\n",
      "    4:                                Control          1  57           NA\n",
      "    5:                                Control          1  59           NA\n",
      "   ---                                                                   \n",
      "16853:   Genetically-enriched-cohort Affected          2  NA           NA\n",
      "16854: Genetidally-enriched-cohort Unaffected          1  57           NA\n",
      "16855:   Genetically-enriched-cohort Affected          1  60           NA\n",
      "16856:   Genetically-enriched-cohort Affected          2  68           NA\n",
      "16857:   Genetically-enriched-cohort Affected          1  NA           NA\n",
      "       age_at_diagnosis age_at_death race_for_qc family_history_for_qc\n",
      "    1:               NA           NA       Asian          Not Reported\n",
      "    2:               NA           NA       Asian          Not Reported\n",
      "    3:               NA           NA       Asian          Not Reported\n",
      "    4:               NA           NA       Asian          Not Reported\n",
      "    5:               NA           NA       Asian          Not Reported\n",
      "   ---                                                                \n",
      "16853:               NA           NA       White          Not Reported\n",
      "16854:               NA           NA       White          Not Reported\n",
      "16855:               55           55       White          Not Reported\n",
      "16856:               57           57       White          Not Reported\n",
      "16857:               NA           NA       White          Not Reported\n",
      "       region_for_qc     study pruned  pruned_reason label related\n",
      "    1:           KAZ SYNAPS-KZ      0                  CAS      NA\n",
      "    2:           KAZ SYNAPS-KZ      0                  CAS      NA\n",
      "    3:           KAZ SYNAPS-KZ      0                  CAS      NA\n",
      "    4:           KAZ SYNAPS-KZ      0                  CAS      NA\n",
      "    5:           KAZ SYNAPS-KZ      0                  CAS      NA\n",
      "   ---                                                            \n",
      "16853:           ISR      PPMI      0                   AJ      NA\n",
      "16854:           ISR      PPMI      0                   AJ      NA\n",
      "16855:           ISR      PPMI      1 callrate_prune            NA\n",
      "16856:           ISR      PPMI      0                   AJ       1\n",
      "16857:           ISR      PPMI      0                   AJ      NA\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "colnames(covs)[2]  <- \"IID\"  \n",
    "covs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                      IID              GP2ID manifest_id phenotype pheno_for_qc\n",
      " 1:         BCM_000051_s1         BCM_000051          m1        PD            2\n",
      " 2:         BCM_000158_s1         BCM_000158          m1        PD            2\n",
      " 3:     BIOFIND_000049_s1     BIOFIND_000049          m1        PD            2\n",
      " 4:     BIOFIND_000136_s1     BIOFIND_000136          m1        PD            2\n",
      " 5:     BIOFIND_000205_s1     BIOFIND_000205          m1        PD            2\n",
      " 6:     CORIELL_000100_s1     CORIELL_000100          m3        PD            2\n",
      " 7:     CORIELL_000121_s1     CORIELL_000121          m3        PD            2\n",
      " 8:     CORIELL_000297_s1     CORIELL_000297          m3        PD            2\n",
      " 9:     CORIELL_000341_s1     CORIELL_000341          m3        PD            2\n",
      "10:     CORIELL_000527_s1     CORIELL_000527          m3        PD            2\n",
      "11:     CORIELL_000547_s1     CORIELL_000547          m3        PD            2\n",
      "12:     CORIELL_000772_s1     CORIELL_000772          m3        PD            2\n",
      "13:     CORIELL_000795_s1     CORIELL_000795          m3        PD            2\n",
      "14:     CORIELL_000876_s1     CORIELL_000876          m3        PD            2\n",
      "15:     CORIELL_000983_s1     CORIELL_000983          m3        PD            2\n",
      "16:     CORIELL_000986_s1     CORIELL_000986          m3        PD            2\n",
      "17:     CORIELL_001062_s1     CORIELL_001062          m3        PD            2\n",
      "18:     CORIELL_001089_s1     CORIELL_001089          m3        PD            2\n",
      "19:     CORIELL_001221_s1     CORIELL_001221          m3        PD            2\n",
      "20:     CORIELL_001343_s1     CORIELL_001343          m3        PD            2\n",
      "21:     CORIELL_001409_s1     CORIELL_001409          m3        PD            2\n",
      "22:     CORIELL_001450_s1     CORIELL_001450          m3        PD            2\n",
      "23:     CORIELL_001806_s1     CORIELL_001806          m3        PD            2\n",
      "24:     CORIELL_001822_s1     CORIELL_001822          m3        PD            2\n",
      "25:     CORIELL_001824_s1     CORIELL_001824          m3        PD            2\n",
      "26:     CORIELL_001833_s1     CORIELL_001833          m3        PD            2\n",
      "27:     CORIELL_001875_s1     CORIELL_001875          m3        PD            2\n",
      "28:     CORIELL_001908_s1     CORIELL_001908          m3        PD            2\n",
      "29:     CORIELL_001946_s1     CORIELL_001946          m3        PD            2\n",
      "30:     CORIELL_001980_s1     CORIELL_001980          m3        PD            2\n",
      "31:     CORIELL_002080_s1     CORIELL_002080          m3        PD            2\n",
      "32:     CORIELL_002238_s1     CORIELL_002238          m3        PD            2\n",
      "33:     CORIELL_002305_s1     CORIELL_002305          m3        PD            2\n",
      "34:     CORIELL_002317_s1     CORIELL_002317          m3        PD            2\n",
      "35:     CORIELL_002441_s1     CORIELL_002441          m3        PD            2\n",
      "36:     CORIELL_002522_s1     CORIELL_002522          m3        PD            2\n",
      "37:     CORIELL_002544_s1     CORIELL_002544          m3        PD            2\n",
      "38:     CORIELL_002551_s1     CORIELL_002551          m3        PD            2\n",
      "39:     CORIELL_002566_s1     CORIELL_002566          m3        PD            2\n",
      "40:     CORIELL_002567_s1     CORIELL_002567          m3        PD            2\n",
      "41:     CORIELL_002798_s1     CORIELL_002798          m3        PD            2\n",
      "42:     CORIELL_003099_s1     CORIELL_003099          m3        PD            2\n",
      "43:     CORIELL_003302_s1     CORIELL_003302          m3        PD            2\n",
      "44:     CORIELL_003335_s1     CORIELL_003335          m3        PD            2\n",
      "45:     CORIELL_003791_s1     CORIELL_003791          m3        PD            2\n",
      "46:     CORIELL_003798_s1     CORIELL_003798          m3        PD            2\n",
      "47:     CORIELL_003916_s1     CORIELL_003916          m3        PD            2\n",
      "48:     CORIELL_003933_s1     CORIELL_003933          m3        PD            2\n",
      "49:     CORIELL_003964_s1     CORIELL_003964          m3        PD            2\n",
      "50:     CORIELL_004262_s1     CORIELL_004262          m3        PD            2\n",
      "51:     CORIELL_004300_s1     CORIELL_004300          m4        PD            2\n",
      "52:     CORIELL_004375_s1     CORIELL_004375          m4        PD            2\n",
      "53:     CORIELL_004402_s1     CORIELL_004402          m4        PD            2\n",
      "54:     CORIELL_004407_s1     CORIELL_004407          m4        PD            2\n",
      "55:     CORIELL_004441_s1     CORIELL_004441          m4        PD            2\n",
      "56:         LCC_000002_s1         LCC_000002          m1        PD            2\n",
      "57: MDGAP-KINGS_000070_s1 MDGAP-KINGS_000070          m1        PD            2\n",
      "58:        PAGE_000572_s1        PAGE_000572          m1        PD            2\n",
      "59:        PAGE_001110_s1        PAGE_001110          m1        PD            2\n",
      "60:          S4_000081_s1          S4_000081          m1        PD            2\n",
      "61:         UMD_000047_s1         UMD_000047          m1        PD            2\n",
      "62:         UMD_000252_s1         UMD_000252          m1        PD            2\n",
      "63:         UMD_000346_s1         UMD_000346          m1        PD            2\n",
      "64:         UMD_000364_s1         UMD_000364          m1        PD            2\n",
      "65:         UMD_000410_s1         UMD_000410          m1        PD            2\n",
      "                      IID              GP2ID manifest_id phenotype pheno_for_qc\n",
      "    other_pheno sex_for_qc      age age_of_onset age_at_diagnosis age_at_death\n",
      " 1:          PD          1       NA           NA               NA           NA\n",
      " 2:          PD          1       NA           NA               NA           NA\n",
      " 3:          PD          1 60.00000     54.00000         60.00000           NA\n",
      " 4:          PD          2 71.00000     61.00000         71.00000           NA\n",
      " 5:          PD          1 65.00000     60.00000         62.00000           NA\n",
      " 6:          PD          1 52.00000     44.00000               NA           NA\n",
      " 7:          PD          2 46.00000     44.00000               NA           NA\n",
      " 8:          PD          2 69.00000     66.00000               NA           NA\n",
      " 9:          PD          1 74.00000     70.00000               NA           NA\n",
      "10:          PD          1 69.00000     12.00000               NA           NA\n",
      "11:          PD          1 65.00000     45.00000               NA           NA\n",
      "12:          PD          2 61.00000     35.00000               NA           NA\n",
      "13:          PD          2 59.00000     50.00000               NA           NA\n",
      "14:          PD          2 64.00000     62.00000               NA           NA\n",
      "15:          PD          2 70.00000     43.00000               NA           NA\n",
      "16:          PD          2 53.00000     47.00000               NA           NA\n",
      "17:          PD          1 58.00000     53.00000               NA           NA\n",
      "18:          PD          1 71.00000     67.00000               NA           NA\n",
      "19:          PD          1 53.00000     44.00000               NA           NA\n",
      "20:          PD          1 60.00000     31.00000               NA           NA\n",
      "21:          PD          2 63.00000     63.00000               NA           NA\n",
      "22:          PD          1 59.00000     45.00000               NA           NA\n",
      "23:          PD          1 60.00000     53.00000               NA           NA\n",
      "24:          PD          1 67.00000     57.00000               NA           NA\n",
      "25:          PD          2 54.00000     48.00000               NA           NA\n",
      "26:          PD          1 58.00000     52.00000               NA           NA\n",
      "27:          PD          1 68.00000     63.00000               NA           NA\n",
      "28:          PD          1 63.00000     51.00000               NA           NA\n",
      "29:          PD          1 61.00000     14.00000         55.00000           NA\n",
      "30:          PD          1 82.00000     76.00000               NA           NA\n",
      "31:          PD          1 75.00000     70.00000               NA           NA\n",
      "32:          PD          1 69.00000     64.00000               NA           NA\n",
      "33:          PD          1 62.00000     51.00000               NA           NA\n",
      "34:          PD          1 72.00000     68.00000         68.00000           NA\n",
      "35:          PD          1 74.00000     63.00000               NA           NA\n",
      "36:          PD          2 61.00000     57.00000               NA           NA\n",
      "37:          PD          1 64.00000     61.00000         61.00000           NA\n",
      "38:          PD          2 62.00000     62.00000         62.00000           NA\n",
      "39:          PD          2 67.00000     32.00000               NA           NA\n",
      "40:          PD          2 71.00000     49.00000               NA           NA\n",
      "41:          PD          2 60.00000     53.00000               NA           NA\n",
      "42:          PD          1 66.00000     60.00000         62.00000           NA\n",
      "43:          PD          1 78.00000     75.00000         77.00000           NA\n",
      "44:          PD          1 80.00000     68.00000         60.00000           NA\n",
      "45:          PD          2 73.00000     66.00000               NA           NA\n",
      "46:          PD          1 51.00000     46.00000         49.00000           NA\n",
      "47:          PD          1 72.00000     70.00000         71.00000           NA\n",
      "48:          PD          2 78.00000     60.00000               NA           NA\n",
      "49:          PD          1 65.00000     32.00000         32.00000           NA\n",
      "50:          PD          1 79.00000     50.00000         55.00000           NA\n",
      "51:          PD          1 66.00000     62.00000         65.00000           NA\n",
      "52:          PD          1 61.00000     53.00000         56.00000           NA\n",
      "53:          PD          1 81.00000     76.00000         76.00000           NA\n",
      "54:          PD          1 67.00000     45.00000         45.00000           NA\n",
      "55:          PD          1 78.00000     73.00000         73.00000           NA\n",
      "56:          PD          2       NA     43.00000         44.00000           NA\n",
      "57:          PD          1       NA           NA               NA           77\n",
      "58:          PD          1 76.86242     68.58333         70.58333           NA\n",
      "59:          PD          1 77.99863     65.91667         65.91667           NA\n",
      "60:          PD          1 74.00000           NA         62.00000           NA\n",
      "61:          PD          1       NA           NA               NA           NA\n",
      "62:          PD          2       NA           NA               NA           NA\n",
      "63:          PD          1       NA           NA               NA           NA\n",
      "64:          PD          1       NA           NA               NA           NA\n",
      "65:          PD          1       NA           NA               NA           NA\n",
      "    other_pheno sex_for_qc      age age_of_onset age_at_diagnosis age_at_death\n",
      "     race_for_qc family_history_for_qc region_for_qc       study pruned\n",
      " 1:        White                    No           USA         BCM      0\n",
      " 2:        White                    No           USA         BCM      0\n",
      " 3:        White                   Yes           USA     BIOFIND      0\n",
      " 4:        White                    No           USA     BIOFIND      0\n",
      " 5:        White                    No           USA     BIOFIND      0\n",
      " 6:        White                    No           USA     CORIELL      0\n",
      " 7:        White                   Yes           USA     CORIELL      0\n",
      " 8:        White                    No           USA     CORIELL      0\n",
      " 9:        White                    No           USA     CORIELL      0\n",
      "10:        White                    No           USA     CORIELL      0\n",
      "11:        White                    No           USA     CORIELL      0\n",
      "12:        White                   Yes           USA     CORIELL      0\n",
      "13:        White                   Yes           USA     CORIELL      0\n",
      "14:        White                    No           USA     CORIELL      0\n",
      "15:        White                    No           USA     CORIELL      0\n",
      "16:        White                   Yes           USA     CORIELL      0\n",
      "17:        White                    No           USA     CORIELL      0\n",
      "18:        White                    No           USA     CORIELL      0\n",
      "19:        White                   Yes           USA     CORIELL      0\n",
      "20:        White                   Yes           USA     CORIELL      0\n",
      "21:        White                    No           USA     CORIELL      0\n",
      "22:        White                   Yes           USA     CORIELL      0\n",
      "23:        White                    No           USA     CORIELL      0\n",
      "24:        White                   Yes           USA     CORIELL      0\n",
      "25:        White                    No           USA     CORIELL      0\n",
      "26:        White                    No           USA     CORIELL      0\n",
      "27:        White                   Yes           USA     CORIELL      0\n",
      "28:        White                    No           USA     CORIELL      0\n",
      "29:        White                    No           USA     CORIELL      0\n",
      "30:        White                    No           USA     CORIELL      0\n",
      "31:        White                    No           USA     CORIELL      0\n",
      "32:        White                    No           USA     CORIELL      0\n",
      "33:        White                   Yes           USA     CORIELL      0\n",
      "34:        White                    No           USA     CORIELL      0\n",
      "35:        White                   Yes           USA     CORIELL      0\n",
      "36:        White                    No           USA     CORIELL      0\n",
      "37:        White                   Yes           USA     CORIELL      0\n",
      "38:        White                    No           USA     CORIELL      0\n",
      "39:        White                    No           USA     CORIELL      0\n",
      "40:        White                    No           USA     CORIELL      0\n",
      "41:        White                   Yes           USA     CORIELL      0\n",
      "42:        White                    No           USA     CORIELL      0\n",
      "43:        White                    No           USA     CORIELL      0\n",
      "44:        White                    No           USA     CORIELL      0\n",
      "45:        White                   Yes           USA     CORIELL      0\n",
      "46:        White                    No           USA     CORIELL      0\n",
      "47:        White                    No           USA     CORIELL      0\n",
      "48:        White                   Yes           USA     CORIELL      0\n",
      "49:        White                   Yes           USA     CORIELL      0\n",
      "50:        White                   Yes           USA     CORIELL      0\n",
      "51:        White                    No           USA     CORIELL      0\n",
      "52:        White                   Yes           USA     CORIELL      0\n",
      "53:        White                    No           USA     CORIELL      0\n",
      "54:        White                    No           USA     CORIELL      0\n",
      "55:        White                    No           USA     CORIELL      0\n",
      "56: Not Reported          Not Reported           ISR         LCC      0\n",
      "57:      Unknown          Not Reported           GBR MDGAP-KINGS      0\n",
      "58:        White                    No           USA        PAGE      0\n",
      "59:        White                    No           USA        PAGE      0\n",
      "60:        White          Not Reported           USA          S4      0\n",
      "61: Not Reported                    No           USA         UMD      0\n",
      "62:        White                    No           USA         UMD      0\n",
      "63:        White                    No           USA         UMD      0\n",
      "64:        White                   Yes           USA         UMD      0\n",
      "65:        White                    No           USA         UMD      0\n",
      "     race_for_qc family_history_for_qc region_for_qc       study pruned\n",
      "    pruned_reason label related   V1 FID PAT MAT SEX PHENOTYPE\n",
      " 1:            NA    AJ      NA   12   0   0   0   1         2\n",
      " 2:            NA    AJ      NA   14   0   0   0   1         2\n",
      " 3:            NA    AJ      NA   37   0   0   0   1         2\n",
      " 4:            NA    AJ      NA   48   0   0   0   2         2\n",
      " 5:            NA    AJ      NA   56   0   0   0   1         2\n",
      " 6:            NA    AJ      NA   63   0   0   0   1         2\n",
      " 7:            NA    AJ      NA   66   0   0   0   2         2\n",
      " 8:            NA    AJ      NA   77   0   0   0   2         2\n",
      " 9:            NA    AJ      NA   79   0   0   0   1         2\n",
      "10:            NA    AJ      NA   86   0   0   0   1         2\n",
      "11:            NA    AJ      NA   88   0   0   0   1         2\n",
      "12:            NA    AJ      NA  102   0   0   0   2         2\n",
      "13:            NA    AJ      NA  106   0   0   0   2         2\n",
      "14:            NA    AJ      NA  113   0   0   0   2         2\n",
      "15:            NA    AJ      NA  123   0   0   0   2         2\n",
      "16:            NA    AJ      NA  124   0   0   0   2         2\n",
      "17:            NA    AJ      NA  132   0   0   0   1         2\n",
      "18:            NA    AJ      NA  133   0   0   0   1         2\n",
      "19:            NA    AJ      NA  142   0   0   0   1         2\n",
      "20:            NA    AJ      NA  155   0   0   0   1         2\n",
      "21:            NA    AJ      NA  159   0   0   0   2         2\n",
      "22:            NA    AJ      NA  164   0   0   0   1         2\n",
      "23:            NA    AJ      NA  189   0   0   0   1         2\n",
      "24:            NA    AJ      NA  192   0   0   0   1         2\n",
      "25:            NA    AJ      NA  193   0   0   0   2         2\n",
      "26:            NA    AJ      NA  195   0   0   0   1         2\n",
      "27:            NA    AJ      NA  201   0   0   0   1         2\n",
      "28:            NA    AJ      NA  207   0   0   0   1         2\n",
      "29:            NA    AJ      NA  216   0   0   0   1         2\n",
      "30:            NA    AJ      NA  218   0   0   0   1         2\n",
      "31:            NA    AJ      NA  248   0   0   0   1         2\n",
      "32:            NA    AJ      NA  289   0   0   0   1         2\n",
      "33:            NA    AJ      NA  302   0   0   0   1         2\n",
      "34:            NA    AJ      NA  305   0   0   0   1         2\n",
      "35:            NA    AJ      NA  324   0   0   0   1         2\n",
      "36:            NA    AJ      NA  330   0   0   0   2         2\n",
      "37:            NA    AJ      NA  334   0   0   0   1         2\n",
      "38:            NA    AJ      NA  335   0   0   0   2         2\n",
      "39:            NA    AJ      NA  338   0   0   0   2         2\n",
      "40:            NA    AJ      NA  339   0   0   0   2         2\n",
      "41:            NA    AJ      NA  355   0   0   0   2         2\n",
      "42:            NA    AJ      NA  371   0   0   0   1         2\n",
      "43:            NA    AJ      NA  382   0   0   0   1         2\n",
      "44:            NA    AJ      NA  384   0   0   0   1         2\n",
      "45:            NA    AJ       1  412   0   0   0   2         2\n",
      "46:            NA    AJ      NA  413   0   0   0   1         2\n",
      "47:            NA    AJ      NA  420   0   0   0   1         2\n",
      "48:            NA    AJ      NA  421   0   0   0   2         2\n",
      "49:            NA    AJ      NA  424   0   0   0   1         2\n",
      "50:            NA    AJ       1  444   0   0   0   1         2\n",
      "51:            NA    AJ      NA  451   0   0   0   1         2\n",
      "52:            NA    AJ      NA  460   0   0   0   1         2\n",
      "53:            NA    AJ      NA  467   0   0   0   1         2\n",
      "54:            NA    AJ      NA  469   0   0   0   1         2\n",
      "55:            NA    AJ      NA  478   0   0   0   1         2\n",
      "56:            NA    AJ      NA  637   0   0   0   2         2\n",
      "57:            NA    AJ      NA  657   0   0   0   1         2\n",
      "58:            NA    AJ      NA  692   0   0   0   1         2\n",
      "59:            NA    AJ      NA  727   0   0   0   1         2\n",
      "60:            NA    AJ      NA 1236   0   0   0   1         2\n",
      "61:            NA    AJ      NA 1241   0   0   0   1         2\n",
      "62:            NA    AJ      NA 1257   0   0   0   2         2\n",
      "63:            NA    AJ      NA 1267   0   0   0   1         2\n",
      "64:            NA    AJ      NA 1268   0   0   0   1         2\n",
      "65:            NA    AJ      NA 1277   0   0   0   1         2\n",
      "    pruned_reason label related   V1 FID PAT MAT SEX PHENOTYPE\n",
      "    chr9:129129094:A:G_G chr9:129147457:C:T_T\n",
      " 1:                    0                    1\n",
      " 2:                    0                    1\n",
      " 3:                    0                    1\n",
      " 4:                    0                    1\n",
      " 5:                    0                    1\n",
      " 6:                    0                    1\n",
      " 7:                    0                    1\n",
      " 8:                    0                    1\n",
      " 9:                    0                    1\n",
      "10:                    0                    1\n",
      "11:                    0                    1\n",
      "12:                    0                    1\n",
      "13:                    0                    1\n",
      "14:                    0                    1\n",
      "15:                    0                    1\n",
      "16:                    0                    1\n",
      "17:                    0                    1\n",
      "18:                    0                    1\n",
      "19:                    0                    1\n",
      "20:                    0                    1\n",
      "21:                    0                    1\n",
      "22:                    0                    1\n",
      "23:                    0                    1\n",
      "24:                    0                    1\n",
      "25:                    0                    1\n",
      "26:                    0                    1\n",
      "27:                    0                    1\n",
      "28:                    0                    1\n",
      "29:                    0                    1\n",
      "30:                    0                    1\n",
      "31:                    0                    1\n",
      "32:                    0                    1\n",
      "33:                    0                    1\n",
      "34:                    0                    1\n",
      "35:                    0                    1\n",
      "36:                    0                    1\n",
      "37:                    0                    1\n",
      "38:                    0                    1\n",
      "39:                    0                    1\n",
      "40:                    0                    1\n",
      "41:                    0                    1\n",
      "42:                    0                    1\n",
      "43:                    0                    1\n",
      "44:                    0                    1\n",
      "45:                    0                    1\n",
      "46:                    0                    1\n",
      "47:                    0                    1\n",
      "48:                    1                    1\n",
      "49:                    0                    1\n",
      "50:                    0                    1\n",
      "51:                    0                    1\n",
      "52:                    0                    1\n",
      "53:                    0                    1\n",
      "54:                    0                    1\n",
      "55:                    0                    1\n",
      "56:                    0                    1\n",
      "57:                    0                    1\n",
      "58:                    0                    1\n",
      "59:                    0                    1\n",
      "60:                    0                    1\n",
      "61:                    0                    1\n",
      "62:                    0                    1\n",
      "63:                    0                    1\n",
      "64:                    0                    1\n",
      "65:                    0                    1\n",
      "    chr9:129129094:A:G_G chr9:129147457:C:T_T\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "setwd('/home/jupyter/PP2A_GP2/')\n",
    "library(data.table)\n",
    "PP2A_het_casos_129147457_AJ <- fread(\"PP2A_het_casos_129147457_AJ.csv\", header =T)\n",
    "PP2A_het_casos_129147457_AJ\n",
    "temp <- merge(covs, PP2A_het_casos_129147457_AJ, by=\"IID\")\n",
    "head(temp)\n",
    "write.table(temp, file = \"PP2A_het_casos_129147457_AJ_extracted.txt\", quote = F, row.names = F, sep = \"\\t\")\n",
    "PP2A_het_casos_129147457_AJ_extracted <- fread(\"PP2A_het_casos_129147457_AJ_extracted.txt\", header =T)\n",
    "PP2A_het_casos_129147457_AJ_extracted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                  IID          GP2ID manifest_id phenotype pheno_for_qc\n",
      " 1: BIOFIND_000116_s1 BIOFIND_000116          m1   Control            1\n",
      " 2: BIOFIND_000126_s1 BIOFIND_000126          m1   Control            1\n",
      " 3: CORIELL_004515_s1 CORIELL_004515          m6   Control            1\n",
      " 4: CORIELL_004521_s1 CORIELL_004521          m6   Control            1\n",
      " 5: CORIELL_004628_s1 CORIELL_004628          m6   Control            1\n",
      " 6: CORIELL_005008_s1 CORIELL_005008          m6   Control            1\n",
      " 7: CORIELL_005259_s1 CORIELL_005259          m6   Control            1\n",
      " 8: CORIELL_005327_s1 CORIELL_005327          m6   Control            1\n",
      " 9: CORIELL_005425_s1 CORIELL_005425          m6   Control            1\n",
      "10: CORIELL_005730_s1 CORIELL_005730          m6   Control            1\n",
      "11: CORIELL_005747_s1 CORIELL_005747          m6   Control            1\n",
      "12: CORIELL_005951_s1 CORIELL_005951          m6   Control            1\n",
      "13: CORIELL_006063_s1 CORIELL_006063          m6   Control            1\n",
      "14: CORIELL_006079_s1 CORIELL_006079          m6   Control            1\n",
      "15: CORIELL_006173_s1 CORIELL_006173          m6   Control            1\n",
      "16: CORIELL_006211_s1 CORIELL_006211          m6   Control            1\n",
      "17: CORIELL_006409_s1 CORIELL_006409          m6   Control            1\n",
      "18: CORIELL_006443_s1 CORIELL_006443          m6   Control            1\n",
      "19: CORIELL_006451_s1 CORIELL_006451          m6   Control            1\n",
      "20: CORIELL_006835_s1 CORIELL_006835          m6   Control            1\n",
      "21: CORIELL_007129_s1 CORIELL_007129          m6   Control            1\n",
      "22:     LCC_000015_s1     LCC_000015          m1   Control            1\n",
      "23:    PAGE_000030_s1    PAGE_000030          m1   Control            1\n",
      "24:    PAGE_000860_s1    PAGE_000860          m1   Control            1\n",
      "25:    PAGE_001040_s1    PAGE_001040          m1   Control            1\n",
      "26:    PAGE_001439_s1    PAGE_001439          m1   Control            1\n",
      "27:    PAGE_001773_s1    PAGE_001773          m1   Control            1\n",
      "28:    PAGE_001814_s1    PAGE_001814          m1   Control            1\n",
      "29:    PAGE_001871_s1    PAGE_001871          m1   Control            1\n",
      "30:    PAGE_002073_s1    PAGE_002073          m1   Control            1\n",
      "31:    PAGE_002134_s1    PAGE_002134          m1   Control            1\n",
      "32:    PAGE_002324_s1    PAGE_002324          m1   Control            1\n",
      "33:    PAGE_002355_s1    PAGE_002355          m1   Control            1\n",
      "34:    PAGE_002407_s1    PAGE_002407          m1   Control            1\n",
      "35:    PAGE_002574_s1    PAGE_002574          m1   Control            1\n",
      "36:    PAGE_002680_s1    PAGE_002680          m1   Control            1\n",
      "37:    PAGE_002754_s1    PAGE_002754          m1   Control            1\n",
      "38:    PAGE_002787_s1    PAGE_002787          m1   Control            1\n",
      "39:    PPMI_000079_s1    PPMI_000079          m1   Control            1\n",
      "40:    PPMI_000674_s1    PPMI_000674          m1   Control            1\n",
      "41:     UMD_000386_s1     UMD_000386          m1   Control            1\n",
      "                  IID          GP2ID manifest_id phenotype pheno_for_qc\n",
      "    other_pheno sex_for_qc  age age_of_onset age_at_diagnosis age_at_death\n",
      " 1:     Control          2 72.0           NA               NA           NA\n",
      " 2:     Control          1 81.0           NA               NA           NA\n",
      " 3:     Control          2 65.0           NA               NA           NA\n",
      " 4:     Control          2 55.0           NA               NA           NA\n",
      " 5:     Control          1 91.0           NA               NA           NA\n",
      " 6:     Control          1 57.0           NA               NA           NA\n",
      " 7:     Control          2 83.0           NA               NA           NA\n",
      " 8:     Control          1 55.0           NA               NA           NA\n",
      " 9:     Control          2 59.0           NA               NA           NA\n",
      "10:     Control          2 62.0           NA               NA           NA\n",
      "11:     Control          1 60.0           NA               NA           NA\n",
      "12:     Control          2 55.0           NA               NA           NA\n",
      "13:     Control          2 74.0           NA               NA           NA\n",
      "14:     Control          1 69.0           NA               NA           NA\n",
      "15:     Control          1 77.0           NA               NA           NA\n",
      "16:     Control          2 66.0           NA               NA           NA\n",
      "17:     Control          2 64.0           NA               NA           NA\n",
      "18:     Control          1 56.0           NA               NA           NA\n",
      "19:     Control          1 71.0           NA               NA           NA\n",
      "20:     Control          1 63.0           NA               NA           NA\n",
      "21:     Control          1 84.0           NA               NA           NA\n",
      "22:     Control          2 51.0           NA               NA           NA\n",
      "23:     Control          2   NA           NA               NA           NA\n",
      "24:     Control          1   NA           NA               NA           NA\n",
      "25:     Control          1   NA           NA               NA           NA\n",
      "26:     Control          1   NA           NA               NA           NA\n",
      "27:     Control          1   NA           NA               NA           NA\n",
      "28:     Control          1   NA           NA               NA           NA\n",
      "29:     Control          1   NA           NA               NA           NA\n",
      "30:     Control          1   NA           NA               NA           NA\n",
      "31:     Control          2   NA           NA               NA           NA\n",
      "32:     Control          1   NA           NA               NA           NA\n",
      "33:     Control          1   NA           NA               NA           NA\n",
      "34:     Control          2   NA           NA               NA           NA\n",
      "35:     Control          2   NA           NA               NA           NA\n",
      "36:     Control          2   NA           NA               NA           NA\n",
      "37:     Control          1   NA           NA               NA           NA\n",
      "38:     Control          1   NA           NA               NA           NA\n",
      "39:     Control          2 43.2           NA               NA           NA\n",
      "40:     Control          1 52.1           NA               NA           NA\n",
      "41:     Control          2   NA           NA               NA           NA\n",
      "    other_pheno sex_for_qc  age age_of_onset age_at_diagnosis age_at_death\n",
      "     race_for_qc family_history_for_qc region_for_qc   study pruned\n",
      " 1:        White                    No           USA BIOFIND      0\n",
      " 2:        White                    No           USA BIOFIND      0\n",
      " 3:        White          Not Reported           USA CORIELL      0\n",
      " 4:        White          Not Reported           USA CORIELL      0\n",
      " 5:        White                    No           USA CORIELL      0\n",
      " 6:        White                    No           USA CORIELL      0\n",
      " 7:        White                    No           USA CORIELL      0\n",
      " 8:        White                    No           USA CORIELL      0\n",
      " 9:        White                    No           USA CORIELL      0\n",
      "10:        White                    No           USA CORIELL      0\n",
      "11:        White                    No           USA CORIELL      0\n",
      "12:        White                    No           USA CORIELL      0\n",
      "13:        White                    No           USA CORIELL      0\n",
      "14:        White                    No           USA CORIELL      0\n",
      "15:        White                    No           USA CORIELL      0\n",
      "16:        White                    No           USA CORIELL      0\n",
      "17:        White                    No           USA CORIELL      0\n",
      "18:        White                    No           USA CORIELL      0\n",
      "19:        White                    No           USA CORIELL      0\n",
      "20:        White                    No           USA CORIELL      0\n",
      "21:        White                    No           USA CORIELL      0\n",
      "22: Not Reported          Not Reported           ISR     LCC      0\n",
      "23:        White          Not Reported           USA    PAGE      0\n",
      "24:        White          Not Reported           USA    PAGE      0\n",
      "25:        White          Not Reported           USA    PAGE      0\n",
      "26:        White          Not Reported           USA    PAGE      0\n",
      "27:        White          Not Reported           USA    PAGE      0\n",
      "28:        White          Not Reported           USA    PAGE      0\n",
      "29:        White          Not Reported           USA    PAGE      0\n",
      "30:        White          Not Reported           USA    PAGE      0\n",
      "31:        White          Not Reported           USA    PAGE      0\n",
      "32:        White          Not Reported           USA    PAGE      0\n",
      "33:        White          Not Reported           USA    PAGE      0\n",
      "34:        White          Not Reported           USA    PAGE      0\n",
      "35:        White          Not Reported           USA    PAGE      0\n",
      "36:        White          Not Reported           USA    PAGE      0\n",
      "37:        White          Not Reported           USA    PAGE      0\n",
      "38:        White          Not Reported           USA    PAGE      0\n",
      "39:        White          Not Reported           USA    PPMI      0\n",
      "40:        White          Not Reported           USA    PPMI      0\n",
      "41: Not Reported                    No           USA     UMD      0\n",
      "     race_for_qc family_history_for_qc region_for_qc   study pruned\n",
      "    pruned_reason label related   V1 FID PAT MAT SEX PHENOTYPE\n",
      " 1:            NA    AJ      NA   43   0   0   0   2         1\n",
      " 2:            NA    AJ      NA   45   0   0   0   1         1\n",
      " 3:            NA    AJ      NA  484   0   0   0   2         1\n",
      " 4:            NA    AJ      NA  485   0   0   0   2         1\n",
      " 5:            NA    AJ      NA  487   0   0   0   1         1\n",
      " 6:            NA    AJ      NA  501   0   0   0   1         1\n",
      " 7:            NA    AJ      NA  517   0   0   0   2         1\n",
      " 8:            NA    AJ      NA  520   0   0   0   1         1\n",
      " 9:            NA    AJ      NA  523   0   0   0   2         1\n",
      "10:            NA    AJ      NA  532   0   0   0   2         1\n",
      "11:            NA    AJ      NA  533   0   0   0   1         1\n",
      "12:            NA    AJ      NA  557   0   0   0   2         1\n",
      "13:            NA    AJ      NA  571   0   0   0   2         1\n",
      "14:            NA    AJ      NA  573   0   0   0   1         1\n",
      "15:            NA    AJ      NA  586   0   0   0   1         1\n",
      "16:            NA    AJ      NA  590   0   0   0   2         1\n",
      "17:            NA    AJ      NA  601   0   0   0   2         1\n",
      "18:            NA    AJ      NA  602   0   0   0   1         1\n",
      "19:            NA    AJ      NA  603   0   0   0   1         1\n",
      "20:            NA    AJ      NA  607   0   0   0   1         1\n",
      "21:            NA    AJ      NA  628   0   0   0   1         1\n",
      "22:            NA    AJ      NA  648   0   0   0   2         1\n",
      "23:            NA    AJ      NA  664   0   0   0   2         1\n",
      "24:            NA    AJ      NA  711   0   0   0   1         1\n",
      "25:            NA    AJ      NA  721   0   0   0   1         1\n",
      "26:            NA    AJ      NA  744   0   0   0   1         1\n",
      "27:            NA    AJ      NA  763   0   0   0   1         1\n",
      "28:            NA    AJ      NA  766   0   0   0   1         1\n",
      "29:            NA    AJ      NA  769   0   0   0   1         1\n",
      "30:            NA    AJ      NA  780   0   0   0   1         1\n",
      "31:            NA    AJ      NA  787   0   0   0   2         1\n",
      "32:            NA    AJ      NA  804   0   0   0   1         1\n",
      "33:            NA    AJ      NA  807   0   0   0   1         1\n",
      "34:            NA    AJ      NA  812   0   0   0   2         1\n",
      "35:            NA    AJ      NA  824   0   0   0   2         1\n",
      "36:            NA    AJ      NA  833   0   0   0   2         1\n",
      "37:            NA    AJ      NA  839   0   0   0   1         1\n",
      "38:            NA    AJ      NA  843   0   0   0   1         1\n",
      "39:            NA    AJ      NA  857   0   0   0   2         1\n",
      "40:            NA    AJ      NA 1104   0   0   0   1         1\n",
      "41:            NA    AJ      NA 1273   0   0   0   2         1\n",
      "    pruned_reason label related   V1 FID PAT MAT SEX PHENOTYPE\n",
      "    chr9:129129094:A:G_G chr9:129147457:C:T_T\n",
      " 1:                    0                    1\n",
      " 2:                    0                    1\n",
      " 3:                    0                    1\n",
      " 4:                    0                    1\n",
      " 5:                    0                    1\n",
      " 6:                    0                    1\n",
      " 7:                    0                    1\n",
      " 8:                    0                    1\n",
      " 9:                    0                    1\n",
      "10:                    0                    1\n",
      "11:                    0                    1\n",
      "12:                    0                    1\n",
      "13:                    0                    1\n",
      "14:                    0                    1\n",
      "15:                    0                    1\n",
      "16:                    0                    1\n",
      "17:                    0                    1\n",
      "18:                    0                    1\n",
      "19:                    0                    1\n",
      "20:                    0                    1\n",
      "21:                    0                    1\n",
      "22:                    0                    1\n",
      "23:                    0                    1\n",
      "24:                    0                    1\n",
      "25:                    0                    1\n",
      "26:                    0                    1\n",
      "27:                    0                    1\n",
      "28:                    0                    1\n",
      "29:                    0                    1\n",
      "30:                    0                    1\n",
      "31:                    0                    1\n",
      "32:                    0                    1\n",
      "33:                    0                    1\n",
      "34:                    0                    1\n",
      "35:                    0                    1\n",
      "36:                    0                    1\n",
      "37:                    0                    1\n",
      "38:                    0                    1\n",
      "39:                    0                    1\n",
      "40:                    0                    1\n",
      "41:                    0                    1\n",
      "    chr9:129129094:A:G_G chr9:129147457:C:T_T\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "setwd('/home/jupyter/PP2A_GP2/')\n",
    "library(data.table)\n",
    "PP2A_het_controles_129147457_AJ <- fread(\"PP2A_het_controles_129147457_AJ.csv\", header =T)\n",
    "PP2A_het_controles_129147457_AJ\n",
    "temp <- merge(covs, PP2A_het_controles_129147457_AJ, by=\"IID\")\n",
    "head(temp)\n",
    "write.table(temp, file = \"PTPA_het_129147457_controles_extracted.txt\", quote = F, row.names = F, sep = \"\\t\")\n",
    "PP2A_het_controles_129147457_AJ_extracted <- fread(\"PP2A_het_controles_129147457_AJ_extracted.txt\", header =T)\n",
    "PP2A_het_controles_129147457_AJ_extracted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                 IID          GP2ID manifest_id phenotype pheno_for_qc\n",
      "1: CORIELL_000035_s1 CORIELL_000035          m3        PD            2\n",
      "2: CORIELL_000138_s1 CORIELL_000138          m3        PD            2\n",
      "3: CORIELL_001140_s1 CORIELL_001140          m3        PD            2\n",
      "4: CORIELL_001997_s1 CORIELL_001997          m3        PD            2\n",
      "5: CORIELL_002302_s1 CORIELL_002302          m3        PD            2\n",
      "6: CORIELL_003933_s1 CORIELL_003933          m3        PD            2\n",
      "7: CORIELL_004167_s1 CORIELL_004167          m3        PD            2\n",
      "8:    PAGE_001019_s1    PAGE_001019          m1        PD            2\n",
      "9:     UMD_000157_s1     UMD_000157          m1        PD            2\n",
      "   other_pheno sex_for_qc      age age_of_onset age_at_diagnosis age_at_death\n",
      "1:          PD          2 60.00000     59.00000               NA           NA\n",
      "2:          PD          1 56.00000     53.00000               NA           NA\n",
      "3:          PD          1 52.00000     50.00000               NA           NA\n",
      "4:          PD          1 68.00000     49.00000             50.0           NA\n",
      "5:          PD          1 56.00000     40.00000             41.0           NA\n",
      "6:          PD          2 78.00000     60.00000               NA           NA\n",
      "7:          PD          2 58.00000     43.00000             48.0           NA\n",
      "8:          PD          1 72.81588     55.33333             56.5           NA\n",
      "9:          PD          1       NA           NA               NA           NA\n",
      "   race_for_qc family_history_for_qc region_for_qc   study pruned pruned_reason\n",
      "1:       White                    No           USA CORIELL      0            NA\n",
      "2:       White                    No           USA CORIELL      0            NA\n",
      "3:       White                    No           USA CORIELL      0            NA\n",
      "4:       White                    No           USA CORIELL      0            NA\n",
      "5:       White                    No           USA CORIELL      0            NA\n",
      "6:       White                   Yes           USA CORIELL      0            NA\n",
      "7:       White                    No           USA CORIELL      0            NA\n",
      "8:       White                    No           USA    PAGE      0            NA\n",
      "9:       White                    No           USA     UMD      0            NA\n",
      "   label related   V1 FID PAT MAT SEX PHENOTYPE chr9:129129094:A:G_G\n",
      "1:    AJ      NA   60   0   0   0   2         2                    1\n",
      "2:    AJ      NA   67   0   0   0   1         2                    1\n",
      "3:    AJ      NA  137   0   0   0   1         2                    1\n",
      "4:    AJ      NA  224   0   0   0   1         2                    1\n",
      "5:    AJ      NA  301   0   0   0   1         2                    1\n",
      "6:    AJ      NA  421   0   0   0   2         2                    1\n",
      "7:    AJ      NA  437   0   0   0   2         2                    1\n",
      "8:    AJ      NA  718   0   0   0   1         2                    1\n",
      "9:    AJ      NA 1252   0   0   0   1         2                    1\n",
      "   chr9:129147457:C:T_T\n",
      "1:                    0\n",
      "2:                    0\n",
      "3:                    0\n",
      "4:                    0\n",
      "5:                    0\n",
      "6:                    1\n",
      "7:                    0\n",
      "8:                    0\n",
      "9:                    0\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "setwd('/home/jupyter/PP2A_GP2/')\n",
    "library(data.table)\n",
    "PP2A_het_casos_129129094_AJ <- fread(\"PP2A_het_casos_129129094_AJ.csv\", header =T)\n",
    "PP2A_het_casos_129129094_AJ\n",
    "temp <- merge(covs, PP2A_het_casos_129129094_AJ, by=\"IID\")\n",
    "head(temp)\n",
    "write.table(temp, file = \"PP2A_het_casos_129129094_AJ_extracted.txt\", quote = F, row.names = F, sep = \"\\t\")\n",
    "PP2A_het_casos_129129094_AJ_extracted <- fread(\"PP2A_het_casos_129129094_AJ_extracted.txt\", header =T)\n",
    "PP2A_het_casos_129129094_AJ_extracted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Error in bmerge(i, x, leftcols, rightcols, roll, rollends, nomatch, mult,  : \n",
      "  Incompatible join types: x.IID (logical) and i.IID (character)\n"
     ]
    },
    {
     "ename": "RInterpreterError",
     "evalue": "Failed to parse and evaluate line 'setwd(\\'/home/jupyter/PP2A_GP2/\\')\\nlibrary(data.table)\\nPP2A_het_controles_129129094_AJ <- fread(\"PP2A_het_controles_129129094_AJ.csv\", header =T)\\nPP2A_het_controles_129129094_AJ\\ntemp <- merge(covs, PP2A_het_controles_129129094_AJ, by=\"IID\")\\nhead(temp)\\nwrite.table(temp, file = \"PP2A_het_controles_129129094_AJ_extracted.txt\", quote = F, row.names = F, sep = \"\\\\t\")\\nPP2A_het_controles_129129094_AJ_extracted <- fread(\"PP2A_het_controles_129129094_AJ_extracted.txt\", header =T)\\nPP2A_het_controles_129129094_AJ_extracted\\n'.\nR error message: 'Error in bmerge(i, x, leftcols, rightcols, roll, rollends, nomatch, mult,  : \\n  Incompatible join types: x.IID (logical) and i.IID (character)'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mRRuntimeError\u001b[0m                             Traceback (most recent call last)",
      "\u001b[0;32m~/.local/lib/python3.7/site-packages/rpy2/ipython/rmagic.py\u001b[0m in \u001b[0;36meval\u001b[0;34m(self, code)\u001b[0m\n\u001b[1;32m    401\u001b[0m                 value, visible = ri.evalr_expr_with_visible(\n\u001b[0;32m--> 402\u001b[0;31m                     \u001b[0mr_expr\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    403\u001b[0m                 )\n",
      "\u001b[0;32m~/.local/lib/python3.7/site-packages/rpy2/rinterface.py\u001b[0m in \u001b[0;36mevalr_expr_with_visible\u001b[0;34m(expr, envir)\u001b[0m\n\u001b[1;32m    182\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0merror_occured\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 183\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0membedded\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mRRuntimeError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0m_rinterface\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_geterrmessage\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    184\u001b[0m         \u001b[0mres\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mconversion\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_cdata_to_rinterface\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mr_res\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mRRuntimeError\u001b[0m: Error in bmerge(i, x, leftcols, rightcols, roll, rollends, nomatch, mult,  : \n  Incompatible join types: x.IID (logical) and i.IID (character)\n",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mRInterpreterError\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m/tmp/ipykernel_93/2045526081.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mget_ipython\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrun_cell_magic\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'R'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m''\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'setwd(\\'/home/jupyter/PP2A_GP2/\\')\\nlibrary(data.table)\\nPP2A_het_controles_129129094_AJ <- fread(\"PP2A_het_controles_129129094_AJ.csv\", header =T)\\nPP2A_het_controles_129129094_AJ\\ntemp <- merge(covs, PP2A_het_controles_129129094_AJ, by=\"IID\")\\nhead(temp)\\nwrite.table(temp, file = \"PP2A_het_controles_129129094_AJ_extracted.txt\", quote = F, row.names = F, sep = \"\\\\t\")\\nPP2A_het_controles_129129094_AJ_extracted <- fread(\"PP2A_het_controles_129129094_AJ_extracted.txt\", header =T)\\nPP2A_het_controles_129129094_AJ_extracted\\n'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m/opt/conda/lib/python3.7/site-packages/IPython/core/interactiveshell.py\u001b[0m in \u001b[0;36mrun_cell_magic\u001b[0;34m(self, magic_name, line, cell)\u001b[0m\n\u001b[1;32m   2460\u001b[0m             \u001b[0;32mwith\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbuiltin_trap\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2461\u001b[0m                 \u001b[0margs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mmagic_arg_s\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2462\u001b[0;31m                 \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfn\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2463\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2464\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/conda/lib/python3.7/site-packages/decorator.py\u001b[0m in \u001b[0;36mfun\u001b[0;34m(*args, **kw)\u001b[0m\n\u001b[1;32m    230\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mkwsyntax\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    231\u001b[0m                 \u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkw\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfix\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkw\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msig\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 232\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mcaller\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfunc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mextras\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0margs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkw\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    233\u001b[0m     \u001b[0mfun\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__name__\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfunc\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__name__\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    234\u001b[0m     \u001b[0mfun\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__doc__\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfunc\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__doc__\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/conda/lib/python3.7/site-packages/IPython/core/magic.py\u001b[0m in \u001b[0;36m<lambda>\u001b[0;34m(f, *a, **k)\u001b[0m\n\u001b[1;32m    185\u001b[0m     \u001b[0;31m# but it's overkill for just that one bit of state.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    186\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mmagic_deco\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 187\u001b[0;31m         \u001b[0mcall\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mlambda\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0ma\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mk\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0ma\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mk\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    188\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    189\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mcallable\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.local/lib/python3.7/site-packages/rpy2/ipython/rmagic.py\u001b[0m in \u001b[0;36mR\u001b[0;34m(self, line, cell, local_ns)\u001b[0m\n\u001b[1;32m    976\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstdout\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mendswith\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0me\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0merr\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    977\u001b[0m                 \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0me\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0merr\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 978\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    979\u001b[0m         \u001b[0;32mfinally\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    980\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdevice\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mDEVICES_STATIC\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.local/lib/python3.7/site-packages/rpy2/ipython/rmagic.py\u001b[0m in \u001b[0;36mR\u001b[0;34m(self, line, cell, local_ns)\u001b[0m\n\u001b[1;32m    941\u001b[0m                     \u001b[0mreturn_output\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mFalse\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    942\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 943\u001b[0;31m                 \u001b[0mtext_result\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvisible\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0meval\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcode\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    944\u001b[0m                 \u001b[0mtext_output\u001b[0m \u001b[0;34m+=\u001b[0m \u001b[0mtext_result\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    945\u001b[0m                 \u001b[0;32mif\u001b[0m \u001b[0mvisible\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.local/lib/python3.7/site-packages/rpy2/ipython/rmagic.py\u001b[0m in \u001b[0;36meval\u001b[0;34m(self, code)\u001b[0m\n\u001b[1;32m    406\u001b[0m                 \u001b[0mwarning_or_other_msg\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mflush\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    407\u001b[0m                 raise RInterpreterError(code, str(exception),\n\u001b[0;32m--> 408\u001b[0;31m                                         warning_or_other_msg)\n\u001b[0m\u001b[1;32m    409\u001b[0m             \u001b[0;32mfinally\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    410\u001b[0m                 \u001b[0mro\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_print_deferred_warnings\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mRInterpreterError\u001b[0m: Failed to parse and evaluate line 'setwd(\\'/home/jupyter/PP2A_GP2/\\')\\nlibrary(data.table)\\nPP2A_het_controles_129129094_AJ <- fread(\"PP2A_het_controles_129129094_AJ.csv\", header =T)\\nPP2A_het_controles_129129094_AJ\\ntemp <- merge(covs, PP2A_het_controles_129129094_AJ, by=\"IID\")\\nhead(temp)\\nwrite.table(temp, file = \"PP2A_het_controles_129129094_AJ_extracted.txt\", quote = F, row.names = F, sep = \"\\\\t\")\\nPP2A_het_controles_129129094_AJ_extracted <- fread(\"PP2A_het_controles_129129094_AJ_extracted.txt\", header =T)\\nPP2A_het_controles_129129094_AJ_extracted\\n'.\nR error message: 'Error in bmerge(i, x, leftcols, rightcols, roll, rollends, nomatch, mult,  : \\n  Incompatible join types: x.IID (logical) and i.IID (character)'"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "setwd('/home/jupyter/PP2A_GP2/')\n",
    "library(data.table)\n",
    "PP2A_het_controles_129129094_AJ <- fread(\"PP2A_het_controles_129129094_AJ.csv\", header =T)\n",
    "PP2A_het_controles_129129094_AJ\n",
    "temp <- merge(covs, PP2A_het_controles_129129094_AJ, by=\"IID\")\n",
    "head(temp)\n",
    "write.table(temp, file = \"PP2A_het_controles_129129094_AJ_extracted.txt\", quote = F, row.names = F, sep = \"\\t\")\n",
    "PP2A_het_controles_129129094_AJ_extracted <- fread(\"PP2A_het_controles_129129094_AJ_extracted.txt\", header =T)\n",
    "PP2A_het_controles_129129094_AJ_extracted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                 IID          GP2ID manifest_id phenotype pheno_for_qc\n",
      "1: BIOFIND_000043_s1 BIOFIND_000043          m1        PD            2\n",
      "2: CORIELL_001619_s1 CORIELL_001619          m3        PD            2\n",
      "   other_pheno sex_for_qc age age_of_onset age_at_diagnosis age_at_death\n",
      "1:          PD          1  82           67               68           NA\n",
      "2:          PD          2  62           36               36           NA\n",
      "   race_for_qc family_history_for_qc region_for_qc   study pruned pruned_reason\n",
      "1:       White                   Yes           USA BIOFIND      0            NA\n",
      "2:       White                    No           USA CORIELL      0            NA\n",
      "   label related  V1 FID PAT MAT SEX PHENOTYPE chr9:129129094:A:G_G\n",
      "1:    AJ      NA  35   0   0   0   1         2                    0\n",
      "2:    AJ      NA 176   0   0   0   2         2                    0\n",
      "   chr9:129147457:C:T_T\n",
      "1:                    2\n",
      "2:                    2\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "setwd('/home/jupyter/PP2A_GP2/')\n",
    "library(data.table)\n",
    "PP2A_hom_casos_129147457_AJ <- fread(\"PP2A_hom_casos_129147457_AJ.csv\", header =T)\n",
    "PP2A_hom_casos_129147457_AJ\n",
    "temp <- merge(covs, PP2A_hom_casos_129147457_AJ, by=\"IID\")\n",
    "head(temp)\n",
    "write.table(temp, file = \"PP2A_hom_casos_129147457_AJ_extracted.txt\", quote = F, row.names = F, sep = \"\\t\")\n",
    "PP2A_hom_casos_129147457_AJ_extracted <- fread(\"PP2A_hom_casos_129147457_AJ_extracted.txt\", header =T)\n",
    "PP2A_hom_casos_129147457_AJ_extracted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Save out results..!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR} {WORKSPACE_BUCKET}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.12"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {
    "height": "248.972px",
    "width": "263.062px"
   },
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "268.366px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
